<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Clustering 3K PBMCs with Seurat / Clustering 3K PBMCs with Seurat / Single Cell</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@84fb1485f5a2b0e4f71d6539516636907c91b67f",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Clustering 3K PBMCs with Seurat">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2025-02-13 09:44:22 +0000">
<meta name="DC.creator" content="Marisa Loach"><meta name="description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). ">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Single Cell / Clustering 3K PBMCs with Seurat / Hands-on: Clustering 3K PBMCs with Seurat">
        <meta property="og:description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). ">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.png">
	<script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": [
    {
      "@type": "CreativeWork",
      "name": "Galaxy Training: A Powerful Framework for Teaching!",
      "url": "https://doi.org/10.1371/journal.pcbi.1010752"
    },
    {
      "@type": "CreativeWork",
      "name": "Community-Driven Data Analysis Training for Biology",
      "url": "https://doi.org/10.1016/j.cels.2018.05.012"
    }
  ],
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Clustering 3K PBMCs with Seurat",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [],
  "creativeWorkStatus": "Active",
  "dateModified": "2025-02-13 09:44:22 +0000",
  "datePublished": "2025-01-23 21:33:29 +0000",
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "funder": [],
  "funding": [],
  "identifier": "https://gxy.io/GTN:",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Single Cell",
    "description": "Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!).\n",
    "url": "https://training.galaxyproject.org/training-material/topics/single-cell/"
  },
  "abstract": "Single cell RNA-seq analysis enables us to explore differences in gene expression between cells. It can reveal the heterogenity within cell populations and help us to identify cell types that could play roles in development, disease, or other processes. Single cell omics is a relatively young field, but there are a few commonly-used analysis pipelines that you will often see in the literature. In this tutorial, we will use one of these pipelines, Seurat, to cluster single cell data from a 10X Genomics experiment. You can follow the same analysis using the Scanpy pipeline in the [Clustering 3K PBMCs with Scanpy](http://localhost:4000/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html) tutorial.",
  "learningResourceType": "e-learning",
  "name": "Clustering 3K PBMCs with Seurat",
  "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html",
  "version": 2,
  "timeRequired": "PT8H",
  "teaches": "- Explain the steps involved in clustering single cell data\n- Evaluate the quality of single cell data and filter out low quality cells\n- Prepare single cell data for analysis with Seurat\n- Perform clustering with Seurat\n- Be ready to apply the Seurat pipeline to new datasets",
  "keywords": [
    "Single Cell",
    "10x"
  ],
  "description": "## Abstract\n\nSingle cell RNA-seq analysis enables us to explore differences in gene expression between cells. It can reveal the heterogenity within cell populations and help us to identify cell types that could play roles in development, disease, or other processes. Single cell omics is a relatively young field, but there are a few commonly-used analysis pipelines that you will often see in the literature. In this tutorial, we will use one of these pipelines, Seurat, to cluster single cell data from a 10X Genomics experiment. You can follow the same analysis using the Scanpy pipeline in the [Clustering 3K PBMCs with Scanpy](http://localhost:4000/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html) tutorial.\n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - How can we identify cell types in single cell RNA-Seq data?\n - What are the steps for clustering single cell data with Seurat?\n\n\n## Learning Objectives\n\n- Explain the steps involved in clustering single cell data\n- Evaluate the quality of single cell data and filter out low quality cells\n- Prepare single cell data for analysis with Seurat\n- Perform clustering with Seurat\n- Be ready to apply the Seurat pipeline to new datasets\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-preprocessing/slides.html",
      "name": "Dealing with Cross-Contamination in Fixed Barcode Protocols",
      "description": "Slides for 'Dealing with Cross-Contamination in Fixed Barcode Protocols' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html",
      "name": "Pre-processing of Single-Cell RNA Data",
      "description": "Hands-on for 'Pre-processing of Single-Cell RNA Data' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html",
      "name": "Pre-processing of 10X Single-Cell RNA Datasets",
      "description": "Hands-on for 'Pre-processing of 10X Single-Cell RNA Datasets' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/MarisaJL/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/MarisaJL/",
      "name": "Marisa Loach",
      "image": "https://avatars.githubusercontent.com/MarisaJL",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-6979-6930",
      "orcid": "https://orcid.org/0000-0001-6979-6930"
    }
  ],
  "contributor": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/pavanvidem/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/pavanvidem/",
      "name": "Pavankumar Videm",
      "image": "https://avatars.githubusercontent.com/pavanvidem",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-5192-126X",
      "orcid": "https://orcid.org/0000-0002-5192-126X"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/shiltemann/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/shiltemann/",
      "name": "Saskia Hiltemann",
      "image": "https://avatars.githubusercontent.com/shiltemann",
      "description": "Researcher at Erasmus Medical Center",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/nfdi4plants/",
          "name": "DataPLANT (NFDI4Plants)",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://nfdi4plants.org"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-3803-468X",
      "orcid": "https://orcid.org/0000-0003-3803-468X"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Single Cell",
      "description": "Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!).\n",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/workflows/",
      "name": "Associated Workflows"
    },
    {
      "@type": "Thing",
      "url": "https://bio.tools/tool/seurat",
      "name": "Seurat"
    },
    {
      "@type": "Thing",
      "url": "https://bio.tools/tool/galaxy",
      "name": "Galaxy"
    },
    {
      "@type": "Thing",
      "url": "https://zenodo.org/record/3581213",
      "name": "Associated Training Datasets"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/single-cell" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Single Cell
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/single-cell/faqs/" title="Check our FAQs for the Single Cell topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-single-cell">
    <h1 data-toc-skip>Clustering 3K PBMCs with Seurat</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/MarisaJL/" class="contributor-badge contributor-MarisaJL"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/MarisaJL?s=36" alt="Marisa Loach avatar" width="36" class="avatar" />
    Marisa Loach</a>
		</td>
	</tr>
	

	

	

	

	

	
	<tr>
		<td><abbr title="These people edited the text, either for spelling and grammar, flow, GTN-fit, or other similar editing categories">Editor(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/pavanvidem/" class="contributor-badge contributor-pavanvidem"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/pavanvidem?s=36" alt="Pavankumar Videm avatar" width="36" class="avatar" />
    Pavankumar Videm</a><a href="/training-material/hall-of-fame/shiltemann/" class="contributor-badge contributor-shiltemann"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/shiltemann?s=36" alt="Saskia Hiltemann avatar" width="36" class="avatar" />
    Saskia Hiltemann</a></td>
	</tr>
	

	

	
	<tr class="reviewers">
		<td><abbr title="These people reviewed this material for accuracy and correctness">Reviewers</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/pavanvidem/" class="contributor-badge contributor-badge-small contributor-pavanvidem"><img src="https://avatars.githubusercontent.com/pavanvidem?s=36" alt="Pavankumar Videm avatar" width="36" class="avatar" /></a><a href="/training-material/hall-of-fame/shiltemann/" class="contributor-badge contributor-badge-small contributor-shiltemann"><img src="https://avatars.githubusercontent.com/shiltemann?s=36" alt="Saskia Hiltemann avatar" width="36" class="avatar" /></a></td>
	</tr>
	

	

	

</table>


	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How can we identify cell types in single cell RNA-Seq data?</p>
</li>
        
        <li><p>What are the steps for clustering single cell data with Seurat?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Explain the steps involved in clustering single cell data</p>
</li>
        
        <li><p>Evaluate the quality of single cell data and filter out low quality cells</p>
</li>
        
        <li><p>Prepare single cell data for analysis with Seurat</p>
</li>
        
        <li><p>Perform clustering with Seurat</p>
</li>
        
        <li><p>Be ready to apply the Seurat pipeline to new datasets</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    


        
    
        
        
        
            
                
                    
                        
                            <li>
                              <a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Pre-processing of Single-Cell RNA Data</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Pre-processing of Single-Cell RNA Data</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Pre-processing of 10X Single-Cell RNA Datasets</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 8 hours</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Introductory <span class="visually-hidden">Introductory</span>
<span class="level introductory" title="Introductory Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/3581213">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/workflows/" title="Clustering 3K PBMCs with Seurat workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
                <li class="btn btn-default supporting_material">
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Answer Histories">
        <i class="fas fa-sign-out-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-answer</span>&nbsp;Answer Histories
    </a>
    <ul class="dropdown-menu">
      
        <li>
            <a class="dropdown-item" href="https://usegalaxy.eu/u/marisa_jl/h/clustering-3k-pbmcs-with-seurat---separate-preprocessing---answer-key">
              Using Separate Preprocessing Tools
              
              <br/><span class="text-muted">2023-10-28</span>
              
            </a>
        </li>
      
        <li>
            <a class="dropdown-item" href="https://usegalaxy.eu/u/marisa_jl/h/clustering-3k-pbmcs-with-seurat---sctransform---answer-ke">
              Using SCTransform
              
              <br/><span class="text-muted">2023-10-28</span>
              
            </a>
        </li>
      
      <li class="dropdown-divider"></li>
      <li>
        <a class="dropdown-item" href="/training-material/faqs/gtn/gtn_example_histories.html"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> How to Use This</a>
      </li>
    </ul>

</li>
            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            












            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://mississippi.sorbonne-universite.fr" title="">
			MISSISSIPPI
		</a>
	</li>
    
    

    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Jan 23, 2025 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Feb 13, 2025 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        

	
	

	
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 2 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>Single cell RNA-seq analysis enables us to explore differences in gene expression between cells. It can reveal the heterogenity within cell populations and help us to identify cell types that could play roles in development, disease, or other processes. Single cell omics is a relatively young field, but there are a few commonly-used analysis pipelines that you will often see in the literature. In this tutorial, we will use one of these pipelines, Seurat, to cluster single cell data from a 10X Genomics experiment (<span class="citation"><a href="#seurat2023v5">Hao <i>et al.</i> 2023</a></span>). You can follow the same analysis using the Scanpy pipeline in the <a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">Clustering 3K PBMCs with Scanpy</a> tutorial.</p>

<p>Clustering is typically the first type of analysis we will perform on a single cell dataset. It groups together cells that are expressing similar genes, which makes the data easier to understand and often helps us to identify specific cell types.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

  <p>This tutorial is based on the <a href="https://satijalab.org/seurat/articles/pbmc3k_tutorial">Seurat - Guided Clustering Tutorial</a>. The SCTransform sections also draw from the <a href="https://satijalab.org/seurat/articles/sctransform_vignette.html">Using sctransform in Seurat</a> tutorial.</p>

</blockquote>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#important-tips-for-easier-analysis" id="markdown-toc-important-tips-for-easier-analysis">Important tips for easier analysis</a></li>
  <li><a href="#data" id="markdown-toc-data">Data</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
      <li><a href="#seuratobject" id="markdown-toc-seuratobject">SeuratObject</a></li>
    </ol>
  </li>
  <li><a href="#preprocessing" id="markdown-toc-preprocessing">Preprocessing</a>    <ol>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality Control</a></li>
      <li><a href="#further-preprocessing" id="markdown-toc-further-preprocessing">Further Preprocessing</a></li>
    </ol>
  </li>
  <li><a href="#dimensionality-reduction" id="markdown-toc-dimensionality-reduction">Dimensionality Reduction</a>    <ol>
      <li><a href="#principal-component-analysis" id="markdown-toc-principal-component-analysis">Principal Component Analysis</a></li>
    </ol>
  </li>
  <li><a href="#cluster-the-cells" id="markdown-toc-cluster-the-cells">Cluster the Cells</a>    <ol>
      <li><a href="#find-neighbors" id="markdown-toc-find-neighbors">Find Neighbors</a></li>
      <li><a href="#find-clusters" id="markdown-toc-find-clusters">Find Clusters</a></li>
    </ol>
  </li>
  <li><a href="#run-non-linear-dimensional-reduction-umaptsne" id="markdown-toc-run-non-linear-dimensional-reduction-umaptsne">Run non-linear dimensional reduction (UMAP/tSNE)</a>    <ol>
      <li><a href="#visualise-umap" id="markdown-toc-visualise-umap">Visualise UMAP</a></li>
      <li><a href="#other-ways-to-visualise-clusters" id="markdown-toc-other-ways-to-visualise-clusters">Other ways to visualise clusters</a></li>
    </ol>
  </li>
  <li><a href="#find-marker-genes" id="markdown-toc-find-marker-genes">Find Marker Genes</a>    <ol>
      <li><a href="#find-positive-markers-for-every-cluster-compared-to-the-rest" id="markdown-toc-find-positive-markers-for-every-cluster-compared-to-the-rest">Find Positive Markers for Every Cluster Compared to the Rest</a></li>
      <li><a href="#find-markers-of-cluster-2" id="markdown-toc-find-markers-of-cluster-2">Find Markers of Cluster 2</a></li>
      <li><a href="#find-markers-distinguishing-cluster-5-from-clusters-0-and-3" id="markdown-toc-find-markers-distinguishing-cluster-5-from-clusters-0-and-3">Find Markers Distinguishing Cluster 5 from Clusters 0 and 3</a></li>
      <li><a href="#find-markers-with-the-highest-classification-power" id="markdown-toc-find-markers-with-the-highest-classification-power">Find markers with the highest ‘classification power’</a></li>
    </ol>
  </li>
  <li><a href="#identify-cell-types" id="markdown-toc-identify-cell-types">Identify Cell Types</a>    <ol>
      <li><a href="#known-markers-for-pbmcs" id="markdown-toc-known-markers-for-pbmcs">Known Markers for PBMCs</a></li>
      <li><a href="#canonical-markers-vs-marker-genes" id="markdown-toc-canonical-markers-vs-marker-genes">Canonical Markers vs Marker Genes</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="important-tips-for-easier-analysis">Important tips for easier analysis</h1>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-use-our-single-cell-omics-lab"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-use-our-single-cell-omics-lab" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Use our Single Cell Omics Lab</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Did you know we have a unique Single Cell Omics Lab with all our single cell tools highlighted to make it easier to use on Galaxy? We recommend this site for all your single cell analysis needs, particularly for newer users.</p>   <p>The Single Cell Omics Lab is a different view of the underlying Galaxy server that organises tools and resources better for single-cell users! It also provides a platform for communities to engage and connect; distribute more targeted news and events; and highlight community-specific funding sources.</p>   <p>Try it out!</p>   <ul>   <li><i class="fa fa-tv" aria-hidden="true"></i><span class="visually-hidden">subdomain</span> <a href="https://singlecell.usegalaxy.eu">Europe: Single Cell Omics Lab</a></li>   <li><i class="fa fa-tv" aria-hidden="true"></i><span class="visually-hidden">subdomain</span> <a href="https://singlecell.usegalaxy.org">USA: Single Cell Omics Lab</a></li>   <li><i class="fa fa-tv" aria-hidden="true"></i><span class="visually-hidden">subdomain</span> <a href="https://singlecell.usegalaxy.org.au">Australia: Single Cell Omics Lab</a></li> </ul> </blockquote>
<p><!--END_SNIPPET--></p>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-using-tutorial-mode"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-tutorial-mode" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using tutorial mode</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool. This may NOT be the same tool used in the tutorial you are accessing. Furthermore, if you use a newer tool in one step, and try using an older tool in the next step… this may fail! To ensure you use the same tool versions of a given tutorial, use the <strong>Tutorial mode</strong> feature.</p>   <ul>   <li>Open your Galaxy server</li>   <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>   <li>Navigate to your tutorial</li>   <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>   <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>   <li>You can click anywhere in the grey-ed out area outside of the tutorial box to return back to the Galaxy analytical interface</li> </ul>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-not-all-browsers-work"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Not all browsers work!</div>   <ul>     <li>We’ve had some issues with Tutorial mode on Safari for Mac users.</li>     <li>Try a <strong>different browser</strong> if you aren’t seeing the button.</li>   </ul> </blockquote> </blockquote>
<p><!--END_SNIPPET--></p>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-troubleshooting-errors"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-troubleshooting-errors" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Troubleshooting errors</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>When something goes wrong in Galaxy, there are a number of things you can do to find out what it was. Error messages can help you figure out whether it was a problem with one of the settings of the tool, or with the input data, or maybe there is a bug in the tool itself and the problem should be reported. Below are the steps you can follow to troubleshoot your Galaxy errors.</p>   <ol>   <li><strong>Expand</strong> the red history dataset by clicking on it.     <ul>       <li>Sometimes you can already see an error message here</li>     </ul>   </li>   <li>     <p><strong>View the error message</strong> by clicking on the <strong>bug icon</strong> <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span></p>   </li>   <li><strong>Check the logs.</strong> Output (stdout) and error logs (stderr) of the tool are available:     <ul>       <li>Expand the history item</li>       <li>Click on the <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> icon</li>       <li>Scroll down to the <strong>Job Information</strong> section to view the 2 logs:         <ul>           <li>Tool Standard Output</li>           <li>Tool Standard Error</li>         </ul>       </li>       <li>For more information about specific tool errors, please see the <a href="/training-material/faqs/galaxy/index.html#troubleshooting">Troubleshooting section</a></li>     </ul>   </li>   <li><strong>Submit a bug report!</strong> If you are still unsure what the problem is.     <ul>       <li>Click on the bug icon <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span></li>       <li>Write down any information you think might help solve the problem         <ul>           <li>See <a href="/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html">this FAQ</a> on how to write good bug reports</li>         </ul>       </li>       <li>Click <strong><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span> Report</strong> button</li>     </ul>   </li>   <li><strong>Ask for help!</strong>     <ul>       <li>Where?         <ul>           <li>In the <a href="http://gxy.io/gtn-slack">User community chatspace in Slack</a> in our <a href="https://gtnsmrgsbord.slack.com/archives/C06PBRR40D7">#single-cell-users</a> channel</li>           <li>In the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Matrix Channel</a></li>           <li>In the <a href="https://gitter.im/galaxyproject/Lobby">Galaxy Matrix Channel</a></li>           <li>Browse the <a href="https://help.galaxyproject.org/">Galaxy Help Forum</a> to see if others have encountered the same problem before (or post your question).</li>         </ul>       </li>       <li>When asking for help, it is useful to <a href="/training-material/faqs/galaxy/histories_sharing.html">share a link to your history</a></li>     </ul>   </li> </ol> </blockquote>
<p><!--END_SNIPPET--></p>

<h1 id="data">Data</h1>

<p>For this tutorial, we will analyze a dataset of Peripheral Blood Mononuclear Cells (PBMC) extracted from a healthy donor, which is freely available from 10X Genomics. The dataset contains 2700 single cells sequenced using Illumina NextSeq 500. The raw sequences have been processed by the <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger"><strong>cellranger</strong></a> pipeline from 10X to extract a unique molecular identifier (UMI) count matrix, in a similar way to that explained in the <a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html">Pre-processing of 10X Single-Cell RNA Datasets</a> tutorial.</p>

<p>In this matrix, the values represent the number of each feature (i.e. gene; row) detected in each cell (column). Single cell matrices can be quite large: here there are 2700 columns with 32,738 rows, so for each of our 2700 cells we know how many times we found RNAs matching each of the 32,738 genes. Since most of these genes weren’t detected in most of the cells, the matrix is largely filled with zeros, i.e. it is an extremely sparse matrix. To optimize the storage of such a table and the information about the genes and cells, <strong>cellranger</strong> creates 3 files:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">genes.tsv</code>: a tabular file with information about the 32,738 genes in 2 columns (Ensembl gene id and the gene symbol)</li>
  <li><code class="language-plaintext highlighter-rouge">barcodes.tsv</code>: a tabular file with the barcode for each of the 2700 cells</li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">matrix.mtx</code>: a condensed version of the count matrix (including the non-zero values only)</p>

    <p>The count matrix is represented by its non-zero values - we don’t need to store all of those zeroes as long as we know where our non-zero values are in the matrix. Each non-zero value is represented by its line number (1st column), its column number (2nd column) and its value (3rd column). The first row gives the total number of rows (genes), columns (cells) and non-zero values. More information on the Matrix Market Exchange (mtx) format can be found <a href="https://math.nist.gov/MatrixMarket/formats.html">in this documentation</a></p>
  </li>
</ul>

<h2 id="data-upload">Data upload</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial</p>
    </li>
    <li>
      <p>Import the <code class="language-plaintext highlighter-rouge">genes.tsv</code>, <code class="language-plaintext highlighter-rouge">barcodes.tsv</code> and <code class="language-plaintext highlighter-rouge">matrix.mtx</code> from <a href="https://zenodo.org/record/3581213">Zenodo</a> or from the shared data library</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3581213/files/genes.tsv
https://zenodo.org/record/3581213/files/barcodes.tsv
https://zenodo.org/record/3581213/files/matrix.mtx
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from a data library</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Libraries</strong> (left panel)</li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Rename the datasets as <code class="language-plaintext highlighter-rouge">genes</code>, <code class="language-plaintext highlighter-rouge">barcodes</code>, and <code class="language-plaintext highlighter-rouge">matrix</code> if necessary</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Check the datatypes are correct - the <code class="language-plaintext highlighter-rouge">genes</code> and <code class="language-plaintext highlighter-rouge">barcodes</code> files should be tsv or tabular while the <code class="language-plaintext highlighter-rouge">matrix</code> should be an mtx file</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select  your desired datatype  from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Inspect the <code class="language-plaintext highlighter-rouge">matrix</code> file by clicking on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> icon</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-dataset-snippet"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-dataset-snippet" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Dataset snippet</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>A single Galaxy dataset can either be “collapsed” or “expanded”.</p>   <p><strong>Collapsed dataset view</strong></p>   <p>Datasets in the panel are initially shown in a “collapsed” view:</p>   <p><img src="/training-material/shared/images/history_item_collapsed.png" alt="Collapsed view of a single Galaxy dataset" /></p>   <p>It contains the following elements:</p>   <ul>   <li><strong>Dataset number</strong>: (“1”) order of dataset in the history;</li>   <li><strong>Dataset name</strong>: (“M117-bl_1.fq.gz”) its name;</li>   <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span>: click this to view the dataset contents;</li>   <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span>: click this to edit dataset properties;</li>   <li><i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span>: click this to delete the dataset from the history (<em>don’t worry</em>, you can undo this action!).</li> </ul>   <p>Clicking on a collapsed dataset will expand it.</p>   <blockquote class="details">   <div class="box-title details-title" id="details-some-buttons-can-be-disabled"><button class="gtn-boxify-button details" type="button" aria-controls="details-some-buttons-can-be-disabled" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Some buttons can be disabled.</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Some of the buttons above may be disabled if the dataset is in a state that doesn’t allow the action. For example, the ‘edit’ button is disabled for datasets that are still queued or running</p> </blockquote>   <p><strong>Expanded dataset view</strong></p>   <p>Expanded dataset view adds a preview element and many additional controls.</p>   <p><img src="/training-material/shared/images/history_item_expanded.png" alt="Expanded view of a single Galaxy dataset" /></p>   <p>In addition to the elements described above for the collapsed dataset, its expanded view contains:</p>   <ul>   <li><strong>Add tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span>: click on this to tag this dateset;</li>   <li><strong>Dataset size</strong>: (“2 variants, 18 comments”) lists the size of the dataset. When datasets are small (like in this example) the exact size is shown. For large datasets, Galaxy gives an approximate estimate.</li>   <li><strong>format</strong>: (“VCF”) lists the datatype;</li>   <li><strong>database</strong>: (“?”) lists which genome built this dataset corresponds to. This usually lists “?” unless the genome build is set explicitly or the dataset is derived from another dataset with defined genome build information;</li>   <li><strong>info field</strong>: (“INFO [2024-03-26 12:08:53,435]…”) displays information provided by the tool that generated this dataset. This varies widely and depends on the type of job that generated this dataset.</li>   <li><i class="fas fa-save" aria-hidden="true"></i><span class="visually-hidden">dataset-save</span>: Saves dataset to disk;</li>   <li><i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">dataset-link</span>: Copies dataset link into clipboard;</li>   <li><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">dataset-info</span>: Displays additional details about the dataset in the center pane;</li>   <li><i class="fas fa-arrow-rotate-right" aria-hidden="true"></i><span class="visually-hidden">dataset-rerun</span>: Reruns job that generated this dataset. This button is unavailable for datasets uploaded into history because they were not produced by a Galaxy tool;</li>   <li><i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">dataset-visualize</span>: Displays visualization options for this dataset. The list of options is dependent on the datatype;</li>   <li><i class="fas fa-sitemap" aria-hidden="true"></i><span class="visually-hidden">dataset-related-datasets</span>: Shows datasets related to this dataset. This is useful for tracking down parental datasets - those that were used as inputs into a job that produced this particular dataset.</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<p>The beginning of the file should look like this:</p>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>32738	2700	2286884
32709	1	4
32707	1	1
32706	1	10
32704	1	1
</code></pre></div>  </div>

  <ol>
    <li>How many non-zero values are in the matrix?</li>
    <li>How many counts were found for the 32,706th gene in the 1st cell?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>The first row tells us there are 2,286,884 non-zero values for the 32,738 genes (rows) and 2,700 cells (columns) - so only 2.6% of the 88,392,600 potential values we could have in this matrix are non-zero. Getting rid of all those zeros has made the matrix much more compact.</li>
      <li>10 counts were found for the 32,706th row (gene) and 1st column (cell), so we collected 10 RNAs that the first cell had produced from this particular gene.</li>
    </ol>

  </blockquote>

</blockquote>

<p>Representing the matrix with these three files is convenient for sharing the data, but not for processing them. Different single cell analysis packages have attempted to solve the problem of storage and analysis by inventing their own formats, which has led to the proliferation of many different ‘standards’ in the scRNA-seq package ecosystem.</p>

<h2 id="seuratobject">SeuratObject</h2>

<p>In order to analyse the data using Seurat, we will first need to create a SeuratObject. A SeuratObject can store all our data, including the gene names, cell barcodes and matrix in a single RDS file. Since Seurat was written in the R programming language, SeuratObjects are saved as RData or RDS files. An RDS file is simply a type of RData file that contains a single object.</p>

<p>SeuratObjects can also hold any metadata we might have about our cells or genes, as well as the information we will produce during our analysis, such as our dimensional reductions and cell clusters. A SeuratObject can hold data in multiple layers (also known as slots in earlier version of Seurat). We can have layers of raw counts, normalised data, and scaled data. We could also store multiple assays or types of data in a single object, although in this tutorial we will only be using one assay for RNA-seq data. The SeuratObject is designed to be self-contained so that we don’t have to work with lots of different files, even if we have different versions or types of data.</p>

<p>Creating a SeuratObject in R would require two steps - first, we would need to read in our data, in this case using the <code class="language-plaintext highlighter-rouge">Read10X</code> function, then secondly we would turn it into a SeuratObject using the <code class="language-plaintext highlighter-rouge">CreateSeuratObject</code> function. On Galaxy, we can perform both steps with a single tool. The <code class="language-plaintext highlighter-rouge">CreateSeuratObject</code> function also generates some QC metrics and performs basic filtering of the data.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-create-a-seuratobject"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Create a SeuratObject</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_create/seurat_create/5.0+galaxy1" title="Seurat Create tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Create</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Create Seurat Object</code>
          <ul>
            <li><em>“Select format of input”</em>: <code class="language-plaintext highlighter-rouge">matrix market (for e.g. 10x data)</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Counts matrix with features as rows, cells as columns (.mtx)”</em>: <code class="language-plaintext highlighter-rouge">matrix.mtx</code></li>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“List of gene names (for rows)”</em>: <code class="language-plaintext highlighter-rouge">genes.tsv</code></li>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“List of cell barcodes (for columns)”</em>: <code class="language-plaintext highlighter-rouge">barcodes.tsv</code></li>
              </ul>
            </li>
            <li><em>“Include features detected in at least this many cells”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
            <li><em>“Include cells where at least this many features are detected”</em>: <code class="language-plaintext highlighter-rouge">200</code></li>
            <li><em>“Calculate percentage of mito genes in each cell”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the generated file to <code class="language-plaintext highlighter-rouge">Input 3k PBMC</code></p>
    </li>
    <li>Check that the format is <code class="language-plaintext highlighter-rouge">rds</code></li>
  </ol>
</blockquote>

<p>We can’t look at the RDS file directly as it is designed for computers to read, rather than humans, but the Seurat tools will now be able to interact with the data. We can also use the <span><strong>Seurat Data Management</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> tool to look inside our new SeuratObject.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-inspect-the-input-3k-pbmc-seuratobject"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Inspect the `Input 3k PBMC` SeuratObject</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_data/seurat_data/5.0+galaxy0" title="Seurat Data Management tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Data Management</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Inspect Seurat Object</code>
          <ul>
            <li><em>“Display information about”</em>: <code class="language-plaintext highlighter-rouge">General</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Click the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> on the output in your history to see some information about your SeuratObject</li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How many cells and genes are in the <code class="language-plaintext highlighter-rouge">Input 3k PBMC</code> SeuratObject we just created?</li>
    <li>Is this the same as the number of cells and genes we started out with?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>If we click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> of the output from <span><strong>Seurat Data Management</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> we can see some general information about our SeuratObject. We can see that it contains information about 2700 samples (cells) and 13,714 features (genes).</li>
      <li>When we looked at the <code class="language-plaintext highlighter-rouge">matrix.mtx</code> file previously, we saw that it contained information on the same 2700 cells, but that there were 32,738 genes with non-zero values in the matrix. That means we are missing 19,024 genes - but don’t worry, this is supposed to happen! Remember that we set the <code class="language-plaintext highlighter-rouge">Include features detected in at least this many cells</code> parameter to <code class="language-plaintext highlighter-rouge">3</code> - this got rid of any genes that were found in fewer than three cells. We assume that these genes are just technical noise (e.g. misreads, reads mapped to the wrong gene) rather than real biological features. We also set the <code class="language-plaintext highlighter-rouge">Include cells where at least this many features are detected</code> to <code class="language-plaintext highlighter-rouge">200</code> - this would have removed any cells that had fewer than 200 genes found in them. In this case, we didn’t remove any cells as all 2700 of them must have had at least 200 genes. If we did have cells with fewer than 200 genes, we would want to get rid of them they are probably low quality cells or cell fragments. We would usually expect to find more features in an undamaged cell that was captured and sequenced properly.</li>
    </ol>

  </blockquote>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>If you would like to see the impact of the initial filtering performed by <span><strong>Seurat Create</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>, then you can rerun this step with these filters removed (or changed) and then perform the QC visualisation steps described in the next section on the unfiltered dataset. The Seurat pipeline usually includes some filtering during object creation, which shouldn’t cause any problems as long as we use reasonable thresholds that only filter out the lowest quality cells and genes.  We can always come back and create a new object with different values if we suspect they weren’t right for our dataset once we make our QC plots or if we have problems later on in the analysis.</p>
</blockquote>

<h1 id="preprocessing">Preprocessing</h1>

<p>The standard Seurat pipeline begins with preprocessing, which includes identifying and filtering out low-quality cells, normalising and scaling the data, and selecting highly variable features.</p>

<h2 id="quality-control">Quality Control</h2>

<p>Quality control is an essential step in preparing single cell data for analysis. The results from a single cell experiment can vary in quality due to cell damage during dissociation, capturing more than one cell in a well or droplet, failures in library preparation such as inefficient reverse transcription (copying of captured RNAs to cDNA) or PCR amplification (making more copies of the cDNA so we have enough material for sequencing) or other technical problems. Low quality cells can cause issues for some analysis tools and may contribute to misleading results.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-impact-of-low-quality-cells-on-the-downstream-analyses"><button class="gtn-boxify-button details" type="button" aria-controls="details-impact-of-low-quality-cells-on-the-downstream-analyses" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Impact of low-quality cells on the downstream analyses</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <h4 id="formation-of-their-own-distinct-clusters">Formation of their own distinct cluster(s)</h4>

  <p>The most obvious cause of this problem is that damaged cells can appear more similar to each other because they have increased proportions of mitochondrial genes or are enriched for certain damage-induced genes. We could end up with a cluster made up of different cell types that share similar RNAs because they’ve been damaged. We might think these damaged cells represent an intermediate state between other cell types or end up misinterpreting our data in other ways.</p>

  <h4 id="distortion-of-population-heterogeneity-during-variance-estimation-or-principal-components-analysis">Distortion of population heterogeneity during variance estimation or principal components analysis</h4>

  <p>When we cluster single cell data, we’re looking for the biggest differences between groups of cells. If we have lots of low-quality cells, then the biggest differences we’ll see might be these differences in quality, rather than something more biologically interesting. Differences in quality can have a big impact because low-quality cells often have low total RNA counts. When we perform scaling and normalisation on these cells during preprocessing, this can make the variances for the genes they do express much bigger than for other cells. When we select the most variable genes in the dataset, we’ll end up picking the ones expressed by low-quality cells. We’ll use these in our PCA analysis, likely ending up with top principal components based on cell quality rather than biology, making it harder to detect the differences we’re actually interested in.</p>

  <h4 id="misidentification-of-upregulated-genes">Misidentification of upregulated genes</h4>

  <p>Another problem that arises when we apply our preprocessing steps to small, low quality cells is that the genes we detect in them can appear to be strongly upregulated. Since we didn’t detect many other genes in these cells, even a small difference in the number of transcripts detected can end up becoming much larger after normalisation. For example, contaminating transcripts may be present in all cells at low but constant levels. With the increased scaling and normalization in low-quality cells, the small counts for these transcripts may become large normalized expression values, so we might think we’ve found a population of cells where these genes are upregulated.</p>

</blockquote>

<p>In order to mitigate these problems, we need to remove low-quality cells at the start of the analysis.</p>

<p>The <code class="language-plaintext highlighter-rouge">CreateSeuratObject</code> function we used in the <span><strong>Seurat Create</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> tool automatically calculates some QC metrics and allowed us to filter out the lowest quality cells and features immediately. We will now check the data to see if it requires any further filtering.</p>

<h3 id="computation-of-qc-metrics">Computation of QC metrics</h3>

<p>We already have two useful QC metrics that were calculated when we created our SeuratObject:</p>

<ul>
  <li><strong>nCount_RNA</strong> the total sum of RNAs that were found in each cell.
  The total number of counts is related to cell size, but it can also be an indication of quality. If <code class="language-plaintext highlighter-rouge">nCount_RNA</code> is very high we might be looking at results from a doublet or multiplet - two or more cells that were isolated together during the experiment. If <code class="language-plaintext highlighter-rouge">nCount_RNA</code> is very low, then it is likely that we lost a lot of the RNA due to cell lysis (breakage) or inefficient cDNA capture and amplification - or perhaps we only captured a fragment of a cell.</li>
  <li><strong>nFeature_RNA</strong> the number of unique genes that were detected in each cell.
  We would expect to see some variation in the variety of genes expressed by cells of different sizes, types, and conditions, but if this number is unusually high or low then it could be a sign of poor quality. High <code class="language-plaintext highlighter-rouge">nFeature_RNA</code> could be another sign of a doublet or multiplet - we might have captured cells of different types and processed them together. Low <code class="language-plaintext highlighter-rouge">nFeature_RNA</code> could be due to loss of RNA if <code class="language-plaintext highlighter-rouge">nCount_RNA</code> is also low, or a sign that we have failed to capture the diversity of the transcript population, perhaps due to technical problems in our experiment.</li>
</ul>

<p>One other metric that is often used to assess cell quality is the proportion of reads that came from the mitochondrial genome. The proportion is often higher in low quality, damaged or dying cells. Mitochondrial RNAs, which are protected inside the mitochondrial membranes, can be the last RNAs to be degraded or lost from a damaged cell, so we can end up with higher proportions of them in a low quality cell. We could have calculated the proportion of mitochondrial genes while creating our SeuratObject, but we will calculate it separately here to see how it is done - and how we could do the same for other types of genes.</p>

<p>We can identify mitochondrial genes from their gene symbols. Human genes that are encoded in the mitochondrial DNA (rather than in the cell nucleus) have names beginning with ‘MT-‘. Different naming patterns are used in other species. The <span><strong>Seurat Create</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> tool can identify genes based on a specific naming pattern and then calculate the proportion of reads in each cell that came from these genes. We could use this tool to calculate the proportions of other genes with shared naming patterns too, such as ribosomal genes.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-calculate-the-proportion-of-mitochondrial-reads"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Calculate the Proportion of Mitochondrial Reads</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_create/seurat_create/5.0+galaxy1" title="Seurat Create tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Create</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Add QC Metrics</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Input 3k PBMC</code> (output of <strong>Seurat Create</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Calculate percentage of reads based on”</em>: <code class="language-plaintext highlighter-rouge">Pattern in gene names</code></li>
            <li><em>“Pattern/regex to match in gene names”</em>: <code class="language-plaintext highlighter-rouge">^MT-</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Mitochondrial Annotations</code></li>
  </ol>
</blockquote>

<p>When we use these QC metrics to remove low quality cells, we are assuming they are independent of the biological state of each cell, which isn’t always the case in reality. For example, some cell types with higher energy demands may have higher proportions of mitochondrial reads because they have more mitochondria. We need to think carefully when deciding which metrics to use for filtering, but for now we will assume that these differences in our dataset are being driven by technical factors (e.g. cell damage, loss of RNAs, problems with capture and sequencing) rather than biological processes (e.g. cell types or conditions). We can therefore remove cells with poor values without misrepresenting the biology in downstream analyses.</p>

<h3 id="filtering-of-low-quality-cells">Filtering of low-quality cells</h3>

<p>The simplest approch for identifing low-quality cells is to apply thresholds on the QC metrics. We assume that any cells beyond these thresholds are low quality cells and that this is due to those technical factors. Although this is a simple strategy, we will need to decide where to set our thresholds, and this will depend on the experimental protocol and biological system - there are no standard thresholds that we can use for every dataset.</p>

<p>We can visualise the QC metrics to help us decide where to set our thresholds for filtering out low quality cells. We want to get rid of cells that have unusually high or low numbers of genes or unique features, as well as cells that have higher proportions of mitochondrial genes.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualise-qc-metrics"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualise QC Metrics</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Mitochondrial Annotations</code> (output of <strong>Seurat Create</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Violin Plot with 'VlnPlot'</code>
          <ul>
            <li><em>“Features to plot”</em>: <code class="language-plaintext highlighter-rouge">nFeature_RNA,nCount_RNA,percent.mt</code></li>
            <li>In <em>“Plot Formatting Options”</em>:
              <ul>
                <li><em>“Number of columns to display”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Mitochondrial Annotations</code> (output of <strong>Seurat Create</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Scatter Plot with 'FeatureScatter'</code>
          <ul>
            <li><em>“First feature to plot”</em>: <code class="language-plaintext highlighter-rouge">nCount_RNA</code></li>
            <li><em>“Second feature to plot”</em>: <code class="language-plaintext highlighter-rouge">percent.mt</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Mitochondrial Annotations</code> (output of <strong>Seurat Create</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Scatter Plot with 'FeatureScatter'</code>
          <ul>
            <li><em>“First feature to plot”</em>: <code class="language-plaintext highlighter-rouge">nCount_RNA</code></li>
            <li><em>“Second feature to plot”</em>: <code class="language-plaintext highlighter-rouge">nFeature_RNA</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip"><button class="gtn-boxify-button tip" type="button" aria-controls="tip" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>If you are re-running the same step again or just want to change a couple of parameters for the same tool, then you can click on a previous dataset to expand it in the history and then click the <i class="fas fa-arrow-rotate-right" aria-hidden="true"></i><span class="visually-hidden">dataset-rerun</span>. The tool you used to create that dataset will open in the main panel, with the settings and inputs you used before. Make any changes and run it again.</p>
</blockquote>

<figure id="figure-1" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_violin_QC_before.png" alt="Three violin plots each showing most points grouped together in the lower part of the plot with some outliers at higher values. " width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_violin_QC_before.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Violin Plots showing the unique features (nFeature_RNA), total counts (nCount_RNA) and the proportion of reads coming from mitochondial genes (percent.mt) for all cells</figcaption></figure>

<figure id="figure-2" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_QC_scatter_before.png" alt="Scatter plot A shows clear positive correlation with the number of unique features increasing as total counts increases. A small number of cells have very high values of both. Scatter plot B does not show a relationship between the proportion of mitochondrial genes and total counts. A small number of cells with low total counts have high values of percent.mt. " width="4408" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_QC_scatter_before.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Scatter plots showing the relationships between the total counts (nCount_RNA) and A. the number of unique features (nFeature_RNA) and B. the proportion of mitochondrial reads (percent.mt)</figcaption></figure>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>What do the violin plots tell us about the cell sizes and quality in our dataset?</li>
    <li>What do the scatter plots in tell us about the relationship between cell size and the other QC metrics?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>The violin plots give us an overview of the cell sizes (nCount_RNA), number of unique genes (nFeature_RNA), and proportion of mitochondrial reads for our cells. We can see that most of the cells are grouped together near the bottom of each plot, but there are some outliers that have unusually high or low values of each metric. Cells with higher values of nCount_RNA are likely to be larger as they contained more RNA. Cells with higher values of nFeature_RNA had RNAs produced from lots of different genes, while those with low nFeature_RNA had RNAs produced from a smaller range of genes. Cells with high percent.mt had lots of reads from mitochondrial genes. Although there will be some natural variation in cell size, the range of genes being expressed, and mitochondrial content, we suspect that extreme values of these three metrics reflect low quality cells.</li>
      <li>The scatter plots show us how these QC metrics relate to each other. As expected, the cells with higher total RNA counts also tend to have higher numbers of unique features, but some cells have particularly high values of both - these could be doublets. The proportion of mitochondrial genes doesn’t increase with cell size (total counts) in the same way. We can see a small number of cells that have very high proportions of mitochondrial reads despite having low total reads - these are likely to be damaged cells that have lost a lot of their non-mitochondrial RNAs.</li>
    </ol>

  </blockquote>
</blockquote>

<p>We can now set our QC thresholds based on these plots. Unfortunately, there are no standard thresholds that can be applied to every dataset, so we need to look at our data and make this decision for ourselves.</p>

<p>We’ve already filtered out the cells with the lowest total counts when we created the SeuratObject, so we’ll focus on filtering the number of unique features and the proportion of mitochondrial reads. We saw on the scatter plots that the cells with high values for these two metrics also had the highest nCount_RNA values, so we’ll actually be getting rid of the cells with the highest total counts too.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>If you’ve used Scanpy or followed the <a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">Clustering 3K PBMCs with Scanpy</a> tutorial then you might be wondering why we aren’t visualising and filtering our genes as well as our cells. Seurat enables us to set the minimum number of cells that genes must be present in when we create the SeuratObject, but the pipeline doesn’t usually include any further quality checks for genes. If we did need to do further filtering, we could do this in R by calculating an appropriate QC metric and subsetting the data.</p>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>What threshold would you set for nFeature_RNA?</li>
    <li>What threshold would you set for percent.mt?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>A threshold of 2500 seems sensible for this dataset. The violin plot for nFeature_RNA shows that most of our cells should be under this threshold, so we won’t lose too much of our data. The scatter plot shows that the cells above this threshold also had unusually high values of nCount_RNA, which suggest they could include some doublets.</li>
      <li>A threshold of 5% should work for this dataset. As before, the violin plot shows that the majority of our cells are below this threshold, but in this case the cells above it had low total RNA counts, which suggests these could be damaged cells that had lost a lot of their other RNAs. Although there are few standards to guide us in single cell analysis, you will see the same 5% threshold for mitochondrial content used in many studies. It often works well, but it can filter out energetic cell types such as muscle cells, so we shouldn’t apply this threshold without considering whether it works for our dataset. Some studies don’t filter on mitochondrial reads at all!</li>
    </ol>

  </blockquote>
</blockquote>

<p>We’re setting QC thresholds based on our visual inspections of the data, but we could take different approaches too. We could calculate the interquartile range for our QC metrics and use this to set a threshold (e.g. maximum of 1.5 times the IQ range above the median). We could also try running the analysis with different thresholds to see how it affects the results - this could help us to see if we’re excluding any biological variation, for example if we’re filtering out a specific high-energy cell type with our mitochondrial threshold.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-out-low-quality-cells"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filter Out Low Quality Cells</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_create/seurat_create/5.0+galaxy1" title="Seurat Create tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Create</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Filter cells by QC metrics</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Mitochondrial Annotations</code> (output of <strong>Seurat Create</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Minimum nFeature_RNA”</em>: <code class="language-plaintext highlighter-rouge">200</code></li>
            <li><em>“Maximum nFeature_RNA”</em>: <code class="language-plaintext highlighter-rouge">2500</code></li>
            <li><em>“Maximum percent.mt”</em>: <code class="language-plaintext highlighter-rouge">5.0</code></li>
            <li><em>“Filter by a different metric”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Filtered Dataset</code></li>
  </ol>

</blockquote>

<p>If we produce the same plots again, we can see what has changed in our data.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-re-visualise-qc-metrics"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Re-Visualise QC Metrics</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Filtered Dataset</code> (output of <strong>Seurat Create</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Violin Plot with 'VlnPlot'</code>
          <ul>
            <li><em>“Features to plot”</em>: <code class="language-plaintext highlighter-rouge">nFeature_RNA,nCount_RNA,percent.mt</code></li>
            <li>In <em>“Plot Formatting Options”</em>:
              <ul>
                <li><em>“Number of columns to display”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Filtered Dataset</code> (output of <strong>Seurat Create</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Scatter Plot with 'FeatureScatter'</code>
          <ul>
            <li><em>“First feature to plot”</em>: <code class="language-plaintext highlighter-rouge">nCount_RNA</code></li>
            <li><em>“Second feature to plot”</em>: <code class="language-plaintext highlighter-rouge">percent.mt</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Filtered Dataset</code> (output of <strong>Seurat Create</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Scatter Plot with 'FeatureScatter'</code>
          <ul>
            <li><em>“First feature to plot”</em>: <code class="language-plaintext highlighter-rouge">nCount_RNA</code></li>
            <li><em>“Second feature to plot”</em>: <code class="language-plaintext highlighter-rouge">nFeature_RNA</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<figure id="figure-3" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_violin_after.png" alt="Three violin plots each showing most points grouped together and no extreme outliers. " width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_violin_after.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Violin Plots showing the unique features (nFeature_RNA), total counts (nCount_RNA) and the proportion of reads coming from mitochondial genes for all cells after filtering</figcaption></figure>

<figure id="figure-4" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_QC_scatter_after.png" alt="Scatter plot A shows clear positive correlation with the number of unique features increasing as total counts increases. No extreme outliers. Scatter plot B does not show a relationship between the proportion of mitochondrial genes and total counts. No extreme outliers." width="4408" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_QC_scatter_after.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Scatter plots showing the relationships between the total counts (nCount_RNA) and A. the number of unique features (nFeature_RNA) and B. the proportion of mitochondrial reads (percent.mt) after filtering</figcaption></figure>

<blockquote class="question">
  <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>Have we eliminated the low-quality cells from our data?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>We can see in both the violin and scatter plots that the outliers have been eliminated from our dataset - we’ve removed the cells with high mitochondrial counts and/or unusually high numbers of detected genes. We hope that this means we’ve got rid of the poor quality cells without losing any real biological variation, but we can’t really be sure. Sometimes it is necessary to come back and re-think our filters - for example if you get to the end of the analysis and realise that you’re missing a cell type that should be present in your data!</li>
    </ol>

  </blockquote>
</blockquote>

<h2 id="further-preprocessing">Further Preprocessing</h2>

<p>Now that we’re happy with the cells left in our SeuratObject, we can begin preparing the data for analysis. Seurat provides two routes for preprocessing, one with separate tools for each step and another that combines all these preprocessing steps into a single tool, <code class="language-plaintext highlighter-rouge">SCTransform</code>.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-3"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

  <p><code class="language-plaintext highlighter-rouge">SCTransform</code> makes preprocessing quicker and easier because it combines several steps from the standard Seurat pipeline into one. <code class="language-plaintext highlighter-rouge">SCTransform</code> normalises and scales the data while also selecting highly variable features. It also takes a different approach to preprocessing that can be more effective at removing technical effects from the data.</p>

  <p>Once you are familiar with the separate steps, you might find it more effective and convenient to use <code class="language-plaintext highlighter-rouge">SCTransform</code>. The main difference to be aware of when using <code class="language-plaintext highlighter-rouge">SCTransform</code> is that the results will be stored as a new assay called <code class="language-plaintext highlighter-rouge">SCT</code> in the SeuratObject. You will need to use this assay in the following analysis steps, instead of the original <code class="language-plaintext highlighter-rouge">RNA</code> assay. After running <code class="language-plaintext highlighter-rouge">SCTransform</code>, <code class="language-plaintext highlighter-rouge">SCT</code> will be set as the default assay for the SeuratObject so the tools should automatically use the correct one. The methods and default settings used in <code class="language-plaintext highlighter-rouge">SCTransform</code> also differ from the standard pipeline, so you should expect to see some differences in your results compared to the other route.</p>

</blockquote>

<p>The usual preprocessing steps for single cell data are normalisation, selection of the most variable features, and scaling the data. The same steps are performed in slightly different ways by the separate preprocessing functions and SCTransform.</p>

<ul>
  <li><strong>Normalisation</strong>
  Normalisation deals with variations in the number of reads we counted for each cell that were caused by systemic technical differences. We might have captured more RNAs or produced more copies of them during the PCR amplification step in some cells. If we don’t remove these differences then they could obscure the biological differences we’re trying to explore. We could end up with clusters of cells based on how many RNAs we captured or copied rather than on their cell type.
  The default normalisation method for the separate <code class="language-plaintext highlighter-rouge">NormalizeData</code> tool in Seurat is <code class="language-plaintext highlighter-rouge">LogNormalize</code>, which simply divides the number of counts for each gene by the total counts for the cell, multiplies this by a scale factor (10,000 is the default), and then log-transforms the results. Essentially, we’re working out how many counts we would have for each gene if we had produced 10,000 reads from each cell, instead of getting different numbers of reads from different cells. The log-transformation then increases the impact that genes that showed stronger relative differences in expression between cells will have on our analysis (e.g. a gene that is expressed at an average count of 50 in cell type A and 10 in cell type B should have a bigger impact than a gene that is expressed at an average count of 1100 in A and 1000 in B). The normalised data will be stored in the <code class="language-plaintext highlighter-rouge">data</code> layer of the SeuratObject. The unnormalised data will still be available in the <code class="language-plaintext highlighter-rouge">counts</code> layer.
  Although this form of normalisation is widely-used, it does assume that each cell originally had the same number of RNA molecules, which isn’t the case when our samples contain cells of different types and sizes. SCTransform takes a different approach to normalisation that doesn’t make the same assumption about the cells <span class="citation"><a href="#SCTransform2022">Choudhary and Satija 2022</a></span>. Instead, it creates a model of the UMI counts that enables it to regress out variation in sequencing depth (nFeature_RNA) and pools information from genes with similar abundances to adjust the variances. The corrected counts will be stored in the <code class="language-plaintext highlighter-rouge">counts</code> layer of the new SCT assay. The log1p(counts) will be stored in the <code class="language-plaintext highlighter-rouge">data</code> layer and the pearson residuals will be stored in the <code class="language-plaintext highlighter-rouge">scale.data</code> layer.</li>
  <li><strong>Selection of highly variable features</strong>
  Rather than use the entire dataset in every stage of the analysis, we can focus on the genes that provide the most information - the highly variable genes that showed the biggest differences in expression between cells. We assume that these bigger differences reflect genuine biological differences, while the smaller differences in expression seen in other genes is down to chance or technical noise. Focusing on the most variable genes can make biological differences clearer in single cell analysis.
  Feature selection can have a big impact on our analysis as we will run downstream analyses (e.g. PCA) on these genes. We will be comparing cells based on the expression of these variable genes.
  Seurat’s separate <code class="language-plaintext highlighter-rouge">FindVariableFeatures</code> tool selects the 2000 most variable genes by default, while <code class="language-plaintext highlighter-rouge">SCTransform</code> selects the top 3000 genes. We can select more variable genes when we use SCTransform because this method is better at removing technical effects from the data, so the additional variable features are more likely to represent biological variation.
We may need to change this setting for some datasets to ensure we’re selecting the most useful genes without including too many others.</li>
  <li><strong>Scaling</strong>
  Scaling is a linear transformation that we apply to prepare our data for dimensional reduction.
  The default scaling method for the separate <code class="language-plaintext highlighter-rouge">ScaleData</code> tool in Seurat shifts the expression of each gene so that the mean expression across all the cells is 0. It also scales the expression of each gene so that the variance across all cells is 1. Scaling ensures that highly expressed genes don’t dominate the analysis too much - we’re interested in differences in expression between cells, not in genes that are always highly expressed in all the cells. The results will be stored in the <code class="language-plaintext highlighter-rouge">scale.data</code> layer.
  SCTransform doesn’t scale data in the same way - although it centers the data by default, it won’t scale the data unless you select this option. Instead, SCTransform usually stores the pearson residuals in the <code class="language-plaintext highlighter-rouge">scale.data</code> layer, which don’t need to be scaled to the same variance.
  By default, both approaches only center/scale the highly variable genes that we’ll use for dimensional reduction, but it is possible to scale more genes if required. If you choose to use the separate preprocessing tool in this tutorial, then we’ll use the ScaleData tool to scale all the genes - this means that we can compare or plot the expression of any of the genes, as they will all be scaled in the same way. We won’t perform scaling with SCTransform, as we’re following the <a href="https://satijalab.org/seurat/articles/sctransform_vignette.html">original SCT tutorial</a>, but we will center all genes.</li>
  <li><strong>Regression</strong>
The Seurat pipeline can include another step during preprocessing of our single cell data. We can regress out (or remove) the impact of unwanted sources of variation. We could use this technique to remove the effects of the cell cycle or the differences associated with the proportion of mitochondrial genes. The goal is to reduce differences that are related to factors we are not interested in as this can help the differences we are interested in (like those between cell types or experimental groups) stand out more.
  It is possible to use the <code class="language-plaintext highlighter-rouge">ScaleData</code> function to regress out unwanted variation, but the creators of Seurat recommend using <code class="language-plaintext highlighter-rouge">SCTransform</code> for preprocessing if you want to do any regression. <code class="language-plaintext highlighter-rouge">SCTransform</code> automatically regresses out variation associated with sequencing depth (unique counts or nFeature_RNA) and can also regress out other variables. If you choose to use SCTransform in this tutorial, then you’ll regress out the variation associated with the proportion of mitochondrial content, just like in <a href="https://satijalab.org/seurat/articles/sctransform_vignette.html">Seurat’s original version of this tutorial</a></li>
</ul>

<blockquote class="notranslate hands_on gtn-cyoa" id="gtn-cyoa">
	<div class="box-title" aria-label="hands-on box: choose your own tutorial"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Choose Your Own Tutorial</div>
	
	<p>This is a "Choose Your Own Tutorial" section, where you can select between multiple paths. Click one of the buttons below to select how you want to follow the tutorial</p>
	
	
	<p>
		You can perform each preprocessing step separately, which might give you a better understanding of the different elements involved in preprocessing, or run them all at once using SCTransform.
	</p>
	
	
		<input id="cyoa-opt1" type="radio" name="cyoa" value="Separate-Preprocessing-Steps" onclick="cyoaChoice('Separate-Preprocessing-Steps', 'gtn-cyoa')" />
		<label for="cyoa-opt1" class="select btn btn-secondary">Separate Preprocessing Steps</label>
	
	
		<input id="cyoa-opt2" type="radio" name="cyoa" value="SCTransform" onclick="cyoaChoice('SCTransform', 'gtn-cyoa')" />
		<label for="cyoa-opt2" class="select btn btn-secondary">SCTransform</label>
	
	
	

	
	<script>
		function ready(fn) {
			if (document.readyState != 'loading'){
				fn();
			} else {
				document.addEventListener('DOMContentLoaded', fn);
			}
		}
		ready(function() {
			cyoaDefault("Separate-Preprocessing-Steps", "gtn-cyoa");
		})
	</script>
	
</blockquote>

<div class="Separate-Preprocessing-Steps">
  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-separate-preprocessing-steps"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Separate Preprocessing Steps</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_preprocessing/seurat_preprocessing/5.0+galaxy0" title="Seurat Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Filtered Dataset</code> (output of <strong>Seurat Create</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Normalize with 'NormalizeData'</code>
            <ul>
              <li><em>“Method for normalization”</em>: <code class="language-plaintext highlighter-rouge">LogNormalize</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_preprocessing/seurat_preprocessing/5.0+galaxy0" title="Seurat Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">rds_out</code> (output of previous <strong>Seurat Preprocessing</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Identify highly variable genes with 'FindVariableFeatures'</code>
            <ul>
              <li><em>“Method to select variable features”</em>: <code class="language-plaintext highlighter-rouge">vst</code></li>
              <li><em>“Output list of most variable features”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              <li><em>“Number to show”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_preprocessing/seurat_preprocessing/5.0+galaxy0" title="Seurat Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">rds_out</code> (output of previous <strong>Seurat Preprocessing</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Scale and regress with 'ScaleData'</code>
            <ul>
              <li><em>“Regress out a variable”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
              <li><em>“Features to scale”</em>: <code class="language-plaintext highlighter-rouge">All Features</code></li>
            </ul>
          </li>
        </ul>

        <p>We’ll choose to scale <code class="language-plaintext highlighter-rouge">All Features</code> here, instead of just the highly variable genes.</p>
      </li>
      <li>Rename the output as <code class="language-plaintext highlighter-rouge">Preprocessed Data</code></li>
    </ol>

  </blockquote>
</div>

<div class="SCTransform">
  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-sctransform"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: SCTransform</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_preprocessing/seurat_preprocessing/5.0+galaxy0" title="Seurat Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Filtered Dataset</code> (output of <strong>Seurat Create</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Complete all preprocessing with "SCTransform"</code>
            <ul>
              <li><em>“Genes to calculate residual features for”</em>: <code class="language-plaintext highlighter-rouge">all genes</code>
                <ul>
                  <li><em>“How to set variable features”</em>: <code class="language-plaintext highlighter-rouge">set number of variable features</code></li>
                </ul>
              </li>
              <li><em>“Output list of most variable features”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              <li><em>“Number to show”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
              <li><em>“Variable(s) to regress out”</em>: <code class="language-plaintext highlighter-rouge">percent.mt</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>
        <p>Rename the output as <code class="language-plaintext highlighter-rouge">Preprocessed Data</code></p>

        <p>If we inspect this dataset, we can see that a new assay called <code class="language-plaintext highlighter-rouge">SCT</code> has been created and is set as the default assay. Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span>- of the following output to check this.</p>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_data/seurat_data/5.0+galaxy0" title="Seurat Data Management tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Data Management</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Preprocessed Data</code> (output of <strong>Seurat Preprocessing</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Inspect Seurat Object</code>
            <ul>
              <li><em>“Display information about”</em>: <code class="language-plaintext highlighter-rouge">General</code></li>
            </ul>
          </li>
        </ul>
      </li>
    </ol>

  </blockquote>
</div>

<h3 id="visualise-highly-variable-genes">Visualise Highly Variable Genes</h3>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualise-results"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualise Results</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Preprocessed Data</code> (output of <strong>Seurat Preprocessing</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Plot Variable Genes with 'VariableFeaturePlot'</code>
          <ul>
            <li><em>“Label the top most variable features”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“Number to show”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-5" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_variable_genes.png" alt="Most genes are grouped together at the bottom of the plot with low standardized variance. The selected variable genes are highlighted in red and labelled by name. They have higher standardized variances although most are still quite close to the non variable genes. The labels read PPBP, S100A9, LYZ, IGLL5, GNLY, FTL, PF4, FTH1, GNG11, and FCER1A." width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_variable_genes.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Plot showing the standardized variances of variable and non-variable genes. The top 10 most variable genes are labelled.</figcaption></figure>

</div>

<div class="SCTransform">

  <figure id="figure-6" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_variable_genes_SCT.png" alt="Most genes are closely grouped together, right at the bottom of the plot with low standardized variance. The selected variable genes are highlighted in red and labelled by name. They are higher up on the plot, showing they have higher standardized variances although most are still quite close to the non variable genes. They are S100A9, GNLY, LYZ, S100A8, NKG7, FTL, GZMB, IGLL5, FTH1, and CCL5." width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_variable_genes_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 6</strong>:</span> Plot showing the standardized variances of variable and non-variable genes. The top 10 most variable genes are labelled.</figcaption></figure>

</div>

<blockquote class="question">
  <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>What are the top 10 most variable genes in this dataset?</li>
    <li>Why are we interested in these genes?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>We can check the list of the top variable genes in our history by clicking on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> or see them on our variable features plot.
        <div class="Separate-Preprocessing-Steps">

          <table class="matrix">
            <thead>
              <tr>
                <th> </th>
                <th> </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>1</td>
                <td>PPBP</td>
              </tr>
              <tr>
                <td>2</td>
                <td>LYZ</td>
              </tr>
              <tr>
                <td>3</td>
                <td>S100A9</td>
              </tr>
              <tr>
                <td>4</td>
                <td>IGLL5</td>
              </tr>
              <tr>
                <td>5</td>
                <td>GNLY</td>
              </tr>
              <tr>
                <td>6</td>
                <td>FTL</td>
              </tr>
              <tr>
                <td>7</td>
                <td>PF4</td>
              </tr>
              <tr>
                <td>8</td>
                <td>FTH1</td>
              </tr>
              <tr>
                <td>9</td>
                <td>GNG11</td>
              </tr>
              <tr>
                <td>10</td>
                <td>S100A8</td>
              </tr>
            </tbody>
          </table>

        </div>

        <div class="SCTransform">

          <table class="matrix">
            <thead>
              <tr>
                <th> </th>
                <th> </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>1</td>
                <td>S100A9</td>
              </tr>
              <tr>
                <td>2</td>
                <td>GNLY</td>
              </tr>
              <tr>
                <td>3</td>
                <td>LYZ</td>
              </tr>
              <tr>
                <td>4</td>
                <td>S100A8</td>
              </tr>
              <tr>
                <td>5</td>
                <td>NKG7</td>
              </tr>
              <tr>
                <td>6</td>
                <td>FTL</td>
              </tr>
              <tr>
                <td>7</td>
                <td>GZMB</td>
              </tr>
              <tr>
                <td>8</td>
                <td>IGLL5</td>
              </tr>
              <tr>
                <td>9</td>
                <td>FTH1</td>
              </tr>
              <tr>
                <td>10</td>
                <td>CCL5</td>
              </tr>
            </tbody>
          </table>

        </div>

        <blockquote class="comment">
          <div class="box-title comment-title" id="comment-4"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
          <p>The list of highly variable genes you end up with will depend on which preprocessing route you chose. The two preprocessing routes use different methods to select features, so they won’t always end up with the same genes, although there are likely to be some similarities. <code class="language-plaintext highlighter-rouge">SCTransform</code> also returns 3000 variable genes by default, rather than the 2000 selected by <code class="language-plaintext highlighter-rouge">FindVariableFeatures</code>. We can select more features with <code class="language-plaintext highlighter-rouge">SCTransform</code> because its normalisation method is better at removing technical effects from the data, so we believe that these additional genes reflect subtler biological variations rather than technical differences.</p>
        </blockquote>
      </li>
      <li>Single cell datasets contain a lot of information, including expression data for thousands of different genes. Some of these genes don’t tell us much about the data, for example they might be housekeeping genes that are expressed at similar levels in most of our cells. We want to find the genes that can tell us most about the differences between our cells, so we want to identify the genes whose expression varies most across the dataset - focusing on these highly variable features should help us to uncover the biological differences we’re looking for.</li>
    </ol>

  </blockquote>
</blockquote>

<h1 id="dimensionality-reduction">Dimensionality Reduction</h1>

<p>Dimensional reduction is a key step in single cell analysis that simplifies our big, complex datasets enough to enable us to perform and understand further analyses.</p>

<p>In single cell analysis, we’re comparing cells based on their patterns of gene expression - we’re looking for cells with similar transcriptomic profiles. Each gene represents a dimension of the data. If we only found two genes being expressed in our cells, we could represent them on a 2-dimensional plot. The two axes of the plot would represent the expression levels of the two genes. Each cell would become a point on the plot, positioned according to its expression of these genes. We might spot one group of cells with high levels of both genes and another group that expresses large amounts of one gene and not much of the other.</p>

<p>For datasets with thousands of genes, the concept is essentially the same: each cell’s expression profile defines its location in the high-dimensional expression space. Each gene is still a dimension or axis on the plot and each cell can be positioned along these axes based on their expression. It’s just much harder to imagine a plot with thousands of different axes - that’s why we need dimensional reduction!</p>

<p>We don’t have to think of the expression of each of our thousands of genes individually. The expression of some genes can be correlated if they are affected by the same biological process - cells that express a lot of one gene might also express higher levels of other genes that have similar functions. Instead of keeping the information from each of these genes as separate axes or dimensions, we can compress them into a single dimension (e.g. an ‘eigengene’). We can do the same for all of the correlated groups of genes in our data, going from thousands of individual genes to perhaps 10, 20, or 50 grouped dimensions.</p>

<p>Dimensionality reduction aims to reduce the number of separate dimensions in the data, which:</p>

<ul>
  <li>reduces the computational work in downstream analyses to only a few dimensions</li>
  <li>reduces noise by averaging across multiple genes to obtain a more precise representation of the patterns in the data</li>
  <li>enables effective plotting of the data.</li>
</ul>

<h2 id="principal-component-analysis">Principal Component Analysis</h2>

<p>Principal Component Analysis (PCA) is a dimensionality reduction technique that identifies the axes in high-dimensional space that capture the largest amount of variation. It is a simple, highly effective strategy that is widely used in data science, including for single cell omics.</p>

<p>The axes or dimensions identified by PCA will be ordered based on how much of the variation they explain. The first axis (or Principal Component, PC) is chosen so that it captures the greatest variance across cells. The next PC is the axis orthogonal to (uncorrelated with) the first that captures the greatest amount of the remaining variation across the cells. The third PC will be orthoganol to the first two and capture the greatest amount of the remaining variation, and so on. You can decide how many PCs you want to produce and then how many of these PCs you want to use in downstream analyses.</p>

<p>We assume that the top PCs are more likely to represent real biological variation in the data because this should affect multiple genes in coordinated ways, having a bigger impact on variation. Random technical or biological noise should affect each gene independently, without creating patterns of correlation or explaining much variation, so these effects should be represented in the later PCs. By using just the top PCs in our downstream analysis, we can therefore focus on the biological variation and eliminate some of the unwanted noise while also making our data easier to analyse.</p>

<h3 id="perform-the-pca">Perform the PCA</h3>

<p>We will start by calculating the top 50 PCs and then decide how many of these we want to use.</p>

<p>The standard Seurat pipeline performs the PCA on the Variable Features only, rather than the complete dataset. You can change this using the <code class="language-plaintext highlighter-rouge">features</code> parameter, but you will need to scale the chosen features using <code class="language-plaintext highlighter-rouge">ScaleData</code> before running the PCA, if you didn’t already choose this option during the preprocessing stage as we did above.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-perform-the-pca"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Perform the PCA</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_reduce_dimension/seurat_reduce_dimension/5.0+galaxy0" title="Seurat Run Dimensional Reduction tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Run Dimensional Reduction</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Preprocessed Data</code> (output of <strong>Seurat Preprocessing</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Run a PCA dimensionality reduction using 'RunPCA'</code>
          <ul>
            <li><em>“Output list of top genes”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“Number of PCs to print genes for”</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
            <li><em>“Number of top genes to print for each PC”</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">PCA Results</code></li>
  </ol>

</blockquote>

<h3 id="visualise-the-pca-results">Visualise the PCA Results</h3>

<p>We’ve now reduced our dataset to 50 dimensions or PCs that represent the expression of sets of correlated genes. Since we selected the option to output a list of the top genes, we can use the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> icon on this output in our history to see which genes were most strongly associated with the top five PCs. We can see lists of the genes that had the strongest positive and negative scores for each PC - these are the correlated sets of genes that defined the PCs.</p>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-dataset-snippet-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-dataset-snippet-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Dataset snippet</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>A single Galaxy dataset can either be “collapsed” or “expanded”.</p>   <p><strong>Collapsed dataset view</strong></p>   <p>Datasets in the panel are initially shown in a “collapsed” view:</p>   <p><img src="/training-material/shared/images/history_item_collapsed.png" alt="Collapsed view of a single Galaxy dataset" /></p>   <p>It contains the following elements:</p>   <ul>   <li><strong>Dataset number</strong>: (“1”) order of dataset in the history;</li>   <li><strong>Dataset name</strong>: (“M117-bl_1.fq.gz”) its name;</li>   <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span>: click this to view the dataset contents;</li>   <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span>: click this to edit dataset properties;</li>   <li><i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span>: click this to delete the dataset from the history (<em>don’t worry</em>, you can undo this action!).</li> </ul>   <p>Clicking on a collapsed dataset will expand it.</p>   <blockquote class="details">   <div class="box-title details-title" id="details-some-buttons-can-be-disabled-1"><button class="gtn-boxify-button details" type="button" aria-controls="details-some-buttons-can-be-disabled-1" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Some buttons can be disabled.</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Some of the buttons above may be disabled if the dataset is in a state that doesn’t allow the action. For example, the ‘edit’ button is disabled for datasets that are still queued or running</p> </blockquote>   <p><strong>Expanded dataset view</strong></p>   <p>Expanded dataset view adds a preview element and many additional controls.</p>   <p><img src="/training-material/shared/images/history_item_expanded.png" alt="Expanded view of a single Galaxy dataset" /></p>   <p>In addition to the elements described above for the collapsed dataset, its expanded view contains:</p>   <ul>   <li><strong>Add tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span>: click on this to tag this dateset;</li>   <li><strong>Dataset size</strong>: (“2 variants, 18 comments”) lists the size of the dataset. When datasets are small (like in this example) the exact size is shown. For large datasets, Galaxy gives an approximate estimate.</li>   <li><strong>format</strong>: (“VCF”) lists the datatype;</li>   <li><strong>database</strong>: (“?”) lists which genome built this dataset corresponds to. This usually lists “?” unless the genome build is set explicitly or the dataset is derived from another dataset with defined genome build information;</li>   <li><strong>info field</strong>: (“INFO [2024-03-26 12:08:53,435]…”) displays information provided by the tool that generated this dataset. This varies widely and depends on the type of job that generated this dataset.</li>   <li><i class="fas fa-save" aria-hidden="true"></i><span class="visually-hidden">dataset-save</span>: Saves dataset to disk;</li>   <li><i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">dataset-link</span>: Copies dataset link into clipboard;</li>   <li><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">dataset-info</span>: Displays additional details about the dataset in the center pane;</li>   <li><i class="fas fa-arrow-rotate-right" aria-hidden="true"></i><span class="visually-hidden">dataset-rerun</span>: Reruns job that generated this dataset. This button is unavailable for datasets uploaded into history because they were not produced by a Galaxy tool;</li>   <li><i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">dataset-visualize</span>: Displays visualization options for this dataset. The list of options is dependent on the datatype;</li>   <li><i class="fas fa-sitemap" aria-hidden="true"></i><span class="visually-hidden">dataset-related-datasets</span>: Shows datasets related to this dataset. This is useful for tracking down parental datasets - those that were used as inputs into a job that produced this particular dataset.</li> </ul> </blockquote>
<p><!--END_SNIPPET--></p>

<p>Rather than just looking at a list of genes, we can also produce plots to help us better understand how our cells and genes relate to the PCs we have just computed. Let’s start by finding out more about the genes that were most strongly associated with our top three PCs. We can do this with the <code class="language-plaintext highlighter-rouge">VizDimLoadings</code> plot, which shows the genes associated with each PC, how strongly each gene affected the PC, and whether it was positively or negatively correlated with the PC. The plots will look a bit different depending on which preprocessing approach you followed.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualise-the-pca-results-dimensional-loadings"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualise the PCA Results - Dimensional Loadings</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">PCA Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Visualize Reduction Results with 'VizDimLoadings'</code>
          <ul>
            <li><em>“Number of dimensions to display”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
            <li>In <em>“Plot Formatting Options”</em>:
              <ul>
                <li><em>“Number of columns to display”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Change size of plot”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Width of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">3000</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-7" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_vizdimloadings.png" alt="Three plots with dots for the top 30 genes for each PC. Each plot has dots near the positive and negative ends of the axis and no values close to zero. " width="3000" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_vizdimloadings.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 7</strong>:</span> Plots showing the genes with the highest positive and negative loadings for each of the first three PCs</figcaption></figure>

</div>

<div class="SCTransform">

  <figure id="figure-8" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_vizdimloadings_SCT.png" alt="Three plots with dots for the top 30 genes for each PC. Each plot has dots near the positive and negative ends of the axis and no values close to zero. " width="3000" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_vizdimloadings_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 8</strong>:</span> Plots showing the genes with the highest positive and negative loadings for each of the first three PCs</figcaption></figure>

</div>

<blockquote class="question">
  <div class="box-title question-title" id="question-6"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>What are the top positive and negative genes for the first three PCs?</li>
    <li>Are any of our top 10 highly variable genes associated with the top PCS? Does this surprise you?</li>
    <li>When we plot the cells along the PC axes (in the next step) do you expect to see differences in the expression of these genes along the associated axis?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-6"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-6" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>
        <div class="Separate-Preprocessing-Steps">
          <p>The list produced by the <code class="language-plaintext highlighter-rouge">RunPCA</code> function shows the genes that were most strongly positively and negatively assocated with each PC. The top positively associated genes were CST3 for PC1, CD79A for PC2, and HLA-DQA1 for PC3. The top negatively associated genes were MALAT1 for PC1, NKG7 for PC2, and PPBP for PC3. However, these top genes don’t define the PCs by themselves - they are part of groups of genes that showed correlated patterns of expression. The figures show us more information about the top genes. We can see the top 30 genes in each of these groups on the plots. We can also see how strongly each of these genes was associated with the PC.</p>
        </div>
        <div class="SCTransform">
          <p>The list produced by the <code class="language-plaintext highlighter-rouge">RunPCA</code> function shows the genes that were most strongly positively and negatively assocated with each PC. The top positively associated genes were MALAT1 for PC1, NKG7 for PC2, and S100A8 for PC3, while the top negatively associated genes were FTL for PC1, HLA-DRA for PC2, and CD74 for PC3. However, these top genes don’t define the PCs by themselves - they are part of groups of genes that showed correlated patterns of expression. The figures show us more information about the top genes. We can see the top 30 genes in each of these groups on the plots. We can also see how strongly each of these genes was associated with the PC.</p>
        </div>
      </li>
      <li>We can see that some of our highly variable genes are associated with the top PCs. For example, FTL is one of the top genes associated with PC1. We should expect to see some of our highly variable genes here as we used the features we selected to perform the PCA. It also makes sense that our top 10 variable genes are strongly associated with PCs, because we know these are the genes that varied most across the dataset and the PCA was looking for these strong differences in expression. However, a gene that varied a lot won’t necessarily be associated with a top PC unless its expression correlates with other variable genes - a group of correlated genes is likely to have a stronger impact than a single gene, even if that one gene varies a lot.</li>
      <li>Since these are the genes that most are strongly associated with each PC, we should expect to see strong differences in their expression from one end of the associated axis to the other. The genes positively associated with PC1 should mainly be expressed near the positive end of the PC1 axis while the negatively associated genes should mainly be expressed at the negative end. We would expect to see similar patterns for the other PCs.</li>
    </ol>
  </blockquote>
</blockquote>

<p>Next, let’s see how our cells are distributed along the top PCs. We can use <code class="language-plaintext highlighter-rouge">DimPlot</code> to create a 2-dimensional plot where each axis represents one of the PCs - being able to produce these plots is one of the benefits of dimensional reduction as it makes our data easier to interpret when we can see it on two axes rather than imagining it along thousands. If we’re plotting the top PCs then we should see cells or clumps of cells spread along these axes. If PC1 is really explaining a lot of the variation in the data then we should see that there are cells with both high and low values along this axis - if they’re all grouped together at one end of the axis then the PC wouldn’t be telling us much about the differences between cells. We’d hope to see a similar effect along the next PCs, although it won’t be as strong as for the top PC because that’s the one that explained the most variation in the dataset. We should see similar relationships no matter which preprocessing approach we used, although of course the plots will look a bit different.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualise-the-pca-results-dimplot"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualise the PCA Results - DimPlot</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">PCA Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Visualize Dimensional Reduction with 'DimPlot'</code>
          <ul>
            <li><em>“Name of reduction to use”</em>: <code class="language-plaintext highlighter-rouge">pca</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-9" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_PCA_DimPlot.png" alt="Three large groups appear in different parts of the plot with a scattering of cells in between them. " width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_PCA_DimPlot.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 9</strong>:</span> PCA plot showing the distribution of cells along the first two principal components</figcaption></figure>

  <blockquote class="comment">
    <div class="box-title comment-title" id="comment-5"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
    <p>If you’ve tried the <a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">Clustering 3K PBMCs with Scanpy</a> tutorial then you might notice that there are some differences in the relationships between the genes and PCs. Our top negative genes for PCs 2 and 3 (NKG7 and PPBP) here are among the top positively associated genes there. This is simply a mathematical quirk - in both cases, these genes are strongly associated with the top PCs, but we’re just calculating or drawing our axes in the opposite direction. You should be able to spot some similarities in the plots between these two tutorials, especially in the PC2/PC3 plot, which looks like it has been flipped upside down!</p>
  </blockquote>

</div>

<div class="SCTransform">

  <figure id="figure-10" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_PCA_DimPlot_SCT.png" alt="Three large groups appear in different parts of the plot with a scattering of cells in between them. " width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_PCA_DimPlot_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 10</strong>:</span> PCA plot showing the distribution of cells along the first two principal components</figcaption></figure>

</div>

<blockquote class="question">
  <div class="box-title question-title" id="question-7"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>Can you see any groups or clusters of cells in the the PCA plot?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-7"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-7" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>The cells seem to form three main clusters, although there are also cells spread outside and in between these groups. The cells are distributed along the full length of both axes, which makes sense as these are the PCs that explain the greatest amounts of variation in the data. If you’ve tried the other preprocessing route in this tutorial, then you might notice that while the plots look quite similar, it seems like one of them has been flipped upside down and back to front compared to the other! This explains why MALAT1 is the top positive gene for PC1 in the separate preprocessing route but the top negative gene for PC1 in the SCTransform route - this axis is showing simiar differences, but the other way around.</li>
    </ol>
  </blockquote>
</blockquote>

<p>We don’t have to use PCs 1 and 2 as the axes. We can decide which PCs we want to plot our data along. We can also use the <code class="language-plaintext highlighter-rouge">FeaturePlot</code> function to colour the plots by the expression levels of specific genes to see how this relates to the PCs, so let’s see how the top genes relate to the top three PCs.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualise-the-pca-results-feature-plots"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualise the PCA Results - Feature Plots</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">PCA Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Visualize expression with 'FeaturePlot'</code>
          <ul>
            <li><em>“Features to plot”</em>: Use the top positive and negative genes for PCs 1-3: <span class="Separate-Preprocessing-Steps"> <code class="language-plaintext highlighter-rouge">CST3,CD79A,HLA-DQA1, MALAT1,NKG7,PPBP</code></span><span class="SCTransform"><code class="language-plaintext highlighter-rouge">MALAT1,NKG7,S100A8,FTL,HLA-DRA,CD74</code></span></li>
            <li><em>“Dimension to plot on x axis”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
            <li><em>“Dimension to plot on y axis”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
            <li><em>“Name of reduction to use”</em>: <code class="language-plaintext highlighter-rouge">pca</code></li>
            <li>In <em>“Plot Formatting Options”</em>:
              <ul>
                <li><em>“Number of columns to display”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Change size of plot”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Width of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">3100</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">PCA Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Visualize expression with 'FeaturePlot'</code>
          <ul>
            <li><em>“Features to plot”</em>:  Use the top positive and negative genes for PCs 1-3: <span class="Separate-Preprocessing-Steps"> <code class="language-plaintext highlighter-rouge">CST3,CD79A,HLA-DQA1, MALAT1,NKG7,PPBP</code></span><span class="SCTransform"><code class="language-plaintext highlighter-rouge">MALAT1,NKG7,S100A8,FTL,HLA-DRA,CD74</code></span></li>
            <li><em>“Dimension to plot on x axis”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
            <li><em>“Dimension to plot on y axis”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
            <li><em>“Name of reduction to use”</em>: <code class="language-plaintext highlighter-rouge">pca</code></li>
            <li>In <em>“Plot Formatting Options”</em>:
              <ul>
                <li><em>“Number of columns to display”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Change size of plot”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Width of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">3100</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>You should now have two plots showing the expression of these genes on different axes.</p>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-11" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_PCA_12_featureplots.png" alt="Six versions of the same PCA plot showing three main groups of cells in different parts of the plot. The positively associated genes for each PC are mainly shown as expressed in cells near the positive end of that axis. The opposite is true for the negatively associated genes." width="3100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_PCA_12_featureplots.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 11</strong>:</span> PCA Plots coloured by expression of the top positive and  negative markers for PCs 1 to 3, plotted along the PC1 and PC2 axes</figcaption></figure>

  <figure id="figure-12" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_PCA_23_featureplots.png" alt="Six versions of the same PCA plot showing two main groups of cells and a smaller third group spread vertically along the middle of the plot. The positively associated genes for each PC are mainly shown as expressed in cells near the positive end of that axis. The opposite is true for the negatively associated genes." width="3100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_PCA_23_featureplots.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 12</strong>:</span> PCA Plots coloured by the expression of the top positive and negative markers for PCs 1 to 3, plotted along the PC2 and PC3 axes</figcaption></figure>

  <blockquote class="question">
    <div class="box-title question-title" id="question-8"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
    <ol>
      <li>How does the expression of the top positive and negative genes relate to PCs 1 and 2?</li>
      <li>How does the expression of the top positive and negative genes relate to PCs 2 and 3?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-8"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-8" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
      <ol>
        <li>When we look at the expression of our top genes along PCs 1 and 2 we see that, as expected, the top positively associated gene for PC1, CST3, is almost exclusively expressed in cells at the positive end of the PC1 (horizontal) axis. The pattern is less clear for the negatively associated MALAT1 as most cells are expressing it, but if we look closely we can see that the expression is higher in cells at the negative end of the PC1 axis. We can see similar patterns for the top genes associated with PC2. The top positively associated gene, CD79A is mainly expressed by the cells closest to the top of the PC2 (vertical) axis. The top negative gene NKG7 is mainly expressed by cells along the lower half of the PC2 axis.
We don’t see similar patterns for the genes associated with PC3 in this figure - although these genes are only expressed in parts of the plot, there is no clear relationship with either the PC1 or PC2 axes. This is exactly what we would expect to see - these genes are associated with PC3 so they shouldn’t show any relationship with the axes.</li>
        <li>The picture looks very different when we plot our cells along PC2 and PC3. We can see two clumps of cells close together at the top of the plot and another small group of cells spread down the middle of the plot. This is why we have to be careful when interpreting dimensional reduction plots - the data can look very different depending on which of the many axes we choose to visualise.
Now it is the top genes for PC1 that don’t show any relationship with the axes. The top genes for PC2 still show differences in expression along the PC2 axis (now the horizontal axis) - and these actually look a bit clearer than in the previous plot. We can also see how the top genes for PC3 relate to the PC3 (vertical) axis. The positively associated HLA-DQA1 is mostly expressed by cells at the top of the axis while the negatively associated PPBP is mostly expressed along the negative part of that axis.</li>
      </ol>
    </blockquote>
  </blockquote>

</div>

<div class="SCTransform">

  <figure id="figure-13" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_PCA_12_featureplots_SCT.png" alt="Six versions of the same PCA plot showing three main groups of cells in different parts of the plot. The positively associated genes for each PC are mainly shown as expressed in cells near the positive end of that axis. The opposite is true for the negatively associated genes." width="3100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_PCA_12_featureplots_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 13</strong>:</span> PCA Plots coloured by expression of the top positive and  negative markers for PCs 1 to 3, plotted along the PC1 and PC2 axes</figcaption></figure>

  <figure id="figure-14" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_PCA_23_featureplots_SCT.png" alt="Six versions of the same PCA plot showing two main groups of cells and a smaller third group spread vertically along the middle of the plot. The positively associated genes for each PC are mainly shown as expressed in cells near the positive end of that axis. The opposite is true for the negatively associated genes." width="3100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_PCA_23_featureplots_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 14</strong>:</span> PCA Plots coloured by the expression of the top positive and negative markers for PCs 1 to 3, plotted along the PC2 and PC3 axes</figcaption></figure>

  <blockquote class="question">
    <div class="box-title question-title" id="question-9"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
    <ol>
      <li>How does the expression of the top positive and negative genes relate to PCs 1 and 2?</li>
      <li>How does the expression of the top positive and negative genes relate to PCs 2 and 3?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-9"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-9" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
      <ol>
        <li>When we look at the expression of our top genes along PCs 1 and 2 we see that, as expected, the top positively associated gene for PC1, MALAT1, is expressed at higher levels in cells at the positive end of the PC1 (horizontal) axis while the top negatively associated gene, FTL, is expressed mainly at the negative end. The pattern isn’t as clear for these genes as it is for the other PCs we plotted, because both MALAT1 and FTL are expressed to some extent by most of the cells in our plots. We can see more obvious patterns for the top genes associated with PC2. The top positively associated gene, NKG7 is mainly expressed by the cells at the positive end of the PC2 (vertical) axis. The top negative gene HLA-DRA is mainly expressed by cells along the lower half of the PC2 axis.
We don’t see similar patterns for the genes associated with PC3 in this figure - although these genes are only expressed in parts of the plot, there is no clear relationship with either the PC1 or PC2 axes. This is exactly what we would expect to see - these genes are associated with PC3 so they shouldn’t show any relationship with the PC1 and PC2 axes.</li>
        <li>The picture looks very different when we plot our cells along PC2 and PC3. We can see two long lines of cells forming an upside-down ‘V’. This is why we have to be careful when interpreting dimensional reduction plots - the data can look very different depending on which of the many axes we choose to visualise.
Now it is the top genes for PC1 that don’t show any relationship with the axes. The top genes for PC2 still show differences in expression along the PC2 axis (now the horizontal axis). We can also see how the top genes for PC3 relate to the PC3 (vertical) axis. The positively associated S100A8 is mostly expressed by cells at the top of the axis while the negatively associated CD74 is mostly expressed along the negative part of that axis.</li>
      </ol>
    </blockquote>
  </blockquote>

</div>

<p>Another option for visualising our PCA results is to use <code class="language-plaintext highlighter-rouge">DimHeatmap</code> to produce a heatmap of the PCA scores for the top genes in each cell. We can see which genes scored highly in the same cells and get an idea of how the scores varied across the dataset. We can look for groups of cells that had similar scores for a particular PC, suggesting that they share similar expression profiles - these groups of cells might become our clusters. We can also compare the patterns for different PCs. We would expect to see more similarities (genes with similar scores across cells and more cells grouping together) for the top PCs since these explain more of the variation in the data.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualise-the-pca-results-heatmaps"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualise the PCA Results - Heatmaps</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">PCA Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Visualize Reduction Results with 'DimHeatmap'</code>
          <ul>
            <li><em>“Dimensions from reduction to plot”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
            <li>In <em>“Advanced Options”</em>:
              <ul>
                <li><em>“Number of top cells to plot”</em>: <code class="language-plaintext highlighter-rouge">500</code></li>
                <li><em>“Return an equal number of genes with + and - scores”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">PCA Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Visualize Reduction Results with 'DimHeatmap'</code>
          <ul>
            <li><em>“Dimensions from reduction to plot”</em>: <code class="language-plaintext highlighter-rouge">1:15</code></li>
            <li>In <em>“Advanced Options”</em>:
              <ul>
                <li><em>“Number of top cells to plot”</em>: <code class="language-plaintext highlighter-rouge">500</code></li>
                <li><em>“Return an equal number of genes with + and - scores”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Change size of plot”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Width of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">4100</code></li>
            <li><em>“Height of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">4400</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-15" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_heatmap_PC_1.png" alt="The heatmap is sharply divided into four quarters. The left side of shows PCA scores for groups of cells that have low values for the genes in the top half of the plot that were negatively associated with PC1 and high values for genes in the bottom half that were positively associated with PC1. Cells on the right side have high values in the top half of the plot and low values in the bottom half of the plot. " width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_heatmap_PC_1.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 15</strong>:</span> Heatmap of PCA scores for the top 30 genes associated with PC1</figcaption></figure>

  <figure id="figure-16" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_heatmap_PC_1_15.png" alt="The plot shows 15 heatmaps displaying PCA scores for the top cells and genes for PC1 to PC15. The first three plots are clearly divided into quarters with the pattern becoming less clear for later PCs." width="4100" height="4400" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_heatmap_PC_1_15.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 16</strong>:</span> Heatmap of PCA scores for the top 30 genes associated with PCs 1 to 15</figcaption></figure>

</div>

<div class="SCTransform">

  <figure id="figure-17" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_heatmap_PC_1_SCT.png" alt="The heatmap is sharply divided into four quarters. The left side of shows PCA scores for groups of cells that have low values for the genes in the top half of the plot that were negatively associated with PC1 and high values for genes in the bottom half that were positively associated with PC1. Cells on the right side have high values in the top half of the plot and low values in the bottom half of the plot. " width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_heatmap_PC_1_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 17</strong>:</span> Heatmap of PCA scores for the top 30 genes associated with PC1</figcaption></figure>

  <figure id="figure-18" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_heatmap_PC_1_15_SCT.png" alt="The plot shows 15 heatmaps displaying PCA scores for the top cells and genes for PC1 to PC15. The first two plots are clearly divided into quarters but the pattern becomes less clear in subsequent plots." width="4100" height="4400" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_heatmap_PC_1_15_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 18</strong>:</span> Heatmap of PCA scores for the top 30 genes associated with PCs 1 to 15</figcaption></figure>

</div>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-6"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>Note that the cells aren’t plotted in the same order in all of our heatmaps, so we can’t compare the same cell (column) across multiple plots - the first column in our heatmap doesn’t always represent the same cell. We have plotted the top 500 cells for each PC, which means there are lots of cells that aren’t included in these plots. We are probably looking at different cells and/or cell orders on each plot. The cells that showed the strongest variation along PC1 aren’t necessarily the ones that varied most along PC2, so they won’t appear in the same place on the plots.</p>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-10"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>What does the PC1 heatmap reveal about the expression of the top genes for this PC, <span class="Separate-Preprocessing-Steps">CST3 and MALAT1?</span><span class="SCTransform">MALAT1 and FTL?</span></li>
    <li>Why do the plots become messier for lower PCs?</li>
    <li>Which type of PCA visualisation was most useful for understanding the main sources of variation in the data?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-10"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-10" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>The bottom row of the heatmap shows the PCA scores for the top positive gene associated with PC1, in the top cells for this PC. We can see that most cells on the left side of the heatmap are coloured yellow in this row, indicating they scored highly for this gene. Almost all of the cells on the right side are black or pink in this bottom row, indicating lower scores for the gene. Meanwhile, we see the opposite for the top negative gene associated with PC1, which appears in the topmost row of the heatmap. Cells (columns) that scored highly for the top positive gene had low scores for the top negative gene, while those that had low scores for the top positive gene showed high scores for the top negative gene.
Clearly, there is strong variation in these genes across the cells in the plot - and that variation isn’t just in these two genes, but also in the 28 other genes in this heatmap. These are the two correlated groups of genes that define the first PC. Half the genes were more highly expressed in cells at one end of the PC while the other half were expressed more at the other end of the PC.</li>
      <li>Each of the plots for the top PCs distinguishes very clearly between two groups of cells. For each of the top PCs, one group (the left side of the plot) scored highly for most of the genes positively associated with the PC and usually had low scores for the genes negatively associated with the PC. The second group (the right side of the plot) showed the opposite, with low scores for positively associated genes and high scores for negatively associated genes.
The distinction becomes less clear in later PCs, with more cells showing different patterns to the rest and a less obvious distinction between the two sides of the plots. This makes sense, because the top PCs are the ones that explained the greatest amount of variation in the data, so they are linked to very clear patterns of expression across many cells. The later PCs didn’t explain much variation because they aren’t linked with such obvious patterns.</li>
      <li>The most useful plot might depend on your own preferences as well as the data you’re analysing, but the DimHeatmap is often best for identifying which PCs will be most interesting for downstream analysis. Both cells and genes are ordered by their PCA scores on these plots, so you can see which PCs have the strongest impact on the most numbers of cells while also identifying the sets of features associated with different PCs. You can also limit the number of cells being plotted if you want to focus on the most extreme cells for each PC.</li>
    </ol>
  </blockquote>
</blockquote>

<h3 id="determine-the-dimensionality-of-the-dataset">Determine the ‘dimensionality’ of the dataset</h3>

<p>We have reduced our dataset into 50 dimensions or PCs, but we don’t necessarily need all these PCs to understand the biological variation between our cells. We can simplify our data even more by selecting a smaller number of PCs that still provide a robust representation of the data. Just like when we chose how many features to select, we’re trying to find the number that will best represent the biology while getting rid of as much unwanted noise as we can. Including more PCs might help us capture more of the biological signal, but it will also increase the noise.</p>

<p>The DimHeatmap plots can give you an idea of which PCs are having the strongest effects on your dataset, but an Elbow Plot can provide a simpler option for selecting how many dimensions (PCs) we should use in the following analyses.</p>

<p>The Elbow Plot ranks the PCs based on the percentage of variance that each of them explains. We should see that the top PCs each explain a substantial amount of the data, but that there is a bend or elbow where the dots start lining up along the bottom axis. The PCs before the elbow explain most of the variation while those after the elbow don’t provide much additional information, so we don’t need to include them in our analysis.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-make-an-elbow-plot"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Make an Elbow Plot</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">PCA Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Determine dimensionality with 'ElbowPlot'</code>
          <ul>
            <li><em>“Number of dimensions to plot standard deviation for”</em>: <span class="Separate-Preprocessing-Steps"><code class="language-plaintext highlighter-rouge">30</code></span><span class="SCTransform"><code class="language-plaintext highlighter-rouge">50</code></span></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-19" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_elbowplot.png" alt="Dots showing the standard deviation for each PC start high for PC1 then drop substantially for each following PC. A sharp bend appears around PC10 where the drop from one PC to the next becomes very small. The dots from PC10 onwards are close to the bottom axis, with low standard deviations." width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_elbowplot.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 19</strong>:</span> Elbow Plot showing the standard deviations for the first 30 PCs</figcaption></figure>

</div>

<div class="SCTransform">

  <figure id="figure-20" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_elbowplot_SCT.png" alt="Dots showing the standard deviation for each PC start high for PC1 then drop substantially for each following PC. A sharp bend appears around PC10, with a gentler slope down towards the horizontal axis from this point. The dots from PC30 onwards are close to the bottom axis, with low standard deviations." width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_elbowplot_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 20</strong>:</span> Elbow Plot showing the standard deviations for the first 30 PCs</figcaption></figure>

</div>

<blockquote class="question">
  <div class="box-title question-title" id="question-11"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>How many PCs should we use?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-11"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-11" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <div class="Separate-Preprocessing-Steps">
      <ol>
        <li>As with many decisions in single cell analysis, there isn’t an exact method for deciding how many PCs we should use. The elbow or bend in our plot appears to be around PC9-10, so we’ll use 10 dimensions in this tutorial, but you could justifiably choose anywhere between about PC7 to PC12 on the basis of this plot. It is usually better to err on the higher side than to get rid of PCs that might be useful. Sometimes it is worth repeating the analysis with different numbers of PCs to see how it affects the results.
As always, it is also important to consider biology when making your decision. In this case, an expert might have spotted that the genes strongly associated with PCs 12 and 13 are known markers for certain rare subtypes of immune cells (e.g. MZB1 is a marker for Plasmacytoid Dendritic cells). However, these cells are so rare that we’re unlikely to find many in a dataset of this size, so these PCs might not be that useful here. In a larger dataset or one that was enriched for these cell types, we might decide to include these PCs in our analysis because of these genes. Since we only have 2700 cells, we can’t be sure that this is a true biological signal rather than just noise, so we’ll stick with the top 10 PCs on the basis of our Elbow Plot.
It’s also worth noting that we calcualted 50 PCs earlier, but only plotted 30 of them here as we wouldn’t expect to need all 50 to explain this small dataset (especially after seeing the weaker patterns in those later heatmaps) - if we didn’t see a clear bend in this plot, we could try plotting all 50 PCs instead.</li>
      </ol>
    </div>
    <div class="SCTransform">
      <ol>
        <li>The plot shows a sharp elbow or bend at around PC10, but there is a gradual slope down towards the x-axis after this rather than an immediate drop to the bottom. In order to capture some of this additional variation for our analysis, we might decide to include 30 PCs. The curve of dots is much flatter and closer to the x-axis after this point, so the additional PCs won’t explain much more of the variation.
We can actually use more PCs when we preprocess with SCTransform than if we used the separate preprocessing tools because the more effective normalisation method seems to be better at removing technical effects from the data. When we preprocess with SCTransform, we assume that higher PCs are more likely to represent subtle biological variation rather than technical effects, so we might improve our results by including more of them.</li>
      </ol>
    </div>
  </blockquote>
</blockquote>

<h1 id="cluster-the-cells">Cluster the Cells</h1>

<p>The preprocessing and dimensional reduction we’ve performed were all done in preparation for what we actually want to do to understand our data - clustering. We want to find groups of cells that are expressing similar genes. You might already have spotted some cells grouping together in the DimPlots we produced from the PCA, but we don’t want to rely on our eyes to pick out clusters. We will empirically define subpopulations of cells with similar expression profiles using a clustering algorithm.</p>

<p>Clustering summarises the data and allows us to describe a heterogeneous population of cells using a set of discrete and easily understandable cluster labels - each cell will be assigned to one specific cluster. Once we’ve explored which genes are being expressed in these clusters, we’ll then be able to treat them as proxies for biological identities such as cell types or states. We’ll be able to think of each cluster as a particular type of cell.</p>

<p>We will perform graph-based, unsupervised clustering in two steps to find out which cells are most similar to each other and how these similar cells can be grouped together:</p>

<ol>
  <li>
    <p><strong>Computation of a neighborhood graph</strong></p>

    <p>First, we will build a graph where every cell is a node that is connected to its neighbors by a line or edge. The distances between cells (the lengths of these edges) will be based on a distance metric, which is calculated by comparing the values for our top PCs between each pair of cells. The edges are weighted based on the similarity between the cells, so that more weight is given to the connections between cells that are more similar based on their expression patterns. The cells that share the closest, most heavily weighted connections in the graph are the ones with the most similar expression profiles - they are each others’ nearest neighbours. Seurat uses a K-nearest neighbor (KNN) graph, so it will draw edges (lines) between each cell and a certain number (k) of its nearest neighbours.</p>
  </li>
  <li>
    <p><strong>Clustering of the neighborhood graph</strong></p>

    <p>Second, we will partition this graph into a number of highly interconnected groups, ‘quasi-cliques’, or communities. Seurat provides several different algorithms for clustering, but they all work to find the optimal way to organise the data into groups. Clustering algorithms will look for communities of cells that are more closely connected with each other than with the cells outside their group. Each community represents a cluster that we can use for downstream interpretation.</p>
  </li>
</ol>

<h2 id="find-neighbors">Find Neighbors</h2>

<p>Seurat includes a selection of different methods for computing a neighborhood graph, but we will stick to the default settings of using a Euclidean distance metric and the <code class="language-plaintext highlighter-rouge">annoy</code> method. We will need to decide how many nearest neighbors we want to include in our KNN graph - the value we want to use for <code class="language-plaintext highlighter-rouge">k</code>. The default for Seurat is to look for the 20 nearest neighbors for each cell.</p>

<p>The best approach to building the neighborhood graph and the optimal value for <code class="language-plaintext highlighter-rouge">k</code> will depend on the data we are analysing - sometimes we’ll need to try a few options to find the right fit. We’ll use Seurat’s default value of 20 for now - we’ll know how well it worked when we take a look at our clusters later.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-find-neighbors-by-computing-a-neighborhood-graph"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Find Neighbors by Computing a Neighborhood Graph</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_clustering/seurat_clustering/5.0+galaxy0" title="Seurat Find Clusters tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Find Clusters</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">PCA Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Compute nearest neighbors with 'FindNeighbors'</code>
          <ul>
            <li><em>“Number of dimensions from reduction to use as input”</em>: <span class="Separate-Preprocessing-Steps"><code class="language-plaintext highlighter-rouge">10</code></span><span class="SCTransform"><code class="language-plaintext highlighter-rouge">30</code></span></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="find-clusters">Find Clusters</h2>

<p>Seurat also offers a variety of different clustering algorithms, including SLM, Leiden and Louvain. We will follow the default Seurat pipeline and use the Louvain algorithm.</p>

<p>We need to define a value for the resolution parameter, which determines the size of the clusters the algorithm will find. We’ll set the resolution at <code class="language-plaintext highlighter-rouge">0.5</code> for now, or <code class="language-plaintext highlighter-rouge">0.8</code> if we used SCTransform for preprocessing, but this is one of those parameters that we’ll often want to try a few different values for when working with single cell data. We’ll set a higher resolution on the SCTransform route because we expect it to capture more of the biological variation, so we might be able to find more clusters.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-7"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>Resolution is one of the key parameters you might need to change when performing clustering. It sets the ‘granularity’ of the clustering - you can choose a lower value to arrange your data into bigger clusters or use a higher resolution if you’re looking for lots of little clusters. The best resolution can depend on how varied your cells are - do you think there are only a few different cell types or should your sample contain lots of different populations of cells? The resolution will usually need to be higher for larger datasets too. In most cases, a resolution of between 0.4 and 1.2 will work well - we’re using a fairly low value here as we have a smaller dataset.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-find-clusters-using-the-neighborhood-graph"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Find Clusters using the Neighborhood Graph</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_clustering/seurat_clustering/5.0+galaxy0" title="Seurat Find Clusters tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Find Clusters</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">rds_out</code> (output of <strong>Seurat Find Clusters</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Identify cell clusters with 'FindClusters'</code>
          <ul>
            <li><em>“Resolution”</em>: <span class="Separate-Preprocessing-Steps"><code class="language-plaintext highlighter-rouge">0.5</code></span><span class="SCTransform"><code class="language-plaintext highlighter-rouge">0.8</code></span></li>
            <li><em>“Algorithm for modularity optimization”</em>: <code class="language-plaintext highlighter-rouge">1. Original Louvain</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="run-non-linear-dimensional-reduction-umaptsne">Run non-linear dimensional reduction (UMAP/tSNE)</h1>

<p>We have already performed dimensional reduction using PCA to simplify our data and make it easier to analyse, but PCA plots aren’t always the best way to display our data. We will often want a plot that separates out our clusters more clearly. It can make our clusters easier to interpret and enable us to use plots to show differences in characteristics such as the expression levels of specific genes between clusters. However, we need to be cautious about over-interpreting these plots - they give us an overview, but we can only look at two dimensions at a time, so we aren’t seeing all of the relationships between clusters. The cells can also end up being plotted on top of each other, so some of them can be hidden from view. We shouldn’t try to draw conclusions just by looking at the plots, but rather use them to make it easier to understand and share the results from tests such as differential expression analysis.</p>

<p>Two options are available for non-linear dimensional reduction with Seurat on Galaxy: UMAP and tSNE. Both approaches try to learn the underlying structure of the dataset so that the cells that are most similar to each other can be placed closest to each other in the two-dimensional plots. Since we’ve already identifed clusters of cells with similar expression profiles, we should expect to see cells from the same cluster grouped together in the plots. However, these dimensional reduction techniques have their limitations as there is no way to fully represent our complex single cell data in just two dimensions. Both UMAP and tSNE prioritise preserving local distances in the data to make sure that cells that are very similar will be located close to one another on the plot, but in order to do this they may not be able to show the more global relationships as accurately.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-8"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>If you have followed the <a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">Clustering 3K PBMCs with Scanpy</a> tutorial, you might notice that this step was performed between construction of the neighborhood graph and finding clusters. In the Seurat pipeline, it is usually performed after clustering, but it shouldn’t make any difference to your plots. You should end up with the same result if you run these steps the other way around, except that you won’t be able to show your clusters on the plot if you haven’t found them yet.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-perform-dimensional-reduction-with-umap"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Perform Dimensional Reduction with UMAP</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_reduce_dimension/seurat_reduce_dimension/5.0+galaxy0" title="Seurat Run Dimensional Reduction tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Run Dimensional Reduction</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">rds_out</code> (output of <strong>Seurat Find Clusters</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Run a UMAP dimensional reduction using 'RunUMAP'</code>
          <ul>
            <li><em>“Nem of reduction to use”</em>: <code class="language-plaintext highlighter-rouge">pca</code></li>
            <li><em>“UMAP implementation to run”</em>: <code class="language-plaintext highlighter-rouge">uwot</code></li>
            <li><em>“Run UMAP on dimensions, features, graph or KNN output”</em>: <code class="language-plaintext highlighter-rouge">dims</code>
              <ul>
                <li><em>“Number of dimensions from reduction to use as input”</em>: <span class="Separate-Preprocessing-Steps"><code class="language-plaintext highlighter-rouge">10</code></span><span class="SCTransform"><code class="language-plaintext highlighter-rouge">30</code></span></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">UMAP Results</code></li>
  </ol>

</blockquote>

<h2 id="visualise-umap">Visualise UMAP</h2>

<p>Now we can visualise the UMAP, just as we did with the PCA. We can also colour in the UMAP plot based on the expression of genes we’re interested in - we’ll start by seeing where the cells expressing the top genes associated with PCs 1-3 have ended up in the UMAP plot.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualise-the-umap"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualise the UMAP</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">UMAP Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Visualize Dimensional Reduction with 'DimPlot'</code>
          <ul>
            <li><em>“Name of reduction to use”</em>: <code class="language-plaintext highlighter-rouge">umap</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">UMAP Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Visualize expression with 'FeaturePlot'</code>
          <ul>
            <li><em>“Features to plot”</em>: Use the top positive and negative genes for PCs 1-3: <span class="Separate-Preprocessing-Steps"><code class="language-plaintext highlighter-rouge">CST3,CD79A,HLA-DQA1, MALAT1,NKG7,PPBP</code></span><span class="SCTransform"><code class="language-plaintext highlighter-rouge">MALAT1,NKG7,S100A8,FTL,HLA-DRA,CD74</code></span></li>
            <li><em>“Name of reduction to use”</em>: <code class="language-plaintext highlighter-rouge">umap</code></li>
            <li>In <em>“Plot Formatting Options”</em>:
              <ul>
                <li><em>“Number of columns to display”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Change size of plot”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Width of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">3100</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-21" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_UMAP_DimPlot.png" alt="Plot showing three big groups of cells coloured by cluster, from 0 to 8. The smallest of these three main groups only contains cells coloured as cluster 3. The other two groups are made up of cells from different clusters." width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_UMAP_DimPlot.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 21</strong>:</span> UMAP coloured by cluster</figcaption></figure>

  <figure id="figure-22" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_UMAP_DimPlot_top_genes.png" alt="Six versions of the same UMAP plot, each coloured by the expression of a different gene. MALAT1 shows higher expression in the clusters where CST3 expression is low. CD79A expression is highest in cluster 3 while NKG7 expression is highest in clusters 4 and 6. HLA-DQA1 expression is mainly in clusters 3 and 7 while PPBP expression is mostly in the tiny cluster 8." width="3100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_UMAP_DimPlot_top_genes.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 22</strong>:</span> UMAP plots coloured by expression of the top genes associated with PCs 1-3</figcaption></figure>

</div>

<div class="SCTransform">

  <figure id="figure-23" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_UMAP_DimPlot_SCT.png" alt="Plot showing three big groups of cells coloured by cluster, from 0 to 11. The smallest of these three main groups only contains cells coloured as cluster 3. The other two groups are made up of cells from different clusters." width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_UMAP_DimPlot_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 23</strong>:</span> UMAP coloured by cluster</figcaption></figure>

  <figure id="figure-24" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_UMAP_DimPlot_top_genes_SCT.png" alt="Six versions of the same UMAP plot, each coloured by the expression of a different gene. MALAT1 shows higher expression in the clusters where FTL expression is low. NKG7 expression is highest in clusters 4, 5 and 7 while HLA-DRA expression is highest in clusters 1, 3, 6 and  9. s100A8 expression is mainly in cluster 1 while CD74 expression is highest in cluster 3." width="3100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_UMAP_DimPlot_top_genes_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 24</strong>:</span> UMAP plots coloured by expression of the top genes associated with PCs 1-3</figcaption></figure>

</div>

<blockquote class="question">
  <div class="box-title question-title" id="question-12"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>How many clusters have we identified? How do they relate to the shapes you can see on the plot?</li>
    <li>Are there any relationships between our clusters and expression of the top genes associated with the first three PCs?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-12"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-12" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>We have three main groups of cells in our plot and they are more clearly separated here than the cells in our PCA plot. Since the cells have been coloured by the cluster they were assigned to by the <code class="language-plaintext highlighter-rouge">FindClusters</code> algorithm, we can see that we have identified <span class="Separate-Preprocessing-Steps">9</span><span class="SCTransform">12</span> clusters in our data. The first cluster is numbered as zero. If you have tried both preprocessing routes in this tutorial, you will see that we identified more clusters when using SCTransform (12 of them) than with the separate preprocessing tools (only nine!). It seems that preprocessing with SCTransform enables us to capture more of the biological variation.
Only one of the three main clumps of cells is made up of a single cluster (cluster 3). The other two big groups have been split up into different clusters by the algorithm, suggesting that there are some differences within them - this is one of the reasons we don’t just want to rely on our eyes when identifying clusters. The UMAP can give us a quick overview of the data and our clusters, but we must be careful about overinterpreting it. We can’t see everything about the data just be looking!</li>
      <li>Although this plot looks somewhat different from our PCA plot, we can still see patterns in the expression of the top genes that were associated with our PCs. The top positive and negative genes associated with PCs 1, 2, and 3 are expressed in different parts of the UMAP plot.  This makes sense, because we used the PCA to make the UMAP, so the cells that were at different ends of the PC axes should be in different parts of the UMAP plot too. We don’t see the same relationships between the genes and the axes, because we’re now looking at a UMAP plot not a PCA plot - the UMAP axes are not the same as the PCA axes!
As well as the broader patterns across the plot, we can also see some relationships between the top genes and specific clusters. <span class="Separate-Preprocessing-Steps">The top negative gene for PC2, NKG7, is expressed at the highest level in cluster 6 and at a slightly lower level in the adjacent cluster 4.</span>The top negative gene for PC1, FTL, is expressed in both the neighbouring clusters 1 and 6, but the top positive gene for PC3, S100A8, is only expressed in cluster 1.<span class="SCTransform"></span> This must be one of the genes that caused the clustering algorithm to assign these cells into separate clusters even though they are part of the same larger group on the plot. Again, this makes sense because we used the PCA reduction to create the neighborhood graph that we then used to identify our clusters.</li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="other-ways-to-visualise-clusters">Other ways to visualise clusters</h2>

<p>UMAP plots aren’t the only way to see what is going on with the clusters we’ve just identified - and they aren’t always the best choice when we’re interested in which cells express specific genes. For example, we can create violin plots to show the expression of our top six genes from the PCA across the clusters we’ve just identified.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualise-expression-across-clusters-with-a-violin-plot"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualise Expression across Clusters with a Violin Plot</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">UMAP Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Violin Plot with 'VlnPlot'</code>
          <ul>
            <li><em>“Features to plot”</em>:  Use the top positive and negative genes for PCs 1-3: <span class="Separate-Preprocessing-Steps"><code class="language-plaintext highlighter-rouge">CST3,CD79A,HLA-DQA1, MALAT1,NKG7,PPBP</code></span><span class="SCTransform"><code class="language-plaintext highlighter-rouge">MALAT1,NKG7,S100A8,FTL,HLA-DRA,CD74</code></span></li>
            <li>In <em>“Plot Formatting Options”</em>:
              <ul>
                <li><em>“Number of columns to display”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Change size of plot”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Width of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">3100</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-25" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_topgenes_violin.png" alt="Six violin plots. CST3 is expressed in clusters 1, 5 and 7 and in some cells in cluster 8. MALAT1 is expressed across all clusters except 8. CD79A is expressed in cluster 3. NKG7 is expressed in clusters 4 and 6. HLA-DQA1 is expressed in clusters 3 and 7 with one cell in cluster 1 expressing very high levels. PPBP is expressed in cluster 8." width="3100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_topgenes_violin.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 25</strong>:</span> Violin plots showing expression of the top genes associated with PCs 1-3 in each cluster</figcaption></figure>

</div>

<div class="SCTransform">

  <figure id="figure-26" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_topgenes_violin_SCT.png" alt="Six violin plots. MALAT1 is expressed in most clusters, but there is less expression in clusters 1, 6, 9, and 11, which are the clusters with the highest expression of FTL. NKG7 is expressed in clusters 4, 5, and 7. HLA-DRA is expressed by clusters 1, 3, 6, and 9. S100A8 is expressed at the highest levels in cluster 1 but also appears in clusters 6 and 9. CD74 is most highly expressed in clusters 3 and 9 but is also present in other clusters." width="3100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_topgenes_violin_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 26</strong>:</span> Violin plots showing expression of the top genes associated with PCs 1-3 in each cluster</figcaption></figure>

</div>

<blockquote class="question">
  <div class="box-title question-title" id="question-13"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>Are the top genes for PCs 1-3 expressed in the clusters you expected?</li>
    <li>Do you find it easier to understand gene expression with the violin plots or the UMAPs we coloured by expression in the last step?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-13"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-13" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>We can see the same expression patterns in the violin plots as we did in the UMAP feature plots. <span class="Separate-Preprocessing-Steps">For example, MALAT1 is expressed across most of the clusters while PPBP is only expressed in cluster 8.</span><span class="SCTransform">*For example, S100A8 expression is highest in cluster 1.</span>
We can also see that there is some expression of the top genes outside the clusters they are most highly expressed in, although usually at low levels in a small number of cells. <span class="Separate-Preprocessing-Steps">One exception we can see on this plot is a cell in cluster 1 that expressed very high levels of HLA-DQA1 - perhaps this cell was assigned to the wrong cluster or maybe there was some contamination or error, but it could also just be an unusual cell that really did have higher levels of this gene!</span><span class="SCTransform"></span></li>
      <li>The UMAP plots can give us a good overview of the data that you may find easier to picture and remember, but it can be hard to tell where one cluster ends and another begins. The violin plots can be clearer as each cluster is plotted separately and we can see the whole range of expression - <span class="Separate-Preprocessing-Steps">we would have a hard time picking out that outlier in cluster 1 with high levels of HLA-DQA1 from the UMAP plot.</span><span class="SCTranform">we can spot outliers more easily on this plot.</span></li>
    </ol>
  </blockquote>
</blockquote>

<p>Looking at the expression of these genes can tell us something about our clusters, but the top genes from the PCA aren’t necessarily the ones that will tell us what our clusters represent. Our next step is to find these marker genes.</p>

<h1 id="find-marker-genes">Find Marker Genes</h1>

<p>We have now identified and plotted our clusters, groups of cells that share similar expression profiles, but we don’t yet know which genes are part of these cluster-specific profiles - although looking at the top genes associated with PCs 1-3 has given us some hints. We need to identify the genes that drive separation between clusters. These marker genes can then be used to assign biological sense (e.g. cell type) to each cluster, based on where we know these genes are usually expressed. We can also use them to identify subtler differences between clusters, such as changes in activation or differentiation state, based on the behaviour of genes in the affected pathways.</p>

<p>Marker genes are usually detected by their differential expression (DE) between clusters. We’re looking for the genes that were expressed much more in one cluster than in the other(s). Seurat provides a number of statistical tests for quantifying these differences, with the Wilcoxon rank sum test as the default.</p>

<p>Seurat also provides different tools for finding markers using these tests so that we can ask various questions about how genes are differentially expressed between clusters or experimental groups. We can use <code class="language-plaintext highlighter-rouge">FindAllMarkers</code> to identify the markers of each cluster compared to all the other clusters or we can use <code class="language-plaintext highlighter-rouge">FindMarkers</code> to look for differences between specific clusters or groups. Let’s try out some of these options on our dataset.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-9"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>When we find markers using the Seurat tools on Galaxy, we will get two outputs: a CSV file and an RDS file. Both files contain the same content, a table of marker genes and the relevant statistics from the test, but in different formats. We’ll be able to read the CSV table, while Seurat tools can interact with the RDS file. Seurat doesn’t save the outputs from DE tests into the original SeuratObject by default. This means that we won’t be able to use the RDS outputs from finding markers if we want to perform further analysis of our dataset - we’ll need to use the output from the previous step instead as that is where all our expression data, metadata, reductions and neighborhood graphs are. However, we can use the RDS markers file for plotting and investigating our marker genes.</p>
</blockquote>

<h2 id="find-positive-markers-for-every-cluster-compared-to-the-rest">Find Positive Markers for Every Cluster Compared to the Rest</h2>

<p>Usually, the first question we’ll want to ask is which genes define each of our clusters as this can help us to understand why we’ve ended up with these groups. If we’re trying to learn more about a particular group of cells (e.g. a specific cell type or condition) then this can tell us which cluster or clusters these cells are in.</p>

<p>The <code class="language-plaintext highlighter-rouge">FindAllMarkers</code> option is the simplest way to do this as it will run the DE test for each cluster simultaneously. The results will tell us which genes set each cluster apart from all the rest of the clusters. We might find markers that are known to be expressed in specific cell types, which will enable us to start annotating our clusters.</p>

<p>We can look for both positive and negative markers, or limit the results to just the positive markers for the clusters. We’ll start by looking for the positive markers of all clusters.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-use-findallmarkers-to-compare-all-clusters"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Use `FindAllMarkers` to Compare All Clusters</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_clustering/seurat_clustering/5.0+galaxy0" title="Seurat Find Clusters tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Find Clusters</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">UMAP Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Identify marker genes with 'FindAllMarkers'</code>
          <ul>
            <li><em>“Select test to run”</em>: <code class="language-plaintext highlighter-rouge">wilcox</code></li>
            <li><em>“Limit output to top N markers per cluster”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li>In <em>“Advanced Options”</em>:
              <ul>
                <li><em>“Only return positive markers”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The output is a list of positive markers for each cluster, genes that were significantly more likely to be expressed by cells in that cluster than in the rest of the dataset.</p>

<p>Take a look at the second output, the CSV file. You can click on the dataset to see a quick peek of it in the history panel or use the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> icon to see the full table in the main panel.</p>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-dataset-snippet-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-dataset-snippet-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Dataset snippet</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>A single Galaxy dataset can either be “collapsed” or “expanded”.</p>   <p><strong>Collapsed dataset view</strong></p>   <p>Datasets in the panel are initially shown in a “collapsed” view:</p>   <p><img src="/training-material/shared/images/history_item_collapsed.png" alt="Collapsed view of a single Galaxy dataset" /></p>   <p>It contains the following elements:</p>   <ul>   <li><strong>Dataset number</strong>: (“1”) order of dataset in the history;</li>   <li><strong>Dataset name</strong>: (“M117-bl_1.fq.gz”) its name;</li>   <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span>: click this to view the dataset contents;</li>   <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span>: click this to edit dataset properties;</li>   <li><i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span>: click this to delete the dataset from the history (<em>don’t worry</em>, you can undo this action!).</li> </ul>   <p>Clicking on a collapsed dataset will expand it.</p>   <blockquote class="details">   <div class="box-title details-title" id="details-some-buttons-can-be-disabled-2"><button class="gtn-boxify-button details" type="button" aria-controls="details-some-buttons-can-be-disabled-2" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Some buttons can be disabled.</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Some of the buttons above may be disabled if the dataset is in a state that doesn’t allow the action. For example, the ‘edit’ button is disabled for datasets that are still queued or running</p> </blockquote>   <p><strong>Expanded dataset view</strong></p>   <p>Expanded dataset view adds a preview element and many additional controls.</p>   <p><img src="/training-material/shared/images/history_item_expanded.png" alt="Expanded view of a single Galaxy dataset" /></p>   <p>In addition to the elements described above for the collapsed dataset, its expanded view contains:</p>   <ul>   <li><strong>Add tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span>: click on this to tag this dateset;</li>   <li><strong>Dataset size</strong>: (“2 variants, 18 comments”) lists the size of the dataset. When datasets are small (like in this example) the exact size is shown. For large datasets, Galaxy gives an approximate estimate.</li>   <li><strong>format</strong>: (“VCF”) lists the datatype;</li>   <li><strong>database</strong>: (“?”) lists which genome built this dataset corresponds to. This usually lists “?” unless the genome build is set explicitly or the dataset is derived from another dataset with defined genome build information;</li>   <li><strong>info field</strong>: (“INFO [2024-03-26 12:08:53,435]…”) displays information provided by the tool that generated this dataset. This varies widely and depends on the type of job that generated this dataset.</li>   <li><i class="fas fa-save" aria-hidden="true"></i><span class="visually-hidden">dataset-save</span>: Saves dataset to disk;</li>   <li><i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">dataset-link</span>: Copies dataset link into clipboard;</li>   <li><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">dataset-info</span>: Displays additional details about the dataset in the center pane;</li>   <li><i class="fas fa-arrow-rotate-right" aria-hidden="true"></i><span class="visually-hidden">dataset-rerun</span>: Reruns job that generated this dataset. This button is unavailable for datasets uploaded into history because they were not produced by a Galaxy tool;</li>   <li><i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">dataset-visualize</span>: Displays visualization options for this dataset. The list of options is dependent on the datatype;</li>   <li><i class="fas fa-sitemap" aria-hidden="true"></i><span class="visually-hidden">dataset-related-datasets</span>: Shows datasets related to this dataset. This is useful for tracking down parental datasets - those that were used as inputs into a job that produced this particular dataset.</li> </ul> </blockquote>
<p><!--END_SNIPPET--></p>

<p>The first column is the name of the marker gene. The following columns tell us the:</p>

<ul>
  <li><strong>p_val</strong> - unadjusted p-value</li>
  <li><strong>avg_log2FC</strong> - log fold-change of average expression between the two groups, which is positive if expression was higher in the first group and negative if lower (in this case, the values are all positive as we only looked for positive markers)</li>
  <li><strong>pct.1</strong> - percentage of cells in the cluster that express this gene</li>
  <li><strong>pct.2</strong> - percentage of cells in the rest of the dataset that express this gene</li>
  <li><strong>p_val_adj</strong> - adjusted p-value (Bonferroni correction using all features in dataset)</li>
  <li><strong>cluster</strong> - which cluster the gene is a marker for</li>
  <li>(the first column with the gene names is also repeated at the end!)</li>
</ul>

<p>Although there is a lot of information here, all we need to know for now is that the markers listed for each cluster are the genes that were expressed more by these cells than any of the other clusters. We can search online for these genes to get an idea of what types of cells are in our clusters. We can always search within these results or filter them to include the ones we’re interested in - for example, we could look at the markers for cluster 2.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-14"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>Are the top genes associated with PCs 1-3 in our list of markers? Which clusters are they markers for?</li>
    <li>Do these results match your expectations?</li>
    <li>What were the top 5 markers for cluster 2?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-14"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-14" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>If we search the markers table for our top genes (you can use Ctrl+F to do this but it may take time for the full dataset to load when you view it), we can see that CST3 is a positive marker for clusters 1, 5, and 7 while MALAT1 is a positive marker for clusters 0, 4 and 6. CD79A was a marker for cluster 3 while NKG7 was a marker for clusters 4 and 6. HLA-DQA1 was a marker for clusters 3 and 7 while PPBP was a marker for cluster 8. So, these top genes are differentially expressed by some of our clusters.</li>
      <li>The results make sense, as we would expect the top positive and negative genes for each PC to be expressed in different clusters. The results also match up fairly well with what we saw on the UMAP and violin plots - although we might have thought that MALAT1 could be a marker for clusters 2 and 3 too as it seems to be highly expressed by them. The apparent difference in expression we saw in the plot wasn’t strong enough to show up in this statistical test.
However, the top genes associated with our PCs aren’t necessarily the most significant markers for our clusters (they can appear quite far down the lists) and they are often markers for more than one cluster. Again, this makes sense, because the PCA was looking for the bigger patterns across the whole dataset, while now we’re looking for differences between smaller groups of cells - the clusters.</li>
      <li>We could scroll down through the markers table to find the results for cluster 2, but it can be easier to filter the table instead. The <span class="tool" data-tool="Filter1" title="Filter tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter</strong></span> can be used to filter the Markers List from <code class="language-plaintext highlighter-rouge">FindAllMarkers</code>. We can set the condition to filter on as <code class="language-plaintext highlighter-rouge">c7==2</code> since column 7 contains the cluster numbers - this will filter out only the rows that have <code class="language-plaintext highlighter-rouge">2</code> in this column.
Make sure to enter <code class="language-plaintext highlighter-rouge">1</code> in the <code class="language-plaintext highlighter-rouge">Number of header lines to skip</code> field as we don’t want to cut off the header row because it doesn’t have the right value in column 7!
If you then click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> for the new output in your history, you should see from the <code class="language-plaintext highlighter-rouge">cluster</code> column that we only have markers for cluster 2. The top five markers for this cluster were:</li>
    </ol>

    <div class="Separate-Preprocessing-Steps">

      <blockquote class="matrix">
        <table>
          <thead>
            <tr>
              <th> </th>
              <th> </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>1</td>
              <td>IL32</td>
            </tr>
            <tr>
              <td>2</td>
              <td>LTB</td>
            </tr>
            <tr>
              <td>3</td>
              <td>CD3D</td>
            </tr>
            <tr>
              <td>4</td>
              <td>IL7R</td>
            </tr>
            <tr>
              <td>5</td>
              <td>LDHB</td>
            </tr>
          </tbody>
        </table>
      </blockquote>

    </div>

    <div class="SCTransform">

      <blockquote class="matrix">
        <table>
          <thead>
            <tr>
              <th> </th>
              <th> </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>1</td>
              <td>RPS27</td>
            </tr>
            <tr>
              <td>2</td>
              <td>RPL32</td>
            </tr>
            <tr>
              <td>3</td>
              <td>RPS6</td>
            </tr>
            <tr>
              <td>4</td>
              <td>RPS12</td>
            </tr>
            <tr>
              <td>5</td>
              <td>RPS14</td>
            </tr>
          </tbody>
        </table>
      </blockquote>

    </div>
  </blockquote>
</blockquote>

<h2 id="find-markers-of-cluster-2">Find Markers of Cluster 2</h2>

<p>If we are interested in a specific cluster, we might want to find out which genes set this cluster apart from all the rest. We can look for genes that are more likely to be expressed in this cluster (positive markers) and ones that are less likely to be expressed (negative markers). We might do this if we’re particularly interested in one cell type or group.</p>

<p>Let’s try finding the positive and negative markers of cluster 2.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-use-findmarkers-on-a-single-cluster"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Use `FindMarkers` on a Single Cluster</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_clustering/seurat_clustering/5.0+galaxy0" title="Seurat Find Clusters tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Find Clusters</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">UMAP Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Identify marker genes for specific groups with 'FindMarkers'</code>
          <ul>
            <li><em>“Compare markers between clusters of cells”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
              <ul>
                <li><em>“Identity class to define markers for”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The results from this test will look a little different than when we were comparing markers for all the clusters.</p>

<p>The first column is still the name of the marker gene. The following columns tell us the:</p>

<ul>
  <li><strong>p_val</strong> - unadjusted p-value</li>
  <li><strong>avg_log2FC</strong> - log fold-change of average expression between the two groups, which is positive if expression was higher in the first group and negative if lower</li>
  <li><strong>pct.1</strong> - percentage of cells in the first group that express this gene (in this case, cluster 2)</li>
  <li><strong>pct.2</strong> - percentage of cells in the second group that express this gene (in this case, all other clusters)</li>
  <li><strong>p_val_adj</strong> - adjusted p-value (Bonferroni correction using all features in dataset)</li>
  <li>(again, the gene names are repeated at the end)</li>
</ul>

<p>We don’t have a cluster column this time as we were only testing one group against another (in this case, cluster 2 against all the rest of the clusters).</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-15"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>What are the top five markers of cluster 2 and are they positive or negative markers?</li>
    <li>Are these the same as the top five markers for cluster 2 when we ran <code class="language-plaintext highlighter-rouge">FindAllMarkers</code>?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-15"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-15" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>When we look at the marker table, we can see that the first five genes listed as markers of cluster 2 are:
        <div class="Separate-Preprocessing-Steps">

          <table class="matrix">
            <thead>
              <tr>
                <th> </th>
                <th> </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>1</td>
                <td>IL32</td>
              </tr>
              <tr>
                <td>2</td>
                <td>LTB</td>
              </tr>
              <tr>
                <td>3</td>
                <td>CD3D</td>
              </tr>
              <tr>
                <td>4</td>
                <td>IL7R</td>
              </tr>
              <tr>
                <td>5</td>
                <td>LDHB</td>
              </tr>
            </tbody>
          </table>

        </div>

        <div class="SCTransform">

          <table class="matrix">
            <thead>
              <tr>
                <th> </th>
                <th> </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>1</td>
                <td>RPS27</td>
              </tr>
              <tr>
                <td>2</td>
                <td>RPL32</td>
              </tr>
              <tr>
                <td>3</td>
                <td>S100A4</td>
              </tr>
              <tr>
                <td>4</td>
                <td>RPS6</td>
              </tr>
              <tr>
                <td>5</td>
                <td>RPS12</td>
              </tr>
            </tbody>
          </table>

        </div>

        <p>We can look at the third column, <code class="language-plaintext highlighter-rouge">avg_log2FC</code> to see if these are positive or negative markers. <span class="Separate-Preprocessing-Steps">Although we didn’t limit this test to positive markers, we can see that the avg_log2FC for the five top markers is positive, which means these are all positive markers for cluster 2. Expression of these genes was higher in cluster 2 than in the rest of the dataset.</span><span class="SCTransform">We can see that the <code class="language-plaintext highlighter-rouge">avg_log2fc</code> value for S100A4 is negative, which means this was a negative marker for cluster 2 - it was less likely to be expressed by these cells than in the rest of the dataset. The rest of the top five markers were positive markers.</span></p>
      </li>
      <li>
        <div class="Separate-Preprocessing-Steps">
          <p>If we go back to the filtered list of cluster 2 markers, we will see the same top five markers for this cluster. Since we used <code class="language-plaintext highlighter-rouge">FindMarkers</code> to test cluster 2 against all the rest of the data, we actually performed the same test that <code class="language-plaintext highlighter-rouge">FindAllMarkers</code> does for each cluster in turn. The only difference is that we previously limited <code class="language-plaintext highlighter-rouge">FindAllMarkers</code> to positive markers only. We don’t see a difference in the top five markers as these all happened to be positive markers for cluster 2, but if we keep looking down the marker tables we’ll start to see differences as the negative markers we just found for cluster 2 using <code class="language-plaintext highlighter-rouge">FindMarkers</code> won’t appear in our <code class="language-plaintext highlighter-rouge">FindAllMarkers</code> table. If we hadn’t limited that test to positive markers, then we wouldn’t see any differences.</p>
        </div>

        <div class="SCTransform">
          <p>If we go back to the filtered list of cluster 2 markers, we will see most of the same genes in the top five. The only exception is that one negative marker, S100A4, which didn’t appear in the <code class="language-plaintext highlighter-rouge">FindAllMarkers</code> results because we limited those to positive markers. We should see the same positive markers in both tables because when we used <code class="language-plaintext highlighter-rouge">FindMarkers</code> to test cluster 2 against all the rest of the data, we actually performed the same test that <code class="language-plaintext highlighter-rouge">FindAllMarkers</code> does for each cluster in turn. If we hadn’t limited those results to positive markers, our marker genes would be identical.</p>
        </div>
      </li>
    </ol>

  </blockquote>
</blockquote>

<h2 id="find-markers-distinguishing-cluster-5-from-clusters-0-and-3">Find Markers Distinguishing Cluster 5 from Clusters 0 and 3</h2>

<p>We just used <code class="language-plaintext highlighter-rouge">FindMarkers</code> to run the same test on cluster 2 as <code class="language-plaintext highlighter-rouge">FindAllMarkers</code> performs for all the clusters, but we can also use it to ask different questions. If we are interested in what makes one cluster different from another, then we might want to look for genes that are differentially expressed between two specific clusters. We can also choose to compare expression with more than one cluster. We might do this if we want to focus on differences between two specific cell types, without considering what makes them different from the rest of the dataset - this could be handy if we have two very similar cell types or subtypes.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-use-findmarkers-to-compare-specific-clusters"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Use `FindMarkers` to Compare Specific Clusters</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_clustering/seurat_clustering/5.0+galaxy0" title="Seurat Find Clusters tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Find Clusters</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">UMAP Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Identify marker genes for specific groups with 'FindMarkers'</code>
          <ul>
            <li><em>“Compare markers between clusters of cells”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
              <ul>
                <li><em>“Identity class to define markers for”</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
                <li><em>“Second identity class to compare”</em>: <code class="language-plaintext highlighter-rouge">0,3</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-16"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>What are the top five markers of cluster 5 compared to clusters 0 and 3?</li>
    <li>Are these the same as the top five markers for the cluster when we ran FindAllMarkers?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-16"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-16" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>
        <p>The top five markers in the output table are:</p>

        <div class="Separate-Preprocessing-Steps">

          <table class="matrix">
            <thead>
              <tr>
                <th> </th>
                <th> </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>1</td>
                <td>FCGR3A</td>
              </tr>
              <tr>
                <td>2</td>
                <td>IFITM3</td>
              </tr>
              <tr>
                <td>3</td>
                <td>CFD</td>
              </tr>
              <tr>
                <td>4</td>
                <td>CD68</td>
              </tr>
              <tr>
                <td>5</td>
                <td>RP11-290F20.3</td>
              </tr>
            </tbody>
          </table>

        </div>

        <div class="SCTransform">

          <table class="matrix">
            <thead>
              <tr>
                <th> </th>
                <th> </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>1</td>
                <td>GZMB</td>
              </tr>
              <tr>
                <td>2</td>
                <td>PRF1</td>
              </tr>
              <tr>
                <td>3</td>
                <td>FGFBP2</td>
              </tr>
              <tr>
                <td>4</td>
                <td>GNLY</td>
              </tr>
              <tr>
                <td>5</td>
                <td>GZMA</td>
              </tr>
            </tbody>
          </table>

        </div>
      </li>
      <li>
        <p>If we go back to our <code class="language-plaintext highlighter-rouge">FindAllMarkers</code> table, we’ll see that these aren’t exactly the same as the top five markers for cluster 5 when we compared it to all of the rest of the dataset (remember you can filter the results of <code class="language-plaintext highlighter-rouge">FindAllMarkers</code> again if you get bored of scrolling all the way down to cluster 5!). Only two of these markers are in the top five of both lists, although we can find the other genes further down in the table if we look.
We wouldn’t expect to see the same results because we’re now looking for differences specifically between cluster 5 and clusters 0 and 3. The genes that <code class="language-plaintext highlighter-rouge">FindAllMarkers</code> identified as differentiating cluster 5 from all of the other clusters might not be best at differentiating it specifically from clusters 0 and 3 - some of those markers could actually be expressed by all three of these clusters.</p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<h2 id="find-markers-with-the-highest-classification-power">Find markers with the highest ‘classification power’</h2>

<p>We can also use other methods for DE analyis in Seurat. We can use the ‘ROC’ test to find out the ‘classification power’ of marker genes for our clusters. A classification power of 1 means that the expression level of this gene can perfectly assign cells to this cluster. A classification power of 0 means that the expression of this gene is useless for identifying cells in this particular cluster - it’s completely random!</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-use-findmarkers-to-calculate-classification-power"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Use `FindMarkers` to Calculate Classification Power</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_clustering/seurat_clustering/5.0+galaxy0" title="Seurat Find Clusters tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Find Clusters</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">UMAP Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Identify marker genes for specific groups with 'FindMarkers'</code>
          <ul>
            <li><em>“Compare markers for two groups of cells”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>“Change cell identities before finding markers”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>“Compare markers between clusters of cells”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
              <ul>
                <li><em>“Identity class to define markers for”</em>: <span class="Separate-Preprocessing-Steps"><code class="language-plaintext highlighter-rouge">0</code></span><span class="SCTransform"><code class="language-plaintext highlighter-rouge">2</code></span></li>
              </ul>
            </li>
            <li><em>“Minimum log-fold difference to test”</em>: <code class="language-plaintext highlighter-rouge">0.25</code></li>
            <li><em>“Select test to run”</em>: <code class="language-plaintext highlighter-rouge">roc</code></li>
            <li>In <em>“Advanced Options”</em>:
              <ul>
                <li><em>“Only return positive markers”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-17"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>What do the names of the top markers for this cluster have in common and what does this signify?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-17"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-17" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>The top five markers in the output table are:</li>
    </ol>

    <div class="Separate-Preprocessing-Steps">

      <blockquote class="matrix">
        <table>
          <thead>
            <tr>
              <th> </th>
              <th> </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>1</td>
              <td>RPS12</td>
            </tr>
            <tr>
              <td>2</td>
              <td>RPS6</td>
            </tr>
            <tr>
              <td>3</td>
              <td>RPL27</td>
            </tr>
            <tr>
              <td>4</td>
              <td>RPS32</td>
            </tr>
            <tr>
              <td>5</td>
              <td>RPS14</td>
            </tr>
          </tbody>
        </table>
      </blockquote>

    </div>

    <div class="SCTransform">

      <blockquote class="matrix">
        <table>
          <thead>
            <tr>
              <th> </th>
              <th> </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>1</td>
              <td>RPS27</td>
            </tr>
            <tr>
              <td>2</td>
              <td>RPL32</td>
            </tr>
            <tr>
              <td>3</td>
              <td>RPS6</td>
            </tr>
            <tr>
              <td>4</td>
              <td>RPS12</td>
            </tr>
            <tr>
              <td>5</td>
              <td>RPS14</td>
            </tr>
          </tbody>
        </table>
      </blockquote>

    </div>

    <p>Many of the top markers (including all of the top five) have names starting with RP. In humans, this gene naming patterns indicates that they are ribosomal genes (encoding proteins or RNAs that form ribosomes). This cluster might represent a group of cells that are very busy making new proteins using all these ribosomes. If we expect our data to include a cell type that has lots of ribosomes, then this could be a sign that they’ve formed their own cluster, so we’ll be happy with this result (this is actually the case here, as we’ll see in the next section).
   However, if we don’t expect to see differences in ribosomal content between cells, then we might suspect that we’ve ended up with a cluster based on ribosomal RNA content rather than on cell type. In this case, we might want to go back to the QC steps. We could score the cells for <code class="language-plaintext highlighter-rouge">percent.ribo</code> in the same way we did for <code class="language-plaintext highlighter-rouge">percent.mt</code>. We could then filter out cells with unusually high proportions of ribosomal genes or regress out the variation associated with this characteristic during the scaling step. Just as when we’re filtering cells by mitochondrial RNA proportions, we would need to think carefully about this - we wouldn’t want to eliminate a cell type just because it has higher proportions of ribosomal genes, which is what we could end up doing if we tried it with this particular dataset.</p>

  </blockquote>
</blockquote>

<h1 id="identify-cell-types">Identify Cell Types</h1>

<p>Now that we’ve arranged our data into clusters, we can start to identify the cell types forming each cluster. We can do this by looking for clusters that express genes that we know a certain type of PBMC should express. We can also check that the clusters don’t express genes that we wouldn’t expect to see if we know any negative markers for a particular cell type.</p>

<p>We can take two different approaches when identifying our cells:</p>

<ol>
  <li><strong>Unsupervised:</strong>
 We could start with our list of marker genes and then see if they are known to be expressed in specific cell types. In this case, we are starting with what the data tells us - the list of genes produced by our DE tests.</li>
  <li><strong>Supervised:</strong>
 We could start with a list of known marker genes for each cell type we expect to see in our data, then look to see which clusters are expressing these genes. With this approach, we are starting with a list of genes selected from previous research and then searching from them in our data - we’re choosing or ‘supervising’ the genes we use.</li>
</ol>

<p>If we’re taking the unsupervised approach, then we might want to limit our markers to the ones that showed the biggest differences in expression. We can do this using some of the extra parametes provided by <code class="language-plaintext highlighter-rouge">FindAllMarkers</code>. Let’s refine the parameters to find the top 10 positive markers for each cluster that showed a log fold change of at least 1 (meaning their expression was at least twice as high).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-find-the-top-10-positive-markers-for-each-cluster"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Find the Top 10 Positive Markers for Each Cluster</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_clustering/seurat_clustering/5.0+galaxy0" title="Seurat Find Clusters tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Find Clusters</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">UMAP Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Identify marker genes with 'FindAllMarkers'</code>
          <ul>
            <li><em>“Minimum log-fold difference to test”</em>: <code class="language-plaintext highlighter-rouge">1.0</code></li>
            <li><em>“Select test to run”</em>: <code class="language-plaintext highlighter-rouge">wilcox</code></li>
            <li><em>“Limit output to top N markers per cluster”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
              <ul>
                <li><em>“Number to show”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
              </ul>
            </li>
            <li>In <em>“Advanced Options”</em>:
              <ul>
                <li><em>“Only return positive markers”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the second output (the csv file) as <code class="language-plaintext highlighter-rouge">DE Markers</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<p>We could now search online or in a database such as <a href="https://panglaodb.se/">PanglaoDB</a> or the <a href="https://www.proteinatlas.org/">Human Protein Atlas</a> to find out which cell types express the markers we’ve identified for each cluster. If we look at the top 10 markers for cluster 3, we see that the first gene in the list is CD79A.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-18"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>Which type of cells express CD79A?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-18"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-18" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>A quick search online or in one of the databases should tell you that CD79A is usually expressed by B cells, which are a type of PBMC, so we would expect to find some in this dataset. This suggests that cluster 3 is made up of B cells.</li>
    </ol>

  </blockquote>
</blockquote>

<p>If we were to continue with this unsupervised approach, we would look at more of the top markers for cluster 3 to check that they are expressed by the same cell type as CD79A. We would then repeat the process for the rest of our clusters.</p>

<p>The unsupervised approach might be the only option if we don’t know exactly what we’re looking for - we might be investigating rare cell types, working with samples from an understudied species, or looking at the changes that happen in a particular disease. However, in many cases, we can use what we already know to begin to understand our dataset. We can look for genes that previous research has identified as markers for the cell types we expect to see. We won’t always get all the answers we need from a supervised approach like this, but it can be a quicker way to identify known cell types.</p>

<p>A supervised approach should work well for the current dataset because PBMCs have been very well characterised. We know which cell types should be present in a PBMC sample and which genes each of these cell types should be expressing.</p>

<div class="Separate-Preprocessing-Steps">

  <p>To begin, we’ll need a list of these canonical markers for PBMCs. Let’s use the one provided in the <a href="https://satijalab.org/seurat/articles/pbmc3k_tutorial">original Seurat version</a> of this tutorial:</p>

  <h2 id="known-markers-for-pbmcs">Known Markers for PBMCs</h2>

  <blockquote class="matrix">
    <table>
      <thead>
        <tr>
          <th>Cell Type</th>
          <th>Marker Genes</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>Naive CD4+ T</td>
          <td>IL7R, CCR7</td>
        </tr>
        <tr>
          <td>CD14+ Mono</td>
          <td>CD14, LYZ</td>
        </tr>
        <tr>
          <td>Memory CD4+ T</td>
          <td>IL7R, S100A4</td>
        </tr>
        <tr>
          <td>B</td>
          <td>MS4A1</td>
        </tr>
        <tr>
          <td>CD8+ T</td>
          <td>CD8A</td>
        </tr>
        <tr>
          <td>FCGR3A+ Mono</td>
          <td>FCGR3A, MS4A7</td>
        </tr>
        <tr>
          <td>NK</td>
          <td>GNLY, NKG7</td>
        </tr>
        <tr>
          <td>DC</td>
          <td>FCER1A, CST3</td>
        </tr>
        <tr>
          <td>Platelet</td>
          <td>PPBP</td>
        </tr>
      </tbody>
    </table>
  </blockquote>

</div>

<div class="SCTransform">

  <p>To begin, we’ll need a list of these canonical markers for PBMCs. Let’s use the ones provided in the original Seurat <a href="https://satijalab.org/seurat/articles/pbmc3k_tutorial">clustering</a> and <a href="https://satijalab.org/seurat/articles/sctransform_vignette.html">SCTransform</a> tutorials:</p>

  <blockquote class="matrix">
    <table>
      <thead>
        <tr>
          <th>Cell Type</th>
          <th>Marker Genes</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>CD14+ Mono</td>
          <td>CD14, LYZ</td>
        </tr>
        <tr>
          <td>B</td>
          <td>MS4A1</td>
        </tr>
        <tr>
          <td>FCGR3A+ Mono</td>
          <td>FCGR3A, MS4A7</td>
        </tr>
        <tr>
          <td>NK</td>
          <td>GNLY, NKG7</td>
        </tr>
        <tr>
          <td>DC</td>
          <td>FCER1A, CST3</td>
        </tr>
        <tr>
          <td>Platelet</td>
          <td>PPBP</td>
        </tr>
        <tr>
          <td>Naive CD4+ T</td>
          <td>IL7R, CCR7</td>
        </tr>
        <tr>
          <td>Memory CD4+ T</td>
          <td>IL7R, S100A4</td>
        </tr>
        <tr>
          <td>IFN-activated CD4+ T</td>
          <td>IL7R, ISG15, IL32</td>
        </tr>
        <tr>
          <td>Naive CD8+ T</td>
          <td>CD8A, CCR7</td>
        </tr>
        <tr>
          <td>Memory CD8+ T</td>
          <td>CD8A, CCL5</td>
        </tr>
        <tr>
          <td>Effector CD8+ T</td>
          <td>CD8A, GZMK</td>
        </tr>
      </tbody>
    </table>
  </blockquote>

  <blockquote class="comment">
    <div class="box-title comment-title" id="comment-10"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
    <p>If you’ve already done the other version of this tutorial, using the separate preprocessing tools, you might notice this list includes some additional markers for T cells. SCTransform can reveal more of the biological variation in our dataset - remember that it produced 12 clusters, rather than the nine we would find using the separate preprocessing tools. We can use these extra markers to identify those additional clusters.</p>
  </blockquote>

</div>

<p>The suggested marker gene for B cells isn’t our top marker, CD79A, but MS4A1. If we look back at our table of the top 10 markers, we can see that this was actually the second gene in the list, so if we’d continued with the unsupervised approach we’d have ended up using the same gene to help identify our B cells. Both of these genes are actually very good markers for B cells, so some lists of B cell markers might suggest using CD79A rather than (or in addition to) MS4A1. If we plot the expression of both genes across our clusters, we should be able to see where our B cells are.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-make-violin-plots-of-b-cell-markers"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Make Violin Plots of B Cell Markers</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">UMAP Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Violin Plot with 'VlnPlot'</code>
          <ul>
            <li><em>“Features to plot”</em>: <code class="language-plaintext highlighter-rouge">CD79A,MS4A1</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-27" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_MS4A1_CD79A_Violin.png" alt="Two violin plots showing expression of CD79A and MS4A1 mainly in cluster 3. " width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_MS4A1_CD79A_Violin.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 27</strong>:</span> Violin plots showing expression of CD79A and MS4A1 by cluster</figcaption></figure>

</div>

<div class="SCTransform">

  <figure id="figure-28" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_MS4A1_CD79A_Violin_SCT.png" alt="Two violin plots showing expression of CD79A and MS4A1 mainly in cluster 3. " width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_MS4A1_CD79A_Violin_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 28</strong>:</span> Violin plots showing expression of CD79A and MS4A1 by cluster</figcaption></figure>

</div>

<p>Both CD79A and MS4A1 are mainly expressed in cluster 3, with very little expression in any of the other clusters. Since most cells in cluster 3 are expressing these two well-known markers of B cells and our DE test showed they were expressed more by this cluster than in the rest of the data, we can confidently say that cluster 3 represents our B cell population.</p>

<p>Sometimes the results aren’t quite so clear as markers might be expressed across multiple clusters of the same or different cell types. We might need to use multiple markers to differentiate between these groups - as with T cells in this dataset.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-make-violin-plots-of-t-cell-markers"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Make Violin Plots of T Cell Markers</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">UMAP Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Violin Plot with 'VlnPlot'</code>
          <ul>
            <li><em>“Features to plot”</em>: <span class="Separate-Preprocessing-Steps"><code class="language-plaintext highlighter-rouge">IL7R,CCR7,S100A4,CD8A</code></span><span class="SCTransform"><code class="language-plaintext highlighter-rouge">IL7R,CCR7,S100A4,CD8A,GZMK,CCL5,IL32,ISG15</code></span></li>
            <li>In <em>“Plot Formatting Options”</em>:
              <ul>
                <li><em>“Number of columns to display”</em>: <code class="language-plaintext highlighter-rouge">4</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Change size of plot”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Width of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">3100</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-29" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_violin_T_cell_markers.png" alt="The first violin plot shows IL7R expressed in clusters 0, 2 and 4 (although there is some lower level expression in cluster 7). Second plot shows high CCR7 expression in cluster 0 with some expression in cluster 2. Plot 3 shows S100A4 expression across most clusters, including at high levels in cluster 2. Fourth plot shows CD8A expression in cluster 4." width="3100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_violin_T_cell_markers.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 29</strong>:</span> Violin plots showing expression of IL7R, CCR7, S100A4, and CD8A by cluster</figcaption></figure>

  <p>We can see that IL7R (also known as CD4+) is mainly expressed in clusters 0, 2, and 4, with some expression in cluster 7. If we were using this marker alone, we could only identify these as clusters of T cells - and in the case of cluster 7, we would be making a mistake as once we plot markers of other cell types, we’ll see that these aren’t T cells at all!</p>

  <p>If we look at which other markers are being expressed (or not expressed) by the T cell clusters, we can identify different types of T cells. Expression of CCR7 is associated with naive T cells, while S100A4 is associated with memory T cells, so you should be able to identify clusters of naive and memory CD4+ T cells in these plots. We can also see that some of our cells are expressing CD8A as well as CD4, so these are CD8+ T cells. We’ll label these clusters below after looking at our other cell type markers.</p>

</div>

<div class="SCTransform">

  <figure id="figure-30" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_violin_T_cell_markers_SCT.png" alt="The first violin plot show IL7R expressed in clusters 0, 2, 4, 8, and 10. Plot 2 shows CCR7 expression in clusters 2, 8, and 10. Plot 3 shows S100A4 expression across most clusters, including at high levels in cluster 0. Plot 4 shows expression of CD8A in clusters 4, 7, and 8. Plot 5 shows GZMK expression in cluster 4 with some lower level expression in cluster 7. Plot 6 shows CCL5 expression in clusters 4, 5, 7, and 11. Plot 7 shows IL32 expression across most clusters, including  Plot 8 shows ISG15 expression at low levels in most clusters, with higher expression in cluster 10." width="3100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_violin_T_cell_markers_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 30</strong>:</span> Violin plots showing expression of IL7R, CCR7, S100A4, CD8A, GZMK, and CCL5 by cluster</figcaption></figure>

  <p>We can see that IL7R (also known as CD4+) is mainly expressed in clusters 0, 2, 4, 8, and 10. With this marker alone, we could only identify these as clusters of T cells, but if we look at which other markers are being expressed (or not expressed) by these T cell clusters, we can identify different types of T cells. To start with, we can see that two of these clusters, 4 and 8, are also expressing CD8A, which identifies them as CD8+ T cells. The expression of CD8A also helps us to identify one more cluster of T cells that we wouldn’t have spotted using CD4 alone - cluster 7.</p>

  <p>We can use the rest of our T cell markers to identify different subtypes of CD4+ and CD8+ T cells. For our CD4+ cells, expression of CCR7 is associated with naive T cells, while S100A4 is associated with memory T cells. We also have some CD4+ T cells expressing ISG15 and IL32, which marks them as IFN activated T cells. For the CD8+ clusters, we can again use CCR7 to pick out the naive T cells, but we will use CCL5 to identify the memory T cells. We can also spot a third group of CD8+ T cells that are expressing GZMK, which identifies them as effector T cells.</p>

  <p>You should be able to identify clusters of all six of these T cell subtypes in your plots. Using SCTransform for preprocessing has allowed us to break down our T cells into more clusters than if we used the separate preprocessing tools. It looks like these include additional subtypes of CD8+ T cells, as we have three clusters expressing CD8A alongside different T cell subtype markers, when we would only have one cluster of CD8+ cells if we used the separate preprocessing steps. We’ll label all of these clusters later, after looking at the markers for other cell types.</p>

</div>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-11"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>How do you know which markers to use to identify different cell types?
Unfortunately, there isn’t always an easy answer - it depends on how well defined the cell types you’re interested in are. A quick search online or a look through the literature should give you an idea of which markers other people have used to identify the cell types you expect to see in your sample. You could look for genes that have been identified by previous single cell studies or ones that are used to identify or isolate a specific cell type. For well studied cell types like PBMCs, it should be easy to find lists of canonical markers that are often used.</p>
</blockquote>

<p>To continue with the supervised approach, we can check the expression of the chosen markers in our clusters to see if they match the known expression patterns of specific cell types. We could just look for these genes in our list of the top positive markers, but there are also several options for visualising marker gene expression. We’ll try a few of them here to give you an idea of the types of plots you can use in the future.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-colour-the-umap-plot-by-canonical-marker-expression"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Colour the UMAP Plot by Canonical Marker Expression</div>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">UMAP Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Visualize expression with 'FeaturePlot'</code>
          <ul>
            <li><em>“Features to plot”</em>: <span class="Separate-Preprocessing-Steps"><code class="language-plaintext highlighter-rouge">IL7R,CCR7,CD14,LYZ,S100A4,MS4A1,CD8A,FCGR3A,MS4A7,GNLY,NKG7,FCER1A,CST3,PPBP</code></span><span class="SCTransform"><code class="language-plaintext highlighter-rouge">CD14,LYZ,MS4A1,FCGR3A,MS4A7,GNLY,NKG7,FCER1A,CST3,PPBP,IL7R,CCR7,S100A4,CD8A,GZMK,CCL5,IL32,ISG15</code></span></li>
            <li><em>“Name of reduction to use”</em>: <code class="language-plaintext highlighter-rouge">umap</code></li>
          </ul>
        </li>
        <li><em>“Change size of plot”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Width of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">4100</code></li>
            <li><em>“Height of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">4100</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-31" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_FeaturePlot_CellTypeMarkers.png" alt="14 UMAP Plots coloured by expression of different genes. Plot 1 shows IL7R expression in clusters 0, 2 and part of cluster 4. Plot 2 shows CCR7 expression in cluster 0. Plot 3 shows CD14 expression mainly in cluster 1. Plot 4 shows LYZ expression in clusters 1, 5 and 7. Plot 5 shows high S100A4 expression in clusters 1, 5, and 7 with medium expression in clusters 2, 4 and 6. Plot 6 shows MS4A1 expression in cluster 3. Plot 7 shows CD8A expression in cluster 4. Plot 8 shows FCGR3A expression in clusters 5 and 6. Plot 9 shows MS4A7 expression in cluster 5. Plot 10 shows GNLY expression in cluster 6. Plot 11 shows NKG7 expression in clusters 4 and 6. Plot 12 shows FCER1A expression in cluster 7. Plot 13 shows CST3 expression in clusters 1, 5 and 7. Plot 14 shows PPBP expression in cluster 8." width="4100" height="4100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_FeaturePlot_CellTypeMarkers.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 31</strong>:</span> UMAP plots showing expression of canonical markers for PBMCs</figcaption></figure>

</div>

<div class="SCTransform">

  <figure id="figure-32" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_FeaturePlot_CellTypeMarkers_SCT.png" alt="18 UMAP Plots coloured by expression of different genes. Plot 1 shows CD14 expression mainly in cluster 1. Plot 2 shows LYZ expression in clusters 1, 6, and 9.  Plot 3 shows MS4A1 expression in cluster 3.  Plot 4 shows FCGR3A expression in clusters 5 and 6. Plot 5 shows MS4A7 expression in cluster 6. Plot 6 shows GNLY expression in cluster 5. Plot 7 shows NKG7 expression in clusters 5 and 7. Plot 8 shows FCER1A expression in cluster 9. Plot 9 shows CST3 expression in clusters 1, 6, 9, and 11. Plot 10 shows PPBP expression in cluster 11.  Plot 11 shows IL7R expression in clusters 0, 2, 4, 8, 10 and part of cluster 7. Plot 12 shows CCR7 expression in clusters 2, 8, and 10. Plot 13 shows high S100A4 expression in clusters 1 and 6 with medium expression in clusters 0, 4, 5, 7 and 9. Plot 14 shows CD8A expression in clusters 4, 7, and 8. Plot 15 shows GZMK expression in cluster 4. Plot 16 shows CCL5 expression in clusters 4, 5, and 7. Plot 17 shows IL32 expression in clusters 0, 2, 4, 5, 6, 7, and 11. Plot 18 shows high ISG15 expression in cluster 10 with some expression in clusters 1 and 6. " width="4100" height="5100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_FeaturePlot_CellTypeMarkers_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 32</strong>:</span> UMAP plots showing expression of canonical markers for PBMCs</figcaption></figure>

</div>

<p>We have produced a series of UMAP plots, each coloured according to the expression level of a different marker gene. We can look back at <a href="#figure-13">the UMAP plots we created earlier</a> showing the cluster numbers to see which areas correspond to specific clusters.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-19"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>Are the markers clearly associated with one or more clusters?</li>
    <li>Are the canonical markers only expressed in these clusters?</li>
    <li>How similar are the clusters in terms of their expression of these cell type markers? Are more similar cell types closer together on the UMAP plot?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-19"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-19" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>Each cell type marker is clearly expressed at higher levels in certain parts of the plots - we don’t have the same problem as we did when we plotted the top genes for PCs 1-3 and found that MALAT1 was expressed at quite high levels across the entire plot! When we compare the expression plots to the clusters we previously plotted, we can see that they match up quite well with each other. Most of the markers are expressed more in certain clusters, but there isn’t always a clear boundary, especially for clusters that are close together in the plot. Some markers are associated with a single cluster, such as MS4A1 and PPBP, while others are expressed across multiple clusters.</li>
      <li>Although marker expression mainly occurs in one or more clusters, there is still some expression by cells in other parts of the plot. It could be that these genes are sometimes expressed by other cell types, but these could also be cells of the same type that have simply ended up further away in the plot. It can sometimes be hard to tell which cluster these cells have been assigned to from these plots, especially when you consider that some points could be hidden behind others. Cells of the same type might have been assigned to different clusters, but there could also be a few cells that have been plotted at a distance from where the main part of their assigned cluster appears on this plot. <span class="Separate-Preprocessing-Steps">If we look back at the UMAP coloured by cluster, we can spot a couple of differently coloured cells at the tip of cluster 7 and by cluster 8.</span><span class="SCTransform">If we look back at the UMAP coloured by cluster, we can spot a couple of differently coloured cells and mixed into cluster 3 and at the tip of cluster 10.</span></li>
      <li>We can distinguish between all of our clusters on the basis of which known markers they are shown to express. Each cluster expresses a unique combination of these genes, although the markers aren’t only expressed in the clusters they’re mainly associated with. Some of the clusters are more similar to each other, particularly those that are close together on the plot, forming part of the same larger group of cells. <span class="Separate-Preprocessing-Steps">For example, clusters 1, 5 and 7 are grouped together on the plot and all express a lot of LYZ, S100A4, and CST3.</span><span class="SCTransform">For example, clusters 1 and 6 are grouped together on the plot for and both express a lot of S100A4 and CST3.</span> Since the UMAP was created based on the PCA, which was in turn based on similarities in expression of the highly variable genes, it makes sense that clusters that are plotted closer together are more similar in expression to each other. However, we should always remember that we’re only looking at a 2D plot of the first two UMAP dimensions, so we shouldn’t read too much into what we see! The plot can’t tell us everything about the relationships between cells and clusters.</li>
    </ol>
  </blockquote>
</blockquote>

<p>Another option for visualising marker gene expression across clusters is the Violin Plot, which can sometimes make it easier to distinguish between clusters. Each cluster is plotted separately, so we don’t need to work out which cluster a cell belongs to, which can be tricky when we have similar clusters close together and often overlapping on the UMAP plot.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-use-violin-plots-to-compare-expression-of-canonical-markers-by-cluster"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Use Violin Plots to Compare Expression of Canonical Markers by Cluster</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">UMAP Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Violin Plot with 'VlnPlot'</code>
          <ul>
            <li><em>“Features to plot”</em>: <span class="Separate-Preprocessing-Steps"><code class="language-plaintext highlighter-rouge">IL7R,CCR7,CD14,LYZ,S100A4,MS4A1,CD8A,FCGR3A,MS4A7,GNLY,NKG7,FCER1A,CST3,PPBP</code></span><span class="SCTransform"><code class="language-plaintext highlighter-rouge">CD14,LYZ,MS4A1,FCGR3A,MS4A7,GNLY,NKG7,FCER1A,CST3,PPBP,IL7R,CCR7,S100A4,CD8A,GZMK,CCL5,IL32,ISG15</code></span></li>
          </ul>
        </li>
        <li><em>“Change size of plot”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Width of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">4100</code></li>
            <li><em>“Height of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">4100</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-33" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_Violin_CellTypeMarkers.png" alt="14 Violin plots showing expression of different genes. Plot 1 shows IL7R expression in clusters 0, 2, and 4, with some expression in cluster 7. Plot 2 shows CCR7 expression in cluster 0 and to a lesser extent in cluster 2. Plot 3 shows CD14 expression in cluster 1. Plot 4 shows LYZ expression across all clusters, with the highest expression in clusters 1 and 7. Plot 5 shows S100A4 expression across all clusters, with the highest expression in clusters 1 and 5. Plot 6 shows MS4A1 expression in cluster 3. Plot 7 shows CD8A expression in cluster 4. Plot 8 shows FCGR3A expression in clusters 5 and 6. Plot 9 shows MS4A7 expression in cluster 5 and to a lesser extent in cluster 1. Plot 10 shows GNLY expression in clusters 4 and 6. Plot 11 shows NKG7 expression in clusters 4 and 6, with some expression at lower levels in clusters 5 and 7. Plot 12 shows FCER1A expression in cluster 7. Plot 13 shows CST3 expression in clusters 1, 5, and 7 with some expression in cluster 8. Plot 14 shows PPBP expression in cluster 8." width="4100" height="4100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_Violin_CellTypeMarkers.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 33</strong>:</span> Violin plots showing expression of canonical marker genes by cluster</figcaption></figure>

</div>

<div class="SCTransform">

  <figure id="figure-34" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_Violin_CellTypeMarkers_SCT.png" alt="18 Violin plots showing expression of different genes. Plot 1 shows CD14 expression in cluster 1. Plot 2 shows LYZ expression across all clusters, with high expression in cluster 1 and 9.  Plot 3 shows MS4A1 expression in cluster 3.  Plot 4 shows FCGR3A expression in clusters 5, 6, and 7. Plot 5 shows MS4A7 expression in cluster 6 with some low expression in cluster 1. Plot 6 shows high GNLY expression in cluster 5 as well as expression at various levels in cluster 7. Plot 7 shows NKG7 expression in clusters 4, 5, and 7. Plot 8 shows FCER1A expression in cluster 9. Plot 9 shows CST3 expression in clusters 1, 6, 9, and 11. Plot 10 shows PPBP expression in cluster 11. Plot 11 shows IL7R expression in clusters 0, 2, 4, 8, and 10. Plot 12 shows CCR7 expression in clusters 2, 8, and 10. Plot 13 shows S100A4 expression across all clusters, with the highest expression in clusters 1, 6, and 9. Plot 14 shows CD8A expression in cluster 4, 7, and 8. Plot 15 shows GZMK expression in cluster 4 with some expression in cluster 7. Plot 16 shows CCL5 expression in cluster 4, 5, 7, and 11. Plot 17 shows IL32 expression across clusters 0, 2, 4, 5, 7, 8, and 10 with some expression in cluster 11. Plot 18 shows ISG15 expression across most clusters, with the highest expression in cluster 10." width="4100" height="5100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_Violin_CellTypeMarkers_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 34</strong>:</span> Violin plots showing expression of canonical marker genes by cluster</figcaption></figure>

</div>

<blockquote class="question">
  <div class="box-title question-title" id="question-20"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>Are the markers clearly associated with one or more clusters?</li>
    <li>Are the canonical markers only expressed in these clusters - and are they expressed at the same level by all the cells in these clusters?</li>
    <li>How similar are the clusters to each other in terms of their marker gene expression - are there any clusters that seem particularly alike?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-20"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-20" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>The violin plot can make the associations between clusters and markers look clearer, since each cluster is plotted separately and labelled by number. Some of the markers are clearly expressed mainly in one or a few clusters, while others like LYZ and S100A4 are expressed across most of the clusters.</li>
      <li>None of the markers are unique to a specific cluster - there are always a few dots (or cells) in the other clusters that are expressing these genes. We can also see that the expression level varies within each cluster as there are cells at different positions along the vertical axis. As well as looking at the expression level in individual cells (dots) we can compare the overall expression patterns by looking at the coloured violin shapes, which show us where most of the cells in each cluster are plotted. For example, we can see there was more variation in the expression level of NKG7 in cluster 4 (where the violin shape is longer) than there was in cluster <span class="Separate-Preprocessing-Steps">6</span><span class="SCTransform">5</span> (where there is a short violin shape as all the cells showed high expression of this gene).</li>
      <li>Each cluster has a unique pattern of expression of the canonical markers. Some clusters are very distinct as they express markers that are rarely seen in other clusters. Cluster 3 is the only cluster to express MS4A1 and it doesn’t express much of the other markers. Other clusters seem to be more similar to each other, with multiple clusters of T cells all expressing IL7R, although the expression levels of other genes vary between these cells allowing us to classify them as different subtypes. This makes sense because cluster 3 is far away from all the other clusters in the UMAP while the different T cell clusters are close to each other because of their similarities.</li>
    </ol>
  </blockquote>
</blockquote>

<p>Based on our table of marker genes and these plots, we know which clusters were expressing the canonical markers for each cell type or subtype. We can use this information to annotate our clusters by cell type.</p>

<div class="Separate-Preprocessing-Steps">

  <blockquote class="question">
    <div class="box-title question-title" id="question-21"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
    <ol>
      <li>Can you identify the cell types for each cluster?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-21"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-21" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
      <ol>
        <li>Based on the expression of the known PBMC markers, we can assign the following cell types to our clusters:</li>
      </ol>

      <blockquote class="matrix">
        <table>
          <thead>
            <tr>
              <th>Cell Type</th>
              <th>Marker Genes</th>
              <th>Clusters</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>Naive CD4+ T</td>
              <td>IL7R, CCR7</td>
              <td>0</td>
            </tr>
            <tr>
              <td>CD14+ Mono</td>
              <td>CD14, LYZ</td>
              <td>1</td>
            </tr>
            <tr>
              <td>Memory CD4+ T</td>
              <td>IL7R, S100A4</td>
              <td>2</td>
            </tr>
            <tr>
              <td>B</td>
              <td>MS4A1</td>
              <td>3</td>
            </tr>
            <tr>
              <td>CD8+ T</td>
              <td>CD8A</td>
              <td>4</td>
            </tr>
            <tr>
              <td>FCGR3A+ Mono</td>
              <td>FCGR3A, MS4A7</td>
              <td>5</td>
            </tr>
            <tr>
              <td>NK</td>
              <td>GNLY, NKG7</td>
              <td>6</td>
            </tr>
            <tr>
              <td>DC</td>
              <td>FCER1A, CST3</td>
              <td>7</td>
            </tr>
            <tr>
              <td>Platelet</td>
              <td>PPBP</td>
              <td>8</td>
            </tr>
          </tbody>
        </table>
      </blockquote>
    </blockquote>
  </blockquote>

</div>

<div class="SCTransform">

  <blockquote class="question">
    <div class="box-title question-title" id="question-22"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
    <ol>
      <li>Can you identify the cell types for each cluster?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-22"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-22" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
      <ol>
        <li>Based on the expression of the known PBMC markers, we can assign the following cell types to our clusters:</li>
      </ol>

      <blockquote class="matrix">
        <table>
          <thead>
            <tr>
              <th>Cell Type</th>
              <th>Marker Genes</th>
              <th>Clusters</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>Memory CD4+ T</td>
              <td>IL7R, S100A4</td>
              <td>0</td>
            </tr>
            <tr>
              <td>CD14+ Mono</td>
              <td>CD14, LYZ</td>
              <td>1</td>
            </tr>
            <tr>
              <td>Naive CD4+ T</td>
              <td>IL7R, CCR7</td>
              <td>2</td>
            </tr>
            <tr>
              <td>B</td>
              <td>MS4A1</td>
              <td>3</td>
            </tr>
            <tr>
              <td>Effector CD8+ T</td>
              <td>CD8A, GZMK</td>
              <td>4</td>
            </tr>
            <tr>
              <td>NK</td>
              <td>GNLY, NKG7</td>
              <td>5</td>
            </tr>
            <tr>
              <td>FCGR3A+ Mono</td>
              <td>FCGR3A, MS4A7</td>
              <td>6</td>
            </tr>
            <tr>
              <td>Memory CD8+ T</td>
              <td>CD8A, CCL5</td>
              <td>7</td>
            </tr>
            <tr>
              <td>Naive CD8+ T</td>
              <td>CD8A, CCR7</td>
              <td>8</td>
            </tr>
            <tr>
              <td>DC</td>
              <td>FCER1A, CST3</td>
              <td>9</td>
            </tr>
            <tr>
              <td>IFN-activated CD4+ T</td>
              <td>IL7R, ISG15, IL32</td>
              <td>10</td>
            </tr>
            <tr>
              <td>Platelet</td>
              <td>PPBP</td>
              <td>11</td>
            </tr>
          </tbody>
        </table>
      </blockquote>
    </blockquote>
  </blockquote>

</div>

<p>We can now rename our clusters using these cell names, while keeping a copy of the original idents (the cluster numbers) in case we want to use them again. It will be easier to interpret our plots and any downstream analyses we run if we can use cell names rather than cluster numbers!</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-rename-clusters-with-cell-types"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Rename Clusters with Cell Types</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_data/seurat_data/5.0+galaxy0" title="Seurat Data Management tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Data Management</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">UMAP Results</code> (output of <strong>Seurat Run Dimensional Reduction</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Manipulate Seurat Object</code>
          <ul>
            <li><em>“Manipulation to perform”</em>: <code class="language-plaintext highlighter-rouge">Rename idents</code>
              <ul>
                <li><em>“Rename all idents”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
                  <ul>
                    <li><em>“New names”</em>: <span class="Separate-Preprocessing-Steps"><code class="language-plaintext highlighter-rouge">CD4 Naive T, CD14 Mono, CD4 Memory T, B, CD8 T,FCGR3A Mono, NK, DC, Platelet</code></span><span class="SCTransform"><code class="language-plaintext highlighter-rouge">CD4 Memory T, CD14 Mono, CD4 Naive T, B, CD8 Effector T, NK, FCGR3A Mono, CD8 Memory T, CD8 Naive T,DC, CD4 IFN-activated T,Platelet</code></span></li>
                  </ul>
                </li>
                <li><em>“Save copy of old idents first”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Annotated Clusters</code></li>
  </ol>

</blockquote>

<p>Now we can plot our UMAP again, this time showing the names of our clusters.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-revisualise-the-umap-with-cell-type-annotations"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Revisualise the UMAP with Cell Type Annotations</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Annotated Clusters</code> (output of <strong>Seurat Data Management</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Visualize Dimensional Reduction with 'DimPlot'</code>
          <ul>
            <li><em>“Name of reduction to use”</em>: <code class="language-plaintext highlighter-rouge">umap</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-35" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_UMAP_DimPlot_CellTypes.png" alt="Plot showing three big groups of cells divided into 9 clusters and coloured by cell type. The smallest of these three main groups only contains cells coloured as B cells. The other two groups are made up of cells from different cell types." width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_UMAP_DimPlot_CellTypes.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 35</strong>:</span> UMAP coloured by cell type</figcaption></figure>

</div>

<div class="SCTransform">

  <figure id="figure-36" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_UMAP_DimPlot_CellTypes_SCT.png" alt="Plot showing three big groups of cells divided into 12 clusters and coloured by cell type. The smallest of these three main groups only contains cells coloured as B cells. The other two groups are made up of cells from different cell types." width="2100" height="2100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_UMAP_DimPlot_CellTypes_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 36</strong>:</span> UMAP coloured by cell type</figcaption></figure>

</div>

<h2 id="canonical-markers-vs-marker-genes">Canonical Markers vs Marker Genes</h2>

<p>We’ve now annotated our clusters by cell type using the supervised approach, but let’s try making one more type of plot that we could have used during this process. We can make heatmaps of the canonical markers for each cell type and of the top markers for each cluster to see how expression of these genes varies within and between our clusters.</p>

<p>In order to create a heatmap, we need to prepare a tabular file with a list of the genes we want to plot. For the canonical markers, you could create this table outside of Galaxy and then upload it or follow the first step below to create the file in Galaxy.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-create-heatmaps-to-compare-expression-by-cluster-canonical-markers"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Create Heatmaps to Compare Expression by Cluster - Canonical Markers</div>

  <ol>
    <li>
      <p>Use the Upload Data - Paste/Fetch data option to create a table of genes to plot. Select the input type as <strong>tabular</strong> and enter a list of genes, one on each row. You can copy and paste in the following list.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-file"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-file" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new file</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong> at the bottom</li>   <li>Paste the file contents into the text field</li>   <li>Change <strong>Type</strong> from “Auto-detect” to <code class="language-plaintext highlighter-rouge">tabular</code>* Press <strong>Start</strong> and <strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <div class="Separate-Preprocessing-Steps">

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>IL7R
CCR7
CD14
LYZ
S100A4
MS4A1
CD8A
FCGR3A
MS4A7
GNLY
NKG7
FCER1A
CST3
PPBP
</code></pre></div>        </div>

      </div>

      <div class="SCTransform">

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>IL7R
GZMK
CCL5
CCR7
CD14
IL32
ISG15
LYZ
S100A4
MS4A1
CD8A
FCGR3A
MS4A7
GNLY
NKG7
FCER1A
CST3
PPBP
</code></pre></div>        </div>

      </div>
    </li>
    <li>
      <p>Rename the file as <code class="language-plaintext highlighter-rouge">Canonical Markers</code> when it has finished uploading and make sure the datatype is <strong>tabular</strong></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select  your desired datatype  from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:</p>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Annotated Clusters</code> (output of <strong>Seurat Data Management</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Visualize expression with 'DoHeatmap'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“List of features to plot”</em>: <code class="language-plaintext highlighter-rouge">Canonical Markers</code> (Input dataset)</li>
            <li>In <em>“Plot Formatting Options”</em>:
              <ul>
                <li><em>“Size of text above colour bar”</em>: <code class="language-plaintext highlighter-rouge">3.5</code></li>
                <li><em>“Angle of text above colour bar”</em>: <code class="language-plaintext highlighter-rouge">60</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Change size of plot”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Width of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">4100</code></li>
            <li><em>“Height of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">4100</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-37" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_DoHeatmap_CellType_markers.png" alt="Heatmap showing expression of IL7R mainly in CD4NaiveT, CD4MemoryT, and CD8T cells, expression of CCR7 mainly in CD4NaiveT, CD4MemoryT, and B cells, expression of CD14 in CD14Mono cells, expression of LYZ in clusters CD14Mono, FCGR3Mono, and DC cells, expression of S100A4 in CD14Mono and FCGR3A cells, expression of MS4A1 in B cells, expression of CD8A in CD8T cells, expression of FCGR3A in FCGR3AMono and NK cells, expression of MS4A7 in FCGR3AMono and some CD14Mono cells, expression of GNLY in NK cells and to a lesser extent in CD8T cells, expression of NKG7 in CD8T and NK cells, expression of FCER1A in cluster DC, expression of CST3 in CD14Mono, FCGR3A and DC cells, and expression of PPBP in Platelets." width="4100" height="4100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_DoHeatmap_CellType_markers.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 37</strong>:</span> Heatmap showing expression of known PBMC markers by cluster</figcaption></figure>

</div>

<div class="SCTransform">

  <figure id="figure-38" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_DoHeatmap_CellType_markers_SCT.png" alt="Heatmap showing expression of IL7R mainly in CD4MemoryT, CD4NaiveT, and CD8EffectorT cells, expression of GZMK in CD8EffectorT cells and some CD8MemoryT cells, expression of CCL5 in CD8EffectorT, CD8MemoryT and most NK cells, expression of CCR7 mainly in CD4NaiveT and CD8NaiveT cells, expression of CD14 in CD14Mono cells, expression of IL32 in CD4MemoryT, CD8EffectorT, and CD8MemoryT cells as well as some NK cells, expression of ISG15 in CD4IFN-activatedT cells as well as some CD14Mono and FCGR3AMono cells, expression of LYZ in CD14Mono and DC cells, expression of S100A4 mainly in CD14Mono and FCGR3AMono cells but also in some CD4MemoryT cells, expression of MS4A1 in B cells, expression of CD8A in CD8EffectorT, CD8MemoryT, and CD8NaiveT cells, expression of FCGR3A in FCGR3AMono and NK cells as well as some CD8MemoryT cells, expression of MS4A7 in FCG3RAMono cells with some expression by CD14Mono cells, expression of GNLY in NK cells, expression of NKG7 in NK and CD8MemoryT cells as well as some CD8EffectorT cells, expression of FCER1A in DC, expression of CST3 in CD14Mono, FCGR3AMono and DC cells, and expression of PPBP in Platelets." width="4100" height="4100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_DoHeatmap_CellType_markers_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 38</strong>:</span> Heatmap showing expression of known PBMC markers by cluster</figcaption></figure>

</div>

<p>Rather than looking at the canonical markers, we might want to make a heatmap of the top markers from our DE analysis.
In order to do this, we’ll need to turn the output from <code class="language-plaintext highlighter-rouge">FindAllMarkers</code> into a tabular file type. We can then cut out the column with the list of gene names, getting rid of the column header, and use this as an input for the plot.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-create-heatmaps-to-compare-expression-by-cluster-markers-from-de"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Create Heatmaps to Compare Expression by Cluster - Markers from DE</div>

  <ol>
    <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> pencil icon of the file we renamed as <code class="language-plaintext highlighter-rouge">DE Markers</code> (this was the CSV output from <code class="language-plaintext highlighter-rouge">FindAllMarkers</code>) then
      <ul>
        <li>select <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> Datatypes in the central panel.</li>
        <li>Choose the second option, <code class="language-plaintext highlighter-rouge">Convert to Datatype</code></li>
        <li>make sure <code class="language-plaintext highlighter-rouge">tabular (using </code>Convert CSV to tabular<code class="language-plaintext highlighter-rouge">)</code> is selected in the drop down menu</li>
        <li>press the <code class="language-plaintext highlighter-rouge">Create Dataset</code> button.</li>
        <li>This will create a new, tabular version of the dataset at the top of your history - make sure that this is the version you use in the next step.</li>
      </ul>
    </li>
    <li><span class="tool" data-tool="Cut1" title="Cut tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Cut</strong></span> with the following parameters:
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: <code class="language-plaintext highlighter-rouge">DE Markers</code> (<strong>tabular</strong> output of <strong>Convert CSV to tabular</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li>
      <p>Rename this file as <code class="language-plaintext highlighter-rouge">Input DE Markers</code> - if you take a look at it using the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span>, you should see the column of gene names</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seurat_plot/seurat_plot/5.0+galaxy0" title="Seurat Visualize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Visualize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file with the Seurat object”</em>: <code class="language-plaintext highlighter-rouge">Annotated Clusters</code> (output of <strong>Seurat Data Management</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used”</em>: <code class="language-plaintext highlighter-rouge">Visualize expression with 'DoHeatmap'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“List of features to plot”</em>: <code class="language-plaintext highlighter-rouge">Input DE Markers</code> (output of <strong>Table Compute</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li>In <em>“Plot Formatting Options”</em>:
              <ul>
                <li><em>“Size of text above colour bar”</em>: <code class="language-plaintext highlighter-rouge">3.5</code></li>
                <li><em>“Angle of text above colour bar”</em>: <code class="language-plaintext highlighter-rouge">60</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Change size of plot”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Width of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">4100</code></li>
            <li><em>“Height of plot in pixels”</em>: <code class="language-plaintext highlighter-rouge">4100</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<div class="Separate-Preprocessing-Steps">

  <figure id="figure-39" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_DoHeatmap_TopPositiveMarkers.png" alt="Heatmap showing blocks of higher expression for the top 10 markers in the clusters they are markers for. Most cells in each cluster express the markers for that cluster but the patterns are stronger for some genes than others. Some cells outside the clusters also express the markers. For example the CD8T cells show some expression of the top markers for CD4NaiveT and CD4MemoryT cells, although expression of their own top markers is much higher." width="4100" height="4100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_DoHeatmap_TopPositiveMarkers.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 39</strong>:</span> Heatmap showing expression of the top 10 markers for each cluster</figcaption></figure>

</div>

<div class="SCTransform">

  <figure id="figure-40" style="max-width: 90%;"><img src="../../images/scrna-seurat-pbmc3k/seurat_DoHeatmap_TopPositiveMarkers_SCT.png" alt="Heatmap showing blocks of higher expression for the top 10 markers in the clusters they are markers for. Most cells in each cluster express the markers for that cluster but the patterns are stronger for some genes than others. Some cells outside the clusters also express the markers. For example, the different T cell subtypes often show high expression of the top markers for other T cell groups." width="4100" height="4100" loading="lazy" /><a target="_blank" href="../../images/scrna-seurat-pbmc3k/seurat_DoHeatmap_TopPositiveMarkers_SCT.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 40</strong>:</span> Heatmap showing expression of the top 10 markers for each cluster</figcaption></figure>

</div>

<p>Looking at these plots, we can see that there are clear associations between certain genes and clusters. In both cases, we can also see that there is variation in the expression level of these genes within the associated clusters, with some cells expressing little or none of the gene. We can also see cells outside of the associated clusters that are expressing the genes. This is true for both the known cell type markers and the markers we identified through our DE analysis - even when a statistical test has identified a significant association with one cluster, there are still differences between cells.</p>

<p>Comparing the two plots also shows us why the supervised approach can be faster - we had fewer genes to consider and the patterns can also be a bit clearer. If we know enough about the cells we’re trying to identify, then we can make our lives easier by taking the supervised approach. However, it is always a good idea to try multiple approaches when we can, as we’ll have more confidence in our results if they both end up giving the same answers!</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-23"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>Which plot type was best for annotating clusters?</li>
    <li>Are you happy with this clustering - even for the CD4NaiveT cells cluster, which had a lot of ribosomal genes in its top markers list?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-23"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-23" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>The best type of plot for identifying cell types can depend on your data as well as your own personal preferences. You might find one plot easier to interpret than another. It can also be helpful to create different types of plots as some patterns may be clearer on one type while others are clearer on another. It’s also good to be able to confirm your interpretation on multiple plots!
UMAP plots are often used to provide a quick and memorable overview of the data, but it can be tricky to match the expression patterns to the clusters, especially given the limitations of these plots - some cells can be hidden, adjacent clusters can blend into each other, and we can’t rely on the 2D plot to accurately represent all the relationships between cells and clusters.
Violin plots present each cluster separately, so they can make it easier to differentiate between clusters. Since every cluster is given the same amount of space on the plot, no matter how many cells it contains, violin plots can also make it easier to see what’s going on with smaller clusters. We can also get a clearer idea of how much variation there is within and between clusters, although some of the points may still be hidden behind others.
Heatmaps can give us the best overview of the variation between cells as each cell is given its own little section on the plot. We can see how consistently the markers are expressed within the cluster and how common it is for cells outside the cluster to express the same genes. We can also see the overall patterns as blocks of cells with similar expression profiles, including the clusters that share similar patterns. However, heatmaps can be less useful if we want to focus on individual cells or genes, as it can be hard to pick out details.</li>
      <li>We should be happy with the results of our clustering as they match up with what we already know about PBMCs. We have been able to annotate each cluster as a different cell type based on a supervised approach - and we could do the same using an unsupervised approach. The decisions we made along the way, such as the number of PCs we used and how many nearest neighbors we looked for have worked well. We even identified some subtypes of T cells, although we were able to separate out more of these when we used SCTransform.
Even though many of the top markers for the CD4NaiveT cell cluster were from ribosomal genes, we were still able to identify it as a specific cell type. It represents our population of Naive CD4+ T cells. If we search online to learn a bit more about this cell type, we’ll quickly find that it is known to have lots of ribosomes, so in this case, we can be confident that the high expression of ribosomal genes in these cells is due to real biological differences between cell types, rather than a problem with our data. If we weren’t able to assign a cell type to this cluster, for example because it expressed a mix of markers for different types, then we would come to a different conclusion!
If we couldn’t see strong associations between our clusters and the different cell types that we expect to see in the dataset, then this would be very suspicious - did something go wrong with our experiment or analysis? We would need to go back and try to identify the problem to see if we can fix it. If the problem isn’t too bad, we might just need to change some of the clustering parameters to get clusters that make biological sense - maybe we would need to use more PCs, look for more/fewer nearest neighbors, or simply change the resolution. If the problem is more serious, we might need to recheck the quality of our data or make bigger changes to the analysis.</li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="conclusion">Conclusion</h1>
<p><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Well done, you’ve successfully used Seurat to prepare and cluster single cell data. You might want to check your results against the example histories for the <a href="https://usegalaxy.eu/u/marisa_jl/h/clustering-3k-pbmcs-with-seurat---separate-preprocessing---answer-key">separate preprocessing steps</a> or <a href="https://usegalaxy.eu/u/marisa_jl/h/clustering-3k-pbmcs-with-seurat---sctransform---answer-ke">SCTransform route</a>. You can also take a look at the whole workflow for <a href="https://usegalaxy.eu/u/marisa_jl/w/copy-of-cluster-3k-pbmcs-with-seurat---workflow">the separate steps</a> or <a href="https://usegalaxy.eu/u/marisa_jl/w/cluster-3k-pbmcs-with-seurat---workflow---sctransform-version">SCTransform version</a>.</p>

<p>In this tutorial, we’ve learned about the steps involved in clustering single cell data and how to identify different cell types. We followed a typical clustering workflow:</p>

<ol>
  <li>Preprocessing with
    <ul>
      <li>selection and filtering of cells based on quality metrics</li>
      <li>normalisation and scaling</li>
      <li>selection of features (highly variable genes)</li>
    </ul>
  </li>
  <li>Dimensional reduction
    <ul>
      <li>using PCA and then UMAP</li>
    </ul>
  </li>
  <li>Clustering of the cells by
    <ul>
      <li>computation of a neighborhood graph</li>
      <li>clustering the neighborhood graph into communities</li>
    </ul>
  </li>
  <li>Identification of marker genes for the clusters</li>
  <li>Annotation of the clusters with cell types</li>
</ol>

<p>Now that you know how to perform clustering with Seurat, you might want to try the alternative Scanpy pipeline by following the <a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">Clustering 3K PBMCs with Scanpy</a> tutorial. If you would prefer to stay with Seurat, then you could go back and try using the alternative preprocessing steps described above or try using it to analyse a slightly trickier dataset by following the ‘Filter, plot and explore single cell RNA-seq data with Seurat’ tutorial <a href="/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html">using Galaxy buttons</a> or in an <a href="/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExploreRStudio/tutorial.html">R notebook on Galaxy</a>. You could also try using Seurat to analyse your own data!</p>

<p>This tutorial is part of the https://singlecell.usegalaxy.eu portal (<span class="citation"><a href="#tekman2020single">Tekman <i>et al.</i> 2020</a></span>).</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(typeof plausible !== 'undefined'){
                            // Plausible may be undefined (script blocked)
                            // or it may be defined, but opted-out (select box/DNT),
                            // which means `plausible()` will work but not send data, *nor* execute the callback.
                            plausible('TutorialComplete', {props: {path: document.location.pathname}})
                          }
                          // since the callback is completely cosmetic, we'll just issue it optimistically.
                          tutorial_finish_finish();
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Seurat is a commonly-used pipeline for single cell data analysis</p>
</li>
                        
                        <li><p>Clustering makes single cell datasets easier for us to understand</p>
</li>
                        
                        <li><p>Different tools and parameters should be considered when analysing different datasets</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/single-cell/faqs/">FAQ page for the Single Cell topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

		<!--
                
		-->

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/single-cell#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="tekman2020single">Tekman, M., B. Batut, A. Ostrovsky, C. Antoniewski, D. Clements <i>et al.</i>, 2020 <b>A single-cell RNA-sequencing training and analysis suite using the Galaxy framework</b>. GigaScience 9: giaa102. <a href="https://doi.org/10.1093/gigascience/giaa102">10.1093/gigascience/giaa102</a> <a href="https://academic.oup.com/gigascience/article/9/10/giaa102/5931798">https://academic.oup.com/gigascience/article/9/10/giaa102/5931798</a></li>
<li id="SCTransform2022">Choudhary, S., and R. Satija, 2022 <b>Comparison and evaluation of statistical error models for scrna-seq</b>. Genome Biology 23: <a href="https://doi.org/10.1186/s13059-021-02584-9">10.1186/s13059-021-02584-9</a></li>
<li id="seurat2023v5">Hao, Y., T. Stuart, M. H. Kowalski, S. Choudhary, P. Hoffman <i>et al.</i>, 2023 <b>Dictionary learning for integrative, multimodal and scalable single-cell analysis</b>. Nature Biotechnology. <a href="https://doi.org/10.1038/s41587-023-01767-y">10.1038/s41587-023-01767-y</a></li></ol>
                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=single-cell/scrna-seurat-pbmc3k"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=single-cell/scrna-seurat-pbmc3k">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Marisa Loach,  <b>Clustering 3K PBMCs with Seurat (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html">https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{single-cell-scrna-seurat-pbmc3k,
author = "Marisa Loach",
	title = "Clustering 3K PBMCs with Seurat (Galaxy Training Materials)",
	year = "",
	month = "",
	day = "",
	url = "\url{https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

        


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                
                <blockquote class="agenda follow-up">
                    <div class="box-title">Go Further</div>
                    <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
                    <ul>
                        
    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Importing files from public atlases</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


                    </ul>
                </blockquote>
                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: seurat_clustering
  owner: iuc
  revisions: 51eb02d9b17a
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: seurat_create
  owner: iuc
  revisions: af8bdeb89691
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: seurat_data
  owner: iuc
  revisions: 4ec52a83ec82
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: seurat_plot
  owner: iuc
  revisions: fee4b4eb0ef5
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: seurat_preprocessing
  owner: iuc
  revisions: 6bccf5f85f92
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: seurat_reduce_dimension
  owner: iuc
  revisions: 46ed93e8da30
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
		   </table>
		   </p>
		
        
        
            <p>No feedback has been recieved yet for this training. Be the first one by filling in the feedback form from <a href="#gtn-feedback">above</a>. </p>
        
		</blockquote>




		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/84fb1485f5a2b0e4f71d6539516636907c91b67f">84fb148</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>