<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Multi-sample batch correction with Harmony and SnapATAC2 / Multi-sample batch correction with Harmony and SnapATAC2 / Single Cell</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@8ab532ae844a7d4d5ab89f0c20772e0036c67efc",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-batch-correction-snapatac2/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Multi-sample batch correction with Harmony and SnapATAC2">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2025-02-28 07:41:57 +0000">
<meta name="DC.creator" content="Timon Schlegel"><meta name="description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). ">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Single Cell / Multi-sample batch correction with Harmony and SnapATAC2 / Hands-on: Multi-sample batch correction with Harmony and SnapATAC2">
        <meta property="og:description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). ">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/single-cell/tutorials/scatac-batch-correction-snapatac2/tutorial.png">
	<script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": [
    {
      "@type": "CreativeWork",
      "name": "Galaxy Training: A Powerful Framework for Teaching!",
      "url": "https://doi.org/10.1371/journal.pcbi.1010752"
    },
    {
      "@type": "CreativeWork",
      "name": "Community-Driven Data Analysis Training for Biology",
      "url": "https://doi.org/10.1016/j.cels.2018.05.012"
    }
  ],
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Multi-sample batch correction with Harmony and SnapATAC2",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [],
  "creativeWorkStatus": "Active",
  "dateModified": "2025-02-28 07:41:57 +0000",
  "datePublished": "2025-02-28 07:41:57 +0000",
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "funder": [],
  "funding": [],
  "identifier": "https://gxy.io/GTN:",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Single Cell",
    "description": "Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!).\n",
    "url": "https://training.galaxyproject.org/training-material/topics/single-cell/"
  },
  "abstract": "<!-- This is a comment. -->",
  "learningResourceType": "e-learning",
  "name": "Multi-sample batch correction with Harmony and SnapATAC2",
  "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-batch-correction-snapatac2/tutorial.html",
  "version": 1,
  "timeRequired": "PT4H",
  "teaches": "- Perform batch correction on a dataset collection of multiple single cell ATAC-seq samples.\n- Learn how Harmony and other batch correction algorithms remove batch effects.",
  "keywords": [
    "Single Cell",
    "10x",
    "epigenetics"
  ],
  "description": "## Abstract\n\n<!-- This is a comment. -->\n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - Why is batch correction important during the analysis of data from multiple samples?\n - How is batch correction performed on single cell ATAC-seq data?\n\n\n## Learning Objectives\n\n- Perform batch correction on a dataset collection of multiple single cell ATAC-seq samples.\n- Learn how Harmony and other batch correction algorithms remove batch effects.\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html",
      "name": "Pre-processing of 10X Single-Cell ATAC-seq Datasets",
      "description": "Hands-on for 'Pre-processing of 10X Single-Cell ATAC-seq Datasets' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html",
      "name": "Single-cell ATAC-seq standard processing with SnapATAC2",
      "description": "Hands-on for 'Single-cell ATAC-seq standard processing with SnapATAC2' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/timonschlegel/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/timonschlegel/",
      "name": "Timon Schlegel",
      "image": "https://avatars.githubusercontent.com/timonschlegel",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0009-0001-3228-105X",
      "orcid": "https://orcid.org/0009-0001-3228-105X"
    }
  ],
  "contributor": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/dianichj/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/dianichj/",
      "name": "Diana Chiang Jurado",
      "image": "https://avatars.githubusercontent.com/dianichj",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-5857-1477",
      "orcid": "https://orcid.org/0000-0002-5857-1477"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/shiltemann/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/shiltemann/",
      "name": "Saskia Hiltemann",
      "image": "https://avatars.githubusercontent.com/shiltemann",
      "description": "Researcher at Erasmus Medical Center",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/nfdi4plants/",
          "name": "DataPLANT (NFDI4Plants)",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://nfdi4plants.org"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-3803-468X",
      "orcid": "https://orcid.org/0000-0003-3803-468X"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/timonschlegel/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/timonschlegel/",
      "name": "Timon Schlegel",
      "image": "https://avatars.githubusercontent.com/timonschlegel",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0009-0001-3228-105X",
      "orcid": "https://orcid.org/0009-0001-3228-105X"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Single Cell",
      "description": "Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!).\n",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/"
    }
  ],
  "educationalLevel": "Intermediate",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-batch-correction-snapatac2/workflows/",
      "name": "Associated Workflows"
    },
    {
      "@type": "Thing",
      "url": "https://bio.tools/tool/compute_sequence_length",
      "name": "compute_sequence_length"
    },
    {
      "@type": "Thing",
      "url": "https://bio.tools/tool/galaxy",
      "name": "Galaxy"
    },
    {
      "@type": "Thing",
      "url": "https://zenodo.org/records/12683310",
      "name": "Associated Training Datasets"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/single-cell" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Single Cell
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/single-cell/faqs/" title="Check our FAQs for the Single Cell topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scatac-batch-correction-snapatac2/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/single-cell/tutorials/scatac-batch-correction-snapatac2/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-single-cell">
    <h1 data-toc-skip>Multi-sample batch correction with Harmony and SnapATAC2</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/timonschlegel/" class="contributor-badge contributor-timonschlegel"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/timonschlegel?s=36" alt="Timon Schlegel avatar" width="36" class="avatar" />
    Timon Schlegel</a>
		</td>
	</tr>
	

	

	

	

	

	
	<tr>
		<td><abbr title="These people edited the text, either for spelling and grammar, flow, GTN-fit, or other similar editing categories">Editor(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/dianichj/" class="contributor-badge contributor-dianichj"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/dianichj?s=36" alt="Diana Chiang Jurado avatar" width="36" class="avatar" />
    Diana Chiang Jurado</a></td>
	</tr>
	

	

	
	<tr class="reviewers">
		<td><abbr title="These people reviewed this material for accuracy and correctness">Reviewers</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/shiltemann/" class="contributor-badge contributor-badge-small contributor-shiltemann"><img src="https://avatars.githubusercontent.com/shiltemann?s=36" alt="Saskia Hiltemann avatar" width="36" class="avatar" /></a><a href="/training-material/hall-of-fame/timonschlegel/" class="contributor-badge contributor-badge-small contributor-timonschlegel"><img src="https://avatars.githubusercontent.com/timonschlegel?s=36" alt="Timon Schlegel avatar" width="36" class="avatar" /></a><a href="/training-material/hall-of-fame/dianichj/" class="contributor-badge contributor-badge-small contributor-dianichj"><img src="https://avatars.githubusercontent.com/dianichj?s=36" alt="Diana Chiang Jurado avatar" width="36" class="avatar" /></a></td>
	</tr>
	

	

	

</table>


	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Why is batch correction important during the analysis of data from multiple samples?</p>
</li>
        
        <li><p>How is batch correction performed on single cell ATAC-seq data?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Perform batch correction on a dataset collection of multiple single cell ATAC-seq samples.</p>
</li>
        
        <li><p>Learn how Harmony and other batch correction algorithms remove batch effects.</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    


        
    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Pre-processing of 10X Single-Cell ATAC-seq Datasets</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Single-cell ATAC-seq standard processing with SnapATAC2</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 4 hours</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Intermediate <span class="visually-hidden">Intermediate</span>
<span class="level intermediate" title="Intermediate Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/records/12683310">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/single-cell/tutorials/scatac-batch-correction-snapatac2/workflows/" title="Multi-sample batch correction with Harmony and SnapATAC2 workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/single-cell/tutorials/scatac-batch-correction-snapatac2/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            












            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main) <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
    
    

    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Feb 28, 2025 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Feb 28, 2025 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        

	
	

	
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 1 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <!-- This is a comment. -->
<p>Performing experiments in replicates is a cornerstone of modern biological science. However, when integrating data from multiple single-cell sequencing experiments, technical confounders might impact the results.
To reduce technical confounders, such as different experimenters, experimental protocols, sequencing lanes or sequencing technologies, batch correction is often beneficial.</p>

<p>In this tutorial, we will perform batch correction on five datasets of Single-cell Assay for Transposase-Accessible Chromatin using sequencing (scATAC-seq) data with the three algorithms <em>Harmony</em> (<span class="citation"><a href="#Korsunsky2019">Korsunsky <i>et al.</i> 2019</a></span>), <em>Scanorama</em> (<span class="citation"><a href="#Hie2019">Hie <i>et al.</i> 2019</a></span>) and the <em>mutual nearest neighbor-based</em> (<span class="citation"><a href="#Haghverdi2018">Haghverdi <i>et al.</i> 2018</a></span>) algorithm <em>MNC-correct</em> (<span class="citation"><a href="#Zhang2024">Zhang <i>et al.</i> 2024</a></span>). The <abbr title="Single-cell Assay for Transposase-Accessible Chromatin using sequencing">scATAC-seq</abbr> analysis will be performed with the tool suite <a href="https://kzhang.org/SnapATAC2/version/2.5/index.html"><strong>SnapATAC2</strong></a> (<span class="citation"><a href="#Zhang2024">Zhang <i>et al.</i> 2024</a></span>).</p>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-use-our-single-cell-omics-lab"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-use-our-single-cell-omics-lab" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Use our Single Cell Omics Lab</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Did you know we have a unique Single Cell Omics Lab with all our single cell tools highlighted to make it easier to use on Galaxy? We recommend this site for all your single cell analysis needs, particularly for newer users.</p>   <p>The Single Cell Omics Lab is a different view of the underlying Galaxy server that organises tools and resources better for single-cell users! It also provides a platform for communities to engage and connect; distribute more targeted news and events; and highlight community-specific funding sources.</p>   <p>Try it out!</p>   <ul>   <li><i class="fa fa-tv" aria-hidden="true"></i><span class="visually-hidden">subdomain</span> <a href="https://singlecell.usegalaxy.eu">Europe: Single Cell Omics Lab</a></li>   <li><i class="fa fa-tv" aria-hidden="true"></i><span class="visually-hidden">subdomain</span> <a href="https://singlecell.usegalaxy.org">USA: Single Cell Omics Lab</a></li>   <li><i class="fa fa-tv" aria-hidden="true"></i><span class="visually-hidden">subdomain</span> <a href="https://singlecell.usegalaxy.org.au">Australia: Single Cell Omics Lab</a></li> </ul> </blockquote>
<p><!--END_SNIPPET--></p>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-using-tutorial-mode"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-tutorial-mode" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using tutorial mode</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool. This may NOT be the same tool used in the tutorial you are accessing. Furthermore, if you use a newer tool in one step, and try using an older tool in the next step… this may fail! To ensure you use the same tool versions of a given tutorial, use the <strong>Tutorial mode</strong> feature.</p>   <ul>   <li>Open your Galaxy server</li>   <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>   <li>Navigate to your tutorial</li>   <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>   <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>   <li>You can click anywhere in the grey-ed out area outside of the tutorial box to return back to the Galaxy analytical interface</li> </ul>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-not-all-browsers-work"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Not all browsers work!</div>   <ul>     <li>We’ve had some issues with Tutorial mode on Safari for Mac users.</li>     <li>Try a <strong>different browser</strong> if you aren’t seeing the button.</li>   </ul> </blockquote> </blockquote>
<p><!--END_SNIPPET--></p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

  <p>This tutorial is significantly based on the <a href="https://kzhang.org/SnapATAC2/version/2.5/tutorials/integration.html">“Multi-sample Pipeline” tutorial</a> from <strong>SnapATAC2</strong>.</p>

  <p>The data analysis is performed with the same tools shown in the tutorial <a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">Single-cell ATAC-seq standard processing with SnapATAC2</a>.
That tutorial also explains the steps of the ATAC-seq analysis with SnapATAC2 in more detail.
We recommend completing that tutorial before continuing with batch correction using SnapATAC2.</p>

</blockquote>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#data" id="markdown-toc-data">Data</a>    <ol>
      <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
    </ol>
  </li>
  <li><a href="#snapatac2-preprocessing-and-filtering" id="markdown-toc-snapatac2-preprocessing-and-filtering">SnapATAC2 preprocessing and filtering</a>    <ol>
      <li><a href="#filtering-the-count-matrices" id="markdown-toc-filtering-the-count-matrices">Filtering the count matrices</a></li>
    </ol>
  </li>
  <li><a href="#concatenate-the-collection" id="markdown-toc-concatenate-the-collection">Concatenate the Collection</a>    <ol>
      <li><a href="#extracting-datasets-from-a-collection" id="markdown-toc-extracting-datasets-from-a-collection">Extracting datasets from a collection</a></li>
      <li><a href="#concatenating-all-anndata-files" id="markdown-toc-concatenating-all-anndata-files">Concatenating all AnnData files</a></li>
    </ol>
  </li>
  <li><a href="#dimension-reduction" id="markdown-toc-dimension-reduction">Dimension Reduction</a>    <ol>
      <li><a href="#control-without-batch-correction" id="markdown-toc-control-without-batch-correction">Control Without Batch Correction</a></li>
    </ol>
  </li>
  <li><a href="#batch-correction" id="markdown-toc-batch-correction">Batch correction</a></li>
  <li><a href="#clustering-of-the-batch-corrected-samples" id="markdown-toc-clustering-of-the-batch-corrected-samples">Clustering of the batch corrected samples</a></li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="data">Data</h1>

<p>In this tutorial we will analyze colon samples from multiple donors, provided by the <a href="https://kzhang.org/SnapATAC2/version/2.5/tutorials/integration.html">SnapATAC2 documentation</a>. The <code class="language-plaintext highlighter-rouge">chrom_sizes</code> file and the <code class="language-plaintext highlighter-rouge">gene_annotation</code> file are identical to the previous tutorial <a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">Single-cell ATAC-seq standard processing with SnapATAC2</a>.
The five <code class="language-plaintext highlighter-rouge">colon_multisample</code> files have been generated by the <strong>Cell Ranger ATAC 2.0.0</strong> pipeline from 10X to generate a <a href="https://support.10xgenomics.com/single-cell-atac/software/pipelines/latest/output/fragments"><em>Fragments File</em></a>.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-chromosome-sizes"><button class="gtn-boxify-button details" type="button" aria-controls="details-chromosome-sizes" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Chromosome sizes </span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <ul>
    <li>A chromosome sizes file can be generated using the tool <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fasta_compute_length/fasta_compute_length/1.0.3" title="Compute sequence length tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Compute sequence length</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.3)</span>.</li>
    <li>The reference genome can either be selected from cached genomes or uploaded to the galaxy history.</li>
  </ul>

</blockquote>

<p>First, we will import the datasets into Galaxy. Next, we will organize all <code class="language-plaintext highlighter-rouge">colon_multisample</code> datasets into a collection. Using collections helps streamline the workflow, making the subsequent analysis steps more efficient.</p>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-datasets-versus-collections"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-datasets-versus-collections" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Datasets versus collections</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>Datasets versus collections</strong></p>   <p>In Galaxy’s history datasets can be present as individual entries or they can be combined into <em>Collections</em>. Why do we need collections? Collections combine multiple individual datasets into a single entity which is easy to manage. Galaxy tools can use collections directly as inputs. Collection can be <strong>simple</strong> or <strong>nested</strong>.</p>   <p><strong>Simple collections</strong></p>   <p>Imagine that you’ve uploaded a hundred FASTQ files corresponding to a hundred samples. These will appear as a hundred individual datasets in your history making it very long. But the chances are that when you analyze these data you will do the same thing on each dataset.</p>   <p>To simplify this process you can combine all hundred datasets into a single entity called a <em>dataset collection</em> (or simply a <em>collection</em> or a <em>list</em>). It will appear as a single box in your history making it much easier to understand. Galaxy tools are designed to take collections as inputs. So, for example, if you want to map each of these datasets against a reference genome using, say, <span><strong>Minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>, you will need to provide <code class="language-plaintext highlighter-rouge">minmap2</code> with just one input, the collection, and it will automatically start 100 jobs behind the scenes and will combine all outputs into a single collection containing BAM files.</p>   <p><img src="/training-material/shared/images/simple_collection.svg" alt="A simple collection is a container containing individual datasets" /></p>   <p>There is a number of situations when simple collections are not sufficient to reflect the complexity of the data. To deal with this situation Galaxy allows for <strong>nested</strong> collections.</p>   <p><strong>Nested collections</strong></p>   <p>Probably the most common example of this is paired end data when each sample is represented by two files: one containing forward reads and another containing reverse reads. In Galaxy you can create <strong>nested</strong> collection that reflects the hierarchy of the data. In the case of paired data Galaxy supports <strong>paired</strong> collections.</p>   <p><img src="/training-material/shared/images/paired_collection.svg" alt="A paired collection is a container containing individual datasets and preserving their hierarchy" /></p>   <!-- Original editable image for simple collections = https://docs.google.com/drawings/d/1A-tRerNLzC4FJfShUFT327wMvSSX4Y8AAdaxD_Fwaa0/edit?usp=sharing -->   <!-- Original editable image for paired collection = https://docs.google.com/drawings/d/1Bbx4UmIYdDAqK3KSm6LtLQ8zwEXxDglHmVbb0MK-mbQ/edit?usp=sharing --> </blockquote>
<p><!--END_SNIPPET--></p>

<h2 id="get-data">Get data</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands On:  Data Upload</div>

  <ol>
    <li>Create a new history for this tutorial</li>
    <li>
      <p>Import the files from <a href="https://zenodo.org/records/12683310">Zenodo</a> or from
the shared data library</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/records/12683310/files/colon_multisample_01.gz
https://zenodo.org/records/12683310/files/colon_multisample_02.gz
https://zenodo.org/records/12683310/files/colon_multisample_03.gz
https://zenodo.org/records/12683310/files/colon_multisample_04.gz
https://zenodo.org/records/12683310/files/colon_multisample_05.gz
https://zenodo.org/records/12683310/files/chrom_sizes.txt
https://zenodo.org/records/12683310/files/gencode.v46.annotation.gtf.gz
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from a data library</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Libraries</strong> (left panel)</li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>

      <blockquote class="warning">
        <div class="box-title warning-title" id="warning-large-file-sizes"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Large file sizes!</div>
        <ul>
          <li>The <code class="language-plaintext highlighter-rouge">colon_multisample</code> datasets are quite large. The entire tutorial requires approximately 50 GB of storage.</li>
          <li>To reduce storage you can change the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>Upload Configuration</strong> to <em>Defer dataset resolution</em> for the upload of the <code class="language-plaintext highlighter-rouge">colon_multisample</code> datasets.</li>
          <li>This will delete the datasets after the first analysis step and helps in reducing storage.</li>
          <li>
            <p>You can also permanently delete datasets, which are no longer required.</p>

            <!--SNIPPET-->
            <blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-delete-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-delete-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to delete datasets?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>Deleting datasets individually</strong></p>   <p>To delete datasets individually simply click the <i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span> button with dataset’s box. That’s it! This action is reversible: datasets can be undeleted.</p>   <p><strong>Deleting datasets in bulk</strong></p>   <p>To delete multiple datasets at once:</p>   <ul>   <li>Click <i class="fas fa-check-square" aria-hidden="true"></i><span class="visually-hidden">history-select-multiple</span> icon at the top of the history pane;</li>   <li>Select datasets you want to delete;</li>   <li>Click the dropdown that would appear at the top of the history;</li>   <li>Select <strong>“Delete”</strong> option.</li> </ul>   <p>This action is also reversible: datasets can be undeleted.</p>   <p><img src="/training-material/shared/images/datasets_deleting.gif" alt="An animated gif showing how to delete datasets" /></p>   <p><strong>Deleting datasets permanently</strong> <font color="red"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">warning</span> <b>Danger zone!</b></font></p>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-permanent-is-permanent"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Permanent is ... PERMANENT!</div>   <p>Datasets deleted in this fashion CANNOT be undeleted!</p> </blockquote>   <p>To delete multiple datasets <font color="red">PERMANENTLY</font>:</p>   <ul>   <li>Click <i class="fas fa-check-square" aria-hidden="true"></i><span class="visually-hidden">history-select-multiple</span> icon at the top of the history pane;</li>   <li>Select datasets you want to delete;</li>   <li>Click the dropdown that would appear at the top of the history;</li>   <li>Select <strong>“Delete (permanently)”</strong> option.</li> </ul> </blockquote>
            <p><!--END_SNIPPET--></p>
          </li>
        </ul>

      </blockquote>
    </li>
    <li>Rename the datasets
      <ul>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>Rename</strong> the file <code class="language-plaintext highlighter-rouge">gencode.v46.annotation.gtf.gz</code> to <code class="language-plaintext highlighter-rouge">gene_annotation.gtf.gz</code></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Check that the datatypes of the <code class="language-plaintext highlighter-rouge">colon_multisample</code> files are set to <code class="language-plaintext highlighter-rouge">bed</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select <code class="language-plaintext highlighter-rouge">bed</code> from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Create a dataset collection with all <code class="language-plaintext highlighter-rouge">colon_multisample</code> datasets and rename the collection to <code class="language-plaintext highlighter-rouge">Colon Multisample Fragments</code>.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-dataset-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-dataset-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a dataset collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> <strong>Select Items</strong> at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Select Items button" /></li>   <li>Check all the datasets in your history you would like to include</li>   <li>     <p>Click <strong>n of N selected</strong> and choose <strong>Build Dataset List</strong></p>     <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>   </li>   <li>Enter a name for your collection  to Colon Multisample Fragments</li>   <li>Click <strong>Create collection</strong> to build your collection</li>   <li>Click on the checkmark icon at the top of your history again</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<h1 id="snapatac2-preprocessing-and-filtering">SnapATAC2 preprocessing and filtering</h1>

<p>With our data imported and the collection assembled, we can now begin the <abbr title="Single-cell Assay for Transposase-Accessible Chromatin using sequencing">scATAC-seq</abbr> data preprocessing with SnapATAC2.</p>

<p>The first step is to import the datasets into an AnnData object using the <em>pp.import_data</em> tool. Next, the <code class="language-plaintext highlighter-rouge">transcription start site enrichment (TSSe)</code> will be calculated, serving as a <code class="language-plaintext highlighter-rouge">quality control (QC)</code> metric to selectively filter droplets containing high-quality cells.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-anndata-format"><button class="gtn-boxify-button details" type="button" aria-controls="details-anndata-format" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: AnnData format </span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <ul>
    <li>The <a href="https://anndata.readthedocs.io/en/latest/"><strong>AnnData</strong></a> format was initially developed for the <a href="https://scanpy.readthedocs.io/en/stable/index.html"><strong>Scanpy</strong></a> package and is now a widely accepted data format to store annotated data matrices in a space-efficient manner.</li>
  </ul>

  <figure id="figure-1" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scatac-standard-snapatac2/anndata_schema.svg" alt="Anndata format. " width="200" height="195" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scatac-standard-snapatac2/anndata_schema.svg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> AnnData format stores a count matrix `X` together with annotations of observations (i.e., cells) `obs`, variables (i.e., genes) `var`, and unstructured annotations `uns`.</figcaption></figure>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-preprocessing-and-qc"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands On:  Preprocessing and QC</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Import data fragment files and compute basic QC metrics, using 'pp.import_data'</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Fragment file, optionally compressed with gzip or zstd”</em>: <code class="language-plaintext highlighter-rouge">Colon Multisample Fragments</code> (Input dataset collection)</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“A tabular file containing chromosome names and sizes”</em>: <code class="language-plaintext highlighter-rouge">chrom_sizes.txt</code> (Input dataset)</li>
            <li><em>“Number of unique fragments threshold used to filter cells”</em>: <code class="language-plaintext highlighter-rouge">1000</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Compute the TSS enrichment score (TSSe) for each cell, using 'metrics.tsse'</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Colon Multisample AnnDatas</code> (dataset collection output of <strong>pp.import_data</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GTF/GFF file containing the gene annotation”</em>: <code class="language-plaintext highlighter-rouge">gene_annotation</code> (Input dataset)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> Rename the generated collection to <code class="language-plaintext highlighter-rouge">Colon Multisample AnnDatas TSSe</code>.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_plotting/snapatac2_plotting/2.6.4+galaxy1" title="SnapATAC2 Plotting tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Plotting</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for plotting”</em>: <code class="language-plaintext highlighter-rouge">Plot the TSS enrichment vs. number of fragments density figure, using 'pl.tsse'</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Colon Multisample AnnDatas TSSe</code> (output of <strong>metrics.tsse</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect a few exemplary <code class="language-plaintext highlighter-rouge">.png</code> outputs of the collection</li>
  </ol>

  <figure id="figure-2" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/tsse-colon01-colon02.png" alt="TSSe plots against number of unique fragments. " width="4158" height="1429" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/tsse-colon01-colon02.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Examplary plots of TSSe from <code>colon_multisample_01</code> and <code>colon_multisample_02</code></figcaption></figure>

  <p>High-quality cells can be identified in the plot of <abbr title="transcription start site enrichment">TSSe</abbr> scores against a number of unique fragments for each cell.</p>

  <blockquote class="question">
    <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>Where are high-quality cells typically located on a <abbr title="transcription start site enrichment">TSSe</abbr> plot?</li>
      <li>Based on these plots, how should the filtering threshold be set?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ol>
        <li>The cells in the upper right are high-quality cells, enriched for transcription start sites (TSS). Fragments in the lower left represent low-quality cells or empty droplets and should be filtered out.</li>
        <li>Setting the minimum <abbr title="transcription start site enrichment">TSSe</abbr> to 7.0 will filter out the lowest quality droplets without loosing too much data.</li>
      </ol>

    </blockquote>

  </blockquote>

</blockquote>

<h2 id="filtering-the-count-matrices">Filtering the count matrices</h2>

<p>The <abbr title="transcription start site enrichment">TSSe</abbr> threshold balances data retention and quality control. In our dataset, the <abbr title="transcription start site enrichment">TSSe</abbr> distributions indicate substantial differences in sample quality between batches. To retain as much biological data as possible, we will use a broader filter (e.g., minimum <abbr title="transcription start site enrichment">TSSe</abbr> = 7.0).</p>

<p>Standard <abbr title="transcription start site enrichment">TSSe</abbr> Thresholds</p>
<ul>
  <li><strong>Strict (TSSe ≥ 10–15)</strong>: High-confidence cells.</li>
  <li><strong>Moderate (TSSe ≥ 7–10)</strong>: Balances data retention and quality but may include lower-quality cells.</li>
  <li><strong>Broad (TSSe ≥ 5–7)</strong>: Maximizes retention but increases the risk of including lower-quality data.</li>
</ul>

<p>Choosing the Right Cutoff</p>
<ul>
  <li>If sample quality varies, a <strong>moderate threshold (≥7.0)</strong> helps retain more data.</li>
  <li>If high-quality data is required, <strong>stricter filtering (≥10.0)</strong> is recommended.</li>
  <li>Inspect <abbr title="transcription start site enrichment">TSSe</abbr> distributions to determine the most suitable threshold for your dataset.</li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filtering"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands On:  Filtering</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Filter cell outliers based on counts and numbers of genes expressed, using 'pp.filter_cells'</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Colon Multisample AnnDatas TSSe</code> (output of <strong>metrics.tsse</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Minimum TSS enrichemnt score required for a cell to pass filtering”</em>: <code class="language-plaintext highlighter-rouge">7.0</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Generate cell by bin count matrix, using 'pp.add_tile_matrix'</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Colon Multisample AnnDatas filtered</code> (output of <strong>pp.filter_cells</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“The size of consecutive genomic regions used to record the counts”</em>: <code class="language-plaintext highlighter-rouge">5000</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Perform feature selection, using 'pp.select_features'</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Colon Multisample AnnDatas tile_matrix</code> (output of <strong>pp.add_tile_matrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Number of features to keep”</em>: <code class="language-plaintext highlighter-rouge">50000</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="details">
        <div class="box-title details-title" id="details-bin-size-and-features"><button class="gtn-boxify-button details" type="button" aria-controls="details-bin-size-and-features" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details:  Bin size and features </span><span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ul>
          <li><em>pp.add_tile_matrix</em> divides the genome into a specified number of bins, depending on the bin size (f.ex. 5000bp). For each bin, the ATAC-seq reads of individual cells are checked to determine if a read is located in the bin. This is counted as a feature and stored under <code class="language-plaintext highlighter-rouge">n_vars</code> in the AnnData object.
            <ul>
              <li>Increasing the bin size greatly reduces compute time at the cost of some biological data.</li>
              <li>For this reason, the bin size 5000bp has been selected for the colon datasets. In the <a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html#feature-selection">previous tutorial</a>, the lower bin size of 500bp was chosen, since fewer cells were analyzed.</li>
            </ul>
          </li>
          <li><em>pp.select_features</em> uses the previously identified features to select the most accessible features for further analysis.
            <ul>
              <li>The parameter <em>“Number of features to keep”</em> determines the upper limit of features which can be selected.</li>
              <li>Similarly to the <em>bin_size</em>, the <em>Number of Features to keep</em> can also impact downstream clustering. This was demonstrated in the <a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html#feature-selection">previous tutorial</a>.</li>
            </ul>
          </li>
        </ul>
      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Compute probability of being a doublet using the scrublet algorithm, using 'pp.scrublet'</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Colon Multisample AnnDatas features</code> (output of <strong>pp.select_features</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Remove doublets according to the doublet probability or doublet score, using 'pp.filter_doublets'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Colon Multisample AnnDatas scrublet</code> (output of <strong>pp.scrublet</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="concatenate-the-collection">Concatenate the Collection</h1>
<p>Before we can continue with the analysis and batch correction, we need to extract the datasets out of the collection and merge them into a single AnnData object.</p>

<h2 id="extracting-datasets-from-a-collection">Extracting datasets from a collection</h2>
<p>This is achieved in two steps:</p>
<ol>
  <li>The first dataset is extracted from the collection using <em>Extract dataset</em></li>
  <li>Afterwards, the first dataset will be removed from the collection. This is done by extracting element identifiers and filtering the collection with the name of the first dataset.</li>
</ol>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment:</div>
  <ul>
    <li>It is also possible to manually remove a dataset from a collection.</li>
    <li>However, manually removing datasets can not be implemented in Galaxy workflows. In contrast the automatic method, shown here, can be represented in a workflow and is therefore much better scalable.</li>
  </ul>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-extract-datasets"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands On:  Extract datasets</div>

  <ol>
    <li><span class="tool" data-tool="__EXTRACT_DATASET__" title="Extract element from collection tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Extract element from collection</strong></span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input List”</em>: <code class="language-plaintext highlighter-rouge">Colon Multisample AnnDatas filtered_doublets</code> (output of <strong>pp.filter_doublets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“How should a dataset be selected?”</em>: <code class="language-plaintext highlighter-rouge">The first dataset</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2" title="Extract element identifiers tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Extract element identifiers</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: <code class="language-plaintext highlighter-rouge">Colon Multisample AnnDatas filtered_doublets</code> (output of <strong>pp.filter_doublets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li><span class="tool" data-tool="Show beginning1" title="Select first tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Select first</strong></span> with the following parameters:
      <ul>
        <li><em>“Select first”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“from”</em>: <code class="language-plaintext highlighter-rouge">Element identifiers</code> (output of <strong>Extract element identifiers</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li><span class="tool" data-tool="__FILTER_FROM_FILE__" title="Filter collection tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter collection</strong></span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input Collection”</em>: <code class="language-plaintext highlighter-rouge">Colon Multisample AnnDatas filtered_doublets</code> (output of <strong>pp.filter_doublets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“How should the elements to remove be determined?”</em>: <code class="language-plaintext highlighter-rouge">Remove if identifiers are PRESENT in file</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter out identifiers present in”</em>: <code class="language-plaintext highlighter-rouge">select_first</code> (output of <strong>Select first</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> Rename the filtered collection to <code class="language-plaintext highlighter-rouge">Colon Multisample 02-05</code></li>
  </ol>

</blockquote>

<h2 id="concatenating-all-anndata-files">Concatenating all AnnData files</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-concatenate"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands On:  Concatenate</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.3+galaxy0" title="Manipulate AnnData tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Manipulate AnnData</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.10.3+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">colon_multisample_01</code> (output of <strong>Extract dataset</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Function to manipulate the object”</em>: <code class="language-plaintext highlighter-rouge">Concatenate along the observations axis</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Annotated data matrix to add”</em>: <code class="language-plaintext highlighter-rouge">Colon Multisample 02-05</code> (output of <strong>Filter collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Join method”</em>: <code class="language-plaintext highlighter-rouge">Intersection of variables</code></li>
            <li><em>“Key to add the batch annotation to obs”</em>: <code class="language-plaintext highlighter-rouge">batch</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="details">
        <div class="box-title details-title" id="details-issues-with-concatenation"><button class="gtn-boxify-button details" type="button" aria-controls="details-issues-with-concatenation" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Issues with concatenation</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
        <ul>
          <li>The runtime of this operation depends on the dataset size. For <code class="language-plaintext highlighter-rouge">colon_multisample</code> datasets, concatenation may take up to 1 hour.</li>
          <li>
            <p>For larger datasets, the tool’s allocated memory may be insufficient, causing the operation to fail. In that case, you will see the following error message:</p>

            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>             Fatal error: Exit code 137 ()
</code></pre></div>            </div>
          </li>
          <li>
            <p>In this case, report the issue so that administrators can increase the memory limit, allowing the job to complete successfully.</p>

            <!--SNIPPET-->
            <blockquote class="tip">   <div class="box-title tip-title" id="tip-what-information-should-i-include-when-reporting-a-problem"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-information-should-i-include-when-reporting-a-problem" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What information should I include when reporting a problem?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Writing bug reports is a good skill to have as bioinformaticians, and a key point is that you should include enough information from the first message to help the process of resolving your issue more efficient and a better experience for everyone.</p>   <p><strong>What to include</strong></p>   <ol>   <li>Which commands did you run, precisely, we want details. Which flags did you set?</li>   <li>Which server(s) did you run those commands on?</li>   <li>What account/username did you use?</li>   <li>Where did it go wrong?</li>   <li>What were the stdout/stderr of the tool that failed? Include the text.</li>   <li>Did you try any workarounds? What results did those produce?</li>   <li>(If relevant) screenshot(s) that show exactly the problem, if it cannot be described in text. Is there a details panel you could include too?</li>   <li>If there are job IDs, please include them as text so administrators don’t have to manually transcribe the job ID in your picture.</li> </ol>   <p>It makes the process of answering ‘bug reports’ much smoother for us, as we will have to ask you these questions anyway. If you provide this information from the start, we can get straight to answering your question!</p>   <p><strong>What does a GOOD bug report look like?</strong></p>   <p>The people who provide support for Galaxy are largely volunteers in this community, so try and provide as much information up front to avoid wasting their time:</p>   <blockquote class="quote">   <p>I encountered an issue: I was working on (this server&gt; and trying to run (tool)+(version number) but all of the output files were empty. My username is jane-doe.</p>   <p>Here is everything that I know:</p>   <ul>     <li>The dataset is green, the job did not fail</li>     <li>This is the standard output/error of the tool that I found in the information page (insert it here)</li>     <li>I have read it but I do not understand what X/Y means.</li>     <li>The job ID from the output information page is 123123abdef.</li>     <li>I tried re-running the job and changing parameter Z but it did not change the result.</li>   </ul>   <p>Could you help me?</p> </blockquote> </blockquote>
            <p><!--END_SNIPPET--></p>
          </li>
        </ul>

      </blockquote>
    </li>
    <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> Rename the AnnData output to <code class="language-plaintext highlighter-rouge">Multisample AnnData</code></li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the general information of <code class="language-plaintext highlighter-rouge">Multisample AnnData</code></p>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-inspecting-anndata-objects"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-inspecting-anndata-objects" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Inspecting AnnData objects</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
        <ul>
          <li>Many toolsets producing outputs in <em>AnnData</em> formats in Galaxy, provide the general information by default:
            <ul>
              <li>Click on the name of the dataset in the history to expand it. The general Anndata information will be given in the expanded box.</li>
              <li>Alternatively, expand the dataset and click on <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span><em>Dataset Details</em>. Scroll to Job Information and inspect the Tool Standard Output.</li>
            </ul>
          </li>
          <li>If a tool does not provide general AnnData information or a more specific query is needed, you can use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.3+galaxy0" title="Inspect AnnData tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Inspect AnnData</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.10.3+galaxy0)</span>.</li>
        </ul>
      </blockquote>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>AnnData object with n_obs × n_vars = 34372 × 606219
 obs: 'n_fragment', 'frac_dup', 'frac_mito', 'tsse', 'doublet_probability', 'doublet_score', 'batch'
 var: 'count-0', 'selected-0', 'count-1', 'selected-1', 'count-2', 'selected-2', 'count-3', 'selected-3', 'count-4', 'selected-4'
 obsm: 'fragment_paired'
</code></pre></div>      </div>
    </li>
  </ol>

  <blockquote class="question">
    <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>How many colon cells are present in this AnnData object?</li>
      <li>What does the ‘batch’ annotation indicate?</li>
      <li>What do the ‘count-‘ and ‘selected-‘ annotations represent?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ol>
        <li>There are 34,372 cells.</li>
        <li>The ‘batch’ annotation labels cells according to their sample number (0–4). This annotation will later be used by batch correction algorithms to generate clusters from multiple samples.</li>
        <li>‘count’ and ‘selected’ are variable annotations representing detected and selected features. The sample number (0–4) indicates the dataset that generated these annotations.</li>
      </ol>

    </blockquote>
  </blockquote>

</blockquote>

<h1 id="dimension-reduction">Dimension Reduction</h1>
<p>Now that all samples have been concatenated into a single AnnData object, the most accessible features of the combined count matrix must be selected. The previously chosen features were sample-specific and cannot be used for dimensionality reduction. Therefore, the most accessible features will be reselected, followed by dimensionality reduction using <em>matrix-free spectral embedding</em>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-spectral-embedding"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands On:  Spectral Embedding</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Perform feature selection, using 'pp.select_features'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Multisample AnnData</code> (output of <strong>Manipulate AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Number of features to keep”</em>: <code class="language-plaintext highlighter-rouge">50000</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1" title="SnapATAC2 Clustering tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Clustering</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Dimension reduction and clustering”</em>: <code class="language-plaintext highlighter-rouge">Perform dimensionality reduction using Laplacian Eigenmap ('tl.spectral')</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Multisample AnnData features</code> (output of <strong>pp.select_features</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Distance metric”</em>: <code class="language-plaintext highlighter-rouge">cosine</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> Rename the output AnnData object to <code class="language-plaintext highlighter-rouge">Multisample AnnData spectral</code></li>
  </ol>

</blockquote>

<h2 id="control-without-batch-correction">Control Without Batch Correction</h2>
<p>Batch effects can be visualized using a <code class="language-plaintext highlighter-rouge">Uniform Manifold Approximation and Projection (UMAP)</code> projection, where samples are colored by their batch annotation (<code class="language-plaintext highlighter-rouge">obs: 'batch'</code>).</p>
<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-umap-projection-without-batch-correction"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands On:  UMAP Projection Without Batch Correction</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1" title="SnapATAC2 Clustering tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Clustering</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Dimension reduction and Clustering”</em>: <code class="language-plaintext highlighter-rouge">Compute UMAP using 'tl.umap'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Multisample AnnData spectral</code> (output of <strong>tl.spectral</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> Rename the output AnnData object to <code class="language-plaintext highlighter-rouge">Multisample AnnData UMAP</code>.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_plotting/snapatac2_plotting/2.6.4+galaxy1" title="SnapATAC2 Plotting tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Plotting</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for plotting”</em>: <code class="language-plaintext highlighter-rouge">Plot the UMAP embedding using 'pl.umap'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Multisample AnnData UMAP</code> (output of <strong>tl.umap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Color”</em>: <code class="language-plaintext highlighter-rouge">batch</code></li>
            <li><em>“Height of the plot”</em>: <code class="language-plaintext highlighter-rouge">500</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> Rename the generated image to <code class="language-plaintext highlighter-rouge">spectral-UMAP-No Batch Correction</code>.</p>
    </li>
    <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the <code class="language-plaintext highlighter-rouge">.png</code> output.</li>
  </ol>

  <p><a href="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/batch-umap-no_harmony-spectral.png" rel="noopener noreferrer"><img src="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/batch-umap-no_harmony-spectral.png" alt="UMAP plot before batch correction. " width="600" height="500" loading="lazy" /></a></p>

  <blockquote class="question">
    <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>How do batch effects appear in this plot?</li>
      <li>What would a plot without batch effects look like?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ol>
        <li>Batch effects appear as distinct clusters where samples are unevenly distributed. For example, in the upper right corner, each batch forms a separate group. These differences are likely technical artifacts rather than true biological variation.</li>
        <li>A plot without batch effects would show cells from all samples evenly mixed, with no distinct clusters corresponding to individual batches.</li>
      </ol>

    </blockquote>
  </blockquote>

</blockquote>

<h1 id="batch-correction">Batch correction</h1>
<p>After confirming the presence of batch effects, they should be removed using correction algorithms. <strong>SnapATAC2</strong> provides three batch correction methods: <em>Harmony</em>, <em>MNC-correct</em>, and <em>Scanorama</em>. To determine the most suitable algorithm for your dataset, compare the outputs of each method.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-batch-correction-algorithms"><button class="gtn-boxify-button details" type="button" aria-controls="details-batch-correction-algorithms" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details:  Batch correction algorithms </span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <ul>
    <li>Batch correction algorithms adjust the cell-by-feature count matrix to account for batch-specific differences between samples.</li>
    <li>Batch effects can arise from various technical sources, such as sequencing lanes, plates, protocols, and handling. Additionally, biological factors, including tissue types, species, and inter-individual variation, can also contribute to batch effects <span class="citation"><a href="#Luecken2021">Luecken <i>et al.</i> 2021</a></span>.</li>
    <li>Many batch correction algorithms have been developed, with most scATAC-seq methods adapted from scRNA-seq batch removal techniques.</li>
    <li>
      <p><em>Harmony</em> <span class="citation"><a href="#Korsunsky2019">Korsunsky <i>et al.</i> 2019</a></span> is a principal component analysis (PCA)-based method that leverages lower-dimensional data to assign cells to new clusters. It prioritizes multi-sample clustering to integrate datasets by computing linear correction factors for each batch and cluster, iteratively adjusting cell positions until optimal batch correction is achieved.</p>

      <p><a href="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/harmony-graphical-abstract.png" rel="noopener noreferrer"><img src="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/harmony-graphical-abstract.png" alt="Graphical abstract of Harmony batch correction. " width="2736" height="984" loading="lazy" /></a></p>
    </li>
    <li>
      <p><em>Scanorama</em> <span class="citation"><a href="#Hie2019">Hie <i>et al.</i> 2019</a></span> performs panorama stiching, to find and merge overlapping cell types.</p>

      <p><a href="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/scanorama-graphical-abstract.png" rel="noopener noreferrer"><img src="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/scanorama-graphical-abstract.png" alt="Graphical abstract of Scanorama batch correction. " width="2736" height="1316" loading="lazy" /></a></p>
    </li>
    <li>
      <p><em>MNC-correct</em> <span class="citation"><a href="#Zhang2024">Zhang <i>et al.</i> 2024</a></span> is a modified version of the <em>mutual nearest neighbor</em> algorithm <span class="citation"><a href="#Haghverdi2018">Haghverdi <i>et al.</i> 2018</a></span>. It calculates centroids for batch-specific clusters and identifies pairs of mutual nearest centroids (MNC) across batches. Correction vectors then align the batches in the same plane. Additionally, <em>MNC-correct</em> can be run iteratively to refine corrections for optimal alignment.</p>

      <p><a href="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/mnn-graphical-abstract.png" rel="noopener noreferrer"><img src="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/mnn-graphical-abstract.png" alt="Graphical abstract of MNC-correct batch correction. " width="946" height="1243" loading="lazy" /></a></p>
    </li>
  </ul>

</blockquote>

<p>We will first use <em>Harmony</em> to correct batch effects, followed by testing the other methods.</p>
<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-batch-correction-and-visualization"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands On:  Batch correction and visualization</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Use harmonypy to integrate different experiments,using 'pp.harmony'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Multisample AnnData UMAP</code> (output of <strong>tl.umap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1" title="SnapATAC2 Clustering tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Clustering</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Dimension reduction and Clustering”</em>: <code class="language-plaintext highlighter-rouge">Compute Umap, using 'tl.umap'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Multisample AnnData harmony</code> (output of <strong>pp.harmony</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Use the indicated representation in <code class="language-plaintext highlighter-rouge">.obsm</code>“</em>: <code class="language-plaintext highlighter-rouge">X_spectral_harmony</code></li>
            <li><em>“<code class="language-plaintext highlighter-rouge">adata.obs</code> key under which t add cluster labels”</em>: <code class="language-plaintext highlighter-rouge">umap_harmony</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> Rename the AnnData output to <code class="language-plaintext highlighter-rouge">Multisample AnnData harmony UMAP</code></p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-key-for-cluster-labels"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment:  Key for Cluster Labels</div>
        <ul>
          <li>Adding the new <em>UMAP embeddings</em> under the key <code class="language-plaintext highlighter-rouge">umap_harmony</code> preserves the non-batch-corrected embeddings in the AnnData object.
            <ul>
              <li>Alternatively, leaving this parameter empty will overwrite the existing embeddings.</li>
            </ul>
          </li>
        </ul>
      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_plotting/snapatac2_plotting/2.6.4+galaxy1" title="SnapATAC2 Plotting tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Plotting</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for plotting”</em>: <code class="language-plaintext highlighter-rouge">Plot the UMAP embedding, using 'pl.umap'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Multisample AnnData harmony UMAP</code> (output of <strong>tl.umap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Color”</em>: <code class="language-plaintext highlighter-rouge">batch</code></li>
            <li><em>“Use the indicated representation in .obsm”</em>: <code class="language-plaintext highlighter-rouge">X_umap_harmony</code></li>
            <li><em>“Height of the plot”</em>: <code class="language-plaintext highlighter-rouge">500</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> Rename the generated image to <code class="language-plaintext highlighter-rouge">spectral-UMAP-harmony</code></p>
    </li>
    <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the <code class="language-plaintext highlighter-rouge">.png</code> output</li>
  </ol>

  <p><a href="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/batch-umap-harmony-spectral.png" rel="noopener noreferrer"><img src="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/batch-umap-harmony-spectral.png" alt="UMAP plot of Batch correction with Harmony. " width="600" height="500" loading="lazy" /></a></p>

  <blockquote class="question">
    <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>How did <em>Harmony</em> affect the appearance of this plot?</li>
      <li>Are there areas in the plot where batch effects persist?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ol>
        <li><em>Harmony</em> has successfully removed most batch effects. Notably, the center-left groups have merged into a single larger group containing all batches.</li>
        <li>Yes, some batch effects remain, particularly in the upper-right corner, where batch-specific colors do not fully overlap. However, this could also be due to certain samples having fewer cells, leading to underrepresentation in clusters of rare cell types. Therefore, it is unclear whether <em>Harmony</em> has completely eliminated all batch effects.</li>
      </ol>

    </blockquote>
  </blockquote>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-batch-correction-with-scanorama-and-mnc-correct"><button class="gtn-boxify-button details" type="button" aria-controls="details-batch-correction-with-scanorama-and-mnc-correct" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Batch Correction with Scanorama and MNC-correct</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <ul>
    <li>Other batch correction methods can be applied using <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span>, including <em>pp.mnc_correct</em> and <em>pp.scanorama_integrate</em>.
      <ul>
        <li>For <em>MNC-correct</em>, the number of iterations can be adjusted.</li>
      </ul>
    </li>
    <li>To determine the optimal batch correction method, it is recommended to test different algorithms and parameter settings.</li>
    <li>Example outputs from <em>Scanorama</em> and <em>MNC-correct</em> are shown below:</li>
  </ul>

  <figure id="figure-3" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/mnc-correct-scanorama-umap-spectral.png" alt="Batch correction UMAP plots of MNC-correct with different settings and Scanorama. " width="4269" height="3610" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/mnc-correct-scanorama-umap-spectral.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> UMAP plots of batch correction with different methods. (a) Batch correction with MNC-correct and default settings. (b) Batch correction with Scanorama. (c) Batch correction with MNC-correct and 30 iterations.</figcaption></figure>

  <blockquote class="question">
    <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ul>
      <li>Compare these plots with the output of <em>Harmony</em>. Which algorithm is best-suited for the colon datasets?</li>
    </ul>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ul>
        <li><em>Harmony</em> achieved the best batch integration, as it produced the fewest single-batch groups. <em>Scanorama</em> and <em>MNC-correct</em> (with default settings) did not integrate the batches as effectively. However, <em>MNC-correct</em> with 30 iterations significantly reduced batch effects and could also be used for further analysis.</li>
      </ul>

    </blockquote>
  </blockquote>

</blockquote>

<h1 id="clustering-of-the-batch-corrected-samples">Clustering of the batch corrected samples</h1>
<p>The analysis can now continue using the same methods outlined in the <a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">standard pathway</a>. The batch-corrected embeddings are clustered and visualized using the <em>Leiden</em> algorithm.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-leiden-clustering-and-visualization"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands On:  Leiden clustering and visualization</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1" title="SnapATAC2 Clustering tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Clustering</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Dimension reduction and Clustering”</em>: <code class="language-plaintext highlighter-rouge">Compute a neighborhood graph of observations, using 'pp.knn'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Multisample AnnData harmony UMAP</code> (output of <strong>tl.umap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“The key for the matrix”</em>: <code class="language-plaintext highlighter-rouge">X_spectral_harmony</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-matrix-keys"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-matrix-keys" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Matrix Keys</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ul>
          <li>Each batch correction algorithm stores its corrected matrix under a specific key:
            <ul>
              <li><em>Harmony</em>: <code class="language-plaintext highlighter-rouge">X_spectral_harmony</code></li>
              <li><em>MNC-correct</em>: <code class="language-plaintext highlighter-rouge">X_spectral_mnn</code></li>
              <li><em>Scanorama</em>: <code class="language-plaintext highlighter-rouge">X_spectral_scanorama</code></li>
            </ul>
          </li>
          <li>These keys are stored in the AnnData object under <code class="language-plaintext highlighter-rouge">'obsm'</code>.</li>
        </ul>

      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1" title="SnapATAC2 Clustering tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Clustering</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Dimension reduction and Clustering”</em>: <code class="language-plaintext highlighter-rouge">Cluster cells into subgroups, using 'tl.leiden'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Multisample AnnData harmony knn</code> (output of <strong>pp.knn</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Whether to use the Constant Potts Model (CPM) or modularity”</em>: <code class="language-plaintext highlighter-rouge">modularity</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> Rename the AnnData output to <code class="language-plaintext highlighter-rouge">Multisample AnnData harmony leiden</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_plotting/snapatac2_plotting/2.6.4+galaxy1" title="SnapATAC2 Plotting tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Plotting</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for plotting”</em>: <code class="language-plaintext highlighter-rouge">Plot the UMAP embedding, using 'pl.umap'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">anndata_out</code> (output of <strong>tl.leiden</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Color”</em>: <code class="language-plaintext highlighter-rouge">leiden</code></li>
            <li><em>“Use the indicated representation in .obsm”</em>: <code class="language-plaintext highlighter-rouge">X_umap_harmony</code></li>
            <li><em>“Height of the plot”</em>: <code class="language-plaintext highlighter-rouge">500</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> Rename the generated image to <code class="language-plaintext highlighter-rouge">spectral-UMAP-harmony-leiden</code></p>
    </li>
    <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the <code class="language-plaintext highlighter-rouge">.png</code> output</li>
  </ol>

  <p><a href="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/batch-umap-harmony-leiden.png" rel="noopener noreferrer"><img src="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/batch-umap-harmony-leiden.png" alt="UMAP plot of Batch-corrected leiden clusters. " width="600" height="500" loading="lazy" /></a></p>

</blockquote>

<p>After integrating the datasets and clustering the cells, the scATAC-seq analysis can proceed with downstream analysis. This includes <a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html#cell-cluster-annotation">cell cluster annotation</a> and differential peak analysis.</p>
<blockquote class="comment">
  <div class="box-title comment-title" id="comment-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

  <p>The <strong>SnapATAC2</strong> tools for differential peak analysis are available in Galaxy, but no GTN training materials are currently provided. Until a tutorial is available, you can refer to the <strong>SnapATAC2</strong> documentation for a <a href="https://kzhang.org/SnapATAC2/version/2.6/tutorials/diff.html">tutorial on differential peak analysis</a>.</p>

  <p>The tools are available in Galaxy under <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_peaks_and_motif/snapatac2_peaks_and_motif/2.6.4+galaxy1" title="SnapATAC2 Peaks and Motif tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Peaks and Motif</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span>.</p>

</blockquote>

<h1 id="conclusion">Conclusion</h1>
<p><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Well done, you’ve made it to the end! You might want to consult your results with this <a href="https://usegalaxy.eu/u/timonschlegel/h/multisample-batch-correction-with-harmony-and-snapatac2">control history</a>, or check out the <a href="https://usegalaxy.eu/u/timonschlegel/w/multisample-batch-correction-with-snapatac2-and-harmony">full workflow</a> for this tutorial.</p>

<p>In this tutorial, we integrated five <abbr title="Single-cell Assay for Transposase-Accessible Chromatin using sequencing">scATAC-seq</abbr> colon samples using a scalable Galaxy workflow. We compared different batch integration algorithms to identify the most suitable method for our data. Finally, we clustered the cells to prepare the data for downstream analysis.</p>

<p><a href="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/Batch-correction-pipeline-overview.png" rel="noopener noreferrer"><img src="/training-material/topics/single-cell/images/scatac-batch-correction-snapatac2/Batch-correction-pipeline-overview.png" alt="SnapATAC2 batch correction pipeline. " width="4063" height="1833" loading="lazy" /></a></p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(typeof plausible !== 'undefined'){
                            // Plausible may be undefined (script blocked)
                            // or it may be defined, but opted-out (select box/DNT),
                            // which means `plausible()` will work but not send data, *nor* execute the callback.
                            plausible('TutorialComplete', {props: {path: document.location.pathname}})
                          }
                          // since the callback is completely cosmetic, we'll just issue it optimistically.
                          tutorial_finish_finish();
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Batch correction is important for the integration of data from multiple experiments</p>
</li>
                        
                        <li><p>Batch correction algorithms identify similar cells and move them closer together through appropriate correction vectors.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Have a look at the available FAQ pages and support channels
                <ul>
                
                  <li><a href="faqs/">Tutorial FAQs</a></li>
                
                
                  <li><a href="/training-material/topics/single-cell/faqs/">Topic  FAQs</a></li>
                
                  <li><a href="/training-material/faqs/galaxy/">Galaxy FAQs</a> </li>
                  <li><a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Matrix Chat</a></li>
                  <li><a href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
                </ul>

		<!--
                
		-->

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/single-cell#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Haghverdi2018">Haghverdi, L., A. T. L. Lun, M. D. Morgan, and J. C. Marioni, 2018 <b>Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors</b>. Nature Biotechnology 36: 421–427. <a href="https://doi.org/10.1038/nbt.4091">10.1038/nbt.4091</a></li>
<li id="Hie2019">Hie, B., B. Bryson, and B. Berger, 2019 <b>Efficient integration of heterogeneous single-cell transcriptomes using Scanorama</b>. Nature Biotechnology 37: 685–691. <a href="https://doi.org/10.1038/s41587-019-0113-3">10.1038/s41587-019-0113-3</a></li>
<li id="Korsunsky2019">Korsunsky, I., N. Millard, J. Fan, K. Slowikowski, F. Zhang <i>et al.</i>, 2019 <b>Fast,  sensitive and accurate integration of single-cell data with Harmony</b>. Nature Methods 16: 1289–1296. <a href="https://doi.org/10.1038/s41592-019-0619-0">10.1038/s41592-019-0619-0</a></li>
<li id="Luecken2021">Luecken, M. D., M. Büttner, K. Chaichoompu, A. Danese, M. Interlandi <i>et al.</i>, 2021 <b>Benchmarking atlas-level data integration in single-cell genomics</b>. Nature Methods 19: 41–50. <a href="https://doi.org/10.1038/s41592-021-01336-8">10.1038/s41592-021-01336-8</a></li>
<li id="Zhang2024">Zhang, K., N. R. Zemke, E. J. Armand, and B. Ren, 2024 <b>A fast,  scalable and versatile tool for analysis of single-cell omics data</b>. Nature Methods 21: 217–227. <a href="https://doi.org/10.1038/s41592-023-02139-9">10.1038/s41592-023-02139-9</a></li></ol>
                

                
                <h1 data-toc-skip>Glossary</h1>
                <dl>
                    
                    
                    <dt>QC</dt>
                    <dd>quality control</dd>
                    
                    <dt>TSS</dt>
                    <dd>transcription start sites</dd>
                    
                    <dt>TSSe</dt>
                    <dd>transcription start site enrichment</dd>
                    
                    <dt>UMAP</dt>
                    <dd>Uniform Manifold Approximation and Projection</dd>
                    
                    <dt>scATAC-seq</dt>
                    <dd>Single-cell Assay for Transposase-Accessible Chromatin using sequencing</dd>
                    
                </dl>
                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=single-cell/scatac-batch-correction-snapatac2"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=single-cell/scatac-batch-correction-snapatac2">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Timon Schlegel,  <b>Multi-sample batch correction with Harmony and SnapATAC2 (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-batch-correction-snapatac2/tutorial.html">https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-batch-correction-snapatac2/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{single-cell-scatac-batch-correction-snapatac2,
author = "Timon Schlegel",
	title = "Multi-sample batch correction with Harmony and SnapATAC2 (Galaxy Training Materials)",
	year = "",
	month = "",
	day = "",
	url = "\url{https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-batch-correction-snapatac2/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

        


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/single-cell/tutorials/scatac-batch-correction-snapatac2/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: fasta_compute_length
  owner: devteam
  revisions: 7d37cfda8e00
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: anndata_inspect
  owner: iuc
  revisions: 6f0d0c784f09
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: anndata_manipulate
  owner: iuc
  revisions: ed4996a16f7f
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: collection_element_identifiers
  owner: iuc
  revisions: d3c07d270a50
  tool_panel_section_label: Collection Operations
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snapatac2_clustering
  owner: iuc
  revisions: d3ea0ba3d066
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snapatac2_peaks_and_motif
  owner: iuc
  revisions: 64fa083411f1
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snapatac2_plotting
  owner: iuc
  revisions: 05bd4db20227
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snapatac2_preprocessing
  owner: iuc
  revisions: c5c94e01a1b5
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
		   </table>
		   </p>
		
        
        
            <p>No feedback has been recieved yet for this training. Be the first one by filling in the feedback form. </p>
        
		</blockquote>




		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scatac-batch-correction-snapatac2/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/single-cell/tutorials/scatac-batch-correction-snapatac2/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/8ab532ae844a7d4d5ab89f0c20772e0036c67efc">8ab532a</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>