<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Single Cell Proteomics data analysis with bioconductor-scp / Single Cell Proteomics data analysis with bioconductor-scp / Proteomics</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@c67c7ffc2850bbbb9e9637f431a6810beb66f7de",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/bioconductor-scp/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Single Cell Proteomics data analysis with bioconductor-scp">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2025-02-10 07:36:58 +0000">
<meta name="DC.creator" content="Kristina Gomoryova">
<meta name="DC.creator" content="Helge Hecht"><meta name="description" content="Training material for proteomics workflows in Galaxy">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Proteomics / Single Cell Proteomics data analysis with bioconductor-scp / Hands-on: Single Cell Proteomics data analysis with bioconductor-scp">
        <meta property="og:description" content="Training material for proteomics workflows in Galaxy">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/proteomics/tutorials/bioconductor-scp/tutorial.png">
	<script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": [
    {
      "@type": "CreativeWork",
      "name": "Galaxy Training: A Powerful Framework for Teaching!",
      "url": "https://doi.org/10.1371/journal.pcbi.1010752"
    },
    {
      "@type": "CreativeWork",
      "name": "Community-Driven Data Analysis Training for Biology",
      "url": "https://doi.org/10.1016/j.cels.2018.05.012"
    }
  ],
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Single Cell Proteomics data analysis with bioconductor-scp",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [],
  "creativeWorkStatus": "Active",
  "dateModified": "2025-02-10 07:36:58 +0000",
  "datePublished": "2025-02-07 12:22:04 +0000",
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "funder": [],
  "funding": [],
  "identifier": "https://gxy.io/GTN:",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Proteomics",
    "description": "Training material for proteomics workflows in Galaxy",
    "url": "https://training.galaxyproject.org/training-material/topics/proteomics/"
  },
  "abstract": "Single-cell proteomics (SCP) is a cutting-edge field that focuses on analyzing the protein composition of individual cells. Unlike bulk proteomics, which measures average protein levels across large cell populations, single-cell proteomics captures the unique protein expression of each cell. This approach reveals cellular heterogeneity, uncovers rare cell subpopulations, and provides deeper insights into complex biological processes, disease progression, and treatment responses. Advances in mass spectrometry and data analysis tools have made it possible to study proteins at single-cell resolution, opening new opportunities for discoveries in biomedicine,,.",
  "learningResourceType": "e-learning",
  "name": "Single Cell Proteomics data analysis with bioconductor-scp",
  "url": "https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/bioconductor-scp/tutorial.html",
  "version": 2,
  "timeRequired": "PT3H",
  "teaches": "- To learn about the individual steps in the SCP workflow, and which steps to pay attention to.\n- To learn how to perform quality control checks during the workflow.\n- To analyze a subset of a real biological dataset.",
  "keywords": [
    "Proteomics"
  ],
  "description": "## Abstract\n\nSingle-cell proteomics (SCP) is a cutting-edge field that focuses on analyzing the protein composition of individual cells. Unlike bulk proteomics, which measures average protein levels across large cell populations, single-cell proteomics captures the unique protein expression of each cell. This approach reveals cellular heterogeneity, uncovers rare cell subpopulations, and provides deeper insights into complex biological processes, disease progression, and treatment responses. Advances in mass spectrometry and data analysis tools have made it possible to study proteins at single-cell resolution, opening new opportunities for discoveries in biomedicine,,.\n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - Which biological questions can single cell proteomics (SCP) help find us answers to?\n - Which tools can be used to analyze the SCP data?\n - Which are the keys steps in the analysis of SCP data?\n\n\n## Learning Objectives\n\n- To learn about the individual steps in the SCP workflow, and which steps to pay attention to.\n- To learn how to perform quality control checks during the workflow.\n- To analyze a subset of a real biological dataset.\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/KristinaGomoryova/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/KristinaGomoryova/",
      "name": "Kristina Gomoryova",
      "image": "https://avatars.githubusercontent.com/KristinaGomoryova",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-4407-3917",
      "orcid": "https://orcid.org/0000-0003-4407-3917"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/hechth/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/hechth/",
      "name": "Helge Hecht",
      "image": "https://avatars.githubusercontent.com/hechth",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-6744-996X",
      "orcid": "https://orcid.org/0000-0001-6744-996X"
    }
  ],
  "contributor": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/pavanvidem/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/pavanvidem/",
      "name": "Pavankumar Videm",
      "image": "https://avatars.githubusercontent.com/pavanvidem",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-5192-126X",
      "orcid": "https://orcid.org/0000-0002-5192-126X"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/hechth/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/hechth/",
      "name": "Helge Hecht",
      "image": "https://avatars.githubusercontent.com/hechth",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-6744-996X",
      "orcid": "https://orcid.org/0000-0001-6744-996X"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/KristinaGomoryova/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/KristinaGomoryova/",
      "name": "Kristina Gomoryova",
      "image": "https://avatars.githubusercontent.com/KristinaGomoryova",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-4407-3917",
      "orcid": "https://orcid.org/0000-0003-4407-3917"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Proteomics",
      "description": "Training material for proteomics workflows in Galaxy",
      "url": "https://training.galaxyproject.org/training-material/topics/proteomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0121",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0121",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0121"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://bio.tools/tool/scp",
      "name": "scp"
    },
    {
      "@type": "Thing",
      "url": "https://bio.tools/tool/galaxy",
      "name": "Galaxy"
    },
    {
      "@type": "Thing",
      "url": "https://zenodo.org/records/14650887",
      "name": "Associated Training Datasets"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Proteomics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/proteomics/faqs/" title="Check our FAQs for the Proteomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/proteomics/tutorials/bioconductor-scp/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/proteomics/tutorials/bioconductor-scp/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-proteomics">
    <h1 data-toc-skip>Single Cell Proteomics data analysis with bioconductor-scp</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/KristinaGomoryova/" class="contributor-badge contributor-KristinaGomoryova"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/KristinaGomoryova?s=36" alt="Kristina Gomoryova avatar" width="36" class="avatar" />
    Kristina Gomoryova</a><a href="/training-material/hall-of-fame/hechth/" class="contributor-badge contributor-hechth"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/hechth?s=36" alt="Helge Hecht avatar" width="36" class="avatar" />
    Helge Hecht</a>
		</td>
	</tr>
	

	

	

	

	

	
	<tr>
		<td><abbr title="These people edited the text, either for spelling and grammar, flow, GTN-fit, or other similar editing categories">Editor(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/pavanvidem/" class="contributor-badge contributor-pavanvidem"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/pavanvidem?s=36" alt="Pavankumar Videm avatar" width="36" class="avatar" />
    Pavankumar Videm</a></td>
	</tr>
	

	

	
	<tr class="reviewers">
		<td><abbr title="These people reviewed this material for accuracy and correctness">Reviewers</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/hechth/" class="contributor-badge contributor-badge-small contributor-hechth"><img src="https://avatars.githubusercontent.com/hechth?s=36" alt="Helge Hecht avatar" width="36" class="avatar" /></a><a href="/training-material/hall-of-fame/KristinaGomoryova/" class="contributor-badge contributor-badge-small contributor-KristinaGomoryova"><img src="https://avatars.githubusercontent.com/KristinaGomoryova?s=36" alt="Kristina Gomoryova avatar" width="36" class="avatar" /></a><a href="/training-material/hall-of-fame/pavanvidem/" class="contributor-badge contributor-badge-small contributor-pavanvidem"><img src="https://avatars.githubusercontent.com/pavanvidem?s=36" alt="Pavankumar Videm avatar" width="36" class="avatar" /></a></td>
	</tr>
	

	

	

</table>


	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Which biological questions can single cell proteomics (SCP) help find us answers to?</p>
</li>
        
        <li><p>Which tools can be used to analyze the SCP data?</p>
</li>
        
        <li><p>Which are the keys steps in the analysis of SCP data?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>To learn about the individual steps in the SCP workflow, and which steps to pay attention to.</p>
</li>
        
        <li><p>To learn how to perform quality control checks during the workflow.</p>
</li>
        
        <li><p>To analyze a subset of a real biological dataset.</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    


        

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 3 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/records/14650887">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/proteomics/tutorials/bioconductor-scp/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            












            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
    
    

    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Feb 7, 2025 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Feb 10, 2025 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        

	
	

	
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 2 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>Single-cell proteomics (SCP) is a cutting-edge field that focuses on analyzing the protein composition of individual cells. Unlike bulk proteomics, which measures average protein levels across large cell populations, single-cell proteomics captures the unique protein expression of each cell. This approach reveals cellular heterogeneity, uncovers rare cell subpopulations, and provides deeper insights into complex biological processes, disease progression, and treatment responses. Advances in mass spectrometry and data analysis tools have made it possible to study proteins at single-cell resolution, opening new opportunities for discoveries in biomedicine <span class="citation"><a href="#Lin2024">Lin <i>et al.</i> 2024</a></span>, <span class="citation"><a href="#Gatto2023">Gatto <i>et al.</i> 2023</a></span>, <span class="citation"><a href="#Ahmad2023">Ahmad and Budnik 2023</a></span>.</p>

<p>Due to the complexity and variability of single-cell data, reproducible and standardized analysis pipelines are essential. They ensure consistent, accurate results across studies and help overcome challenges related to data processing, interpretation, and integration. The scp package addresses these needs by providing a robust and well-tested framework for reliable SCP data analysis. It was originally developed in the laboratory of Prof. Laurent Gatto as an <a href="https://github.com/UCLouvain-CBIO/scp">R package</a> and this training covers its Galaxy implementation, the <strong>bioconductor-scp</strong> Galaxy tool.</p>

<p>The scp-bioconductor tool provides the same functionality as the scp R package. However, currently we support only the analysis of MaxQuant output, with the more options to be added in the future. All R functions are packaged into a single tool, meaning that all settings for the whole workflow are pre-defined at the very beginning. We finally opted for this approach due to the usage of complex data structures under the hood, where at the beginning of the workflow the simple text file (MaxQuant output) is converted to a data structure called SingleCellExperiment, which is an <a href="http://adv-r.had.co.nz/S4.html">S4 object</a>. The SingleCellExperiment object then holds all the data manipulation intermediate steps at various levels of aggregation - you can imagine that as a huge list of all data tables which are produced during the data processing - in one object we store everything from the peptide-to-spectrum match (PSM) level, transformed, imputed peptide level to protein level. In order to avoid multiple export and import from the SingleCellExperiment object to a text file, we therefore start with a text file, do all the data structures manipulation internally, and then output plots and resulting tables in a tabular format.</p>

<p>However, we do realize that it is very important to check also intermediate results and adjust the settings accordingly. We encourage the users not to blindly trust the default settings, but indeed perform quality controls checks, and if needed, change the settings and re-run the tool. It is especially important if there was any batch effect present in the data, which has to be removed or at least considered (which is very common in the SCP analyses). Careful and thorough data exploration at main steps of the workflow is crucial for the correct analysis and interpretation of the data.</p>

<p>Of note, all functions within the <strong>bioconductor-scp</strong> tool are sorted ‘chronologically’: we start on the PSM level, where first quality control filtering is done. Next, PSMs are aggregated to the peptide level where normalization and transformation are performed. Finally, peptides are aggregated into proteins and on the protein level another round of normalization, as well as imputation of missing values, batch effect correction and dimensionality reduction are performed.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-psms-peptides-proteins"><button class="gtn-boxify-button details" type="button" aria-controls="details-psms-peptides-proteins" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details:  PSMs - peptides - proteins </span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>The first layer we end up with after database searches (using tools as MaxQuant or DIA-NN) are <strong>peptide-to-spectrum matches (PSMs)</strong>. These represent the association between a particular mass spectrum and a peptide sequence: how well the measured spectrum matches 
the theoretical spectrum. A <strong>peptide</strong> can have multiple PSMs due to the different charge states, fragmentation variations or 
repeated detection. Aggregation of the individual PSMs brings us to the peptide level. A peptide, in comparison to the PSM, which is 
a raw spectral evidence, is already identified sequence after some preliminary validation (e.g. False Discovery Rate (FDR) control). 
Finally, the last layer is the <strong>protein</strong> level, where multiple peptides (can) match one protein.</p>
</blockquote>

<p>The whole tutorial uses only two tools: <span class="tool" data-tool="Grouping1" title="Group - data by a column and perform aggregate operation on other columns. tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Group - data by a column and perform aggregate operation on other columns.</strong></span> tool and <strong>bioconductor-scp</strong> tool <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. It is structured in a way that parameters associated with the particular data processing step are set in the Hands-on section and the background is explained. However, the <strong>bioconductor-scp</strong> tool will successfully run only after setting all the parameters, so please Run it only after setting of the whole form is completed.</p>

<p>In our tutorial, we will analyze the data included in the scp package, similarly as it’s in the scp vignette. The dataset is a subset of a data included in the publication by Specht et al. <span class="citation"><a href="#Specht2021">Specht <i>et al.</i> 2021</a></span> and is publicly available <a href="https://doi.org/10.5281/zenodo.14650887">on Zenodo</a>. Data come from the SCoPE2 method (next generation Single Cell ProtEomics, firstly described in the abovementioned publication), where single cells are firstly isolated, lysed and proteins released into the solvent are then enzymatically digested. Individual peptides are then barcoded with the tandem-mass-tags (TMT) and labelled peptides from multiple single cells are combined into a single mixture and analyzed by LC-MS/MS. As a model system, monocytes from human cell line U-937 differentiated into macrophage-like cells in the presence of an agonist of protein kinase C, phorbol-12-myristate-13-acetate (PMA), were used here. The main question was whether the macrophage population will be as homogenous as the monocyte one they originated from or more heterogenous.</p>

<figure id="figure-1" style="max-width: 90%;"><img src="../../images/bioconductor-scp_SCOPE2.png" alt="SCOPE2_method. " width="2141" height="646" loading="lazy" /><a target="_blank" href="../../images/bioconductor-scp_SCOPE2.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Scheme of the SCoPE2 method. Adapted from the <a href="https://scope2.slavovlab.net/">Slavov lab website</a>.</figcaption></figure>

<p>There are 5 different types of samples in the SCoPE2 data:</p>

<ul>
  <li><em>Carrier channel</em>: contains 200 cell equivalents and its main purpose is to boost the peptide identification rate and reduce the sample loss,</li>
  <li><em>Reference channel</em>: contains 5 cell equivalents and it is used for normalization purposes, to partially correct for the between-run variation,</li>
  <li><em>Unused channel</em>: is empty due to the isobaric cross-contamination by the carrier channel,</li>
  <li><em>Blank channel</em>: serves as a negative control, they are processed identically to the single cell channels, but do not contain any cells,</li>
  <li><em>Single cell channel</em>: contain the single cell samples - in our case Monocytes and Macrophages.</li>
</ul>

<p>Having now background to both the SCP method and the data example origin, let’s dive into the analysis itself!</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#data-preparation-and-filtering" id="markdown-toc-data-preparation-and-filtering">Data preparation and filtering</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
      <li><a href="#dataset-exploration" id="markdown-toc-dataset-exploration">Dataset exploration</a></li>
      <li><a href="#data-loading" id="markdown-toc-data-loading">Data loading</a></li>
      <li><a href="#psms-filtering" id="markdown-toc-psms-filtering">PSMs filtering</a></li>
    </ol>
  </li>
  <li><a href="#peptide-level-operations" id="markdown-toc-peptide-level-operations">Peptide-level operations</a>    <ol>
      <li><a href="#aggregation-to-peptides" id="markdown-toc-aggregation-to-peptides">Aggregation to peptides</a></li>
      <li><a href="#peptide-filtering" id="markdown-toc-peptide-filtering">Peptide filtering</a></li>
      <li><a href="#peptide-normalization" id="markdown-toc-peptide-normalization">Peptide normalization</a></li>
      <li><a href="#log-transformation" id="markdown-toc-log-transformation">Log transformation</a></li>
      <li><a href="#high-percentage-missing-values-peptides-removal" id="markdown-toc-high-percentage-missing-values-peptides-removal">High percentage missing values peptides removal</a></li>
    </ol>
  </li>
  <li><a href="#protein-level-operations" id="markdown-toc-protein-level-operations">Protein-level operations</a>    <ol>
      <li><a href="#aggregation-to-proteins" id="markdown-toc-aggregation-to-proteins">Aggregation to proteins</a></li>
      <li><a href="#protein-normalization" id="markdown-toc-protein-normalization">Protein normalization</a></li>
      <li><a href="#missing-values-imputation" id="markdown-toc-missing-values-imputation">Missing values imputation</a></li>
      <li><a href="#batch-effect-correction" id="markdown-toc-batch-effect-correction">Batch effect correction</a></li>
      <li><a href="#dimensionality-reduction" id="markdown-toc-dimensionality-reduction">Dimensionality reduction</a></li>
      <li><a href="#data-export" id="markdown-toc-data-export">Data export</a></li>
    </ol>
  </li>
  <li><a href="#summary-tool-setting" id="markdown-toc-summary-tool-setting">Summary tool setting</a></li>
</ol>

</blockquote>

<h1 id="data-preparation-and-filtering">Data preparation and filtering</h1>

<p>There are two files required on the input:</p>

<ul>
  <li><strong>Evidence.txt file</strong>. Evidence file is one of the outputs from MaxQuant, containing all the information about identified PSMs. In general, we don’t need the whole file, which often contains tens of columns, but only a minimal set is required:
    <ul>
      <li><em>Modified.sequence</em>: provides the peptide identifiers</li>
      <li><em>Leading.razor.protein</em>: provides the protein identifiers</li>
      <li><em>runCol/Raw.file identifier</em>: specifies the MS acquisition runs or files (has to be present also in the second required file, SampleAnnotation)</li>
      <li><em>PIF</em>: provides spectral purity</li>
      <li><em>PEP or dart_PEP</em>: provides the peptide posterior error probabilities</li>
      <li><em>Reporter.intensity columns</em>: contain quantification values. Any other intensity-based quantification columns are allowed.</li>
      <li><em>Reverse</em>: marks the reverse peptides from the decoy database (labeled with +)</li>
      <li><em>Potential.contaminant</em>: marks the contaminant peptides (labeled with +)</li>
    </ul>
  </li>
  <li><strong>sampleAnnotation.txt file</strong>. SampleAnnotation file serves as a metadata information about the experiment itself and it’s up to the user to create it. It should contain all important information about e.g. the batch allocation, sample types, etc. Required columns are:
    <ul>
      <li><em>runCol/Raw.file identifier</em>: specifies the MS acquisition runs or files and corresponds to identically labelled column in the Evidence file.</li>
      <li><em>quantCols</em>: specifies the quantification data column names (e.g. Reporter.intensity.1)</li>
      <li><em>SampleType</em>: provides the type of sample that is acquired (carrier, reference, single cell types - e.g. Monocyte, Macrophage, etc.). It is needed only in case of multiplexed experiments.</li>
      <li>If relevant, also <em>batch specification data</em>: batch, sorting day, digestion type, etc.</li>
    </ul>
  </li>
</ul>

<p>As mentioned, we will use the subset of the data from <span class="citation"><a href="#Specht2021">Specht <i>et al.</i> 2021</a></span>. Both example files, evidences.txt and sampleAnnotation.txt, can be imported from Zenodo.</p>

<h2 id="data-upload">Data upload</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-upload-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Upload data</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a reasonable name.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To create a new history simply click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel:</p>   <p><img src="/training-material/shared/images/history_create_new.svg" alt="UI for creating new history" /></p>   <!-- the original drawing can be found here https://docs.google.com/drawings/d/1cCBrLAo4kDGic5QyB70rRiWJAKTenTU8STsKDaLcVU8/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Import the files from <a href="https://zenodo.org/records/14650887">Zenodo</a> and set the data type to tabular:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/records/14650887/files/evidences.txt
https://zenodo.org/records/14650887/files/sampleAnnotation.txt
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Set the data type to tabular for both evidence and sampleAnnotation files.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select <code class="language-plaintext highlighter-rouge">tabular</code> from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>
</blockquote>

<h2 id="dataset-exploration">Dataset exploration</h2>

<p>Before we start the analysis itself, let’s quickly explore the dataset we have. The ‘evidences.txt’ has 1362 rows and 150 columns. The ‘sampleAnnotation.txt’ file has 65 rows and 7 columns. We can see that the <code class="language-plaintext highlighter-rouge">runCol</code> column from the sampleAnnotation file holds the names of runs in our experiment. If we inspect it more, there are 4 unique values: 
190222S_LCA9_X_FP94BM, 190321S_LCA10_X_FP97AG, 190914S_LCB3_X_16plex_Set_21 and 190321S_LCA10_X_FP97_blank_01. In the <code class="language-plaintext highlighter-rouge">quantCols</code> column, we can see that there are always 16 reporter intensities for each run. <code class="language-plaintext highlighter-rouge">SampleType</code> column contains the sample characterization: there are single-cell channels Macrophage and Monocyte; Blank; Carrier and Unused channels. Finally, there are columns related to the lcbatch, sorting day and digestion.</p>

<p>What might be important to know is how many features (PSMs) we have quantified per run. To have at least a rough estimate, we can use the <span class="tool" data-tool="Grouping1" title="Group - data by a column and perform aggregate operation on other columns. tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Group - data by a column and perform aggregate operation on other columns.</strong></span> tool in Galaxy: set the <code class="language-plaintext highlighter-rouge">Select data</code> to ‘evidences.txt’ and <code class="language-plaintext highlighter-rouge">Group by column</code> to ‘19’, which corresponds to the ‘Raw.file’ column. In the <code class="language-plaintext highlighter-rouge">Operation</code>, set then <code class="language-plaintext highlighter-rouge">Type</code> to ‘Count’ and <code class="language-plaintext highlighter-rouge">On column</code> to ‘Column: 1’. Now we can see that the 190321S_LCA10_X_FP97_blank_01 run has approximately three times less features than other runs, so it might be beneficial to filter it once we perform the analysis itself.</p>

<h2 id="data-loading">Data loading</h2>

<p>Find the <strong>bioconductor-scp</strong> tool <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. Upload the datasets using the Upload Data button. Select the evidence and sample annotation tables and set the data type as tabular.</p>

<p>Once the datasets are uploaded (green in the right menu), set the <code class="language-plaintext highlighter-rouge">Input evidence table</code> to ‘evidences.txt’ and <code class="language-plaintext highlighter-rouge">Sample annotations table</code> to ‘sampleAnnotations.txt’. Important is to set the correct run identifier column, in our case <code class="language-plaintext highlighter-rouge">Raw.file</code> - this column has to be present in both sampleAnnotation and evidences files and fields have to match.</p>

<p><code class="language-plaintext highlighter-rouge">Remove empty columns</code> setting concerns the columns which contain only missing values. They typically occur if samples acquired with different multiplexing labels are merged into a single table. We suggest to set it to ‘Yes’. QC plots are generated in multiple places in the workflow and help to decide whether to change any current settings or estimate whether a batch effect is present in the data. As they can be very informative, we suggest to set the option to generate them to ‘Yes’.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-loading"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data loading</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/bioconductor_scp/bioconductor_scp/1.16.0+galaxy0" title="bioconductor-scp tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bioconductor-scp</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.0+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Input evidence table”</em>: <code class="language-plaintext highlighter-rouge">evidences.txt</code></li>
        <li><em>“Sample annotations table”</em>: <code class="language-plaintext highlighter-rouge">sampleAnnotation.txt</code></li>
        <li><em>“Which column specifies the run identifier and batch name?”</em>: <code class="language-plaintext highlighter-rouge">c19: Raw.file</code></li>
        <li><em>“Remove empty columns”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Generate QC plots?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-do-not-run-the-bioconductor-scp-tool-yet"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Do not run the bioconductor-scp tool yet!</div>
  <p>We will learn and set the remaining tool parameters in the next steps of the tutorial. Once we set all the parameters, we will run the tool.</p>
</blockquote>

<h2 id="psms-filtering">PSMs filtering</h2>

<p>At the level of PSMs we will perform the first round of quality control filtering. Firstly, we will remove the contaminant proteins, which are denoted by the <code class="language-plaintext highlighter-rouge">Reverse</code> and <code class="language-plaintext highlighter-rouge">Potential.contaminant</code> columns. Next we set the threshold for the <code class="language-plaintext highlighter-rouge">Parent ion fraction (PIF)</code>: PIF indicates the fraction the target peak makes up of the total intensity in the inclusion window, so it is an indicator of the purity of a spectrum.</p>

<p>We will also remove the samples which contain way too little PSMs. From the dataset exploration, we already know that the 190321S_LCA10_X_FP97_blank_01 run has approximately three times less features than other runs - although keep in mind, that these are counts for the unfiltered data, after the contaminants removal there will be actually even less features. Therefore, we would remove this one with some reasonable threshold - in our case, we can use e.g. 150. This means that if a run has less than 150 features, it will be removed.</p>

<p>Additionally, we will compute the <strong>sample to carrier ratio (SCR)</strong>, which is a metric computed as division of the reporter intensity in the single-cell sample by the reporter ion intensity in the carrier channel within the same batch. To compute this, we need to know what are the single cell channels (in our case the Monocyte and Macrophage) and also how many cells were present in the carrier channel (in our case there were 200 cells). We expect, that the carrier intensities will be much higher than the single cell intensities and also we do expect that that ratio between the single cells and the carrier will be 1/200 which is 0.005. After running this workflow, a QC figure titled “QC_plot_SCR” or “QC_plot_SCR_col” will appear within the Plots collection. We can see the histogram of the MeanSCR values and we can see that opposite to the expected ratio 0.005 the peak is around 0.01. There are however few outlier PSMs which have much higher ratios and should be filtered out above 0.1. Therefore we will set the threshold for the filtering to 0.1.</p>

<p>Finally, we will apply the filtering based on the <strong>q-values</strong>. In the evidences.txt table, there is already a posterior error probability (PEP) calculated. However, PEP is in general considered more stringent than the q-values as PEP measures the direct probability of an error in a single PSM, whereas q-value measures the error across a collection of PSMs, informing us whether the global false discovery rate is acceptable. Therefore, scp package contains the pep2qvalue function, where we just need to specify whether we want to compute the q-values on the PSM, peptide or protein level. We want to be quite stringent, so we will control the FDR at 0.01 meaning we approve that 1% of the PSMs will be identified incorrectly and 99% correctly.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-filtering"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data filtering</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/bioconductor_scp/bioconductor_scp/1.16.0+galaxy0" title="bioconductor-scp tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bioconductor-scp</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.0+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Filter reverse sequence”</em>: <code class="language-plaintext highlighter-rouge">True</code></li>
        <li><em>“Filter potential contaminant”</em>: <code class="language-plaintext highlighter-rouge">True</code></li>
        <li><em>“Parent ion fraction”</em>: <code class="language-plaintext highlighter-rouge">0.8</code></li>
        <li><em>“Minimum number of PSMs per assay required”</em>: <code class="language-plaintext highlighter-rouge">150</code></li>
        <li><em>“Number of cells in the carrier channel”</em>: <code class="language-plaintext highlighter-rouge">200</code></li>
        <li><em>“Single-cell channels present in the experiment”</em>: <code class="language-plaintext highlighter-rouge">Monocyte</code> and <code class="language-plaintext highlighter-rouge">Macrophage</code></li>
        <li><em>“Mean SCR threshold”</em>: <code class="language-plaintext highlighter-rouge">0.1</code></li>
        <li><em>“Filter based on PSM, peptide or protein-level q-values”</em>: <code class="language-plaintext highlighter-rouge">protein</code></li>
        <li><em>“q-value threshold”</em>: <code class="language-plaintext highlighter-rouge">0.01</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-scope2-data-specific-settings"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: SCoPE2 data-specific settings</div>
  <p>If you are analysing the SCoPE2 data, set the <code class="language-plaintext highlighter-rouge">Compute relative reporter ion intensities</code> to ‘True’. These are
computed in a way that intensities in the single cell channels are divided by the reference channel.</p>
</blockquote>

<p>Having the data cleaned, we can now follow with the aggregation of PSMs to the peptide level and peptide-level data processing.</p>

<h1 id="peptide-level-operations">Peptide-level operations</h1>

<p>In general, we don’t want to stay on the PSM level, but want to get to the peptide or further to the protein level, where we can follow with the biological analysis. Therefore, in the next step we will aggregate the PSMs to the peptides.</p>

<h2 id="aggregation-to-peptides">Aggregation to peptides</h2>

<p>By default, the aggregation will be done on the <strong>Modified.sequence</strong> column (found in the evidences table), which contains the peptide sequence along with its possible posttranslational modifications. All PSMs sharing the same modified.sequence will be aggregated by the function we choose - here we will use the colMedians, meaning that the aggregation will be done sample-wise by computing the column medians.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-aggregation-to-the-peptide-level"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Aggregation to the peptide level</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/bioconductor_scp/bioconductor_scp/1.16.0+galaxy0" title="bioconductor-scp tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bioconductor-scp</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.0+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Which function to use for the aggregation?”</em>: <code class="language-plaintext highlighter-rouge">Aggregate using the median of each sample: colMedians</code></li>
        <li><em>“Which column should be used for the PSM to peptide aggregation?”</em>: <code class="language-plaintext highlighter-rouge">c6: Modified.sequence</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="peptide-filtering">Peptide filtering</h2>

<p>Let’s apply further filtering criteria to remove the low-quality cells. At this point, we can already filter out the Carrier and Reference channels as they won’t be used in any more calculations. We will however keep the single cell channels, Monocyte and Macrophage, and also the Blank which is used for the median coefficient of variation (MedianCV) calculation.</p>

<p>In this tutorial, we will use only filtering for the MedianCV, but it is possible to also filter on the median intensity (where we expect that the Blank sample will contain only background signal and the single channels will be of higher intensity). MedianCV value assesses the consistency of peptide quantification within each protein. Therefore, we would exclude the proteins with high MedianCV. We can set here how many observed peptides per protein are there required for the computation of MedianCV. To decide which threshold to use, we can take advantage of the QC plot which will be produced once the whole workflow ran. We can use the Blank sample where we expect no reliable quantifications to serve as a baseline noise level.</p>

<p>After the filtering, we can remove the blank sample if we haven’t done it already (in case you would omit the filtering for the median intensity or medianCV step).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-peptide-filtering"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Peptide filtering</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/bioconductor_scp/bioconductor_scp/1.16.0+galaxy0" title="bioconductor-scp tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bioconductor-scp</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.0+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Which samples to keep?”</em>: <code class="language-plaintext highlighter-rouge">Monocyte</code>, <code class="language-plaintext highlighter-rouge">Macrophage</code> and <code class="language-plaintext highlighter-rouge">Blank</code></li>
        <li><em>“Filter based on median intensity?”</em>: <code class="language-plaintext highlighter-rouge">no</code></li>
        <li><em>“Filter based on median CV?”</em>: <code class="language-plaintext highlighter-rouge">yes</code></li>
        <li><em>“How many peptides per protein are required to compute CV?”</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
        <li><em>“Median CV threshold”</em>: <code class="language-plaintext highlighter-rouge">0.65</code></li>
        <li><em>“Remove blank sample?”</em>: <code class="language-plaintext highlighter-rouge">yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="peptide-normalization">Peptide normalization</h2>

<p>The next step is normalization, which helps us to remove the technical bias and variation from the samples, but still preserves the biological variation. In contrast to the common untargeted proteomics workflow, in the SCP experiment we perform dual normalization, firstly on columns (meaning samples) and second on rows (meaning features, so peptides). We provide two methods of normalization, either on means or medians, with the preferred options set as default.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-peptide-normalization"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Peptide normalization</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/bioconductor_scp/bioconductor_scp/1.16.0+galaxy0" title="bioconductor-scp tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bioconductor-scp</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.0+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Normalization method columns”</em>: <code class="language-plaintext highlighter-rouge">colMedians</code></li>
        <li><em>“Normalization method rows”</em>: <code class="language-plaintext highlighter-rouge">rowMeans</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="log-transformation">Log transformation</h2>

<p>Due to the non-normal distribution which is common for the proteomics data, we usually follow with the log transformation and base of 2. There are certain reasons why base 2 is preferred over base 10, as better interpretation of the fold changes (where a log2FC = 1 means a 2-fold increase, log2FC = -1 means 2-fold decrease, and such symmetric interpretation makes it intuitive to compare the upregulation and downregulation), variance stabilization and handling data skewness (log2 transformation helps to reduce skewness and compresses large values while still preserving relative differences). After the transformation, peptide intensities shall follow roughly the normal distribution and will be therefore suitable for a statistical test.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-log-transformation-of-intensities"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Log transformation of intensities</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/bioconductor_scp/bioconductor_scp/1.16.0+galaxy0" title="bioconductor-scp tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bioconductor-scp</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.0+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Log transformation base”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="high-percentage-missing-values-peptides-removal">High percentage missing values peptides removal</h2>

<p>Finally, we want to get rid of the peptides which do have a high missing rate in order to keep only the most reliable identifications. We set the percent of missing values, meaning that peptides exhibiting a higher number of missing values will be removed from the analysis. Here, the pNA is set to 99, meaning peptides containing more than 99% of missing data will be excluded.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-na-values-removal"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: NA values removal</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/bioconductor_scp/bioconductor_scp/1.16.0+galaxy0" title="bioconductor-scp tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bioconductor-scp</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.0+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Remove peptides with high missing rate?”</em>: <code class="language-plaintext highlighter-rouge">yes</code></li>
        <li><em>”% of NA values filtering threshold”</em>: <code class="language-plaintext highlighter-rouge">99</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>This leaves us with a reliable list of peptide identifications. The next step is aggregation to the protein level.</p>

<h1 id="protein-level-operations">Protein-level operations</h1>

<h2 id="aggregation-to-proteins">Aggregation to proteins</h2>

<p>For the aggregation of peptides to the protein level, we will use again the colMedians function, and as a column specifying the protein identifiers, we will use <strong>Leading.razor.protein</strong>. The peptides which share the same Leading razor protein ID will be aggregated together and based on the function chosen, in our case the colMedians (meaning computation of sample-wise medians).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-aggregation-to-the-protein-level"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Aggregation to the protein level</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/bioconductor_scp/bioconductor_scp/1.16.0+galaxy0" title="bioconductor-scp tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bioconductor-scp</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.0+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Which function to use for the aggregation?”</em>: <code class="language-plaintext highlighter-rouge">Aggregate using the median of each sample: colMedians</code></li>
        <li><em>“Which column should be used for the peptide to protein aggregation?”</em>: <code class="language-plaintext highlighter-rouge">c17: Leading.razor.protein</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="protein-normalization">Protein normalization</h2>

<p>Next, we will also normalize the intensities on the protein level - again firstly in the column-wise manner and then in the row-wise manner. For the columns, we will use the colMedians and for the rows the rowMeans.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-protein-normalization"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Protein normalization</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/bioconductor_scp/bioconductor_scp/1.16.0+galaxy0" title="bioconductor-scp tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bioconductor-scp</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.0+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Normalization method columns”</em>: <code class="language-plaintext highlighter-rouge">colMedians</code></li>
        <li><em>“Normalization method rows”</em>: <code class="language-plaintext highlighter-rouge">rowMeans</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="missing-values-imputation">Missing values imputation</h2>

<p>Finally, we have to deal with the missing values as their presence is problematic for some multivariate analyses such as Principal Component Analysis (PCA) and leads to reduced statistical power as incomplete data limits the ability to detect significant differences between experimental groups. The most common way how to approach them is by the process called <strong>missing values imputation</strong>. By the imputation, we can replace the missing values with the artificial counterparts. Missing values can be of two major types: missing completely at random (MCAR) and missing not at random (MNAR) and they should be treated separately. In the SCP data, the MCAR type is more prevalent, and for MCAR data k-Nearest Neighbors (kNN) approach is often employed. The kNN approach will replace the missing values based on the weighted average of k-most similar proteins - it assumes that the missingness is due to a stochastic factor (e.g. instrumentation problem, but the protein is actually present). For the kNN, we have to choose the number of k-nearest neighbors beforehand, which we can set to 3.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-missing-values-imputation"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Missing values imputation</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/bioconductor_scp/bioconductor_scp/1.16.0+galaxy0" title="bioconductor-scp tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bioconductor-scp</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.0+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Which k to use for the kNN imputation”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-k-nearest-neighbors"><button class="gtn-boxify-button details" type="button" aria-controls="details-k-nearest-neighbors" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details:  k-Nearest Neighbors </span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>kNN stands for the k-Nearest Neighbors, which is a supervised machine learning method often used as a classification
tool. It is a non-parametric algorithm meaning that is has no underlying assumptions about the data distribution. In
proteomics, it is very often used to impute the missing completely at random (MCAR) data, which are of a stochastic
nature and missing values occur across whole range of intensities and are not biased towards the low-intensity
peptides/proteins. The first step is to compute the similarity between the protein with the missing values and all
other proteins without missing values - common similarity (=distance) measures include Euclidean distance, Pearson,
Manhattan or Minkowski. Then the k most similar proteins based on the distance metric are selected - usually it is
recommended to use odd numbers as k. Finally, missing values are replaced by weighted average of the corresponding
values from the k-nearest neighbors. The choice of k is crucial for correct handling of the missing values in our
data, however there is no easy way how to choose the ‘best’ k. Generally accepted is to set the k as the square root
of the number of samples in the dataset.</p>
</blockquote>

<p>After the imputation, we end up with a full numeric matrix without any missing values. The next step is to check for the batch effect as it is unfortunately very common for the SCP experiments.</p>

<h2 id="batch-effect-correction">Batch effect correction</h2>

<p>Batch effects are unavoidable in SCP experiments as only a few multiplexed single cells can be acquired at once. However, having an untreated batch effect could compromise statistical analysis and lead to incorrect biological interpretation. There are two main approaches how to deal with it: either we can include the batch as a covariate in a statistical model and model it, or we can estimate it and remove it using tools as <strong>ComBat</strong>.</p>

<p>In the bioconductor-scp, there are two batch removal methods implemented: already mentioned <strong>ComBat</strong> from the sva R package and <strong>removeBatchEffect</strong> method from limma R package. In principle, both methods perform well for the SCP data: ComBat is preferred if the sample size is small (&lt;10) <span class="citation"><a href="#Goh2017">Goh <i>et al.</i> 2017</a></span>, whereas removeBatchEffect is preferred for large sample sizes, but in general it is up to the users preferences or previous experience which method they will choose. A good strategy would be also to run both methods and then check the PCA and Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP) plots or inspect the batch corrected data. In this tutorial, we will use ComBat. The only thing we need to select is the batch variable to be corrected (e.g. day when the samples were run, sorting day, digestion, … - in general column in the sampleAnnotation table) - here we will use the RunCol.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-batch-effect-correction"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Batch effect correction</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/bioconductor_scp/bioconductor_scp/1.16.0+galaxy0" title="bioconductor-scp tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bioconductor-scp</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.0+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Which batch correction method to use?”</em>: <code class="language-plaintext highlighter-rouge">ComBat</code></li>
        <li><em>“Which column is the technical variable to be corrected?”</em>: <code class="language-plaintext highlighter-rouge">c2: runCol</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-combat"><button class="gtn-boxify-button details" type="button" aria-controls="details-combat" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details:  ComBat </span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>ComBat method (Combatting batch effects when combining batches of microarray data) was originally developed for the
microarray data, but quickly found an application also in the proteomics domain. It uses an empirical Bayes method
to adjust for the systematic known batch effects. It removes the batch effects which impact both means and variances
of each gene within a processing batch. ComBat performs well also for the imbalanced data,
however, it requires a full data matrix on the input - meaning a strict pre-filtering or imputation of missing
values is required prior running it. Also, importantly, to run ComBat we need to know the batch covariate - i.e.
which variable is responsible for the batch effect. The output is then a batch-corrected data matrix. <span class="citation"><a href="#Phua2022">Phua <i>et al.</i> 2022</a></span>, <span class="citation"><a href="#Zhang2018">Zhang <i>et al.</i> 2018</a></span></p>
</blockquote>

<h2 id="dimensionality-reduction">Dimensionality reduction</h2>

<p>The next step is the dimensionality reduction. We will firstly use the PCA to reduce the dimensions, followed by the UMAP. Because PCA cannot deal with the missing values, we will use the imputed and batch-corrected data. We will select the number of components to obtain [is there any optimal number to choose?] and we can also choose according to which variable the PCA plot will be colored by. Usually, we expect that the first few components will account for the most of the variability (the long accepted approach was to choose the number of components retained would be equal to the number of variables being analyzed <span class="citation"><a href="#number-pca-components">O’Brien</a></span>).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-batch-effect-correction-pca"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Batch effect correction: PCA</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/bioconductor_scp/bioconductor_scp/1.16.0+galaxy0" title="bioconductor-scp tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bioconductor-scp</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.0+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Run principal component analysis (PCA)?”</em>: <code class="language-plaintext highlighter-rouge">yes</code></li>
        <li><em>“Number of components”</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
        <li><em>“What column to color the PCA according to?”</em>: <code class="language-plaintext highlighter-rouge">c4: SampleType</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>In contrast to PCA, which is a linear dimension reduction, UMAP is a non-linear approach. Similarly to the PCA, we need to select the number of components to retain.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-batch-effect-correction-umap"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Batch effect correction: UMAP</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/bioconductor_scp/bioconductor_scp/1.16.0+galaxy0" title="bioconductor-scp tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bioconductor-scp</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.0+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Run UMAP on PCA data?”</em>: <code class="language-plaintext highlighter-rouge">yes</code></li>
        <li><em>“Number of components”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-dimensionality-reduction-pca-umap-tsne"><button class="gtn-boxify-button details" type="button" aria-controls="details-dimensionality-reduction-pca-umap-tsne" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details:  Dimensionality reduction: PCA, UMAP, tSNE </span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>Dimensionality reduction (DR) is an important step in the analysis of high dimensional proteomics datasets, where 
thousands of features (proteins; corresponding to variables) are measured across usually fewer samples. The main
purpose is to reduce the noise and identify the most informative features or patterns in our data - to reduce the 
number of dimensions while retaining most of the useful information. There are several methods, which can be used: 
here, we will provide a brief overview of the most used ones, PCA, UMAP and tSNE.</p>

  <p><strong>PCA (Principal Component Analysis)</strong> is a linear method of DR. It creates linear combinations of the original
variables and projects the data onto ortoghonal axes (principal components), which are ordered by the variance
explained (PC1 explaining the most variance, PC2 next highest, etc.). It very well maintains the global structure of
the data. Importantly, data has to be standardized prior running the PCA, meaning that mean is zero and variance is
1 within all variables <span class="citation"><a href="#creative-proteomics">Proteomics</a></span>.</p>

  <p><strong>tSNE (t-Distributed Stochastic Neighbor Embedding)</strong> is an unsupervised non-linear technique modeling the
similarity of the data points in high-dimensional space as probabilities to match these in low-dimensional space.
The non-linearity in this context means that it allows us to separate also data which couldn’t be separated by a
straight line. The more similar objects will be closer together, while the more dissimilar objects will be more
separated. In contrast to PCA, it preservers very well the local structure, but is not so good in preserving the
global structure, meaning the distances between clusters might not be meaningful. Also, tSNE is very sensitive to
parameters tuning, especially perplexity, basically setting how many close neighbors each point has within one
cluster. <span class="citation"><a href="#distill-tSNE">Distill</a></span></p>

  <p><strong>UMAP (Uniform Manifold Approximation and Projection)</strong> is similarly to tSNE a non-linear DR method, which
constructs a high-dimensional graph and optimizes its low-dimensional representation. It can preserve very well both
the global and local structure, meaning that if the two clusters are far apart in the original space, UMAP will try
to maintain the distance also in the low-dimensional space - however, the distance is still not directly
interpretable such as in case of PCA. Compared to tSNE, it also shows a much better performance, however, the
careful fine-tuning of parameters is still crucial in order to correctly interpret the results. <span>ERROR INVALID CITATION &gt; understanding-umap</span>, <span class="citation"><a href="#understanding-umap2">Varma</a></span></p>
</blockquote>

<h2 id="data-export">Data export</h2>

<p>In the end of the workflow, several datasets are exported. By default, a log transformed, normalized, imputed and batch-corrected data on the protein level are provided. Additionally, it is possible to export also intermediate results, meaning peptides, normalized peptides, log transformed peptides, proteins, normalized proteins and imputed proteins. Also, it is possible to export the scp object as an Rdata object, which can be further loaded into the R environment using the R load() function, and the Rscript to reproduce the analysis outside of Galaxy. Both plots and intermediate results will be provided as collections.</p>

<p>In our tutorial, we will get all the QC plots (because we set the <code class="language-plaintext highlighter-rouge">Generate QC plots?</code> to Yes). Let’s a bit describe what we can find in the Plots collection:</p>

<ul>
  <li><em>PCA</em>: PCA plots a scatterplot where PC1 is displayed on the x-axis and PC2 on they y-axis. We can see how much variance is explained by these two principal components and whether there are any clusters of e.g. samples based on how similar they are.</li>
  <li><em>QC_boxplot_peptide</em>: Boxplots of the (non-normalized) log2 transformed peptide intensities for the individual <code class="language-plaintext highlighter-rouge">Runs</code> colored based on the <code class="language-plaintext highlighter-rouge">SampleType</code>.</li>
  <li><em>QC_boxplot_peptide_norm</em>: Boxplots of the normalized and log2 transformed peptide intensities for the individual <code class="language-plaintext highlighter-rouge">Runs</code> colored based on the <code class="language-plaintext highlighter-rouge">SampleType</code>.</li>
  <li><em>QC_boxplot_protein</em>: Boxplots of the (non-normalized) log2 transformed protein intensities for the individual <code class="language-plaintext highlighter-rouge">Runs</code> colored based on the <code class="language-plaintext highlighter-rouge">SampleType</code>.</li>
  <li><em>QC_boxplot_protein_norm</em>: Boxplots of the normalized and log2 transformed protein intensities for the individual <code class="language-plaintext highlighter-rouge">Runs</code> colored based on the <code class="language-plaintext highlighter-rouge">SampleType</code>.</li>
  <li><em>QC_heatmap_proteins</em>: Heatmap of missing values on the protein level (intensities after protein normalization were used). Columns are <code class="language-plaintext highlighter-rouge">Runs</code>, rows are individual proteins. Black means that protein has non-zero intensity, white means the protein has NA value for in the particular run. This can be useful to estimate what type of missingness is prevalent in our data (if the protein is missing in a stochastic manner across conditions, it will likely be MCAR; if protein absence/presence is biased towards specific samples group, it will likely be MNAR).</li>
  <li><em>QC_median_CV</em>: Boxplots of the median coefficient of variation (CV) in the single cell samples and blank. We want to assess the consistency of quantification and filter out the cells with unreliable quantifications. In our example scenario, based on the distribution in Macrophage and Monocyte samples, we would set the threshold to 0.65.</li>
  <li><em>QC_plot_SCR</em>: This plot shows the histogram of the mean sample-to-carrier ratio (SCR), which is computed as reporter ion intensity of a single cell channel divided by the reporter ion intensity of the carrier channel within one batch. We expect that the carrier intensity will be much higher than the single cell intensity. Because we have 200 cells in the carrier channel, we expect the ratio to be 1/200 (=0.005, the dashed line). We want to filter the PSMs which have a very high meanSCR, so in our case we would set the threshold to 0.1, where the long tail begins.</li>
  <li><em>QC_plot_SCR_col</em>: the density plot of meanSCR as described above.</li>
  <li><em>UMAP</em>: UMAP plot based on the PCA data.</li>
</ul>

<p>In our tutorial, we didn’t generate the intermediate outputs. These are the dataframes containing the rowData (Gene.names, Leading.razor.protein, …) along with the particular numerical matrix - i.e. reporter intensities aggregated to the peptide or protein level, after transformation/normalization.</p>

<h1 id="summary-tool-setting">Summary tool setting</h1>

<p>Let’s once again recapitulate how did we set the form: below we will mention only the parameters where we adjusted values, the other parameters were left to the default settings. Now we can finally run the tool!</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-bioconductor-scp-full-example"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: bioconductor-scp: Full example</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/bioconductor_scp/bioconductor_scp/1.16.0+galaxy0" title="bioconductor-scp tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bioconductor-scp</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.0+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Input evidence table”</em>: <code class="language-plaintext highlighter-rouge">evidences.txt</code></li>
        <li><em>“Sample annotations table”</em>: <code class="language-plaintext highlighter-rouge">sampleAnnotation.txt</code></li>
        <li><em>“Which column specifies the run identifier and batch name?”</em>: <code class="language-plaintext highlighter-rouge">c19: Raw.file</code></li>
        <li><em>“Single-cell channels present in the experiment”</em>: <code class="language-plaintext highlighter-rouge">Monocyte</code> and <code class="language-plaintext highlighter-rouge">Macrophage</code></li>
        <li><em>“Which column should be used for the PSM to peptide aggregation?”</em>: <code class="language-plaintext highlighter-rouge">c6: Modified.sequence</code></li>
        <li><em>“Which samples to keep?”</em>: <code class="language-plaintext highlighter-rouge">Monocyte</code>, <code class="language-plaintext highlighter-rouge">Macrophage</code> and <code class="language-plaintext highlighter-rouge">Blank</code></li>
        <li><em>“Which column should be used for the peptide to protein aggregation?”</em>: <code class="language-plaintext highlighter-rouge">c17: Leading.razor.protein</code></li>
        <li><em>“Which column is the technical variable to be corrected?”</em>: <code class="language-plaintext highlighter-rouge">c2: runCol</code></li>
        <li><em>“What column to color the PCA according to?”</em>: <code class="language-plaintext highlighter-rouge">c4: SampleType</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>That’s it! Good luck with the analysis of your own single-cell proteomics data!</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(typeof plausible !== 'undefined'){
                            // Plausible may be undefined (script blocked)
                            // or it may be defined, but opted-out (select box/DNT),
                            // which means `plausible()` will work but not send data, *nor* execute the callback.
                            plausible('TutorialComplete', {props: {path: document.location.pathname}})
                          }
                          // since the callback is completely cosmetic, we'll just issue it optimistically.
                          tutorial_finish_finish();
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Single-cell proteomics can reveal cellular heterogeneity and uncover rare cell populations.</p>
</li>
                        
                        <li><p>Well-tested and standardized tools as bioconductor-scp are essential for the reproducible data analysis.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/proteomics/faqs/">FAQ page for the Proteomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

		<!--
                
		-->

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/proteomics#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Goh2017">Goh, W. W. B., W. Wang, and L. Wong, 2017 <b>Why Batch Effects Matter in Omics Data,  and How to Avoid Them</b>. Trends in Biotechnology 35: 498–507. <a href="https://doi.org/10.1016/j.tibtech.2017.02.012">10.1016/j.tibtech.2017.02.012</a></li>
<li id="Zhang2018">Zhang, Y., D. F. Jenkins, S. Manimaran, and W. E. Johnson, 2018 <b>Alternative empirical Bayes models for adjusting for batch effects in genomic studies</b>. BMC Bioinformatics 19: <a href="https://doi.org/10.1186/s12859-018-2263-6">10.1186/s12859-018-2263-6</a></li>
<li id="Specht2021">Specht, H., E. Emmott, A. A. Petelski, R. G. Huffman, D. H. Perlman <i>et al.</i>, 2021 <b>Single-cell proteomic and transcriptomic analysis of macrophage heterogeneity using SCoPE2</b>. Genome Biology 22: <a href="https://doi.org/10.1186/s13059-021-02267-5">10.1186/s13059-021-02267-5</a></li>
<li id="Phua2022">Phua, S.-X., K.-P. Lim, and W. W.-B. Goh, 2022 <b>Perspectives for better batch effect correction in mass-spectrometry-based proteomics</b>. Computational and Structural Biotechnology Journal 20: 4369–4375. <a href="https://doi.org/10.1016/j.csbj.2022.08.022">10.1016/j.csbj.2022.08.022</a></li>
<li id="Ahmad2023">Ahmad, R., and B. Budnik, 2023 <b>A review of the current state of single-cell proteomics and future perspective</b>. Analytical and Bioanalytical Chemistry 415: 6889–6899. <a href="https://doi.org/10.1007/s00216-023-04759-8">10.1007/s00216-023-04759-8</a></li>
<li id="Gatto2023">Gatto, L., R. Aebersold, J. Cox, V. Demichev, J. Derks <i>et al.</i>, 2023 <b>Initial recommendations for performing,  benchmarking and reporting single-cell proteomics experiments</b>. Nature Methods 20: 375–386. <a href="https://doi.org/10.1038/s41592-023-01785-3">10.1038/s41592-023-01785-3</a></li>
<li id="Lin2024">Lin, H. J. L., K. G. I. Webber, A. J. Nwosu, and R. T. Kelly, 2024 <b>Review and Practical Guide for Getting Started With Single‐Cell Proteomics</b>. PROTEOMICS 25: <a href="https://doi.org/10.1002/pmic.202400021">10.1002/pmic.202400021</a></li>
<li id="distill-tSNE">Distill <b>How to Use t-SNE Effectively</b>. <a href="https://distill.pub/2016/misread-tsne/">https://distill.pub/2016/misread-tsne/</a></li>
<li id="number-pca-components">O’Brien, K. <b>https://rpubs.com/DragonflyStats/pca_components_number</b>. <a href="https://rpubs.com/DragonflyStats/pca_components_number">https://rpubs.com/DragonflyStats/pca_components_number</a></li>
<li id="creative-proteomics">Proteomics, C. <b>Principal Component Analysis</b>. <a href="https://www.creative-proteomics.com/services/pca-service.htm">https://www.creative-proteomics.com/services/pca-service.htm</a></li>
<li id="understanding-umap2">Varma, A. <b>Dimensionality Reduction: PCA, t-SNE, and UMAP</b>. <a href="https://medium.com/@aastha.code/dimensionality-reduction-pca-t-sne-and-umap-41d499da2df2">https://medium.com/@aastha.code/dimensionality-reduction-pca-t-sne-and-umap-41d499da2df2</a></li></ol>
                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=proteomics/bioconductor-scp"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=proteomics/bioconductor-scp">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Kristina Gomoryova, Helge Hecht,  <b>Single Cell Proteomics data analysis with bioconductor-scp (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/bioconductor-scp/tutorial.html">https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/bioconductor-scp/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{proteomics-bioconductor-scp,
author = "Kristina Gomoryova and Helge Hecht",
	title = "Single Cell Proteomics data analysis with bioconductor-scp (Galaxy Training Materials)",
	year = "",
	month = "",
	day = "",
	url = "\url{https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/bioconductor-scp/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

        


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/proteomics/tutorials/bioconductor-scp/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: bioconductor_scp
  owner: recetox
  revisions: cd2f3a280463
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
		   </table>
		   </p>
		
        
        
            <p>No feedback has been recieved yet for this training. Be the first one by filling in the feedback form from <a href="#gtn-feedback">above</a>. </p>
        
		</blockquote>




		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/proteomics/tutorials/bioconductor-scp/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/proteomics/tutorials/bioconductor-scp/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/c67c7ffc2850bbbb9e9637f431a6810beb66f7de">c67c7ff</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>