<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Top Tools</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@ae9d750ff4d4c69ea19d387b01201377aa82bb0b",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/stats/top-tools.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Top Tools">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-12-11 11:18:54 +0000"><meta name="description" content="{% assign groups = site | list_materials_by_tool %}      ToolTutorials      {% for group in groups %}  {% assign match = group[0] | matches: '{{' %}  {% unless match %}     {{ group[0] }}            {% for tutorial in group[1][ tutorials ] %}           {{ tutorial[2] }} /            {{ tutorial[1] }}                {% endfor %}           {% endunless %}  {% endfor %}     ">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Top Tools">
        <meta property="og:description" content="{% assign groups = site | list_materials_by_tool %}      ToolTutorials      {% for group in groups %}  {% assign match = group[0] | matches: '{{' %}  {% unless match %}     {{ group[0] }}            {% for tutorial in group[1][ tutorials ] %}           {{ tutorial[2] }} /            {{ tutorial[1] }}                {% endfor %}           {% endunless %}  {% endfor %}     ">
        <meta property="og:image" content="https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
			

                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/stats/top-tools.html">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/stats/top-tools.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html">
                <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        
<table class="table table-striped">
	<thead>
		<tr><th>Tool</th><th>Tutorials</th></tr>
	</thead>
	<tbody>
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/Cut1.html">Cut1</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-decontamination/tutorial.html">
					Decontamination of a genome assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">
					Creating a new tutorial
					</a>
				</li>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html">
					Infinium Human Methylation BeadChip
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-reports/tutorial.html">
					Workflow Reports
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html">
					Essential genes detection with Transposon insertion sequencing
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">
					Galaxy Basics for everyone
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html">
					From peaks to genes
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-reproduce/tutorial.html">
					How to reproduce published Galaxy analyses
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-4-quantitation/tutorial.html">
					Clinical Metaproteomics 4: Quantitation
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-4-quantitation/tutorial.html">
					Clinical Metaproteomics 4: Quantitation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-label-free/tutorial.html">
					Label-free data analysis using MaxQuant
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html">
					MaxQuant and MSstats for the analysis of label-free data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html">
					metaQuantome 1: Data creation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-7-hla-binding-novel-peptides/tutorial.html">
					Neoantigen 7: IEDB binding PepQuery Validated Neopeptides
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/GO-enrichment/tutorial.html">
					GO Enrichment Analysis on Single-Cell RNA-Seq Data
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html">
					Generating a single cell matrix using Alevin
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html">
					Combining single cell datasets after pre-processing
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-ncbi-anndata/tutorial.html">
					Converting NCBI Data to the AnnData Format
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Seurat
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/gpu_jupyter_lab/tutorial.html">
					A Docker-based interactive Jupyterlab powered by GPU for artificial intelligence in Galaxy
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html">
					Network analysis with Heinz
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html">
					3: RNA-seq genes to pathways
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html">
					Visualization of RNA-Seq results with heatmap2
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html">
					Calling variants in non-diploid systems
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/circos/tutorial.html">
					Visualisation with Circos
					</a>
				</li>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/circos-microbial/tutorial.html">
					Ploting a Microbial Genome with Circos
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/Filter1.html">Filter1</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-decontamination/tutorial.html">
					Decontamination of a genome assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">
					Creating a new tutorial
					</a>
				</li>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/gbif_cleaning/tutorial.html">
					Cleaning GBIF data for the use in Ecology
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html">
					Preparing genomic data for phylogeny reconstruction
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/species-distribution-modeling/tutorial.html">
					Species distribution modeling
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html">
					Essential genes detection with Transposon insertion sequencing
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html">
					Introduction to Genomics and Galaxy
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-analyte-distribution/tutorial.html">
					Mass spectrometry imaging: Examining the spatial distribution of analytes
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html">
					Mass spectrometry imaging: Finding differential analytes
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/beer-data-analysis/tutorial.html">
					Identification of the micro-organisms in a beer using Nanopore sequencing
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_Analysis_MSstats/tutorial.html">
					Statistical analysis of DIA data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html">
					Library Generation for DIA Analysis
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-label-free/tutorial.html">
					Label-free data analysis using MaxQuant
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html">
					MaxQuant and MSstats for the analysis of label-free data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html">
					metaQuantome 1: Data creation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-7-hla-binding-novel-peptides/tutorial.html">
					Neoantigen 7: IEDB binding PepQuery Validated Neopeptides
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html">
					Secretome Prediction
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Seurat
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/gpu_jupyter_lab/tutorial.html">
					A Docker-based interactive Jupyterlab powered by GPU for artificial intelligence in Galaxy
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html">
					De novo transcriptome reconstruction with RNA-Seq
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/goenrichment/tutorial.html">
					GO Enrichment Analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/srna/tutorial.html">
					Differential abundance testing of small RNAs
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dunovo/tutorial.html">
					Calling very rare variants
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html">
					From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html">
					Mutation calling, viral genome reconstruction and lineage/clade assignment from SARS-CoV-2 sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/trio-analysis/tutorial.html">
					Trio Analysis using Synthetic Datasets from RD-Connect GPAP
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/fastqc/fastqc.html">devteam/fastqc/fastqc</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/general-introduction/tutorial.html">
					An Introduction to Genome Assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html">
					Genome Assembly of a bacterial genome (MRSA) sequenced using Illumina MiSeq Data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html">
					Genome Assembly of MRSA from Oxford Nanopore MinION data (and optionally Illumina data)
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/unicycler-assembly/tutorial.html">
					Unicycler Assembly
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ENA_Biodiv_submission/tutorial.html">
					Data submission using ENA upload Tool
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/de-novo-rad-seq/tutorial.html">
					RAD-Seq de-novo data analysis
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
					Identification of the binding sites of the Estrogen receptor
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/methylation-seq/tutorial.html">
					DNA Methylation data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html">
					Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/crispr-screen/tutorial.html">
					CRISPR screen analysis
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-short/tutorial.html">
					A short introduction to Galaxy
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/beer-data-analysis/tutorial.html">
					Identification of the micro-organisms in a beer using Nanopore sequencing
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/nanopore-16S-metagenomics/tutorial.html">
					16S Microbial analysis with Nanopore data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">
					Mapping
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/quality-contamination-control/tutorial.html">
					Quality and contamination control in bacterial isolate using Illumina MiSeq Data
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">
					Quality Control
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/sars-with-galaxy-on-anvil/tutorial.html">
					SARS-CoV-2 Viral Sample Alignment and Variant Visualization
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html">
					Pre-processing of 10X Single-Cell ATAC-seq Datasets
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-umis/tutorial.html">
					Understanding Barcodes
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html">
					De novo transcriptome reconstruction with RNA-Seq
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-interactome/tutorial.html">
					RNA-RNA interactome data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">
					1: RNA-Seq reads to counts
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/srna/tutorial.html">
					Differential abundance testing of small RNAs
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dunovo/tutorial.html">
					Calling very rare variants
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">
					Exome sequencing data analysis for diagnosing a genetic disease
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html">
					Calling variants in non-diploid systems
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/Grep1.html">Grep1</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/climate-101/tutorial.html">
					Getting your hands-on climate data
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/amr-gene-detection/tutorial.html">
					Identification of AMR genes in an assembled bacterial genome
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html">
					Mass spectrometry imaging: Finding differential analytes
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-4-quantitation/tutorial.html">
					Clinical Metaproteomics 4: Quantitation
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-5-data-interpretation/tutorial.html">
					Clinical Metaproteomics 5: Data Interpretation
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_Analysis_MSstats/tutorial.html">
					Statistical analysis of DIA data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html">
					DIA Analysis using OpenSwathWorkflow
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-4-quantitation/tutorial.html">
					Clinical Metaproteomics 4: Quantitation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-5-data-interpretation/tutorial.html">
					Clinical Metaproteomics 5: Data Interpretation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/mass-spectrometry-imaging-loading-exploring-data/tutorial.html">
					Mass spectrometry imaging: Loading and exploring MSI data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-label-free/tutorial.html">
					Label-free data analysis using MaxQuant
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html">
					MaxQuant and MSstats for the analysis of label-free data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html">
					metaQuantome 1: Data creation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-3-fragpipe-discovery/tutorial.html">
					Neoantigen 3: Database merge and FragPipe discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html">
					Peptide and Protein ID using SearchGUI and PeptideShaker
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html">
					Secretome Prediction
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html">
					Removal of human reads from SARS-CoV-2 sequencing data
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html">
					From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/circos/tutorial.html">
					Visualisation with Circos
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/multiqc/multiqc.html">iuc/multiqc/multiqc</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/general-introduction/tutorial.html">
					An Introduction to Genome Assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/unicycler-assembly/tutorial.html">
					Unicycler Assembly
					</a>
				</li>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html">
					Creating content in Markdown
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ENA_Biodiv_submission/tutorial.html">
					Data submission using ENA upload Tool
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/crispr-screen/tutorial.html">
					CRISPR screen analysis
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/vsi_qc/tutorial.html">
					Very Short Introductions: QC
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/nanopore-16S-metagenomics/tutorial.html">
					16S Microbial analysis with Nanopore data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">
					Mapping
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">
					Quality Control
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html">
					Pre-processing of Single-Cell RNA Data
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html">
					Pre-processing of 10X Single-Cell RNA Datasets
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">
					1: RNA-Seq reads to counts
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">
					Exome sequencing data analysis for diagnosing a genetic disease
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html">
					From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_cat.html">bgruening/text_processing/tp_cat</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-decontamination/tutorial.html">
					Decontamination of a genome assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/zauberkugel/tutorial.html">
					Protein target prediction of a bioactive ligand with Align-it and ePharmaLib
					</a>
				</li>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">
					Creating a new tutorial
					</a>
				</li>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html">
					SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/lncrna/tutorial.html">
					Long non-coding RNAs (lncRNAs) annotation with FEELnc
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html">
					Introduction to Genomics and Galaxy
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html">
					Predicting EI+ mass spectra with QCxMS
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/upload1.html">upload1</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/tutorial.html">
					Chloroplast genome assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html">
					Genome Assembly of a bacterial genome (MRSA) sequenced using Illumina MiSeq Data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html">
					Genome Assembly of MRSA from Oxford Nanopore MinION data (and optionally Illumina data)
					</a>
				</li>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html">
					Creating content in Markdown
					</a>
				</li>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/bacterial-isolate-species-contamination-checking/tutorial.html">
					Checking expected species and contamination in bacterial isolate
					</a>
				</li>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_phylogeny/tutorial.html">
					Tree thinking for tuberculosis evolution and epidemiology
					</a>
				</li>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-reports/tutorial.html">
					Workflow Reports
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/amr-gene-detection/tutorial.html">
					Identification of AMR genes in an assembled bacterial genome
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/bacterial-genome-annotation/tutorial.html">
					Bacterial Genome Annotation
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-101/tutorial.html">
					Galaxy Basics for genomics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">
					Galaxy Basics for everyone
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html">
					From peaks to genes
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-reproduce/tutorial.html">
					How to reproduce published Galaxy analyses
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/quality-contamination-control/tutorial.html">
					Quality and contamination control in bacterial isolate using Illumina MiSeq Data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html">
					Mutation calling, viral genome reconstruction and lineage/clade assignment from SARS-CoV-2 sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/circos/tutorial.html">
					Visualisation with Circos
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/Remove beginning1.html">Remove beginning1</a></td>
		<td>
			<ul>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/biodiversity-data-exploration/tutorial.html">
					Biodiversity data exploration
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html">
					Infinium Human Methylation BeadChip
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-reports/tutorial.html">
					Workflow Reports
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html">
					Secondary metabolite discovery
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">
					Galaxy Basics for everyone
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-reproduce/tutorial.html">
					How to reproduce published Galaxy analyses
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/nanopore-16S-metagenomics/tutorial.html">
					16S Microbial analysis with Nanopore data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html">
					metaQuantome 1: Data creation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-7-hla-binding-novel-peptides/tutorial.html">
					Neoantigen 7: IEDB binding PepQuery Validated Neopeptides
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/classification_machinelearning/tutorial.html">
					Classification in Machine Learning
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/regression_machinelearning/tutorial.html">
					Regression in Machine Learning
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/circos/tutorial.html">
					Visualisation with Circos
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nml/collapse_collections/collapse_dataset.html">nml/collapse_collections/collapse_dataset</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html">
					Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html">
					Preparing genomic data for phylogeny reconstruction
					</a>
				</li>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html">
					Using dataset collections
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-automation/tutorial.html">
					Automating Galaxy workflows using the command line
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html">
					Secondary metabolite discovery
					</a>
				</li>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/hela-screen-analysis/tutorial.html">
					Analyse HeLa fluorescence siRNA screen
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-3-fragpipe-discovery/tutorial.html">
					Neoantigen 3: Database merge and FragPipe discovery
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_cut_tool.html">bgruening/text_processing/tp_cut_tool</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">
					Creating a new tutorial
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Ecoregionalization_tutorial/tutorial.html">
					Ecoregionalization workflow tutorial
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/biodiversity-data-exploration/tutorial.html">
					Biodiversity data exploration
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/obisindicators/tutorial.html">
					Obis marine indicators
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/species-distribution-modeling/tutorial.html">
					Species distribution modeling
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html">
					Predicting EI+ mass spectra with QCxMS
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html">
					Biomarker candidate identification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html">
					Secretome Prediction
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html">
					Bulk matrix to ESet | Creating the bulk RNA-seq dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/fruit_360/tutorial.html">
					Image classification in Galaxy with fruit 360 dataset
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html">
					2: RNA-seq counts to genes
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html">
					3: RNA-seq genes to pathways
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html">
					From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/query_tabular/query_tabular.html">iuc/query_tabular/query_tabular</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html">
					SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html">
					Metaproteomics tutorial
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html">
					metaQuantome 1: Data creation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-1-fusion-database-generation/tutorial.html">
					Neoantigen 1: Fusion-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-3-fragpipe-discovery/tutorial.html">
					Neoantigen 3: Database merge and FragPipe discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-4-peptide-verification/tutorial.html">
					Neoantigen 4: PepQuery2 Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-5-variant-annotation/tutorial.html">
					Neoantigen 5: Variant Annotation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-6-predicting-hla-binding/tutorial.html">
					Neoantigen 6: Predicting HLA Binding
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-7-hla-binding-novel-peptides/tutorial.html">
					Neoantigen 7: IEDB binding PepQuery Validated Neopeptides
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html">
					Proteogenomics 2: Database Search
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.html">
					Proteogenomics 3: Novel peptide analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-interactome/tutorial.html">
					RNA-RNA interactome data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/Grouping1.html">Grouping1</a></td>
		<td>
			<ul>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-reports/tutorial.html">
					Workflow Reports
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/bacterial-genome-annotation/tutorial.html">
					Bacterial Genome Annotation
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">
					Galaxy Basics for everyone
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html">
					From peaks to genes
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-reproduce/tutorial.html">
					How to reproduce published Galaxy analyses
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-4-quantitation/tutorial.html">
					Clinical Metaproteomics 4: Quantitation
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/bioconductor-scp/tutorial.html">
					Single Cell Proteomics data analysis with bioconductor-scp
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-4-quantitation/tutorial.html">
					Clinical Metaproteomics 4: Quantitation
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/lparsons/cutadapt/cutadapt.html">lparsons/cutadapt/cutadapt</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">
					Creating a new tutorial
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ENA_Biodiv_submission/tutorial.html">
					Data submission using ENA upload Tool
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/crispr-screen/tutorial.html">
					CRISPR screen analysis
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html">
					Essential genes detection with Transposon insertion sequencing
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">
					Quality Control
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-interactome/tutorial.html">
					RNA-RNA interactome data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">
					1: RNA-Seq reads to counts
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/regex_find_replace/regexColumn1.html">galaxyp/regex_find_replace/regexColumn1</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/pangeo/tutorial.html">
					Pangeo ecosystem 101 for everyone - Introduction to Xarray Galaxy Tools
					</a>
				</li>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Ecoregionalization_tutorial/tutorial.html">
					Ecoregionalization workflow tutorial
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/biodiversity-data-exploration/tutorial.html">
					Biodiversity data exploration
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-5-variant-annotation/tutorial.html">
					Neoantigen 5: Variant Annotation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html">
					Importing files from public atlases
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/trio-analysis/tutorial.html">
					Trio Analysis using Synthetic Datasets from RD-Connect GPAP
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/join1.html">join1</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html">
					Generating a single cell matrix using Alevin
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html">
					3: RNA-seq genes to pathways
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html">
					Visualization of RNA-Seq results with heatmap2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/jbrowse/jbrowse.html">iuc/jbrowse/jbrowse</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/tutorial.html">
					Chloroplast genome assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/amr-gene-detection/tutorial.html">
					Identification of AMR genes in an assembled bacterial genome
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html">
					Genome annotation with Maker
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker-short/tutorial.html">
					Genome annotation with Maker (short)
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-prokka/tutorial.html">
					Genome annotation with Prokka
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/apollo/tutorial.html">
					Refining Genome Annotations with Apollo (prokaryotes)
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/apollo-euk/tutorial.html">
					Refining Genome Annotations with Apollo (eukaryotes)
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/bacterial-genome-annotation/tutorial.html">
					Bacterial Genome Annotation
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html">
					Genome annotation with Funannotate
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/helixer/tutorial.html">
					Genome annotation with Helixer
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">
					Mapping
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/sars-with-galaxy-on-anvil/tutorial.html">
					SARS-CoV-2 Viral Sample Alignment and Variant Visualization
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/microbial-variants/tutorial.html">
					Microbial Variant Calling
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/jbrowse/tutorial.html">
					Genomic Data Visualisation with JBrowse
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/column_maker/Add_a_column1.html">devteam/column_maker/Add_a_column1</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html">
					Essential genes detection with Transposon insertion sequencing
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html">
					From peaks to genes
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-label-free/tutorial.html">
					Label-free data analysis using MaxQuant
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html">
					3: RNA-seq genes to pathways
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/fastp/fastp.html">iuc/fastp/fastp</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html">
					Genome Assembly of a bacterial genome (MRSA) sequenced using Illumina MiSeq Data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html">
					Genome Assembly of MRSA from Oxford Nanopore MinION data (and optionally Illumina data)
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/vsi_qc/tutorial.html">
					Very Short Introductions: QC
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/beer-data-analysis/tutorial.html">
					Identification of the micro-organisms in a beer using Nanopore sequencing
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/nanopore-16S-metagenomics/tutorial.html">
					16S Microbial analysis with Nanopore data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/quality-contamination-control/tutorial.html">
					Quality and contamination control in bacterial isolate using Illumina MiSeq Data
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html">
					From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_sort_header_tool.html">bgruening/text_processing/tp_sort_header_tool</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html">
					Essential genes detection with Transposon insertion sequencing
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-101/tutorial.html">
					Galaxy Basics for genomics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html">
					From peaks to genes
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/mass-spectrometry-imaging-loading-exploring-data/tutorial.html">
					Mass spectrometry imaging: Loading and exploring MSI data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-label-free/tutorial.html">
					Label-free data analysis using MaxQuant
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html">
					3: RNA-seq genes to pathways
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.html">
					Small Non-coding RNA Clustering using BlockClust
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/bwa/bwa_mem.html">devteam/bwa/bwa_mem</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/tutorial.html">
					Chloroplast genome assembly
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html">
					Using dataset collections
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html">
					Removal of human reads from SARS-CoV-2 sequencing data
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/sars-with-galaxy-on-anvil/tutorial.html">
					SARS-CoV-2 Viral Sample Alignment and Variant Visualization
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html">
					Pre-processing of 10X Single-Cell ATAC-seq Datasets
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dunovo/tutorial.html">
					Calling very rare variants
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">
					Exome sequencing data analysis for diagnosing a genetic disease
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html">
					Calling variants in non-diploid systems
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html">
					From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_awk_tool.html">bgruening/text_processing/tp_awk_tool</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/climate-101/tutorial.html">
					Getting your hands-on climate data
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/methylation-seq/tutorial.html">
					DNA Methylation data analysis
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/crispr-screen/tutorial.html">
					CRISPR screen analysis
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/hpc-for-lsgc/tutorial.html">
					From small to large-scale genome comparison
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html">
					Secondary metabolite discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-1-fusion-database-generation/tutorial.html">
					Neoantigen 1: Fusion-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-6-predicting-hla-binding/tutorial.html">
					Neoantigen 6: Predicting HLA Binding
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html">
					Bulk matrix to ESet | Creating the bulk RNA-seq dataset for deconvolution
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/cat1.html">cat1</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html">
					Introduction to Genomics and Galaxy
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html">
					Removing the effects of the cell cycle
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/datamash_ops/datamash_ops.html">iuc/datamash_ops/datamash_ops</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/climate-101/tutorial.html">
					Getting your hands-on climate data
					</a>
				</li>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-reports/tutorial.html">
					Workflow Reports
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-101/tutorial.html">
					Galaxy Basics for genomics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">
					Galaxy Basics for everyone
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-reproduce/tutorial.html">
					How to reproduce published Galaxy analyses
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html">
					MaxQuant and MSstats for the analysis of label-free data
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-raceid/tutorial.html">
					Downstream Single-cell RNA analysis with RaceID
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Scanpy
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html">
					Mutation calling, viral genome reconstruction and lineage/clade assignment from SARS-CoV-2 sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_replace_in_line.html">bgruening/text_processing/tp_replace_in_line</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html">
					Preparing genomic data for phylogeny reconstruction
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/nanopore-16S-metagenomics/tutorial.html">
					16S Microbial analysis with Nanopore data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html">
					metaQuantome 1: Data creation
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html">
					Removal of human reads from SARS-CoV-2 sequencing data
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html">
					2: RNA-seq counts to genes
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/regex_find_replace/regex1.html">galaxyp/regex_find_replace/regex1</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ENA_Biodiv_submission/tutorial.html">
					Data submission using ENA upload Tool
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/marine_omics_bgc/tutorial.html">
					Marine Omics identifying biosynthetic gene clusters
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/openrefine_gbif/tutorial.html">
					Cleaning GBIF data using OpenRefine
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html">
					Preparing genomic data for phylogeny reconstruction
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html">
					metaQuantome 1: Data creation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-1-fusion-database-generation/tutorial.html">
					Neoantigen 1: Fusion-Database-Generation
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html">
					Bulk matrix to ESet | Creating the bulk RNA-seq dataset for deconvolution
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences.html">galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-1-database-generation/tutorial.html">
					Clinical Metaproteomics 1: Database-Generation
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-1-database-generation/tutorial.html">
					Clinical Metaproteomics 1: Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/database-handling/tutorial.html">
					Protein FASTA Database Handling
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-3-fragpipe-discovery/tutorial.html">
					Neoantigen 3: Database merge and FragPipe discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/tabular_to_fasta/tab2fasta.html">devteam/tabular_to_fasta/tab2fasta</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/database-handling/tutorial.html">
					Protein FASTA Database Handling
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-1-fusion-database-generation/tutorial.html">
					Neoantigen 1: Fusion-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-4-peptide-verification/tutorial.html">
					Neoantigen 4: PepQuery2 Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-5-variant-annotation/tutorial.html">
					Neoantigen 5: Variant Annotation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html">
					Proteogenomics 2: Database Search
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_find_and_replace.html">bgruening/text_processing/tp_find_and_replace</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/fates/tutorial.html">
					Functionally Assembled Terrestrial Ecosystem Simulator (FATES)
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/species-distribution-modeling/tutorial.html">
					Species distribution modeling
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html">
					Importing files from public atlases
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html">
					Mapping and molecular identification of phenotype-causing mutations
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_sorted_uniq.html">bgruening/text_processing/tp_sorted_uniq</a></td>
		<td>
			<ul>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">
					Creating a new tutorial
					</a>
				</li>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-reports/tutorial.html">
					Workflow Reports
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">
					Galaxy Basics for everyone
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-reproduce/tutorial.html">
					How to reproduce published Galaxy analyses
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html">
					Secretome Prediction
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html">
					Pre-processing of 10X Single-Cell ATAC-seq Datasets
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/quast/quast.html">iuc/quast/quast</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-quality-control/tutorial.html">
					Genome Assembly Quality Control
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/flye-assembly/tutorial.html">
					Genome assembly using PacBio data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/general-introduction/tutorial.html">
					An Introduction to Genome Assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html">
					Genome Assembly of a bacterial genome (MRSA) sequenced using Illumina MiSeq Data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html">
					Genome Assembly of MRSA from Oxford Nanopore MinION data (and optionally Illumina data)
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/unicycler-assembly/tutorial.html">
					Unicycler Assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_workflow_training/tutorial.html">
					Using the VGP workflows to assemble a vertebrate genome with HiFi and Hi-C data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_replace_in_column.html">bgruening/text_processing/tp_replace_in_column</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/bacterial-genome-annotation/tutorial.html">
					Bacterial Genome Annotation
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html">
					From peaks to genes
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html">
					MaxQuant and MSstats for the analysis of label-free data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html">
					metaQuantome 1: Data creation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html">
					Combining single cell datasets after pre-processing
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-ncbi-anndata/tutorial.html">
					Converting NCBI Data to the AnnData Format
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/anndata_inspect/anndata_inspect.html">iuc/anndata_inspect/anndata_inspect</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html">
					Pre-processing of 10X Single-Cell ATAC-seq Datasets
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html">
					Combining single cell datasets after pre-processing
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html">
					Removing the effects of the cell cycle
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-ncbi-anndata/tutorial.html">
					Converting NCBI Data to the AnnData Format
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/busco/busco.html">iuc/busco/busco</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-quality-control/tutorial.html">
					Genome Assembly Quality Control
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/flye-assembly/tutorial.html">
					Genome assembly using PacBio data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html">
					Preparing genomic data for phylogeny reconstruction
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html">
					Genome annotation with Maker
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker-short/tutorial.html">
					Genome annotation with Maker (short)
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html">
					Genome annotation with Funannotate
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/helixer/tutorial.html">
					Genome annotation with Helixer
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bandage/bandage_image.html">iuc/bandage/bandage_image</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/tutorial.html">
					Chloroplast genome assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html">
					Genome Assembly of a bacterial genome (MRSA) sequenced using Illumina MiSeq Data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html">
					Genome Assembly of MRSA from Oxford Nanopore MinION data (and optionally Illumina data)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/plasmid-metagenomics-nanopore/tutorial.html">
					Antibiotic resistance detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_sed_tool.html">bgruening/text_processing/tp_sed_tool</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-decontamination/tutorial.html">
					Decontamination of a genome assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">
					Creating a new tutorial
					</a>
				</li>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html">
					Bulk matrix to ESet | Creating the bulk RNA-seq dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html">
					Pre-processing of Single-Cell RNA Data
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-raceid/tutorial.html">
					Downstream Single-cell RNA analysis with RaceID
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/bowtie2/bowtie2.html">devteam/bowtie2/bowtie2</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">
					Creating a new tutorial
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
					Identification of the binding sites of the Estrogen receptor
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/amr-gene-detection/tutorial.html">
					Identification of AMR genes in an assembled bacterial genome
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">
					Mapping
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_easyjoin_tool.html">bgruening/text_processing/tp_easyjoin_tool</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html">
					Essential genes detection with Transposon insertion sequencing
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html">
					MaxQuant and MSstats for the analysis of label-free data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html">
					Secretome Prediction
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html">
					Bulk matrix to ESet | Creating the bulk RNA-seq dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html">
					Removing the effects of the cell cycle
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_grep_tool.html">bgruening/text_processing/tp_grep_tool</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/md-simulation-gromacs/tutorial.html">
					Running molecular dynamics simulations using GROMACS
					</a>
				</li>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html">
					Importing files from public atlases
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/trio-analysis/tutorial.html">
					Trio Analysis using Synthetic Datasets from RD-Connect GPAP
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/sort1.html">sort1</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/zauberkugel/tutorial.html">
					Protein target prediction of a bioactive ligand with Align-it and ePharmaLib
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html">
					Bulk matrix to ESet | Creating the bulk RNA-seq dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html">
					Removing the effects of the cell cycle
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/ggplot2_point/ggplot2_point.html">iuc/ggplot2_point/ggplot2_point</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/climate-101/tutorial.html">
					Getting your hands-on climate data
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/fates/tutorial.html">
					Functionally Assembled Terrestrial Ecosystem Simulator (FATES)
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/pangeo/tutorial.html">
					Pangeo ecosystem 101 for everyone - Introduction to Xarray Galaxy Tools
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-reports/tutorial.html">
					Workflow Reports
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">
					Galaxy Basics for everyone
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-reproduce/tutorial.html">
					How to reproduce published Galaxy analyses
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/clustering_machinelearning/tutorial.html">
					Clustering in Machine Learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/anndata_manipulate/anndata_manipulate.html">iuc/anndata_manipulate/anndata_manipulate</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html">
					Combining single cell datasets after pre-processing
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html">
					Removing the effects of the cell cycle
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_trajectories/tutorial.html">
					Inferring single cell trajectories with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-ncbi-anndata/tutorial.html">
					Converting NCBI Data to the AnnData Format
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-plant/tutorial.html">
					Analysis of plant scRNA-Seq Data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_rstudio.html">interactive_tool_rstudio</a></td>
		<td>
			<ul>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/bash-git/tutorial.html">
					Version Control with Git
					</a>
				</li>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/r-advanced/tutorial.html">
					Advanced R in Galaxy
					</a>
				</li>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/r-basics/tutorial.html">
					R basics in Galaxy
					</a>
				</li>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_phylogeny/tutorial.html">
					Tree thinking for tuberculosis evolution and epidemiology
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/rstudio/tutorial.html">
					RStudio in Galaxy
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/intro-to-ml-with-r/tutorial.html">
					Introduction to Machine Learning using R
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.html">
					RNA Seq Counts to Viz in R
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot-r/tutorial.html">
					Visualization of RNA-Seq results with Volcano Plot in R
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/Count1.html">Count1</a></td>
		<td>
			<ul>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/gbif_cleaning/tutorial.html">
					Cleaning GBIF data for the use in Ecology
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/maxquant/maxquant.html">galaxyp/maxquant/maxquant</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-4-quantitation/tutorial.html">
					Clinical Metaproteomics 4: Quantitation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html">
					Library Generation for DIA Analysis
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-4-quantitation/tutorial.html">
					Clinical Metaproteomics 4: Quantitation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-label-free/tutorial.html">
					Label-free data analysis using MaxQuant
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html">
					MaxQuant and MSstats for the analysis of label-free data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-tmt/tutorial.html">
					MaxQuant and MSstats for the analysis of TMT data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/msconvert/msconvert.html">galaxyp/msconvert/msconvert</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html">
					Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html">
					DIA Analysis using OpenSwathWorkflow
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/encyclopedia/tutorial.html">
					EncyclopeDIA
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html">
					metaQuantome 1: Data creation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-4-peptide-verification/tutorial.html">
					Neoantigen 4: PepQuery2 Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/kraken2/kraken2.html">iuc/kraken2/kraken2</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-decontamination/tutorial.html">
					Decontamination of a genome assembly
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/bacterial-isolate-species-contamination-checking/tutorial.html">
					Checking expected species and contamination in bacterial isolate
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/beer-data-analysis/tutorial.html">
					Identification of the micro-organisms in a beer using Nanopore sequencing
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/nanopore-16S-metagenomics/tutorial.html">
					16S Microbial analysis with Nanopore data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/taxonomic-profiling/tutorial.html">
					Taxonomic Profiling and Visualization of Metagenomic Data
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/quality-contamination-control/tutorial.html">
					Quality and contamination control in bacterial isolate using Illumina MiSeq Data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/collection_column_join/collection_column_join.html">iuc/collection_column_join/collection_column_join</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html">
					Pre-processing of Single-Cell RNA Data
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">
					1: RNA-Seq reads to counts
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/pjbriggs/trimmomatic/trimmomatic.html">pjbriggs/trimmomatic/trimmomatic</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html">
					Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)
					</a>
				</li>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html">
					Removal of human reads from SARS-CoV-2 sequencing data
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html">
					De novo transcriptome reconstruction with RNA-Seq
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">
					Exome sequencing data analysis for diagnosing a genetic disease
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/samtools_view/samtools_view.html">iuc/samtools_view/samtools_view</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">
					Exome sequencing data analysis for diagnosing a genetic disease
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/__EXTRACT_DATASET__.html">__EXTRACT_DATASET__</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/fates/tutorial.html">
					Functionally Assembled Terrestrial Ecosystem Simulator (FATES)
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/pangeo/tutorial.html">
					Pangeo ecosystem 101 for everyone - Introduction to Xarray Galaxy Tools
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html">
					Essential genes detection with Transposon insertion sequencing
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/crs4/taxonomy_krona_chart/taxonomy_krona_chart.html">crs4/taxonomy_krona_chart/taxonomy_krona_chart</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/beer-data-analysis/tutorial.html">
					Identification of the micro-organisms in a beer using Nanopore sequencing
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/nanopore-16S-metagenomics/tutorial.html">
					16S Microbial analysis with Nanopore data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/taxonomic-profiling/tutorial.html">
					Taxonomic Profiling and Visualization of Metagenomic Data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/imgteam/unzip/unzip.html">imgteam/unzip/unzip</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Ecoregionalization_tutorial/tutorial.html">
					Ecoregionalization workflow tutorial
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/hela-screen-analysis/tutorial.html">
					Analyse HeLa fluorescence siRNA screen
					</a>
				</li>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html">
					Introduction to Image Analysis using Galaxy
					</a>
				</li>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html">
					Object tracking using CellProfiler
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-ncbi-anndata/tutorial.html">
					Converting NCBI Data to the AnnData Format
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/Paste1.html">Paste1</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ENA_Biodiv_submission/tutorial.html">
					Data submission using ENA upload Tool
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-editor/tutorial.html">
					Creating, Editing and Importing Galaxy Workflows
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html">
					Combining single cell datasets after pre-processing
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-ncbi-anndata/tutorial.html">
					Converting NCBI Data to the AnnData Format
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/datamash_transpose/datamash_transpose.html">iuc/datamash_transpose/datamash_transpose</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-raceid/tutorial.html">
					Downstream Single-cell RNA analysis with RaceID
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html">
					Visualization of RNA-Seq results with heatmap2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/dbbuilder/dbbuilder.html">galaxyp/dbbuilder/dbbuilder</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-1-database-generation/tutorial.html">
					Clinical Metaproteomics 1: Database-Generation
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-1-database-generation/tutorial.html">
					Clinical Metaproteomics 1: Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/database-handling/tutorial.html">
					Protein FASTA Database Handling
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/fasta_to_tabular/fasta2tab.html">devteam/fasta_to_tabular/fasta2tab</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/database-handling/tutorial.html">
					Protein FASTA Database Handling
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html">
					MaxQuant and MSstats for the analysis of label-free data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/samtools_stats/samtools_stats.html">devteam/samtools_stats/samtools_stats</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">
					Mapping
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html">
					Pre-processing of Single-Cell RNA Data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html">
					From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/filter_tabular/filter_tabular.html">iuc/filter_tabular/filter_tabular</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Ecoregionalization_tutorial/tutorial.html">
					Ecoregionalization workflow tutorial
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/nanoplot/nanoplot.html">iuc/nanoplot/nanoplot</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/tutorial.html">
					Chloroplast genome assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html">
					Genome Assembly of MRSA from Oxford Nanopore MinION data (and optionally Illumina data)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/plasmid-metagenomics-nanopore/tutorial.html">
					Antibiotic resistance detection
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">
					Quality Control
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/minimap2/minimap2.html">iuc/minimap2/minimap2</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/plasmid-metagenomics-nanopore/tutorial.html">
					Antibiotic resistance detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/table_compute/table_compute.html">iuc/table_compute/table_compute</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-7-hla-binding-novel-peptides/tutorial.html">
					Neoantigen 7: IEDB binding PepQuery Validated Neopeptides
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html">
					Bulk matrix to ESet | Creating the bulk RNA-seq dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html">
					Removing the effects of the cell cycle
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_tail_tool.html">bgruening/text_processing/tp_tail_tool</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/methylation-seq/tutorial.html">
					DNA Methylation data analysis
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/bacterial-genome-annotation/tutorial.html">
					Bacterial Genome Annotation
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html">
					From peaks to genes
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/picard/picard_MarkDuplicates.html">devteam/picard/picard_MarkDuplicates</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">
					1: RNA-Seq reads to counts
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html">
					Calling variants in non-diploid systems
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html">
					From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/scanpy_plot/scanpy_plot.html">iuc/scanpy_plot/scanpy_plot</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html">
					Pre-processing of 10X Single-Cell ATAC-seq Datasets
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html">
					Removing the effects of the cell cycle
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-plant/tutorial.html">
					Analysis of plant scRNA-Seq Data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/anndata_ops/anndata_ops.html">ebi-gxa/anndata_ops/anndata_ops</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html">
					Importing files from public atlases
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html">
					Combining single cell datasets after pre-processing
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-ncbi-anndata/tutorial.html">
					Converting NCBI Data to the AnnData Format
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/openbabel_compound_convert/openbabel_compound_convert.html">bgruening/openbabel_compound_convert/openbabel_compound_convert</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html">
					Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html">
					Data management in Medicinal Chemistry
					</a>
				</li>
				
				<li>
					FAIR Data, Workflows, and Research /
					<a href="/training-material/topics/fair/tutorials/med-chem-data/tutorial.html">
					Data management in Medicinal Chemistry
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html">
					Predicting EI+ mass spectra with QCxMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/hisat2/hisat2.html">iuc/hisat2/hisat2</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html">
					De novo transcriptome reconstruction with RNA-Seq
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">
					1: RNA-Seq reads to counts
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/srna/tutorial.html">
					Differential abundance testing of small RNAs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/peptideshaker/search_gui.html">galaxyp/peptideshaker/search_gui</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html">
					Metaproteomics tutorial
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html">
					metaQuantome 1: Data creation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html">
					Peptide and Protein ID using SearchGUI and PeptideShaker
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html">
					Proteogenomics 2: Database Search
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/peptideshaker/peptide_shaker.html">galaxyp/peptideshaker/peptide_shaker</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html">
					Metaproteomics tutorial
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html">
					metaQuantome 1: Data creation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html">
					Peptide and Protein ID using SearchGUI and PeptideShaker
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html">
					Proteogenomics 2: Database Search
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/macs2/macs2_callpeak.html">iuc/macs2/macs2_callpeak</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
					Identification of the binding sites of the Estrogen receptor
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html">
					Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html">
					Pre-processing of 10X Single-Cell ATAC-seq Datasets
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/deeptools_compute_matrix/deeptools_compute_matrix.html">bgruening/deeptools_compute_matrix/deeptools_compute_matrix</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
					Identification of the binding sites of the Estrogen receptor
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/methylation-seq/tutorial.html">
					DNA Methylation data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html">
					Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/rgrnastar/rna_star.html">iuc/rgrnastar/rna_star</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html">
					Genome annotation with Funannotate
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-1-fusion-database-generation/tutorial.html">
					Neoantigen 1: Fusion-Database-Generation
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html">
					Pre-processing of Single-Cell RNA Data
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/collection_element_identifiers/collection_element_identifiers.html">iuc/collection_element_identifiers/collection_element_identifiers</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ENA_Biodiv_submission/tutorial.html">
					Data submission using ENA upload Tool
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/samtools_idxstats/samtools_idxstats.html">devteam/samtools_idxstats/samtools_idxstats</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
					Identification of the binding sites of the Estrogen receptor
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html">
					Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">
					1: RNA-Seq reads to counts
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/bamtools_filter/bamFilter.html">devteam/bamtools_filter/bamFilter</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">
					Mapping
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html">
					Pre-processing of Single-Cell RNA Data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html">
					Calling variants in non-diploid systems
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/snpeff/snpEff.html">iuc/snpeff/snpEff</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dip/tutorial.html">
					Calling variants in diploid systems
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">
					Exome sequencing data analysis for diagnosing a genetic disease
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html">
					Mapping and molecular identification of phenotype-causing mutations
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/trio-analysis/tutorial.html">
					Trio Analysis using Synthetic Datasets from RD-Connect GPAP
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/get_pdb/get_pdb.html">bgruening/get_pdb/get_pdb</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/md-simulation-gromacs/tutorial.html">
					Running molecular dynamics simulations using GROMACS
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html">
					Data management in Medicinal Chemistry
					</a>
				</li>
				
				<li>
					FAIR Data, Workflows, and Research /
					<a href="/training-material/topics/fair/tutorials/med-chem-data/tutorial.html">
					Data management in Medicinal Chemistry
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/model_prediction/model_prediction.html">bgruening/model_prediction/model_prediction</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/CNN/tutorial.html">
					Deep Learning (Part 3) - Convolutional neural networks (CNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/FNN/tutorial.html">
					Deep Learning (Part 1) - Feedforward neural networks (FNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/RNN/tutorial.html">
					Deep Learning (Part 2) - Recurrent neural networks (RNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/fruit_360/tutorial.html">
					Image classification in Galaxy with fruit 360 dataset
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/intro_deep_learning/tutorial.html">
					Introduction to deep learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/keras_train_and_eval/keras_train_and_eval.html">bgruening/keras_train_and_eval/keras_train_and_eval</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/CNN/tutorial.html">
					Deep Learning (Part 3) - Convolutional neural networks (CNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/FNN/tutorial.html">
					Deep Learning (Part 1) - Feedforward neural networks (FNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/RNN/tutorial.html">
					Deep Learning (Part 2) - Recurrent neural networks (RNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/fruit_360/tutorial.html">
					Image classification in Galaxy with fruit 360 dataset
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/intro_deep_learning/tutorial.html">
					Introduction to deep learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/keras_model_config/keras_model_config.html">bgruening/keras_model_config/keras_model_config</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/CNN/tutorial.html">
					Deep Learning (Part 3) - Convolutional neural networks (CNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/FNN/tutorial.html">
					Deep Learning (Part 1) - Feedforward neural networks (FNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/RNN/tutorial.html">
					Deep Learning (Part 2) - Recurrent neural networks (RNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/fruit_360/tutorial.html">
					Image classification in Galaxy with fruit 360 dataset
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/intro_deep_learning/tutorial.html">
					Introduction to deep learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/keras_model_builder/keras_model_builder.html">bgruening/keras_model_builder/keras_model_builder</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/CNN/tutorial.html">
					Deep Learning (Part 3) - Convolutional neural networks (CNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/FNN/tutorial.html">
					Deep Learning (Part 1) - Feedforward neural networks (FNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/RNN/tutorial.html">
					Deep Learning (Part 2) - Recurrent neural networks (RNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/fruit_360/tutorial.html">
					Image classification in Galaxy with fruit 360 dataset
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/intro_deep_learning/tutorial.html">
					Introduction to deep learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/split_file_on_column/tp_split_on_column.html">bgruening/split_file_on_column/tp_split_on_column</a></td>
		<td>
			<ul>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/GO-enrichment/tutorial.html">
					GO Enrichment Analysis on Single-Cell RNA-Seq Data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/gffread/gffread.html">devteam/gffread/gffread</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html">
					Genome annotation with Maker
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker-short/tutorial.html">
					Genome annotation with Maker (short)
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/helixer/tutorial.html">
					Genome annotation with Helixer
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/lncrna/tutorial.html">
					Long non-coding RNAs (lncRNAs) annotation with FEELnc
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/annotatemyids/annotatemyids.html">iuc/annotatemyids/annotatemyids</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html">
					Bulk matrix to ESet | Creating the bulk RNA-seq dataset for deconvolution
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html">
					2: RNA-seq counts to genes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/stringtie/stringtie.html">iuc/stringtie/stringtie</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/lncrna/tutorial.html">
					Long non-coding RNAs (lncRNAs) annotation with FEELnc
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html">
					De novo transcriptome reconstruction with RNA-Seq
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/Summary_Statistics1.html">Summary_Statistics1</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/gbif_cleaning/tutorial.html">
					Cleaning GBIF data for the use in Ecology
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html">
					Mass spectrometry imaging: Finding differential analytes
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html">
					MaxQuant and MSstats for the analysis of label-free data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/ncbi_blast_plus/ncbi_blastp_wrapper.html">devteam/ncbi_blast_plus/ncbi_blastp_wrapper</a></td>
		<td>
			<ul>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/online-resources-gene/tutorial.html">
					Learning about one gene across biological resources and formats
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/functional/tutorial.html">
					Functional annotation of protein sequences
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-4-peptide-verification/tutorial.html">
					Neoantigen 4: PepQuery2 Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.html">
					Proteogenomics 3: Novel peptide analysis
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/ncbi-blast-against-the-madland/tutorial.html">
					NCBI BLAST+ against the MAdLand
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/CONVERTER_gz_to_uncompressed.html">CONVERTER_gz_to_uncompressed</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-1-fusion-database-generation/tutorial.html">
					Neoantigen 1: Fusion-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/trio-analysis/tutorial.html">
					Trio Analysis using Synthetic Datasets from RD-Connect GPAP
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/sra_tools/fasterq_dump.html">iuc/sra_tools/fasterq_dump</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html">
					SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html">
					From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bandage/bandage_info.html">iuc/bandage/bandage_info</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/tutorial.html">
					Chloroplast genome assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html">
					Genome Assembly of a bacterial genome (MRSA) sequenced using Illumina MiSeq Data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/fasta_stats/fasta-stats.html">iuc/fasta_stats/fasta-stats</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/tutorial.html">
					Chloroplast genome assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/debruijn-graph-assembly/tutorial.html">
					De Bruijn Graph Assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/flye-assembly/tutorial.html">
					Genome assembly using PacBio data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html">
					Genome annotation with Maker
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/flye/flye.html">bgruening/flye/flye</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/tutorial.html">
					Chloroplast genome assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/flye-assembly/tutorial.html">
					Genome assembly using PacBio data
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html">
					Genome Assembly of MRSA from Oxford Nanopore MinION data (and optionally Illumina data)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/seqtk/seqtk_subseq.html">iuc/seqtk/seqtk_subseq</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">
					Creating a new tutorial
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html">
					Removal of human reads from SARS-CoV-2 sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/random_lines1.html">random_lines1</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">
					Creating a new tutorial
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/circos/tutorial.html">
					Visualisation with Circos
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bedtools/bedtools_intersectbed.html">iuc/bedtools/bedtools_intersectbed</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html">
					Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-101/tutorial.html">
					Galaxy Basics for genomics
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_head_tool.html">bgruening/text_processing/tp_head_tool</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-parameters/tutorial.html">
					Using Workflow Parameters
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-101/tutorial.html">
					Galaxy Basics for genomics
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-raceid/tutorial.html">
					Downstream Single-cell RNA analysis with RaceID
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/trim_galore/trim_galore.html">bgruening/trim_galore/trim_galore</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
					Identification of the binding sites of the Estrogen receptor
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">
					Mapping
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/srna/tutorial.html">
					Differential abundance testing of small RNAs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/deeptools_plot_heatmap/deeptools_plot_heatmap.html">bgruening/deeptools_plot_heatmap/deeptools_plot_heatmap</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
					Identification of the binding sites of the Estrogen receptor
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html">
					Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/deeptools_bam_coverage/deeptools_bam_coverage.html">bgruening/deeptools_bam_coverage/deeptools_bam_coverage</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
					Identification of the binding sites of the Estrogen receptor
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html">
					Essential genes detection with Transposon insertion sequencing
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html">
					De novo transcriptome reconstruction with RNA-Seq
					</a>
				</li>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/circos-microbial/tutorial.html">
					Ploting a Microbial Genome with Circos
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/featurecounts/featurecounts.html">iuc/featurecounts/featurecounts</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html">
					Pre-processing of Single-Cell RNA Data
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html">
					De novo transcriptome reconstruction with RNA-Seq
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">
					1: RNA-Seq reads to counts
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/deseq2/deseq2.html">iuc/deseq2/deseq2</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html">
					De novo transcriptome reconstruction with RNA-Seq
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html">
					Network analysis with Heinz
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/srna/tutorial.html">
					Differential abundance testing of small RNAs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/__MERGE_COLLECTION__.html">__MERGE_COLLECTION__</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/pangeo/tutorial.html">
					Pangeo ecosystem 101 for everyone - Introduction to Xarray Galaxy Tools
					</a>
				</li>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/__FLATTEN__.html">__FLATTEN__</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html">
					Pre-processing of Single-Cell RNA Data
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/csv_to_tabular.html">csv_to_tabular</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-reports/tutorial.html">
					Workflow Reports
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">
					Galaxy Basics for everyone
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-reproduce/tutorial.html">
					How to reproduce published Galaxy analyses
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Seurat
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/clustering_machinelearning/tutorial.html">
					Clustering in Machine Learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/freebayes/freebayes.html">devteam/freebayes/freebayes</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dip/tutorial.html">
					Calling variants in diploid systems
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">
					Exome sequencing data analysis for diagnosing a genetic disease
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html">
					Calling variants in non-diploid systems
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/gemini_load/gemini_load.html">iuc/gemini_load/gemini_load</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dip/tutorial.html">
					Calling variants in diploid systems
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">
					Exome sequencing data analysis for diagnosing a genetic disease
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/trio-analysis/tutorial.html">
					Trio Analysis using Synthetic Datasets from RD-Connect GPAP
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_jupyter_notebook.html">interactive_tool_jupyter_notebook</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/fates-jupyterlab/tutorial.html">
					Functionally Assembled Terrestrial Ecosystem Simulator (FATES) with Galaxy Climate JupyterLab
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Ecoregionalization_tutorial/tutorial.html">
					Ecoregionalization workflow tutorial
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/jupyterlab/tutorial.html">
					JupyterLab in Galaxy
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html">
					Secondary metabolite discovery
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/ml_visualization_ex/ml_visualization_ex.html">bgruening/ml_visualization_ex/ml_visualization_ex</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/CNN/tutorial.html">
					Deep Learning (Part 3) - Convolutional neural networks (CNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/RNN/tutorial.html">
					Deep Learning (Part 2) - Recurrent neural networks (RNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/fruit_360/tutorial.html">
					Image classification in Galaxy with fruit 360 dataset
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/intro_deep_learning/tutorial.html">
					Introduction to deep learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/scanpy_inspect/scanpy_inspect.html">iuc/scanpy_inspect/scanpy_inspect</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html">
					Removing the effects of the cell cycle
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-plant/tutorial.html">
					Analysis of plant scRNA-Seq Data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/music_construct_eset/music_construct_eset.html">bgruening/music_construct_eset/music_construct_eset</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music/tutorial.html">
					Bulk RNA Deconvolution with MuSiC
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html">
					Bulk matrix to ESet | Creating the bulk RNA-seq dataset for deconvolution
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/retrieve_scxa/retrieve_scxa.html">ebi-gxa/retrieve_scxa/retrieve_scxa</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html">
					Importing files from public atlases
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html">
					Filter, plot, and explore single cell RNA-seq data with Seurat
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html">
					Combining single cell datasets after pre-processing
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/lecorguille/xcms_xcmsset/abims_xcms_xcmsSet.html">lecorguille/xcms_xcmsset/abims_xcms_xcmsSet</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html">
					Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gcms/tutorial.html">
					Mass spectrometry : GC-MS analysis with metaMS package
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms/tutorial.html">
					Mass spectrometry: LC-MS analysis
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html">
					Mass spectrometry: LC-MS preprocessing with XCMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/lecorguille/xcms_merge/xcms_merge.html">lecorguille/xcms_merge/xcms_merge</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html">
					Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gcms/tutorial.html">
					Mass spectrometry : GC-MS analysis with metaMS package
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms/tutorial.html">
					Mass spectrometry: LC-MS analysis
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html">
					Mass spectrometry: LC-MS preprocessing with XCMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/lecorguille/xcms_group/abims_xcms_group.html">lecorguille/xcms_group/abims_xcms_group</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html">
					Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gcms/tutorial.html">
					Mass spectrometry : GC-MS analysis with metaMS package
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms/tutorial.html">
					Mass spectrometry: LC-MS analysis
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html">
					Mass spectrometry: LC-MS preprocessing with XCMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/lecorguille/xcms_fillpeaks/abims_xcms_fillPeaks.html">lecorguille/xcms_fillpeaks/abims_xcms_fillPeaks</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html">
					Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gcms/tutorial.html">
					Mass spectrometry : GC-MS analysis with metaMS package
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms/tutorial.html">
					Mass spectrometry: LC-MS analysis
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html">
					Mass spectrometry: LC-MS preprocessing with XCMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/lecorguille/msnbase_readmsdata/msnbase_readmsdata.html">lecorguille/msnbase_readmsdata/msnbase_readmsdata</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html">
					Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gcms/tutorial.html">
					Mass spectrometry : GC-MS analysis with metaMS package
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms/tutorial.html">
					Mass spectrometry: LC-MS analysis
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html">
					Mass spectrometry: LC-MS preprocessing with XCMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/rnateam/mafft/rbc_mafft.html">rnateam/mafft/rbc_mafft</a></td>
		<td>
			<ul>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/abc_intro_phylo/tutorial.html">
					Phylogenetics - Back to basics
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/Show beginning1.html">Show beginning1</a></td>
		<td>
			<ul>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-editor/tutorial.html">
					Creating, Editing and Importing Galaxy Workflows
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_panoply.html">interactive_tool_panoply</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/fates/tutorial.html">
					Functionally Assembled Terrestrial Ecosystem Simulator (FATES)
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/panoply/tutorial.html">
					Visualize Climate data with Panoply netCDF viewer
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/sentinel5_data/tutorial.html">
					Sentinel 5P data visualisation
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/panoply_ebv/tutorial.html">
					Visualize EBV cube data with Panoply netCDF viewer
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/xarray_metadata_info/xarray_metadata_info.html">ecology/xarray_metadata_info/xarray_metadata_info</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/argo_pangeo/tutorial.html">
					Analyse Argo data
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/fates/tutorial.html">
					Functionally Assembled Terrestrial Ecosystem Simulator (FATES)
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/pangeo/tutorial.html">
					Pangeo ecosystem 101 for everyone - Introduction to Xarray Galaxy Tools
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/x-array-map-plot/tutorial.html">
					Visualization of Climate Data using  NetCDF xarray Map Plotting
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/snpsift/snpSift_extractFields.html">iuc/snpsift/snpSift_extractFields</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html">
					Using dataset collections
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html">
					From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bcftools_consensus/bcftools_consensus.html">iuc/bcftools_consensus/bcftools_consensus</a></td>
		<td>
			<ul>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-automation/tutorial.html">
					Automating Galaxy workflows using the command line
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/split_file_to_collection/split_file_to_collection.html">bgruening/split_file_to_collection/split_file_to_collection</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html">
					Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/zauberkugel/tutorial.html">
					Protein target prediction of a bioactive ligand with Align-it and ePharmaLib
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html">
					Predicting EI+ mass spectra with QCxMS
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/newick_utils/newick_display.html">iuc/newick_utils/newick_display</a></td>
		<td>
			<ul>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/abc_intro_phylo/tutorial.html">
					Phylogenetics - Back to basics
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/gffcompare/gffcompare.html">iuc/gffcompare/gffcompare</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html">
					De novo transcriptome reconstruction with RNA-Seq
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/uniprotxml_downloader/uniprotxml_downloader.html">galaxyp/uniprotxml_downloader/uniprotxml_downloader</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-1-database-generation/tutorial.html">
					Clinical Metaproteomics 1: Database-Generation
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-1-database-generation/tutorial.html">
					Clinical Metaproteomics 1: Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/unipept/unipept.html">galaxyp/unipept/unipept</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-5-data-interpretation/tutorial.html">
					Clinical Metaproteomics 5: Data Interpretation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-5-data-interpretation/tutorial.html">
					Clinical Metaproteomics 5: Data Interpretation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html">
					Metaproteomics tutorial
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html">
					metaQuantome 1: Data creation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/pepquery2/pepquery2.html">galaxyp/pepquery2/pepquery2</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-3-verification/tutorial.html">
					Clinical Metaproteomics 3: Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-4-peptide-verification/tutorial.html">
					Neoantigen 4: PepQuery2 Verification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-7-hla-binding-novel-peptides/tutorial.html">
					Neoantigen 7: IEDB binding PepQuery Validated Neopeptides
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/diamond/bg_diamond.html">bgruening/diamond/bg_diamond</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/functional/tutorial.html">
					Functional annotation of protein sequences
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/ncbi-blast-against-the-madland/tutorial.html">
					NCBI BLAST+ against the MAdLand
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/Convert characters1.html">Convert characters1</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-5-variant-annotation/tutorial.html">
					Neoantigen 5: Variant Annotation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html">
					Secretome Prediction
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/porechop/porechop.html">iuc/porechop/porechop</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html">
					Genome Assembly of MRSA from Oxford Nanopore MinION data (and optionally Illumina data)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/beer-data-analysis/tutorial.html">
					Identification of the micro-organisms in a beer using Nanopore sequencing
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/nanopore-16S-metagenomics/tutorial.html">
					16S Microbial analysis with Nanopore data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/ncbi_blast_plus/ncbi_blastn_wrapper.html">devteam/ncbi_blast_plus/ncbi_blastn_wrapper</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-decontamination/tutorial.html">
					Decontamination of a genome assembly
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ENA_Biodiv_submission/tutorial.html">
					Data submission using ENA upload Tool
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/unique/bg_uniq.html">bgruening/unique/bg_uniq</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ENA_Biodiv_submission/tutorial.html">
					Data submission using ENA upload Tool
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html">
					Secretome Prediction
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/circos/circos.html">iuc/circos/circos</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/circos/tutorial.html">
					Visualisation with Circos
					</a>
				</li>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/circos-microbial/tutorial.html">
					Ploting a Microbial Genome with Circos
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/merge_cols/mergeCols1.html">devteam/merge_cols/mergeCols1</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Ecoregionalization_tutorial/tutorial.html">
					Ecoregionalization workflow tutorial
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html">
					2: RNA-seq counts to genes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/meryl/meryl.html">iuc/meryl/meryl</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-quality-control/tutorial.html">
					Genome Assembly Quality Control
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/deeptools_plot_fingerprint/deeptools_plot_fingerprint.html">bgruening/deeptools_plot_fingerprint/deeptools_plot_fingerprint</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
					Identification of the binding sites of the Estrogen receptor
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html">
					Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/deeptools_plot_correlation/deeptools_plot_correlation.html">bgruening/deeptools_plot_correlation/deeptools_plot_correlation</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
					Identification of the binding sites of the Estrogen receptor
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html">
					Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/deeptools_multi_bam_summary/deeptools_multi_bam_summary.html">bgruening/deeptools_multi_bam_summary/deeptools_multi_bam_summary</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
					Identification of the binding sites of the Estrogen receptor
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html">
					Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nilesh/rseqc/rseqc_read_distribution.html">nilesh/rseqc/rseqc_read_distribution</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">
					1: RNA-Seq reads to counts
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nilesh/rseqc/rseqc_geneBody_coverage.html">nilesh/rseqc/rseqc_geneBody_coverage</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">
					1: RNA-Seq reads to counts
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/pygenometracks/pygenomeTracks.html">iuc/pygenometracks/pygenomeTracks</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/sklearn_ensemble/sklearn_ensemble.html">bgruening/sklearn_ensemble/sklearn_ensemble</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/age-prediction-with-ml/tutorial.html">
					Age prediction using machine learning
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/classification_machinelearning/tutorial.html">
					Classification in Machine Learning
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/classification_regression/tutorial.html">
					Machine learning: classification and regression
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/regression_machinelearning/tutorial.html">
					Regression in Machine Learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/plotly_regression_performance_plots/plotly_regression_performance_plots.html">bgruening/plotly_regression_performance_plots/plotly_regression_performance_plots</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/FNN/tutorial.html">
					Deep Learning (Part 1) - Feedforward neural networks (FNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/age-prediction-with-ml/tutorial.html">
					Age prediction using machine learning
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/classification_regression/tutorial.html">
					Machine learning: classification and regression
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/regression_machinelearning/tutorial.html">
					Regression in Machine Learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/histogram/histogram_rpy.html">devteam/histogram/histogram_rpy</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_Analysis_MSstats/tutorial.html">
					Statistical analysis of DIA data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-label-free/tutorial.html">
					Label-free data analysis using MaxQuant
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/freebayes/bamleftalign.html">devteam/freebayes/bamleftalign</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dunovo/tutorial.html">
					Calling very rare variants
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html">
					Calling variants in non-diploid systems
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/samtools_rmdup/samtools_rmdup.html">devteam/samtools_rmdup/samtools_rmdup</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">
					Exome sequencing data analysis for diagnosing a genetic disease
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/qualimap_bamqc/qualimap_bamqc.html">iuc/qualimap_bamqc/qualimap_bamqc</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/snpeff/snpEff_download.html">iuc/snpeff/snpEff_download</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dip/tutorial.html">
					Calling variants in diploid systems
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">
					Exome sequencing data analysis for diagnosing a genetic disease
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html">
					Mapping and molecular identification of phenotype-causing mutations
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/jcvi_gff_stats/jcvi_gff_stats.html">iuc/jcvi_gff_stats/jcvi_gff_stats</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html">
					Genome annotation with Maker
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker-short/tutorial.html">
					Genome annotation with Maker (short)
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/helixer/tutorial.html">
					Genome annotation with Helixer
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/md_converter/md_converter.html">chemteam/md_converter/md_converter</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html">
					Data management in Medicinal Chemistry
					</a>
				</li>
				
				<li>
					FAIR Data, Workflows, and Research /
					<a href="/training-material/topics/fair/tutorials/med-chem-data/tutorial.html">
					Data management in Medicinal Chemistry
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/qed/ctb_silicos_qed.html">bgruening/qed/ctb_silicos_qed</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html">
					Data management in Medicinal Chemistry
					</a>
				</li>
				
				<li>
					FAIR Data, Workflows, and Research /
					<a href="/training-material/topics/fair/tutorials/med-chem-data/tutorial.html">
					Data management in Medicinal Chemistry
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html">
					Secondary metabolite discovery
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/openbabel_svg_depiction/openbabel_svg_depiction.html">bgruening/openbabel_svg_depiction/openbabel_svg_depiction</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html">
					Data management in Medicinal Chemistry
					</a>
				</li>
				
				<li>
					FAIR Data, Workflows, and Research /
					<a href="/training-material/topics/fair/tutorials/med-chem-data/tutorial.html">
					Data management in Medicinal Chemistry
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/chembl/chembl.html">bgruening/chembl/chembl</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html">
					Data management in Medicinal Chemistry
					</a>
				</li>
				
				<li>
					FAIR Data, Workflows, and Research /
					<a href="/training-material/topics/fair/tutorials/med-chem-data/tutorial.html">
					Data management in Medicinal Chemistry
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/spocc_occ/spocc_occ.html">ecology/spocc_occ/spocc_occ</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/earth_system/tutorial.html">
					Getting your hands-on earth data
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/gbif_cleaning/tutorial.html">
					Cleaning GBIF data for the use in Ecology
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/species-distribution-modeling/tutorial.html">
					Species distribution modeling
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/interproscan/interproscan.html">bgruening/interproscan/interproscan</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/marine_omics_bgc/tutorial.html">
					Marine Omics identifying biosynthetic gene clusters
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html">
					Genome annotation with Funannotate
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/functional/tutorial.html">
					Functional annotation of protein sequences
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/__RELABEL_FROM_FILE__.html">__RELABEL_FROM_FILE__</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ENA_Biodiv_submission/tutorial.html">
					Data submission using ENA upload Tool
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/fastq_groomer/fastq_groomer.html">devteam/fastq_groomer/fastq_groomer</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ENA_Biodiv_submission/tutorial.html">
					Data submission using ENA upload Tool
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/sklearn_searchcv/sklearn_searchcv.html">bgruening/sklearn_searchcv/sklearn_searchcv</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/age-prediction-with-ml/tutorial.html">
					Age prediction using machine learning
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/classification_machinelearning/tutorial.html">
					Classification in Machine Learning
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/regression_machinelearning/tutorial.html">
					Regression in Machine Learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/sklearn_build_pipeline/sklearn_build_pipeline.html">bgruening/sklearn_build_pipeline/sklearn_build_pipeline</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/age-prediction-with-ml/tutorial.html">
					Age prediction using machine learning
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/classification_machinelearning/tutorial.html">
					Classification in Machine Learning
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/regression_machinelearning/tutorial.html">
					Regression in Machine Learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2.html">iuc/snpeff_sars_cov_2/snpeff_sars_cov_2</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html">
					SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html">
					From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/lofreq_call/lofreq_call.html">iuc/lofreq_call/lofreq_call</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html">
					Using dataset collections
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html">
					From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/scanpy_cluster_reduce_dimension/scanpy_cluster_reduce_dimension.html">iuc/scanpy_cluster_reduce_dimension/scanpy_cluster_reduce_dimension</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html">
					Removing the effects of the cell cycle
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-plant/tutorial.html">
					Analysis of plant scRNA-Seq Data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/anndata_import/anndata_import.html">iuc/anndata_import/anndata_import</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-ncbi-anndata/tutorial.html">
					Converting NCBI Data to the AnnData Format
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-plant/tutorial.html">
					Analysis of plant scRNA-Seq Data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/scanpy_normalize/scanpy_normalize.html">iuc/scanpy_normalize/scanpy_normalize</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-plant/tutorial.html">
					Analysis of plant scRNA-Seq Data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/scanpy_filter/scanpy_filter.html">iuc/scanpy_filter/scanpy_filter</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-plant/tutorial.html">
					Analysis of plant scRNA-Seq Data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/sceasy_convert/sceasy_convert.html">iuc/sceasy_convert/sceasy_convert</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html">
					Importing files from public atlases
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html">
					Generating a single cell matrix using Alevin
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_read_10x/scanpy_read_10x.html">ebi-gxa/scanpy_read_10x/scanpy_read_10x</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html">
					Importing files from public atlases
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html">
					Combining single cell datasets after pre-processing
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_plot_embed/scanpy_plot_embed.html">ebi-gxa/scanpy_plot_embed/scanpy_plot_embed</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scanpy_parameter_iterator/tutorial.html">
					Scanpy Parameter Iterator
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_trajectories/tutorial.html">
					Inferring single cell trajectories with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_find_cluster/scanpy_find_cluster.html">ebi-gxa/scanpy_find_cluster/scanpy_find_cluster</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scanpy_parameter_iterator/tutorial.html">
					Scanpy Parameter Iterator
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_trajectories/tutorial.html">
					Inferring single cell trajectories with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_filter_cells/scanpy_filter_cells.html">ebi-gxa/scanpy_filter_cells/scanpy_filter_cells</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-ncbi-anndata/tutorial.html">
					Converting NCBI Data to the AnnData Format
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_compute_graph/scanpy_compute_graph.html">ebi-gxa/scanpy_compute_graph/scanpy_compute_graph</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scanpy_parameter_iterator/tutorial.html">
					Scanpy Parameter Iterator
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_trajectories/tutorial.html">
					Inferring single cell trajectories with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/seurat_read10x/seurat_read10x.html">ebi-gxa/seurat_read10x/seurat_read10x</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html">
					Importing files from public atlases
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html">
					Filter, plot, and explore single cell RNA-seq data with Seurat
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/lecorguille/xcms_retcor/abims_xcms_retcor.html">lecorguille/xcms_retcor/abims_xcms_retcor</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html">
					Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms/tutorial.html">
					Mass spectrometry: LC-MS analysis
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html">
					Mass spectrometry: LC-MS preprocessing with XCMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/snippy/snippy.html">iuc/snippy/snippy</a></td>
		<td>
			<ul>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/microbial-variants/tutorial.html">
					Microbial Variant Calling
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/wc_gnu.html">wc_gnu</a></td>
		<td>
			<ul>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/tabular_to_csv.html">tabular_to_csv</a></td>
		<td>
			<ul>
				
				<li>
					Foundations of Data Science /
					<a href="/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/species-distribution-modeling/tutorial.html">
					Species distribution modeling
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">
					Data Manipulation Olympics
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/xarray_coords_info/xarray_coords_info.html">ecology/xarray_coords_info/xarray_coords_info</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/argo_pangeo/tutorial.html">
					Analyse Argo data
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/pangeo/tutorial.html">
					Pangeo ecosystem 101 for everyone - Introduction to Xarray Galaxy Tools
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/x-array-map-plot/tutorial.html">
					Visualization of Climate Data using  NetCDF xarray Map Plotting
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_odv.html">interactive_tool_odv</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/argo_pangeo/tutorial.html">
					Analyse Argo data
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/ocean-data-view/tutorial.html">
					Ocean Data View (ODV)
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/ocean-variables/tutorial.html">
					Ocean's variables study
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bracken/est_abundance.html">iuc/bracken/est_abundance</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/bacterial-isolate-species-contamination-checking/tutorial.html">
					Checking expected species and contamination in bacterial isolate
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/taxonomic-profiling/tutorial.html">
					Taxonomic Profiling and Visualization of Metagenomic Data
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/quality-contamination-control/tutorial.html">
					Quality and contamination control in bacterial isolate using Illumina MiSeq Data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/__UNZIP_COLLECTION__.html">__UNZIP_COLLECTION__</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/taxonomic-profiling/tutorial.html">
					Taxonomic Profiling and Visualization of Metagenomic Data
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html">
					Removal of human reads from SARS-CoV-2 sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/krakentools_kreport2krona/krakentools_kreport2krona.html">iuc/krakentools_kreport2krona/krakentools_kreport2krona</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/beer-data-analysis/tutorial.html">
					Identification of the micro-organisms in a beer using Nanopore sequencing
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/taxonomic-profiling/tutorial.html">
					Taxonomic Profiling and Visualization of Metagenomic Data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/compose_text_param/compose_text_param.html">iuc/compose_text_param/compose_text_param</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-automation/tutorial.html">
					Automating Galaxy workflows using the command line
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bcftools_norm/bcftools_norm.html">iuc/bcftools_norm/bcftools_norm</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">
					Exome sequencing data analysis for diagnosing a genetic disease
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/trio-analysis/tutorial.html">
					Trio Analysis using Synthetic Datasets from RD-Connect GPAP
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/clustalw/clustalw.html">devteam/clustalw/clustalw</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html">
					Preparing genomic data for phylogeny reconstruction
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/hpc-for-lsgc/tutorial.html">
					From small to large-scale genome comparison
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/param_value_from_file.html">param_value_from_file</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html">
					Predicting EI+ mass spectra with QCxMS
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/metaphlan/metaphlan.html">iuc/metaphlan/metaphlan</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/taxonomic-profiling/tutorial.html">
					Taxonomic Profiling and Visualization of Metagenomic Data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/volcanoplot/volcanoplot.html">iuc/volcanoplot/volcanoplot</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/crispr-screen/tutorial.html">
					CRISPR screen analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot/tutorial.html">
					Visualization of RNA-Seq results with Volcano Plot
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot-r/tutorial.html">
					Visualization of RNA-Seq results with Volcano Plot in R
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/umi_tools_extract/umi_tools_extract.html">iuc/umi_tools_extract/umi_tools_extract</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html">
					Pre-processing of Single-Cell RNA Data
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-umis/tutorial.html">
					Understanding Barcodes
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/__SORTLIST__.html">__SORTLIST__</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/addValue.html">addValue</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/database-handling/tutorial.html">
					Protein FASTA Database Handling
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/msstatstmt/msstatstmt.html">galaxyp/msstatstmt/msstatstmt</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-5-data-interpretation/tutorial.html">
					Clinical Metaproteomics 5: Data Interpretation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-5-data-interpretation/tutorial.html">
					Clinical Metaproteomics 5: Data Interpretation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-tmt/tutorial.html">
					MaxQuant and MSstats for the analysis of TMT data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/cardinal_quality_report/cardinal_quality_report.html">galaxyp/cardinal_quality_report/cardinal_quality_report</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-analyte-distribution/tutorial.html">
					Mass spectrometry imaging: Examining the spatial distribution of analytes
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html">
					Mass spectrometry imaging: Finding differential analytes
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/mass-spectrometry-imaging-loading-exploring-data/tutorial.html">
					Mass spectrometry imaging: Loading and exploring MSI data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/proteore/proteore_filter_keywords_values/MQoutputfilter.html">proteore/proteore_filter_keywords_values/MQoutputfilter</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html">
					Biomarker candidate identification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteome_annotation/tutorial.html">
					Annotating a protein list identified by LC-MS/MS experiments
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/sqlite_to_tabular/sqlite_to_tabular.html">iuc/sqlite_to_tabular/sqlite_to_tabular</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html">
					Metaproteomics tutorial
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/merqury/merqury.html">iuc/merqury/merqury</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-quality-control/tutorial.html">
					Genome Assembly Quality Control
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bwa_mem2/bwa_mem2.html">iuc/bwa_mem2/bwa_mem2</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html">
					Genome Assembly of MRSA from Oxford Nanopore MinION data (and optionally Illumina data)
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/unicycler/unicycler.html">iuc/unicycler/unicycler</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/unicycler-assembly/tutorial.html">
					Unicycler Assembly
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/plasmid-metagenomics-nanopore/tutorial.html">
					Antibiotic resistance detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/samtools_fastx/samtools_fastx.html">iuc/samtools_fastx/samtools_fastx</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html">
					Removal of human reads from SARS-CoV-2 sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/__ZIP_COLLECTION__.html">__ZIP_COLLECTION__</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html">
					Removal of human reads from SARS-CoV-2 sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/gfastats/gfastats.html">bgruening/gfastats/gfastats</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-decontamination/tutorial.html">
					Decontamination of a genome assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/megahit/megahit.html">iuc/megahit/megahit</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-binning/tutorial.html">
					Binning of metagenomic sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/fasta_compute_length/fasta_compute_length.html">devteam/fasta_compute_length/fasta_compute_length</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/add_value/addValue.html">devteam/add_value/addValue</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html">
					Removing the effects of the cell cycle
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/genomescope/genomescope.html">iuc/genomescope/genomescope</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/bwa/bwa.html">devteam/bwa/bwa</a></td>
		<td>
			<ul>
				
				<li>
					Galaxy Server administration /
					<a href="/training-material/topics/admin/tutorials/reference-genomes/tutorial.html">
					Reference Data with Data Managers
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html">
					Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/wig_to_bigWig.html">wig_to_bigWig</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bedtools/bedtools_bamtobed.html">iuc/bedtools/bedtools_bamtobed</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/fastq_to_tabular/fastq_to_tabular.html">devteam/fastq_to_tabular/fastq_to_tabular</a></td>
		<td>
			<ul>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">
					Creating a new tutorial
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bedtools/bedtools_sortbed.html">iuc/bedtools/bedtools_sortbed</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html">
					Pre-processing of 10X Single-Cell ATAC-seq Datasets
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/deeptools_bam_compare/deeptools_bam_compare.html">bgruening/deeptools_bam_compare/deeptools_bam_compare</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
					Identification of the binding sites of the Estrogen receptor
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html">
					Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nilesh/rseqc/rseqc_infer_experiment.html">nilesh/rseqc/rseqc_infer_experiment</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">
					1: RNA-Seq reads to counts
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/goseq/goseq.html">iuc/goseq/goseq</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html">
					3: RNA-seq genes to pathways
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/ggplot2_heatmap2/ggplot2_heatmap2.html">iuc/ggplot2_heatmap2/ggplot2_heatmap2</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html">
					MaxQuant and MSstats for the analysis of label-free data
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html">
					Visualization of RNA-Seq results with heatmap2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/__FILTER_FROM_FILE__.html">__FILTER_FROM_FILE__</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/sklearn_svm_classifier/sklearn_svm_classifier.html">bgruening/sklearn_svm_classifier/sklearn_svm_classifier</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/classification_machinelearning/tutorial.html">
					Classification in Machine Learning
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/classification_regression/tutorial.html">
					Machine learning: classification and regression
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/machinelearning/tutorial.html">
					Basics of machine learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/fastq_paired_end_interlacer/fastq_paired_end_interlacer.html">devteam/fastq_paired_end_interlacer/fastq_paired_end_interlacer</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/general-introduction/tutorial.html">
					An Introduction to Genome Assembly
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_unique_seqs/mothur_unique_seqs.html">iuc/mothur_unique_seqs/mothur_unique_seqs</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_summary_seqs/mothur_summary_seqs.html">iuc/mothur_summary_seqs/mothur_summary_seqs</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_screen_seqs/mothur_screen_seqs.html">iuc/mothur_screen_seqs/mothur_screen_seqs</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_pre_cluster/mothur_pre_cluster.html">iuc/mothur_pre_cluster/mothur_pre_cluster</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_make_shared/mothur_make_shared.html">iuc/mothur_make_shared/mothur_make_shared</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_filter_seqs/mothur_filter_seqs.html">iuc/mothur_filter_seqs/mothur_filter_seqs</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_count_seqs/mothur_count_seqs.html">iuc/mothur_count_seqs/mothur_count_seqs</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_cluster_split/mothur_cluster_split.html">iuc/mothur_cluster_split/mothur_cluster_split</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_classify_seqs/mothur_classify_seqs.html">iuc/mothur_classify_seqs/mothur_classify_seqs</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_classify_otu/mothur_classify_otu.html">iuc/mothur_classify_otu/mothur_classify_otu</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/crs4/prokka/prokka.html">crs4/prokka/prokka</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/tutorial.html">
					Chloroplast genome assembly
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/unicycler-assembly/tutorial.html">
					Unicycler Assembly
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-prokka/tutorial.html">
					Genome annotation with Prokka
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/msstats/msstats.html">galaxyp/msstats/msstats</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_Analysis_MSstats/tutorial.html">
					Statistical analysis of DIA data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-label-free/tutorial.html">
					Label-free data analysis using MaxQuant
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html">
					MaxQuant and MSstats for the analysis of label-free data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/intersect/gops_intersect_1.html">devteam/intersect/gops_intersect_1</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html">
					From peaks to genes
					</a>
				</li>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html">
					Introduction to Genomics and Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/beacon2_cnv/beacon2_cnv.html">iuc/beacon2_cnv/beacon2_cnv</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/beacon_cnv_query/tutorial.html">
					Querying the University of Bradford GDC Beacon Database for Copy Number Variants (CNVs)
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/beaconise_1000hg/tutorial.html">
					Working with Beacon V2: A Comprehensive Guide to Creating, Uploading, and Searching for Variants with Beacons
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/melpetera/intensity_checks/intens_check.html">melpetera/intensity_checks/intens_check</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-dataprocessing/tutorial.html">
					Mass spectrometry: LC-MS data processing
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html">
					Mass spectrometry: LC-MS preprocessing with XCMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/mdanalysis_cosine_analysis/mdanalysis_cosine_analysis.html">chemteam/mdanalysis_cosine_analysis/mdanalysis_cosine_analysis</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/analysis-md-simulations/tutorial.html">
					Analysis of molecular dynamics simulations
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/gmx_solvate/gmx_solvate.html">chemteam/gmx_solvate/gmx_solvate</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/md-simulation-gromacs/tutorial.html">
					Running molecular dynamics simulations using GROMACS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/gmx_sim/gmx_sim.html">chemteam/gmx_sim/gmx_sim</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/md-simulation-gromacs/tutorial.html">
					Running molecular dynamics simulations using GROMACS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/gmx_setup/gmx_setup.html">chemteam/gmx_setup/gmx_setup</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/md-simulation-gromacs/tutorial.html">
					Running molecular dynamics simulations using GROMACS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/gmx_em/gmx_em.html">chemteam/gmx_em/gmx_em</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/md-simulation-gromacs/tutorial.html">
					Running molecular dynamics simulations using GROMACS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/gmx_editconf/gmx_editconf.html">chemteam/gmx_editconf/gmx_editconf</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/md-simulation-gromacs/tutorial.html">
					Running molecular dynamics simulations using GROMACS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/bio3d_rmsf/bio3d_rmsf.html">chemteam/bio3d_rmsf/bio3d_rmsf</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/analysis-md-simulations/tutorial.html">
					Analysis of molecular dynamics simulations
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/bio3d_rmsd/bio3d_rmsd.html">chemteam/bio3d_rmsd/bio3d_rmsd</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/analysis-md-simulations/tutorial.html">
					Analysis of molecular dynamics simulations
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/bio3d_pca/bio3d_pca.html">chemteam/bio3d_pca/bio3d_pca</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/analysis-md-simulations/tutorial.html">
					Analysis of molecular dynamics simulations
					</a>
				</li>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/gemini_inheritance/gemini_inheritance.html">iuc/gemini_inheritance/gemini_inheritance</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">
					Exome sequencing data analysis for diagnosing a genetic disease
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/trio-analysis/tutorial.html">
					Trio Analysis using Synthetic Datasets from RD-Connect GPAP
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/samtools_calmd/samtools_calmd.html">devteam/samtools_calmd/samtools_calmd</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/ivar_consensus/ivar_consensus.html">iuc/ivar_consensus/ivar_consensus</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/gemini_query/gemini_query.html">iuc/gemini_query/gemini_query</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dip/tutorial.html">
					Calling variants in diploid systems
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/mbernt/fasta_regex_finder/fasta_regex_finder.html">mbernt/fasta_regex_finder/fasta_regex_finder</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html">
					Mutation calling, viral genome reconstruction and lineage/clade assignment from SARS-CoV-2 sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/eggnog_mapper/eggnog_mapper.html">galaxyp/eggnog_mapper/eggnog_mapper</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html">
					Genome annotation with Funannotate
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/functional/tutorial.html">
					Functional annotation of protein sequences
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/gga/apollo_iframe/iframe.html">gga/apollo_iframe/iframe</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/apollo/tutorial.html">
					Refining Genome Annotations with Apollo (prokaryotes)
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/apollo-euk/tutorial.html">
					Refining Genome Annotations with Apollo (eukaryotes)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/gga/apollo_create_or_update/create_or_update.html">gga/apollo_create_or_update/create_or_update</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/apollo/tutorial.html">
					Refining Genome Annotations with Apollo (prokaryotes)
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/apollo-euk/tutorial.html">
					Refining Genome Annotations with Apollo (eukaryotes)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/tbl2gff3/tbl2gff3.html">iuc/tbl2gff3/tbl2gff3</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/amr-gene-detection/tutorial.html">
					Identification of AMR genes in an assembled bacterial genome
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/bacterial-genome-annotation/tutorial.html">
					Bacterial Genome Annotation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bakta/bakta.html">iuc/bakta/bakta</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/amr-gene-detection/tutorial.html">
					Identification of AMR genes in an assembled bacterial genome
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/bacterial-genome-annotation/tutorial.html">
					Bacterial Genome Annotation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/natural_product_likeness/ctb_np-likeness-calculator.html">bgruening/natural_product_likeness/ctb_np-likeness-calculator</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html">
					Secondary metabolite discovery
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/chemfp/ctb_chemfp_mol2fps.html">bgruening/chemfp/ctb_chemfp_mol2fps</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html">
					Secondary metabolite discovery
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/maker_map_ids/maker_map_ids.html">iuc/maker_map_ids/maker_map_ids</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html">
					Genome annotation with Maker
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker-short/tutorial.html">
					Genome annotation with Maker (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/maker/maker.html">iuc/maker/maker</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html">
					Genome annotation with Maker
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker-short/tutorial.html">
					Genome annotation with Maker (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/ctb_im_rxn_maker/ctb_im_rxn_maker.html">bgruening/ctb_im_rxn_maker/ctb_im_rxn_maker</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html">
					Data management in Medicinal Chemistry
					</a>
				</li>
				
				<li>
					FAIR Data, Workflows, and Research /
					<a href="/training-material/topics/fair/tutorials/med-chem-data/tutorial.html">
					Data management in Medicinal Chemistry
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/obis_data/obis_data.html">ecology/obis_data/obis_data</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/earth_system/tutorial.html">
					Getting your hands-on earth data
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/obisindicators/tutorial.html">
					Obis marine indicators
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/stacks_procrad/stacks_procrad.html">iuc/stacks_procrad/stacks_procrad</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/de-novo-rad-seq/tutorial.html">
					RAD-Seq de-novo data analysis
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/stacks_populations/stacks_populations.html">iuc/stacks_populations/stacks_populations</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/de-novo-rad-seq/tutorial.html">
					RAD-Seq de-novo data analysis
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/stacks_denovomap/stacks_denovomap.html">iuc/stacks_denovomap/stacks_denovomap</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/de-novo-rad-seq/tutorial.html">
					RAD-Seq de-novo data analysis
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/genetic-map-rad-seq/tutorial.html">
					RAD-Seq to construct genetic maps
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/peterjc/seq_filter_by_id/seq_filter_by_id.html">peterjc/seq_filter_by_id/seq_filter_by_id</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-umis/tutorial.html">
					Understanding Barcodes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/funannotate_predict/funannotate_predict.html">iuc/funannotate_predict/funannotate_predict</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html">
					Preparing genomic data for phylogeny reconstruction
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html">
					Genome annotation with Funannotate
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/repeat_masker/repeatmasker_wrapper.html">bgruening/repeat_masker/repeatmasker_wrapper</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html">
					Preparing genomic data for phylogeny reconstruction
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/repeatmasker/tutorial.html">
					Masking repeats with RepeatMasker
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/fastqtofasta/fastq_to_fasta_python.html">devteam/fastqtofasta/fastq_to_fasta_python</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ENA_Biodiv_submission/tutorial.html">
					Data submission using ENA upload Tool
					</a>
				</li>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html">
					SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/recentrifuge/recentrifuge.html">iuc/recentrifuge/recentrifuge</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/bacterial-isolate-species-contamination-checking/tutorial.html">
					Checking expected species and contamination in bacterial isolate
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/quality-contamination-control/tutorial.html">
					Quality and contamination control in bacterial isolate using Illumina MiSeq Data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/ab1_fastq_converter/ab1_fastq_converter.html">ecology/ab1_fastq_converter/ab1_fastq_converter</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ENA_Biodiv_submission/tutorial.html">
					Data submission using ENA upload Tool
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_ml_jupyter_notebook.html">interactive_tool_ml_jupyter_notebook</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/fine_tuning_protTrans/tutorial.html">
					Fine tune large protein model (ProtTrans) using HuggingFace
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/gpu_jupyter_lab/tutorial.html">
					A Docker-based interactive Jupyterlab powered by GPU for artificial intelligence in Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/sklearn_generalized_linear/sklearn_generalized_linear.html">bgruening/sklearn_generalized_linear/sklearn_generalized_linear</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/classification_machinelearning/tutorial.html">
					Classification in Machine Learning
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/regression_machinelearning/tutorial.html">
					Regression in Machine Learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/sklearn_to_categorical/sklearn_to_categorical.html">bgruening/sklearn_to_categorical/sklearn_to_categorical</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/CNN/tutorial.html">
					Deep Learning (Part 3) - Convolutional neural networks (CNN)
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/fruit_360/tutorial.html">
					Image classification in Galaxy with fruit 360 dataset
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/lofreq_viterbi/lofreq_viterbi.html">iuc/lofreq_viterbi/lofreq_viterbi</a></td>
		<td>
			<ul>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html">
					From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/lofreq_indelqual/lofreq_indelqual.html">iuc/lofreq_indelqual/lofreq_indelqual</a></td>
		<td>
			<ul>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">
					NGS data logistics
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html">
					From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/scanpy_remove_confounders/scanpy_remove_confounders.html">iuc/scanpy_remove_confounders/scanpy_remove_confounders</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-plant/tutorial.html">
					Analysis of plant scRNA-Seq Data with Scanpy
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/dropletutils_read_10x/dropletutils_read_10x.html">ebi-gxa/dropletutils_read_10x/dropletutils_read_10x</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html">
					Generating a single cell matrix using Alevin
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/music_manipulate_eset/music_manipulate_eset.html">bgruening/music_manipulate_eset/music_manipulate_eset</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html">
					Bulk matrix to ESet | Creating the bulk RNA-seq dataset for deconvolution
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/monocle3_create/monocle3_create.html">ebi-gxa/monocle3_create/monocle3_create</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/dropletutils/dropletutils.html">iuc/dropletutils/dropletutils</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html">
					Pre-processing of 10X Single-Cell RNA Datasets
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/music_deconvolution/music_deconvolution.html">bgruening/music_deconvolution/music_deconvolution</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music/tutorial.html">
					Bulk RNA Deconvolution with MuSiC
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/column_arrange_by_header/bg_column_arrange_by_header.html">bgruening/column_arrange_by_header/bg_column_arrange_by_header</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_run_umap/scanpy_run_umap.html">ebi-gxa/scanpy_run_umap/scanpy_run_umap</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scanpy_parameter_iterator/tutorial.html">
					Scanpy Parameter Iterator
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_run_tsne/scanpy_run_tsne.html">ebi-gxa/scanpy_run_tsne/scanpy_run_tsne</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scanpy_parameter_iterator/tutorial.html">
					Scanpy Parameter Iterator
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/melpetera/generic_filter/generic_filter.html">melpetera/generic_filter/generic_filter</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms/tutorial.html">
					Mass spectrometry: LC-MS analysis
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-dataprocessing/tutorial.html">
					Mass spectrometry: LC-MS data processing
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/melpetera/batchcorrection/Batch_correction.html">melpetera/batchcorrection/Batch_correction</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms/tutorial.html">
					Mass spectrometry: LC-MS analysis
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-dataprocessing/tutorial.html">
					Mass spectrometry: LC-MS data processing
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/lecorguille/xcms_plot_chromatogram/xcms_plot_chromatogram.html">lecorguille/xcms_plot_chromatogram/xcms_plot_chromatogram</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms/tutorial.html">
					Mass spectrometry: LC-MS analysis
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html">
					Mass spectrometry: LC-MS preprocessing with XCMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/lecorguille/camera_annotate/abims_CAMERA_annotateDiffreport.html">lecorguille/camera_annotate/abims_CAMERA_annotateDiffreport</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms/tutorial.html">
					Mass spectrometry: LC-MS analysis
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html">
					Mass spectrometry: LC-MS preprocessing with XCMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ethevenot/qualitymetrics/quality_metrics.html">ethevenot/qualitymetrics/quality_metrics</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms/tutorial.html">
					Mass spectrometry: LC-MS analysis
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-dataprocessing/tutorial.html">
					Mass spectrometry: LC-MS data processing
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/cardinal_mz_images/cardinal_mz_images.html">galaxyp/cardinal_mz_images/cardinal_mz_images</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-analyte-distribution/tutorial.html">
					Mass spectrometry imaging: Examining the spatial distribution of analytes
					</a>
				</li>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html">
					Mass spectrometry imaging: Finding differential analytes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/pangolin/pangolin.html">iuc/pangolin/pangolin</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-automation/tutorial.html">
					Automating Galaxy workflows using the command line
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html">
					Mutation calling, viral genome reconstruction and lineage/clade assignment from SARS-CoV-2 sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/iqtree/iqtree.html">iuc/iqtree/iqtree</a></td>
		<td>
			<ul>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/abc_intro_phylo/tutorial.html">
					Phylogenetics - Back to basics
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/tbprofiler/tb_profiler_profile.html">iuc/tbprofiler/tb_profiler_profile</a></td>
		<td>
			<ul>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/tb_variant_filter/tb_variant_filter.html">iuc/tb_variant_filter/tb_variant_filter</a></td>
		<td>
			<ul>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/xarray_mapplot/xarray_mapplot.html">ecology/xarray_mapplot/xarray_mapplot</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/pangeo/tutorial.html">
					Pangeo ecosystem 101 for everyone - Introduction to Xarray Galaxy Tools
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/x-array-map-plot/tutorial.html">
					Visualization of Climate Data using  NetCDF xarray Map Plotting
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/climate/climate_stripes/climate_stripes.html">climate/climate_stripes/climate_stripes</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/climate-101/tutorial.html">
					Getting your hands-on climate data
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/pangeo/tutorial.html">
					Pangeo ecosystem 101 for everyone - Introduction to Xarray Galaxy Tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/graphicsmagick_image_montage/graphicsmagick_image_montage.html">bgruening/graphicsmagick_image_montage/graphicsmagick_image_montage</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/pangeo/tutorial.html">
					Pangeo ecosystem 101 for everyone - Introduction to Xarray Galaxy Tools
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/x-array-map-plot/tutorial.html">
					Visualization of Climate Data using  NetCDF xarray Map Plotting
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/xarray_select/xarray_select.html">ecology/xarray_select/xarray_select</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/fates/tutorial.html">
					Functionally Assembled Terrestrial Ecosystem Simulator (FATES)
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/pangeo/tutorial.html">
					Pangeo ecosystem 101 for everyone - Introduction to Xarray Galaxy Tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_copernicus_notebook.html">interactive_tool_copernicus_notebook</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/sentinel5_data/tutorial.html">
					Sentinel 5P data visualisation
					</a>
				</li>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ndvi_openeo/tutorial.html">
					From NDVI data with OpenEO to time series visualisation with Holoviews
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/argo_getdata/argo_getdata.html">ecology/argo_getdata/argo_getdata</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/argo_pangeo/tutorial.html">
					Analyse Argo data
					</a>
				</li>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/earth_system/tutorial.html">
					Getting your hands-on earth data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_save_images/cp_save_images.html">bgruening/cp_save_images/cp_save_images</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html">
					Object tracking using CellProfiler
					</a>
				</li>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_measure_object_size_shape/cp_measure_object_size_shape.html">bgruening/cp_measure_object_size_shape/cp_measure_object_size_shape</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html">
					Object tracking using CellProfiler
					</a>
				</li>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_measure_object_intensity/cp_measure_object_intensity.html">bgruening/cp_measure_object_intensity/cp_measure_object_intensity</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html">
					Object tracking using CellProfiler
					</a>
				</li>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_identify_primary_objects/cp_identify_primary_objects.html">bgruening/cp_identify_primary_objects/cp_identify_primary_objects</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html">
					Object tracking using CellProfiler
					</a>
				</li>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_export_to_spreadsheet/cp_export_to_spreadsheet.html">bgruening/cp_export_to_spreadsheet/cp_export_to_spreadsheet</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html">
					Object tracking using CellProfiler
					</a>
				</li>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_common/cp_common.html">bgruening/cp_common/cp_common</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html">
					Object tracking using CellProfiler
					</a>
				</li>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_cellprofiler/cp_cellprofiler.html">bgruening/cp_cellprofiler/cp_cellprofiler</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html">
					Object tracking using CellProfiler
					</a>
				</li>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/imgteam/binary2labelimage/ip_binary_to_labelimage.html">imgteam/binary2labelimage/ip_binary_to_labelimage</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/hela-screen-analysis/tutorial.html">
					Analyse HeLa fluorescence siRNA screen
					</a>
				</li>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html">
					Introduction to Image Analysis using Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/imgteam/2d_simple_filter/ip_filter_standard.html">imgteam/2d_simple_filter/ip_filter_standard</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/hela-screen-analysis/tutorial.html">
					Analyse HeLa fluorescence siRNA screen
					</a>
				</li>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html">
					Introduction to Image Analysis using Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/imgteam/2d_auto_threshold/ip_threshold.html">imgteam/2d_auto_threshold/ip_threshold</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/hela-screen-analysis/tutorial.html">
					Analyse HeLa fluorescence siRNA screen
					</a>
				</li>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html">
					Introduction to Image Analysis using Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/snpsift/snpSift_filter.html">iuc/snpsift/snpSift_filter</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-automation/tutorial.html">
					Automating Galaxy workflows using the command line
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/fasttree/fasttree.html">iuc/fasttree/fasttree</a></td>
		<td>
			<ul>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/abc_intro_phylo/tutorial.html">
					Phylogenetics - Back to basics
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_multijoin_tool.html">bgruening/text_processing/tp_multijoin_tool</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/add_line_to_file/add_line_to_file.html">bgruening/add_line_to_file/add_line_to_file</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html">
					Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/__FILTER_FAILED_DATASETS__.html">__FILTER_FAILED_DATASETS__</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html">
					Predicting EI+ mass spectra with QCxMS
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/__FILTER_EMPTY_DATASETS__.html">__FILTER_EMPTY_DATASETS__</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ENA_Biodiv_submission/tutorial.html">
					Data submission using ENA upload Tool
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/__BUILD_LIST__.html">__BUILD_LIST__</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/rnateam/sortmerna/bg_sortmerna.html">rnateam/sortmerna/bg_sortmerna</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/humann_unpack_pathways/humann_unpack_pathways.html">iuc/humann_unpack_pathways/humann_unpack_pathways</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/humann_split_stratified_table/humann_split_stratified_table.html">iuc/humann_split_stratified_table/humann_split_stratified_table</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/humann_renorm_table/humann_renorm_table.html">iuc/humann_renorm_table/humann_renorm_table</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/humann_rename_table/humann_rename_table.html">iuc/humann_rename_table/humann_rename_table</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/humann_regroup_table/humann_regroup_table.html">iuc/humann_regroup_table/humann_regroup_table</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/humann/humann.html">iuc/humann/humann</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/graphlan_annotate/graphlan_annotate.html">iuc/graphlan_annotate/graphlan_annotate</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/graphlan/graphlan.html">iuc/graphlan/graphlan</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/export2graphlan/export2graphlan.html">iuc/export2graphlan/export2graphlan</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bebatut/combine_metaphlan2_humann2/combine_metaphlan2_humann2.html">bebatut/combine_metaphlan2_humann2/combine_metaphlan2_humann2</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html">
					Metatranscriptomics analysis using microbiome RNA-seq data (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_venn/mothur_venn.html">iuc/mothur_venn/mothur_venn</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_tree_shared/mothur_tree_shared.html">iuc/mothur_tree_shared/mothur_tree_shared</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_summary_single/mothur_summary_single.html">iuc/mothur_summary_single/mothur_summary_single</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_sub_sample/mothur_sub_sample.html">iuc/mothur_sub_sample/mothur_sub_sample</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_remove_seqs/mothur_remove_seqs.html">iuc/mothur_remove_seqs/mothur_remove_seqs</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_remove_lineage/mothur_remove_lineage.html">iuc/mothur_remove_lineage/mothur_remove_lineage</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_remove_groups/mothur_remove_groups.html">iuc/mothur_remove_groups/mothur_remove_groups</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_rarefaction_single/mothur_rarefaction_single.html">iuc/mothur_rarefaction_single/mothur_rarefaction_single</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_heatmap_sim/mothur_heatmap_sim.html">iuc/mothur_heatmap_sim/mothur_heatmap_sim</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_dist_shared/mothur_dist_shared.html">iuc/mothur_dist_shared/mothur_dist_shared</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_dist_seqs/mothur_dist_seqs.html">iuc/mothur_dist_seqs/mothur_dist_seqs</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_count_groups/mothur_count_groups.html">iuc/mothur_count_groups/mothur_count_groups</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_cluster/mothur_cluster.html">iuc/mothur_cluster/mothur_cluster</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_chimera_vsearch/mothur_chimera_vsearch.html">iuc/mothur_chimera_vsearch/mothur_chimera_vsearch</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/xy_plot/XY_Plot_1.html">devteam/xy_plot/XY_Plot_1</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html">
					16S Microbial Analysis with mothur (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nml/staramr/staramr_search.html">nml/staramr/staramr_search</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/amr-gene-detection/tutorial.html">
					Identification of AMR genes in an assembled bacterial genome
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/plasmid-metagenomics-nanopore/tutorial.html">
					Antibiotic resistance detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/filter_by_fasta_ids/filter_by_fasta_ids.html">galaxyp/filter_by_fasta_ids/filter_by_fasta_ids</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/salmon/salmon.html">bgruening/salmon/salmon</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/srna/tutorial.html">
					Differential abundance testing of small RNAs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/goenrichment/goenrichment.html">iuc/goenrichment/goenrichment</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/GO-enrichment/tutorial.html">
					GO Enrichment Analysis on Single-Cell RNA-Seq Data
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/goenrichment/tutorial.html">
					GO Enrichment Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/stringtie/stringtie_merge.html">iuc/stringtie/stringtie_merge</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html">
					De novo transcriptome reconstruction with RNA-Seq
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/gene2exon1.html">gene2exon1</a></td>
		<td>
			<ul>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html">
					Introduction to Genomics and Galaxy
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/limma_voom/limma_voom.html">iuc/limma_voom/limma_voom</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html">
					Pathway analysis with the MINERVA Platform
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html">
					2: RNA-seq counts to genes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/peptideshaker/ident_params.html">galaxyp/peptideshaker/ident_params</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/peptideshaker/fasta_cli.html">galaxyp/peptideshaker/fasta_cli</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html">
					Clinical Metaproteomics 2: Discovery
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/translate_bed/translate_bed.html">galaxyp/translate_bed/translate_bed</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/gffcompare_to_bed/gffcompare_to_bed.html">galaxyp/gffcompare_to_bed/gffcompare_to_bed</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/custom_pro_db/custom_pro_db.html">galaxyp/custom_pro_db/custom_pro_db</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/bed_to_protein_map/bed_to_protein_map.html">galaxyp/bed_to_protein_map/bed_to_protein_map</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-2-non-reference-database-generation/tutorial.html">
					Neoantigen 2: Non-Reference-Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">
					Proteogenomics 1: Database Creation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/cardinal_data_exporter/cardinal_data_exporter.html">galaxyp/cardinal_data_exporter/cardinal_data_exporter</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-analyte-distribution/tutorial.html">
					Mass spectrometry imaging: Examining the spatial distribution of analytes
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/mass-spectrometry-imaging-loading-exploring-data/tutorial.html">
					Mass spectrometry imaging: Loading and exploring MSI data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/metanovo/metanovo.html">galaxyp/metanovo/metanovo</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/clinical-mp-1-database-generation/tutorial.html">
					Clinical Metaproteomics 1: Database-Generation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/clinical-mp-1-database-generation/tutorial.html">
					Clinical Metaproteomics 1: Database-Generation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/jay/pdaug_sequence_property_based_descriptors/pdaug_sequence_property_based_descriptors.html">jay/pdaug_sequence_property_based_descriptors/pdaug_sequence_property_based_descriptors</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ml-modeling-of-anti-cancer-peptides/tutorial.html">
					Machine Learning Modeling of Anticancer Peptides
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/peptide-library-data-analysis/tutorial.html">
					Peptide Library Data Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/jay/pdaug_merge_dataframes/pdaug_merge_dataframes.html">jay/pdaug_merge_dataframes/pdaug_merge_dataframes</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ml-modeling-of-anti-cancer-peptides/tutorial.html">
					Machine Learning Modeling of Anticancer Peptides
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/peptide-library-data-analysis/tutorial.html">
					Peptide Library Data Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/jay/pdaug_basic_plots/pdaug_basic_plots.html">jay/pdaug_basic_plots/pdaug_basic_plots</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ml-modeling-of-anti-cancer-peptides/tutorial.html">
					Machine Learning Modeling of Anticancer Peptides
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/peptide-library-data-analysis/tutorial.html">
					Peptide Library Data Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/jay/pdaug_addclasslabel/pdaug_addclasslabel.html">jay/pdaug_addclasslabel/pdaug_addclasslabel</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ml-modeling-of-anti-cancer-peptides/tutorial.html">
					Machine Learning Modeling of Anticancer Peptides
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/peptide-library-data-analysis/tutorial.html">
					Peptide Library Data Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/proteore/proteore_venn_diagram/Jvenn.html">proteore/proteore_venn_diagram/Jvenn</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html">
					Biomarker candidate identification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteome_annotation/tutorial.html">
					Annotating a protein list identified by LC-MS/MS experiments
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/proteore/proteore_tissue_specific_expression_data/retrieve_from_hpa.html">proteore/proteore_tissue_specific_expression_data/retrieve_from_hpa</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html">
					Biomarker candidate identification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteome_annotation/tutorial.html">
					Annotating a protein list identified by LC-MS/MS experiments
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/proteore/proteore_id_converter/IDconverter.html">proteore/proteore_id_converter/IDconverter</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html">
					Biomarker candidate identification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteome_annotation/tutorial.html">
					Annotating a protein list identified by LC-MS/MS experiments
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/proteore/proteore_expression_rnaseq_abbased/rna_abbased_data.html">proteore/proteore_expression_rnaseq_abbased/rna_abbased_data</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html">
					Biomarker candidate identification
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteome_annotation/tutorial.html">
					Annotating a protein list identified by LC-MS/MS experiments
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/pep_pointer/pep_pointer.html">galaxyp/pep_pointer/pep_pointer</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-5-variant-annotation/tutorial.html">
					Neoantigen 5: Variant Annotation
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.html">
					Proteogenomics 3: Novel peptide analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/metaquantome_viz/metaquantome_viz.html">galaxyp/metaquantome_viz/metaquantome_viz</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-function/tutorial.html">
					metaQuantome 2: Function
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-taxonomy/tutorial.html">
					metaQuantome 3: Taxonomy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/metaquantome_stat/metaquantome_stat.html">galaxyp/metaquantome_stat/metaquantome_stat</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-function/tutorial.html">
					metaQuantome 2: Function
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-taxonomy/tutorial.html">
					metaQuantome 3: Taxonomy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/metaquantome_sample/metaquantome_sample.html">galaxyp/metaquantome_sample/metaquantome_sample</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-function/tutorial.html">
					metaQuantome 2: Function
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-taxonomy/tutorial.html">
					metaQuantome 3: Taxonomy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/metaquantome_filter/metaquantome_filter.html">galaxyp/metaquantome_filter/metaquantome_filter</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-function/tutorial.html">
					metaQuantome 2: Function
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-taxonomy/tutorial.html">
					metaQuantome 3: Taxonomy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/metaquantome_expand/metaquantome_expand.html">galaxyp/metaquantome_expand/metaquantome_expand</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-function/tutorial.html">
					metaQuantome 2: Function
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-taxonomy/tutorial.html">
					metaQuantome 3: Taxonomy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/metaquantome_db/metaquantome_db.html">galaxyp/metaquantome_db/metaquantome_db</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-function/tutorial.html">
					metaQuantome 2: Function
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-taxonomy/tutorial.html">
					metaQuantome 3: Taxonomy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/uniprot_rest_interface/uniprot.html">bgruening/uniprot_rest_interface/uniprot</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html">
					MaxQuant and MSstats for the analysis of label-free data
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html">
					Secretome Prediction
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_xtandemadapter/XTandemAdapter.html">galaxyp/openms_xtandemadapter/XTandemAdapter</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_textexporter/TextExporter.html">galaxyp/openms_textexporter/TextExporter</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_peptideindexer/PeptideIndexer.html">galaxyp/openms_peptideindexer/PeptideIndexer</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_idscoreswitcher/IDScoreSwitcher.html">galaxyp/openms_idscoreswitcher/IDScoreSwitcher</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_idposteriorerrorprobability/IDPosteriorErrorProbability.html">galaxyp/openms_idposteriorerrorprobability/IDPosteriorErrorProbability</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_fileinfo/FileInfo.html">galaxyp/openms_fileinfo/FileInfo</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_fidoadapter/FidoAdapter.html">galaxyp/openms_fidoadapter/FidoAdapter</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_falsediscoveryrate/FalseDiscoveryRate.html">galaxyp/openms_falsediscoveryrate/FalseDiscoveryRate</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/chromeister/chromeister.html">iuc/chromeister/chromeister</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-quality-control/tutorial.html">
					Genome Assembly Quality Control
					</a>
				</li>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/hpc-for-lsgc/tutorial.html">
					From small to large-scale genome comparison
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/pretext_snapshot/pretext_snapshot.html">iuc/pretext_snapshot/pretext_snapshot</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/pretext_map/pretext_map.html">iuc/pretext_map/pretext_map</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/gfa_to_fa/gfa_to_fa.html">iuc/gfa_to_fa/gfa_to_fa</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/plasmid-metagenomics-nanopore/tutorial.html">
					Antibiotic resistance detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bellerophon/bellerophon.html">iuc/bellerophon/bellerophon</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nml/metaspades/metaspades.html">nml/metaspades/metaspades</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/ngsutils_bam_filter/ngsutils_bam_filter.html">iuc/ngsutils_bam_filter/ngsutils_bam_filter</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/megahit_contig2fastg/megahit_contig2fastg.html">iuc/megahit_contig2fastg/megahit_contig2fastg</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/coverm_contig/coverm_contig.html">iuc/coverm_contig/coverm_contig</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/bamtools/bamtools.html">devteam/bamtools/bamtools</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html">
					Assembly of metagenomic sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/fasta_filter_by_length/fasta_filter_by_length.html">devteam/fasta_filter_by_length/fasta_filter_by_length</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/medaka_consensus_pipeline/medaka_consensus_pipeline.html">iuc/medaka_consensus_pipeline/medaka_consensus_pipeline</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/racon/racon.html">bgruening/racon/racon</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/plasmid-metagenomics-nanopore/tutorial.html">
					Antibiotic resistance detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/barchart_gnuplot.html">barchart_gnuplot</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
					Large genome assembly and polishing
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/meme_chip/meme_chip.html">iuc/meme_chip/meme_chip</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/extract_genomic_dna/Extract genomic DNA 1.html">iuc/extract_genomic_dna/Extract genomic DNA 1</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/replace_column_by_key_value_file/replace_column_with_key_value_file.html">bgruening/replace_column_by_key_value_file/replace_column_with_key_value_file</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/methylation-seq/tutorial.html">
					DNA Methylation data analysis
					</a>
				</li>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/deeptools_compute_gc_bias/deeptools_compute_gc_bias.html">bgruening/deeptools_compute_gc_bias/deeptools_compute_gc_bias</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
					Identification of the binding sites of the Estrogen receptor
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html">
					Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/pe_histogram/pe_histogram.html">iuc/pe_histogram/pe_histogram</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">
					ATAC-Seq data analysis
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/bam_to_sam/bam_to_sam.html">devteam/bam_to_sam/bam_to_sam</a></td>
		<td>
			<ul>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">
					Creating a new tutorial
					</a>
				</li>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
					Identification of the binding sites of the Estrogen receptor
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/climate/c3s/c3s.html">climate/c3s/c3s</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/earth_system/tutorial.html">
					Getting your hands-on earth data
					</a>
				</li>
				
				<li>
					Contributing to the Galaxy Training Material /
					<a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">
					Creating a new tutorial
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/comp1.html">comp1</a></td>
		<td>
			<ul>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-101/tutorial.html">
					Galaxy Basics for genomics
					</a>
				</li>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html">
					Secretome Prediction
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/gtftobed12/gtftobed12.html">iuc/gtftobed12/gtftobed12</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/plotly_ml_performance_plots/plotly_ml_performance_plots.html">bgruening/plotly_ml_performance_plots/plotly_ml_performance_plots</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/classification_machinelearning/tutorial.html">
					Classification in Machine Learning
					</a>
				</li>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/classification_regression/tutorial.html">
					Machine learning: classification and regression
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_merge_files/mothur_merge_files.html">iuc/mothur_merge_files/mothur_merge_files</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_make_biom/mothur_make_biom.html">iuc/mothur_make_biom/mothur_make_biom</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_align_seqs/mothur_align_seqs.html">iuc/mothur_align_seqs/mothur_align_seqs</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/picard/picard_MergeSamFiles.html">devteam/picard/picard_MergeSamFiles</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html">
					Genome annotation with Funannotate
					</a>
				</li>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html">
					Calling variants in non-diploid systems
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/cnv_vcf2json/cnv_vcf2json.html">iuc/cnv_vcf2json/cnv_vcf2json</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/beaconise_1000hg/tutorial.html">
					Working with Beacon V2: A Comprehensive Guide to Creating, Uploading, and Searching for Variants with Beacons
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/cnv_phenopacket/cnv_phenopacket.html">iuc/cnv_phenopacket/cnv_phenopacket</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/beaconise_1000hg/tutorial.html">
					Working with Beacon V2: A Comprehensive Guide to Creating, Uploading, and Searching for Variants with Beacons
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/beacon2_import/beacon2_import.html">iuc/beacon2_import/beacon2_import</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/beaconise_1000hg/tutorial.html">
					Working with Beacon V2: A Comprehensive Guide to Creating, Uploading, and Searching for Variants with Beacons
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/vapor/vapor.html">iuc/vapor/vapor</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bamtools_split_ref/bamtools_split_ref.html">iuc/bamtools_split_ref/bamtools_split_ref</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
					Avian influenza viral strain analysis from gene segment sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_qgis.html">interactive_tool_qgis</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/QGIS_Web_Feature_Services/tutorial.html">
					QGIS Web Feature Services
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_openrefine.html">interactive_tool_openrefine</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/openrefine_gbif/tutorial.html">
					Cleaning GBIF data using OpenRefine
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/ncbi_blast_plus/ncbi_blastx_wrapper.html">devteam/ncbi_blast_plus/ncbi_blastx_wrapper</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/ncbi-blast-against-the-madland/tutorial.html">
					NCBI BLAST+ against the MAdLand
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/chopin2/chopin2.html">iuc/chopin2/chopin2</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/hyperdimensional_computing/tutorial.html">
					Supervised Learning with Hyperdimensional Computing
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/marie-tremblay-metatoul/normalization/normalization.html">marie-tremblay-metatoul/normalization/normalization</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-dataprocessing/tutorial.html">
					Mass spectrometry: LC-MS data processing
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ethevenot/multivariate/Multivariate.html">ethevenot/multivariate/Multivariate</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-dataprocessing/tutorial.html">
					Mass spectrometry: LC-MS data processing
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/graphicsmagick_image_convert/graphicsmagick_image_convert.html">bgruening/graphicsmagick_image_convert/graphicsmagick_image_convert</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-posters/tutorial.html">
					Creating high resolution images of Galaxy Workflows
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/checkm_lineage_wf/checkm_lineage_wf.html">iuc/checkm_lineage_wf/checkm_lineage_wf</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metagenomics-binning/tutorial.html">
					Binning of metagenomic sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/melpetera/idchoice/idchoice.html">melpetera/idchoice/idchoice</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html">
					Mass spectrometry: LC-MS preprocessing with XCMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/lecorguille/xcms_summary/abims_xcms_summary.html">lecorguille/xcms_summary/abims_xcms_summary</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html">
					Mass spectrometry: LC-MS preprocessing with XCMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/lecorguille/xcms_export_samplemetadata/xcms_export_samplemetadata.html">lecorguille/xcms_export_samplemetadata/xcms_export_samplemetadata</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html">
					Mass spectrometry: LC-MS preprocessing with XCMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/sra_tools/fastq_dump.html">iuc/sra_tools/fastq_dump</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html">
					SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/jq/jq.html">iuc/jq/jq</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html">
					SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/data_manager_fetch_genome_dbkeys_all_fasta/data_manager_fetch_genome_all_fasta_dbkey.html">devteam/data_manager_fetch_genome_dbkeys_all_fasta/data_manager_fetch_genome_all_fasta_dbkey</a></td>
		<td>
			<ul>
				
				<li>
					Galaxy Server administration /
					<a href="/training-material/topics/admin/tutorials/reference-genomes/tutorial.html">
					Reference Data with Data Managers
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/data_manager_bwa_mem_index_builder/bwa_mem_index_builder_data_manager.html">devteam/data_manager_bwa_mem_index_builder/bwa_mem_index_builder_data_manager</a></td>
		<td>
			<ul>
				
				<li>
					Galaxy Server administration /
					<a href="/training-material/topics/admin/tutorials/reference-genomes/tutorial.html">
					Reference Data with Data Managers
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/rdconf/rdconf.html">bgruening/rdconf/rdconf</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/zauberkugel/tutorial.html">
					Protein target prediction of a bioactive ligand with Align-it and ePharmaLib
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/openbabel_addh/openbabel_addh.html">bgruening/openbabel_addh/openbabel_addh</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/zauberkugel/tutorial.html">
					Protein target prediction of a bioactive ligand with Align-it and ePharmaLib
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/align_it/ctb_alignit.html">bgruening/align_it/ctb_alignit</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/zauberkugel/tutorial.html">
					Protein target prediction of a bioactive ligand with Align-it and ePharmaLib
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/setup.html">setup</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/md-simulation-namd/tutorial.html">
					Running molecular dynamics simulations using NAMD
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/namd_nvt.html">namd_nvt</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/md-simulation-namd/tutorial.html">
					Running molecular dynamics simulations using NAMD
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/namd_npt.html">namd_npt</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/md-simulation-namd/tutorial.html">
					Running molecular dynamics simulations using NAMD
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/minimizer.html">minimizer</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/md-simulation-namd/tutorial.html">
					Running molecular dynamics simulations using NAMD
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/xchem_pose_scoring.html">xchem_pose_scoring</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html">
					Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/xchem_transfs_scoring/xchem_transfs_scoring.html">bgruening/xchem_transfs_scoring/xchem_transfs_scoring</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html">
					Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/sucos_max_score/sucos_max_score.html">bgruening/sucos_max_score/sucos_max_score</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html">
					Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/rxdock_rbdock/rxdock_rbdock.html">bgruening/rxdock_rbdock/rxdock_rbdock</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html">
					Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/rxdock_rbcavity/rxdock_rbcavity.html">bgruening/rxdock_rbcavity/rxdock_rbcavity</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html">
					Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/rdock_rbdock/rdock_rbdock.html">bgruening/rdock_rbdock/rdock_rbdock</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html">
					Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/rdock_rbcavity/rdock_rbcavity.html">bgruening/rdock_rbcavity/rdock_rbcavity</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html">
					Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/enumerate_charges/enumerate_charges.html">bgruening/enumerate_charges/enumerate_charges</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html">
					Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/ctb_frankenstein_ligand/ctb_frankenstein_ligand.html">bgruening/ctb_frankenstein_ligand/ctb_frankenstein_ligand</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html">
					Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/sdf_to_tab/sdf_to_tab.html">bgruening/sdf_to_tab/sdf_to_tab</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/openbabel/ctb_remIons.html">bgruening/openbabel/ctb_remIons</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/openbabel/ctb_remDuplicates.html">bgruening/openbabel/ctb_remDuplicates</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/openbabel/ctb_obgrep.html">bgruening/openbabel/ctb_obgrep</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/openbabel/ctb_filter.html">bgruening/openbabel/ctb_filter</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/openbabel/ctb_compound_convert.html">bgruening/openbabel/ctb_compound_convert</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/openbabel/ctb_change_title.html">bgruening/openbabel/ctb_change_title</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/fpocket/fpocket.html">bgruening/fpocket/fpocket</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/chemical_data_sources/ctb_online_data_fetch.html">bgruening/chemical_data_sources/ctb_online_data_fetch</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/chemfp/ctb_chemfp_nxn_clustering.html">bgruening/chemfp/ctb_chemfp_nxn_clustering</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/chemfp/ctb_chemfp_butina_clustering.html">bgruening/chemfp/ctb_chemfp_butina_clustering</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/autodock_vina_prepare_receptor/prepare_receptor.html">bgruening/autodock_vina_prepare_receptor/prepare_receptor</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/autodock_vina_prepare_box/prepare_box.html">bgruening/autodock_vina_prepare_box/prepare_box</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/autodock_vina/docking.html">bgruening/autodock_vina/docking</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">
					Protein-ligand docking
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/mdanalysis_rdf/mdanalysis_rdf.html">chemteam/mdanalysis_rdf/mdanalysis_rdf</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/analysis-md-simulations/tutorial.html">
					Analysis of molecular dynamics simulations
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/mdanalysis_ramachandran_plot/mdanalysis_ramachandran_plot.html">chemteam/mdanalysis_ramachandran_plot/mdanalysis_ramachandran_plot</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/analysis-md-simulations/tutorial.html">
					Analysis of molecular dynamics simulations
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/mdanalysis_distance/mdanalysis_distance.html">chemteam/mdanalysis_distance/mdanalysis_distance</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/analysis-md-simulations/tutorial.html">
					Analysis of molecular dynamics simulations
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/mdanalysis_dihedral/mdanalysis_dihedral.html">chemteam/mdanalysis_dihedral/mdanalysis_dihedral</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/analysis-md-simulations/tutorial.html">
					Analysis of molecular dynamics simulations
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/vmd_hbonds/vmd_hbonds.html">chemteam/vmd_hbonds/vmd_hbonds</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/gmx_merge_topology_files/gmx_merge_topology_files.html">chemteam/gmx_merge_topology_files/gmx_merge_topology_files</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/gmx_energy/gmx_energy.html">chemteam/gmx_energy/gmx_energy</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/bio3d_pca_visualize/bio3d_pca_visualize.html">chemteam/bio3d_pca_visualize/bio3d_pca_visualize</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/chemteam/ambertools_acpype/ambertools_acpype.html">chemteam/ambertools_acpype/ambertools_acpype</a></td>
		<td>
			<ul>
				
				<li>
					Computational chemistry /
					<a href="/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html">
					High Throughput Molecular Dynamics and Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/geneiobio/gene_iobio_display_generation_iframe.html">iuc/geneiobio/gene_iobio_display_generation_iframe</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/trio-analysis/tutorial.html">
					Trio Analysis using Synthetic Datasets from RD-Connect GPAP
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/ega_download_client/pyega3.html">iuc/ega_download_client/pyega3</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/trio-analysis/tutorial.html">
					Trio Analysis using Synthetic Datasets from RD-Connect GPAP
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bcftools_merge/bcftools_merge.html">iuc/bcftools_merge/bcftools_merge</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/trio-analysis/tutorial.html">
					Trio Analysis using Synthetic Datasets from RD-Connect GPAP
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/CONVERTER_uncompressed_to_gz.html">CONVERTER_uncompressed_to_gz</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/trio-analysis/tutorial.html">
					Trio Analysis using Synthetic Datasets from RD-Connect GPAP
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/control_freec/control_freec.html">iuc/control_freec/control_freec</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_text_file_with_recurring_lines.html">bgruening/text_processing/tp_text_file_with_recurring_lines</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
					Somatic Variant Discovery from WES Data Using Control-FREEC
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/varscan_somatic/varscan_somatic.html">iuc/varscan_somatic/varscan_somatic</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/gemini_annotate/gemini_annotate.html">iuc/gemini_annotate/gemini_annotate</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
					Identification of somatic and germline variants from tumor and normal sample pairs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/wolma/mimodd_main/mimodd_info.html">wolma/mimodd_main/mimodd_info</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/ucsc_fatovcf/fatovcf.html">iuc/ucsc_fatovcf/fatovcf</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/samtools_merge/samtools_merge.html">iuc/samtools_merge/samtools_merge</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/ivar_trim/ivar_trim.html">iuc/ivar_trim/ivar_trim</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/emboss_5/EMBOSS: maskseq51.html">devteam/emboss_5/EMBOSS: maskseq51</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/__APPLY_RULES__.html">__APPLY_RULES__</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html">
					Pox virus genome analysis from tiled-amplicon sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/gemini_db_info/gemini_db_info.html">iuc/gemini_db_info/gemini_db_info</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dip/tutorial.html">
					Calling variants in diploid systems
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/vcfallelicprimitives/vcfallelicprimitives.html">devteam/vcfallelicprimitives/vcfallelicprimitives</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dip/tutorial.html">
					Calling variants in diploid systems
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/nextclade/nextclade.html">iuc/nextclade/nextclade</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html">
					Mutation calling, viral genome reconstruction and lineage/clade assignment from SARS-CoV-2 sequencing data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/tbvcfreport/tbvcfreport.html">iuc/tbvcfreport/tbvcfreport</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mosdepth/mosdepth.html">iuc/mosdepth/mosdepth</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/jvarkit_wgscoverageplotter/jvarkit_wgscoverageplotter.html">iuc/jvarkit_wgscoverageplotter/jvarkit_wgscoverageplotter</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/emboss_5/EMBOSS:%20seqret84.html">devteam/emboss_5/EMBOSS:%20seqret84</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/emboss_5/EMBOSS: seqret84.html">devteam/emboss_5/EMBOSS: seqret84</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html">
					M. tuberculosis Variant Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/wolma/mimodd_main/mimodd_vcf_filter.html">wolma/mimodd_main/mimodd_vcf_filter</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html">
					Mapping and molecular identification of phenotype-causing mutations
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/wolma/mimodd_main/mimodd_varreport.html">wolma/mimodd_main/mimodd_varreport</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html">
					Mapping and molecular identification of phenotype-causing mutations
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/wolma/mimodd_main/mimodd_varextract.html">wolma/mimodd_main/mimodd_varextract</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html">
					Mapping and molecular identification of phenotype-causing mutations
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/wolma/mimodd_main/mimodd_varcall.html">wolma/mimodd_main/mimodd_varcall</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html">
					Mapping and molecular identification of phenotype-causing mutations
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/wolma/mimodd_main/mimodd_map.html">wolma/mimodd_main/mimodd_map</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html">
					Mapping and molecular identification of phenotype-causing mutations
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/gecko/gecko.html">iuc/gecko/gecko</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/hpc-for-lsgc/tutorial.html">
					From small to large-scale genome comparison
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/omark/omark.html">iuc/omark/omark</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/helixer/tutorial.html">
					Genome annotation with Helixer
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/genouest/helixer/helixer.html">genouest/helixer/helixer</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/helixer/tutorial.html">
					Genome annotation with Helixer
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/feelnc/feelnc.html">iuc/feelnc/feelnc</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/lncrna/tutorial.html">
					Long non-coding RNAs (lncRNAs) annotation with FEELnc
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/Extract_features1.html">Extract_features1</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/lncrna/tutorial.html">
					Long non-coding RNAs (lncRNAs) annotation with FEELnc
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/simon-gladman/fasta_stats/fasta-stats.html">simon-gladman/fasta_stats/fasta-stats</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker-short/tutorial.html">
					Genome annotation with Maker (short)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/funannotate_compare/funannotate_compare.html">iuc/funannotate_compare/funannotate_compare</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html">
					Genome annotation with Funannotate
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/funannotate_annotate/funannotate_annotate.html">iuc/funannotate_annotate/funannotate_annotate</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html">
					Genome annotation with Funannotate
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/aegean_parseval/aegean_parseval.html">iuc/aegean_parseval/aegean_parseval</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html">
					Genome annotation with Funannotate
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/plasmidfinder/plasmidfinder.html">iuc/plasmidfinder/plasmidfinder</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/bacterial-genome-annotation/tutorial.html">
					Bacterial Genome Annotation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/isescan/isescan.html">iuc/isescan/isescan</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/bacterial-genome-annotation/tutorial.html">
					Bacterial Genome Annotation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/integron_finder/integron_finder.html">iuc/integron_finder/integron_finder</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/bacterial-genome-annotation/tutorial.html">
					Bacterial Genome Annotation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/rapidnj/rapidnj.html">iuc/rapidnj/rapidnj</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/orfipy/orfipy.html">iuc/orfipy/orfipy</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/diamond/bg_diamond_view.html">bgruening/diamond/bg_diamond_view</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/diamond/bg_diamond_makedb.html">bgruening/diamond/bg_diamond_makedb</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html">
					Comparative gene analysis in unannotated genomes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/red/red.html">iuc/red/red</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/repeatmasker/tutorial.html">
					Masking repeats with RepeatMasker
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bedtools/bedtools_maskfastabed.html">iuc/bedtools/bedtools_maskfastabed</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/repeatmasker/tutorial.html">
					Masking repeats with RepeatMasker
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/csbl/repeatmodeler/repeatmodeler.html">csbl/repeatmodeler/repeatmodeler</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/repeatmasker/tutorial.html">
					Masking repeats with RepeatMasker
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mageck_test/mageck_test.html">iuc/mageck_test/mageck_test</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/crispr-screen/tutorial.html">
					CRISPR screen analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mageck_mle/mageck_mle.html">iuc/mageck_mle/mageck_mle</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/crispr-screen/tutorial.html">
					CRISPR screen analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mageck_count/mageck_count.html">iuc/mageck_count/mageck_count</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/crispr-screen/tutorial.html">
					CRISPR screen analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/gga/apollo_list_organism/list_organism.html">gga/apollo_list_organism/list_organism</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/apollo/tutorial.html">
					Refining Genome Annotations with Apollo (prokaryotes)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/gga/apollo_create_account/create_account.html">gga/apollo_create_account/create_account</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/apollo/tutorial.html">
					Refining Genome Annotations with Apollo (prokaryotes)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/ncbi_acc_download/ncbi_acc_download.html">iuc/ncbi_acc_download/ncbi_acc_download</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html">
					Secondary metabolite discovery
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bp_genbank2gff3/bp_genbank2gff3.html">iuc/bp_genbank2gff3/bp_genbank2gff3</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html">
					Secondary metabolite discovery
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/cpt/cpt_gbk_to_gff/edu.tamu.cpt.gff3.customGbkToGff.html">cpt/cpt_gbk_to_gff/edu.tamu.cpt.gff3.customGbkToGff</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html">
					Secondary metabolite discovery
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/simsearch/ctb_simsearch.html">bgruening/simsearch/ctb_simsearch</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html">
					Secondary metabolite discovery
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/openbabel_remduplicates/openbabel_remDuplicates.html">bgruening/openbabel_remduplicates/openbabel_remDuplicates</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html">
					Secondary metabolite discovery
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/antismash/antismash.html">bgruening/antismash/antismash</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html">
					Secondary metabolite discovery
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/snap_training/snap_training.html">iuc/snap_training/snap_training</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html">
					Genome annotation with Maker
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/augustus_training/augustus_training.html">bgruening/augustus_training/augustus_training</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html">
					Genome annotation with Maker
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/stacks_genotypes/stacks_genotypes.html">iuc/stacks_genotypes/stacks_genotypes</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/genetic-map-rad-seq/tutorial.html">
					RAD-Seq to construct genetic maps
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/kellrott/regex_replace/regex_replace.html">kellrott/regex_replace/regex_replace</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/stacks_refmap/stacks_refmap.html">iuc/stacks_refmap/stacks_refmap</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/bwa_wrappers/bwa_wrapper.html">devteam/bwa_wrappers/bwa_wrapper</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">
					RAD-Seq Reference-based data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/gdal_ogr2ogr/gdal_ogr2ogr.html">ecology/gdal_ogr2ogr/gdal_ogr2ogr</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/gbif_cleaning/tutorial.html">
					Cleaning GBIF data for the use in Ecology
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/tool_anonymization/tool_anonymization.html">ecology/tool_anonymization/tool_anonymization</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/biodiversity-data-exploration/tutorial.html">
					Biodiversity data exploration
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/ecology_stat_presence_abs/ecology_stat_presence_abs.html">ecology/ecology_stat_presence_abs/ecology_stat_presence_abs</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/biodiversity-data-exploration/tutorial.html">
					Biodiversity data exploration
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/ecology_presence_abs_abund/ecology_presence_abs_abund.html">ecology/ecology_presence_abs_abund/ecology_presence_abs_abund</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/biodiversity-data-exploration/tutorial.html">
					Biodiversity data exploration
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/ecology_link_between_var/ecology_link_between_var.html">ecology/ecology_link_between_var/ecology_link_between_var</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/biodiversity-data-exploration/tutorial.html">
					Biodiversity data exploration
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/ecology_homogeneity_normality/ecology_homogeneity_normality.html">ecology/ecology_homogeneity_normality/ecology_homogeneity_normality</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/biodiversity-data-exploration/tutorial.html">
					Biodiversity data exploration
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/ecology_beta_diversity/ecology_beta_diversity.html">ecology/ecology_beta_diversity/ecology_beta_diversity</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/biodiversity-data-exploration/tutorial.html">
					Biodiversity data exploration
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/obisindicators/obisindicators.html">ecology/obisindicators/obisindicators</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/obisindicators/tutorial.html">
					Obis marine indicators
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/pampa_presabs/pampa_presabs.html">ecology/pampa_presabs/pampa_presabs</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/pampa_plotglm/pampa_plotglm.html">ecology/pampa_plotglm/pampa_plotglm</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/pampa_glmsp/pampa_glmsp.html">ecology/pampa_glmsp/pampa_glmsp</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/pampa_glmcomm/pampa_glmcomm.html">ecology/pampa_glmcomm/pampa_glmcomm</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/pampa_communitymetrics/pampa_communitymetrics.html">ecology/pampa_communitymetrics/pampa_communitymetrics</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
					Compute and analyze biodiversity metrics with PAMPA toolsuite
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_wallace.html">interactive_tool_wallace</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/species-distribution-modeling/tutorial.html">
					Species distribution modeling
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_metashark.html">interactive_tool_metashark</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/MetaShARK_tutorial/tutorial.html">
					Creating metadata using Ecological Metadata Language (EML) standard with EML Assembly Line functionalities
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/srs_spectral_indices/srs_spectral_indices.html">ecology/srs_spectral_indices/srs_spectral_indices</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/remote-sensing/tutorial.html">
					Sentinel 2 biodiversity
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/srs_process_data/srs_process_data.html">ecology/srs_process_data/srs_process_data</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/remote-sensing/tutorial.html">
					Sentinel 2 biodiversity
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/srs_preprocess_s2/srs_preprocess_s2.html">ecology/srs_preprocess_s2/srs_preprocess_s2</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/remote-sensing/tutorial.html">
					Sentinel 2 biodiversity
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/srs_pca/srs_pca.html">ecology/srs_pca/srs_pca</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/remote-sensing/tutorial.html">
					Sentinel 2 biodiversity
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/srs_global_indices/srs_global_indices.html">ecology/srs_global_indices/srs_global_indices</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/remote-sensing/tutorial.html">
					Sentinel 2 biodiversity
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/srs_diversity_maps/srs_diversity_maps.html">ecology/srs_diversity_maps/srs_diversity_maps</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/remote-sensing/tutorial.html">
					Sentinel 2 biodiversity
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/mnhn65mo/tabular_to_csv/tab2csv_R.html">mnhn65mo/tabular_to_csv/tab2csv_R</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/mnhn65mo/regionalgam/flight-curve.html">mnhn65mo/regionalgam/flight-curve</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/mnhn65mo/csv_to_tabular/csv2tab_R.html">mnhn65mo/csv_to_tabular/csv2tab_R</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/regionalgam_plot_trend/regionalgam_plot_trend.html">ecology/regionalgam_plot_trend/regionalgam_plot_trend</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/regionalgam_gls/regionalgam_gls.html">ecology/regionalgam_gls/regionalgam_gls</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/regionalgam_glmmpql/regionalgam_glmmpql.html">ecology/regionalgam_glmmpql/regionalgam_glmmpql</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/regionalgam_flight_curve/regionalgam_flight_curve.html">ecology/regionalgam_flight_curve/regionalgam_flight_curve</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/regionalgam_autocor_acf/regionalgam_autocor_acf.html">ecology/regionalgam_autocor_acf/regionalgam_autocor_acf</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/regionalgam_ab_index/regionalgam_ab_index.html">ecology/regionalgam_ab_index/regionalgam_ab_index</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/mergeCols1.html">mergeCols1</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html">
					Regional GAM
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/ecoregion_taxa_seeker/ecoregion_taxa_seeker.html">ecology/ecoregion_taxa_seeker/ecoregion_taxa_seeker</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Ecoregionalization_tutorial/tutorial.html">
					Ecoregionalization workflow tutorial
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/ecoregion_geonearestneighbor/ecoregion_GeoNearestNeighbor.html">ecology/ecoregion_geonearestneighbor/ecoregion_GeoNearestNeighbor</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Ecoregionalization_tutorial/tutorial.html">
					Ecoregionalization workflow tutorial
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/ecoregion_eco_map/ecoregion_eco_map.html">ecology/ecoregion_eco_map/ecoregion_eco_map</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Ecoregionalization_tutorial/tutorial.html">
					Ecoregionalization workflow tutorial
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/ecoregion_cluster_estimate/ecoregion_cluster_estimate.html">ecology/ecoregion_cluster_estimate/ecoregion_cluster_estimate</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Ecoregionalization_tutorial/tutorial.html">
					Ecoregionalization workflow tutorial
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/ecoregion_clara_cluster/ecoregion_clara_cluster.html">ecology/ecoregion_clara_cluster/ecoregion_clara_cluster</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Ecoregionalization_tutorial/tutorial.html">
					Ecoregionalization workflow tutorial
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/ecoregion_brt_analysis/ecoregion_brt_analysis.html">ecology/ecoregion_brt_analysis/ecoregion_brt_analysis</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Ecoregionalization_tutorial/tutorial.html">
					Ecoregionalization workflow tutorial
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/obi_uniq/obi_uniq.html">iuc/obi_uniq/obi_uniq</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/obi_tab/obi_tab.html">iuc/obi_tab/obi_tab</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/obi_stat/obi_stat.html">iuc/obi_stat/obi_stat</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/obi_ngsfilter/obi_ngsfilter.html">iuc/obi_ngsfilter/obi_ngsfilter</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/obi_illumina_pairend/obi_illumina_pairend.html">iuc/obi_illumina_pairend/obi_illumina_pairend</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/obi_grep/obi_grep.html">iuc/obi_grep/obi_grep</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/obi_clean/obi_clean.html">iuc/obi_clean/obi_clean</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/obi_annotate/obi_annotate.html">iuc/obi_annotate/obi_annotate</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">
					Metabarcoding/eDNA through Obitools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/padge/phykit/phykit_alignment_based.html">padge/phykit/phykit_alignment_based</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html">
					Preparing genomic data for phylogeny reconstruction
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/padge/clipkit/clipkit.html">padge/clipkit/clipkit</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html">
					Preparing genomic data for phylogeny reconstruction
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/proteinortho_grab_proteins/proteinortho_grab_proteins.html">iuc/proteinortho_grab_proteins/proteinortho_grab_proteins</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html">
					Preparing genomic data for phylogeny reconstruction
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/proteinortho/proteinortho.html">iuc/proteinortho/proteinortho</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html">
					Preparing genomic data for phylogeny reconstruction
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/glimmer_gbk_to_orf/glimmer_gbk_to_orf.html">bgruening/glimmer_gbk_to_orf/glimmer_gbk_to_orf</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html">
					Preparing genomic data for phylogeny reconstruction
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/prodigal/prodigal.html">iuc/prodigal/prodigal</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/marine_omics_bgc/tutorial.html">
					Marine Omics identifying biosynthetic gene clusters
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/sanntis_marine/sanntis_marine.html">ecology/sanntis_marine/sanntis_marine</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/marine_omics_bgc/tutorial.html">
					Marine Omics identifying biosynthetic gene clusters
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/SanntiS_marine/SanntiS_marine.html">ecology/SanntiS_marine/SanntiS_marine</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/marine_omics_bgc/tutorial.html">
					Marine Omics identifying biosynthetic gene clusters
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_metashrimps.html">interactive_tool_metashrimps</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/Metashrimps_tutorial/tutorial.html">
					Creating FAIR Quality assessment reports and draft of Data Papers from EML metadata with MetaShRIMPS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/fastq_filter/fastq_filter.html">devteam/fastq_filter/fastq_filter</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ENA_Biodiv_submission/tutorial.html">
					Data submission using ENA upload Tool
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_holoviz.html">interactive_tool_holoviz</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/ndvi_openeo/tutorial.html">
					From NDVI data with OpenEO to time series visualisation with Holoviews
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/climate/cdo_operations/cdo_operations.html">climate/cdo_operations/cdo_operations</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/x-array-map-plot/tutorial.html">
					Visualization of Climate Data using  NetCDF xarray Map Plotting
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/cb_ivr/cb_ivr.html">ecology/cb_ivr/cb_ivr</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/champs-blocs/tutorial.html">
					Champs blocs indicators
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/cb_div/cb_div.html">ecology/cb_div/cb_div</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/champs-blocs/tutorial.html">
					Champs blocs indicators
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/cb_dissim/cb_dissim.html">ecology/cb_dissim/cb_dissim</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/champs-blocs/tutorial.html">
					Champs blocs indicators
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/cb_qecb.html">cb_qecb</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/champs-blocs/tutorial.html">
					Champs blocs indicators
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/cb_ivr.html">cb_ivr</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/champs-blocs/tutorial.html">
					Champs blocs indicators
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/cb_eco.html">cb_eco</a></td>
		<td>
			<ul>
				
				<li>
					Ecology /
					<a href="/training-material/topics/ecology/tutorials/champs-blocs/tutorial.html">
					Champs blocs indicators
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/richard-burhans/ncbi_fcs_adaptor/ncbi_fcs_adaptor.html">richard-burhans/ncbi_fcs_adaptor/ncbi_fcs_adaptor</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/ncbi-fcs/tutorial.html">
					Screening assembled genomes for contamination using NCBI FCS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/ncbi_fcs_gx/ncbi_fcs_gx.html">iuc/ncbi_fcs_gx/ncbi_fcs_gx</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/ncbi-fcs/tutorial.html">
					Screening assembled genomes for contamination using NCBI FCS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/pycoqc/pycoqc.html">iuc/pycoqc/pycoqc</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">
					Quality Control
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/fastqe/fastqe.html">iuc/fastqe/fastqe</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">
					Quality Control
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/seqtk/seqtk_trimfq.html">iuc/seqtk/seqtk_trimfq</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/seqtk/seqtk_mergepe.html">iuc/seqtk/seqtk_mergepe</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/qiime_align_seqs/qiime_align_seqs.html">iuc/qiime_align_seqs/qiime_align_seqs</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_degap_seqs/mothur_degap_seqs.html">iuc/mothur_degap_seqs/mothur_degap_seqs</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/aligned_to_consensus/aligned_to_consensus.html">ecology/aligned_to_consensus/aligned_to_consensus</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/fastx_reverse_complement/cshl_fastx_reverse_complement.html">devteam/fastx_reverse_complement/cshl_fastx_reverse_complement</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/fastq_to_fasta/cshl_fastq_to_fasta.html">devteam/fastq_to_fasta/cshl_fastq_to_fasta</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/fastq_masker_by_quality/fastq_masker_by_quality.html">devteam/fastq_masker_by_quality/fastq_masker_by_quality</a></td>
		<td>
			<ul>
				
				<li>
					Sequence analysis /
					<a href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
					Clean and manage Sanger sequences from raw files to aligned consensus
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/iwtomics_testandplot/iwtomics_testandplot.html">iuc/iwtomics_testandplot/iwtomics_testandplot</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/iwtomics/tutorial.html">
					Interval-Wise Testing for omics data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/iwtomics_plotwithscale/iwtomics_plotwithscale.html">iuc/iwtomics_plotwithscale/iwtomics_plotwithscale</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/iwtomics/tutorial.html">
					Interval-Wise Testing for omics data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/iwtomics_loadandplot/iwtomics_loadandplot.html">iuc/iwtomics_loadandplot/iwtomics_loadandplot</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/iwtomics/tutorial.html">
					Interval-Wise Testing for omics data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/pubmed_by_queries/pubmed_by_queries.html">iuc/pubmed_by_queries/pubmed_by_queries</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/text-mining_simtext/tutorial.html">
					Text-mining with the SimText toolset
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/pmids_to_pubtator_matrix/pmids_to_pubtator_matrix.html">iuc/pmids_to_pubtator_matrix/pmids_to_pubtator_matrix</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/text-mining_simtext/tutorial.html">
					Text-mining with the SimText toolset
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_simtext_app.html">interactive_tool_simtext_app</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/text-mining_simtext/tutorial.html">
					Text-mining with the SimText toolset
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/plotly_parallel_coordinates_plot/plotly_parallel_coordinates_plot.html">bgruening/plotly_parallel_coordinates_plot/plotly_parallel_coordinates_plot</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/age-prediction-with-ml/tutorial.html">
					Age prediction using machine learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/vijay/pancancer_within_disease_analysis/pancancer_within_disease_analysis.html">vijay/pancancer_within_disease_analysis/pancancer_within_disease_analysis</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html">
					PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/vijay/pancancer_visualize_decisions/pancancer_visualize_decisions.html">vijay/pancancer_visualize_decisions/pancancer_visualize_decisions</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html">
					PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/vijay/pancancer_targene_summary_figures/pancancer_targene_summary_figures.html">vijay/pancancer_targene_summary_figures/pancancer_targene_summary_figures</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html">
					PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/vijay/pancancer_targene_pharmacology/pancancer_targene_pharmacology.html">vijay/pancancer_targene_pharmacology/pancancer_targene_pharmacology</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html">
					PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/vijay/pancancer_targene_cell_line_predictions/pancancer_targene_cell_line_predictions.html">vijay/pancancer_targene_cell_line_predictions/pancancer_targene_cell_line_predictions</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html">
					PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/vijay/pancancer_pathway_count_heatmaps/pancancer_pathway_count_heatmaps.html">vijay/pancancer_pathway_count_heatmaps/pancancer_pathway_count_heatmaps</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html">
					PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/vijay/pancancer_map_mutation_class/pancancer_map_mutation_class.html">vijay/pancancer_map_mutation_class/pancancer_map_mutation_class</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html">
					PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/vijay/pancancer_external_sample_status_prediction/pancancer_external_sample_status_prediction.html">vijay/pancancer_external_sample_status_prediction/pancancer_external_sample_status_prediction</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html">
					PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/vijay/pancancer_compare_within_models/pancancer_compare_within_models.html">vijay/pancancer_compare_within_models/pancancer_compare_within_models</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html">
					PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/vijay/pancancer_classifier/pancancer_classifier.html">vijay/pancancer_classifier/pancancer_classifier</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html">
					PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/vijay/pancancer_apply_weights/pancancer_apply_weights.html">vijay/pancancer_apply_weights/pancancer_apply_weights</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html">
					PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/vijay/pancancer_alternative_genes_pathwaymapper/pancancer_alternative_genes_pathwaymapper.html">vijay/pancancer_alternative_genes_pathwaymapper/pancancer_alternative_genes_pathwaymapper</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html">
					PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/paulo_lyra_jr/pycaret_model_comparison/PyCaret_Model_Comparison.html">paulo_lyra_jr/pycaret_model_comparison/PyCaret_Model_Comparison</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/loris_model/tutorial.html">
					Building the LORIS LLR6 PanCancer Model Using PyCaret
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/paulo_lyra_jr/ludwig_applications/ludwig_experiment.html">paulo_lyra_jr/ludwig_applications/ludwig_experiment</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/galaxy-ludwig/tutorial.html">
					Train and Test a Deep learning image classifier with Galaxy-Ludwig
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/sklearn_nn_classifier/sklearn_nn_classifier.html">bgruening/sklearn_nn_classifier/sklearn_nn_classifier</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/classification_machinelearning/tutorial.html">
					Classification in Machine Learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/scater_plot_pca/scater_plot_pca.html">iuc/scater_plot_pca/scater_plot_pca</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-scater-qc/tutorial.html">
					Single-cell quality control with scater
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/scater_plot_dist_scatter/scater_plot_dist_scatter.html">iuc/scater_plot_dist_scatter/scater_plot_dist_scatter</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-scater-qc/tutorial.html">
					Single-cell quality control with scater
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/scater_filter/scater_filter.html">iuc/scater_filter/scater_filter</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-scater-qc/tutorial.html">
					Single-cell quality control with scater
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/scater_create_qcmetric_ready_sce/scater_create_qcmetric_ready_sce.html">iuc/scater_create_qcmetric_ready_sce/scater_create_qcmetric_ready_sce</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-scater-qc/tutorial.html">
					Single-cell quality control with scater
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/column_remove_by_header/column_remove_by_header.html">iuc/column_remove_by_header/column_remove_by_header</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html">
					Bulk matrix to ESet | Creating the bulk RNA-seq dataset for deconvolution
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_regress_variable/scanpy_regress_variable.html">ebi-gxa/scanpy_regress_variable/scanpy_regress_variable</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html">
					Removing the effects of the cell cycle
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/music_inspect_eset/music_inspect_eset.html">bgruening/music_inspect_eset/music_inspect_eset</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music/tutorial.html">
					Bulk RNA Deconvolution with MuSiC
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_parameter_iterator/scanpy_parameter_iterator.html">ebi-gxa/scanpy_parameter_iterator/scanpy_parameter_iterator</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scanpy_parameter_iterator/tutorial.html">
					Scanpy Parameter Iterator
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/sceasy_convert/sceasy_convert.html">ebi-gxa/sceasy_convert/sceasy_convert</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/tabular_to_fastq/tabular_to_fastq.html">devteam/tabular_to_fastq/tabular_to_fastq</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html">
					Converting between common single cell data formats
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/salmon_kallisto_mtx_to_10x/_salmon_kallisto_mtx_to_10x.html">ebi-gxa/salmon_kallisto_mtx_to_10x/_salmon_kallisto_mtx_to_10x</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html">
					Generating a single cell matrix using Alevin
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/gtf2gene_list/_ensembl_gtf2gene_list.html">ebi-gxa/gtf2gene_list/_ensembl_gtf2gene_list</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html">
					Generating a single cell matrix using Alevin
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/dropletutils_empty_drops/dropletutils_empty_drops.html">ebi-gxa/dropletutils_empty_drops/dropletutils_empty_drops</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html">
					Generating a single cell matrix using Alevin
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/droplet_barcode_plot/_dropletBarcodePlot.html">ebi-gxa/droplet_barcode_plot/_dropletBarcodePlot</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html">
					Generating a single cell matrix using Alevin
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/alevin/alevin.html">bgruening/alevin/alevin</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html">
					Generating a single cell matrix using Alevin
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/monocle3_topmarkers/monocle3_topmarkers.html">ebi-gxa/monocle3_topmarkers/monocle3_topmarkers</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/monocle3_reducedim/monocle3_reduceDim.html">ebi-gxa/monocle3_reducedim/monocle3_reduceDim</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/monocle3_preprocess/monocle3_preprocess.html">ebi-gxa/monocle3_preprocess/monocle3_preprocess</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/monocle3_plotcells/monocle3_plotCells.html">ebi-gxa/monocle3_plotcells/monocle3_plotCells</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/monocle3_partition/monocle3_partition.html">ebi-gxa/monocle3_partition/monocle3_partition</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/monocle3_ordercells/monocle3_orderCells.html">ebi-gxa/monocle3_ordercells/monocle3_orderCells</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/monocle3_learngraph/monocle3_learnGraph.html">ebi-gxa/monocle3_learngraph/monocle3_learnGraph</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/monocle3_diffexp/monocle3_diffExp.html">ebi-gxa/monocle3_diffexp/monocle3_diffExp</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">
					Inferring single cell trajectories with Monocle3
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/snapatac2_preprocessing/snapatac2_preprocessing.html">iuc/snapatac2_preprocessing/snapatac2_preprocessing</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/snapatac2_plotting/snapatac2_plotting.html">iuc/snapatac2_plotting/snapatac2_plotting</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/snapatac2_peaks_and_motif/snapatac2_peaks_and_motif.html">iuc/snapatac2_peaks_and_motif/snapatac2_peaks_and_motif</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/snapatac2_clustering/snapatac2_clustering.html">iuc/snapatac2_clustering/snapatac2_clustering</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/cut_columns/Cut1.html">devteam/cut_columns/Cut1</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
					Single-cell ATAC-seq standard processing with SnapATAC2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/rna_starsolo/rna_starsolo.html">iuc/rna_starsolo/rna_starsolo</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html">
					Pre-processing of 10X Single-Cell RNA Datasets
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/seurat_reduce_dimension/seurat_reduce_dimension.html">iuc/seurat_reduce_dimension/seurat_reduce_dimension</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/seurat_preprocessing/seurat_preprocessing.html">iuc/seurat_preprocessing/seurat_preprocessing</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/seurat_plot/seurat_plot.html">iuc/seurat_plot/seurat_plot</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/seurat_data/seurat_data.html">iuc/seurat_data/seurat_data</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/seurat_create/seurat_create.html">iuc/seurat_create/seurat_create</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/seurat_clustering/seurat_clustering.html">iuc/seurat_clustering/seurat_clustering</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-seurat-pbmc3k/tutorial.html">
					Clustering 3K PBMCs with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/sinto_fragments/sinto_fragments.html">iuc/sinto_fragments/sinto_fragments</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html">
					Pre-processing of 10X Single-Cell ATAC-seq Datasets
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/sinto_barcode/sinto_barcode.html">iuc/sinto_barcode/sinto_barcode</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html">
					Pre-processing of 10X Single-Cell ATAC-seq Datasets
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/episcanpy_preprocess/episcanpy_preprocess.html">iuc/episcanpy_preprocess/episcanpy_preprocess</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html">
					Pre-processing of 10X Single-Cell ATAC-seq Datasets
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/episcanpy_build_matrix/episcanpy_build_matrix.html">iuc/episcanpy_build_matrix/episcanpy_build_matrix</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html">
					Pre-processing of 10X Single-Cell ATAC-seq Datasets
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/umi_tools_count/umi_tools_count.html">iuc/umi_tools_count/umi_tools_count</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html">
					Pre-processing of Single-Cell RNA Data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/ggplot2_violin/ggplot2_violin.html">iuc/ggplot2_violin/ggplot2_violin</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/__DUPLICATE_FILE_TO_COLLECTION__.html">__DUPLICATE_FILE_TO_COLLECTION__</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-deconvolution-evaluate/tutorial.html">
					Evaluating Reference Data for Bulk RNA Deconvolution
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/hca_matrix_downloader/hca_matrix_downloader.html">ebi-gxa/hca_matrix_downloader/hca_matrix_downloader</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html">
					Importing files from public atlases
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/raceid_trajectory/raceid_trajectory.html">iuc/raceid_trajectory/raceid_trajectory</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-raceid/tutorial.html">
					Downstream Single-cell RNA analysis with RaceID
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/raceid_inspecttrajectory/raceid_inspecttrajectory.html">iuc/raceid_inspecttrajectory/raceid_inspecttrajectory</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-raceid/tutorial.html">
					Downstream Single-cell RNA analysis with RaceID
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/raceid_inspectclusters/raceid_inspectclusters.html">iuc/raceid_inspectclusters/raceid_inspectclusters</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-raceid/tutorial.html">
					Downstream Single-cell RNA analysis with RaceID
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/raceid_filtnormconf/raceid_filtnormconf.html">iuc/raceid_filtnormconf/raceid_filtnormconf</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-raceid/tutorial.html">
					Downstream Single-cell RNA analysis with RaceID
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/raceid_clustering/raceid_clustering.html">iuc/raceid_clustering/raceid_clustering</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-raceid/tutorial.html">
					Downstream Single-cell RNA analysis with RaceID
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_run_paga/scanpy_run_paga.html">ebi-gxa/scanpy_run_paga/scanpy_run_paga</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_trajectories/tutorial.html">
					Inferring single cell trajectories with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_run_fdg/scanpy_run_fdg.html">ebi-gxa/scanpy_run_fdg/scanpy_run_fdg</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_trajectories/tutorial.html">
					Inferring single cell trajectories with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_run_dpt/scanpy_run_dpt.html">ebi-gxa/scanpy_run_dpt/scanpy_run_dpt</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_trajectories/tutorial.html">
					Inferring single cell trajectories with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_run_diffmap/scanpy_run_diffmap.html">ebi-gxa/scanpy_run_diffmap/scanpy_run_diffmap</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_trajectories/tutorial.html">
					Inferring single cell trajectories with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_plot_trajectory/scanpy_plot_trajectory.html">ebi-gxa/scanpy_plot_trajectory/scanpy_plot_trajectory</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_trajectories/tutorial.html">
					Inferring single cell trajectories with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/music_compare/music_compare.html">bgruening/music_compare/music_compare</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/bulk-music-4-compare/tutorial.html">
					Comparing inferred cell compositions using MuSiC deconvolution
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_scale_data/scanpy_scale_data.html">ebi-gxa/scanpy_scale_data/scanpy_scale_data</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_run_pca/scanpy_run_pca.html">ebi-gxa/scanpy_run_pca/scanpy_run_pca</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_normalise_data/scanpy_normalise_data.html">ebi-gxa/scanpy_normalise_data/scanpy_normalise_data</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_find_variable_genes/scanpy_find_variable_genes.html">ebi-gxa/scanpy_find_variable_genes/scanpy_find_variable_genes</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_find_markers/scanpy_find_markers.html">ebi-gxa/scanpy_find_markers/scanpy_find_markers</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/scanpy_filter_genes/scanpy_filter_genes.html">ebi-gxa/scanpy_filter_genes/scanpy_filter_genes</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_cellxgene.html">interactive_tool_cellxgene</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">
					Filter, plot and explore single-cell RNA-seq data with Scanpy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/seurat_scale_data/seurat_scale_data.html">ebi-gxa/seurat_scale_data/seurat_scale_data</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html">
					Filter, plot, and explore single cell RNA-seq data with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/seurat_run_umap/seurat_run_umap.html">ebi-gxa/seurat_run_umap/seurat_run_umap</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html">
					Filter, plot, and explore single cell RNA-seq data with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/seurat_run_pca/seurat_run_pca.html">ebi-gxa/seurat_run_pca/seurat_run_pca</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html">
					Filter, plot, and explore single cell RNA-seq data with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/seurat_plot/seurat_plot.html">ebi-gxa/seurat_plot/seurat_plot</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html">
					Filter, plot, and explore single cell RNA-seq data with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/seurat_normalise_data/seurat_normalise_data.html">ebi-gxa/seurat_normalise_data/seurat_normalise_data</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html">
					Filter, plot, and explore single cell RNA-seq data with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/seurat_find_variable_genes/seurat_find_variable_genes.html">ebi-gxa/seurat_find_variable_genes/seurat_find_variable_genes</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html">
					Filter, plot, and explore single cell RNA-seq data with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/seurat_find_neighbours/seurat_find_neighbours.html">ebi-gxa/seurat_find_neighbours/seurat_find_neighbours</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html">
					Filter, plot, and explore single cell RNA-seq data with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/seurat_find_markers/seurat_find_markers.html">ebi-gxa/seurat_find_markers/seurat_find_markers</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html">
					Filter, plot, and explore single cell RNA-seq data with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/seurat_find_clusters/seurat_find_clusters.html">ebi-gxa/seurat_find_clusters/seurat_find_clusters</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html">
					Filter, plot, and explore single cell RNA-seq data with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ebi-gxa/seurat_filter_cells/seurat_filter_cells.html">ebi-gxa/seurat_filter_cells/seurat_filter_cells</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html">
					Filter, plot, and explore single cell RNA-seq data with Seurat
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/gprofiler_gost/gprofiler_gost.html">iuc/gprofiler_gost/gprofiler_gost</a></td>
		<td>
			<ul>
				
				<li>
					Single Cell /
					<a href="/training-material/topics/single-cell/tutorials/GO-enrichment/tutorial.html">
					GO Enrichment Analysis on Single-Cell RNA-Seq Data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/muon-spectroscopy-computational-project/pm_uep_opt/pm_uep_opt.html">muon-spectroscopy-computational-project/pm_uep_opt/pm_uep_opt</a></td>
		<td>
			<ul>
				
				<li>
					Materials Science /
					<a href="/training-material/topics/materials-science/tutorials/muon-stopping-sites-muairss-uep/tutorial.html">
					Finding the muon stopping site with pymuon-suite in Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/muon-spectroscopy-computational-project/pm_muairss_read/pm_muairss_read.html">muon-spectroscopy-computational-project/pm_muairss_read/pm_muairss_read</a></td>
		<td>
			<ul>
				
				<li>
					Materials Science /
					<a href="/training-material/topics/materials-science/tutorials/muon-stopping-sites-muairss-uep/tutorial.html">
					Finding the muon stopping site with pymuon-suite in Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/cardinal_spectra_plots/cardinal_spectra_plots.html">galaxyp/cardinal_spectra_plots/cardinal_spectra_plots</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-analyte-distribution/tutorial.html">
					Mass spectrometry imaging: Examining the spatial distribution of analytes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/cardinal_filtering/cardinal_filtering.html">galaxyp/cardinal_filtering/cardinal_filtering</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-analyte-distribution/tutorial.html">
					Mass spectrometry imaging: Examining the spatial distribution of analytes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/recetox/waveica/waveica.html">recetox/waveica/waveica</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html">
					Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/recetox/riassigner/riassigner.html">recetox/riassigner/riassigner</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html">
					Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/recetox/ramclustr_define_experiment/ramclustr_define_experiment.html">recetox/ramclustr_define_experiment/ramclustr_define_experiment</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html">
					Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/recetox/ramclustr/ramclustr.html">recetox/ramclustr/ramclustr</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html">
					Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/recetox/matchms_similarity/matchms_similarity.html">recetox/matchms_similarity/matchms_similarity</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html">
					Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/recetox/matchms_formatter/matchms_formatter.html">recetox/matchms_formatter/matchms_formatter</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html">
					Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/recetox/matchms_filtering/matchms_filtering.html">recetox/matchms_filtering/matchms_filtering</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html">
					Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/fgiacomoni/hmdb_ms_search/wsdl_hmdb.html">fgiacomoni/hmdb_ms_search/wsdl_hmdb</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms/tutorial.html">
					Mass spectrometry: LC-MS analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ethevenot/univariate/Univariate.html">ethevenot/univariate/Univariate</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/lcms/tutorial.html">
					Mass spectrometry: LC-MS analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/recetox/xtb_molecular_optimization/xtb_molecular_optimization.html">recetox/xtb_molecular_optimization/xtb_molecular_optimization</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html">
					Predicting EI+ mass spectra with QCxMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/recetox/qcxms_production_run/qcxms_production_run.html">recetox/qcxms_production_run/qcxms_production_run</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html">
					Predicting EI+ mass spectra with QCxMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/recetox/qcxms_neutral_run/qcxms_neutral_run.html">recetox/qcxms_neutral_run/qcxms_neutral_run</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html">
					Predicting EI+ mass spectra with QCxMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/recetox/qcxms_getres/qcxms_getres.html">recetox/qcxms_getres/qcxms_getres</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html">
					Predicting EI+ mass spectra with QCxMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/ctb_im_conformers/ctb_im_conformers.html">bgruening/ctb_im_conformers/ctb_im_conformers</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html">
					Predicting EI+ mass spectra with QCxMS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/maldi_quant_preprocessing/maldi_quant_preprocessing.html">galaxyp/maldi_quant_preprocessing/maldi_quant_preprocessing</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html">
					Mass spectrometry imaging: Finding differential analytes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/maldi_quant_peak_detection/maldi_quant_peak_detection.html">galaxyp/maldi_quant_peak_detection/maldi_quant_peak_detection</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html">
					Mass spectrometry imaging: Finding differential analytes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/cardinal_segmentations/cardinal_segmentations.html">galaxyp/cardinal_segmentations/cardinal_segmentations</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html">
					Mass spectrometry imaging: Finding differential analytes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/cardinal_preprocessing/cardinal_preprocessing.html">galaxyp/cardinal_preprocessing/cardinal_preprocessing</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html">
					Mass spectrometry imaging: Finding differential analytes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/cardinal_combine/cardinal_combine.html">galaxyp/cardinal_combine/cardinal_combine</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html">
					Mass spectrometry imaging: Finding differential analytes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/cardinal_classification/cardinal_classification.html">galaxyp/cardinal_classification/cardinal_classification</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html">
					Mass spectrometry imaging: Finding differential analytes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/join_files_on_column_fuzzy/join_files_on_column_fuzzy.html">bgruening/join_files_on_column_fuzzy/join_files_on_column_fuzzy</a></td>
		<td>
			<ul>
				
				<li>
					Metabolomics /
					<a href="/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html">
					Mass spectrometry imaging: Finding differential analytes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/intermine_galaxy_exchange/galaxy_intermine_exchange.html">iuc/intermine_galaxy_exchange/galaxy_intermine_exchange</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/intermine/tutorial.html">
					InterMine integration with Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/intermine.html">intermine</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/intermine/tutorial.html">
					InterMine integration with Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/ena_upload/ena_upload.html">iuc/ena_upload/ena_upload</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.html">
					Submitting sequence data to ENA
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ieguinoa/ena_webin_cli/ena_consensus_submit.html">ieguinoa/ena_webin_cli/ena_consensus_submit</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.html">
					Submitting sequence data to ENA
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_tac.html">bgruening/text_processing/tp_tac</a></td>
		<td>
			<ul>
				
				<li>
					Using Galaxy and Managing your Data /
					<a href="/training-material/topics/galaxy-interface/tutorials/workflow-editor/tutorial.html">
					Creating, Editing and Importing Galaxy Workflows
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/mvdbeek/add_input_name_as_column/addName.html">mvdbeek/add_input_name_as_column/addName</a></td>
		<td>
			<ul>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/snp_sites/snp_sites.html">iuc/snp_sites/snp_sites</a></td>
		<td>
			<ul>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/snp_dists/snp_dists.html">iuc/snp_dists/snp_dists</a></td>
		<td>
			<ul>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html">
					Identifying tuberculosis transmission links: from SNPs to transmission clusters
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/raxml/raxml.html">iuc/raxml/raxml</a></td>
		<td>
			<ul>
				
				<li>
					Evolution /
					<a href="/training-material/topics/evolution/tutorials/mtb_phylogeny/tutorial.html">
					Tree thinking for tuberculosis evolution and epidemiology
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/climate/psy_maps/psy_maps.html">climate/psy_maps/psy_maps</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/climate-101/tutorial.html">
					Getting your hands-on climate data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/climate/cds_essential_variability/cds_essential_variability.html">climate/cds_essential_variability/cds_essential_variability</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/climate-101/tutorial.html">
					Getting your hands-on climate data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/xarray_netcdf2netcdf/xarray_netcdf2netcdf.html">ecology/xarray_netcdf2netcdf/xarray_netcdf2netcdf</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/pangeo/tutorial.html">
					Pangeo ecosystem 101 for everyone - Introduction to Xarray Galaxy Tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/divand_full_analysis/divand_full_analysis.html">ecology/divand_full_analysis/divand_full_analysis</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/ocean-variables/tutorial.html">
					Ocean's variables study
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_divand.html">interactive_tool_divand</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/ocean-variables/tutorial.html">
					Ocean's variables study
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_pangeo_notebook.html">interactive_tool_pangeo_notebook</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/pangeo-notebook/tutorial.html">
					Pangeo Notebook in Galaxy - Introduction to Xarray
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/climate/ctsm_fates/ctsm_fates.html">climate/ctsm_fates/ctsm_fates</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/fates/tutorial.html">
					Functionally Assembled Terrestrial Ecosystem Simulator (FATES)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_copernicus.html">interactive_tool_copernicus</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/sentinel5_data/tutorial.html">
					Sentinel 5P data visualisation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ecology/timeseries_extraction/timeseries_extraction.html">ecology/timeseries_extraction/timeseries_extraction</a></td>
		<td>
			<ul>
				
				<li>
					Climate /
					<a href="/training-material/topics/climate/tutorials/argo_pangeo/tutorial.html">
					Analyse Argo data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/imgteam/2d_filter_segmentation_by_features/ip_2d_filter_segmentation_by_features.html">imgteam/2d_filter_segmentation_by_features/ip_2d_filter_segmentation_by_features</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/hela-screen-analysis/tutorial.html">
					Analyse HeLa fluorescence siRNA screen
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/imgteam/2d_feature_extraction/ip_2d_feature_extraction.html">imgteam/2d_feature_extraction/ip_2d_feature_extraction</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/hela-screen-analysis/tutorial.html">
					Analyse HeLa fluorescence siRNA screen
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/perssond/quantification/quantification.html">perssond/quantification/quantification</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.html">
					End-to-End Tissue Microarray Image Analysis with Galaxy-ME
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/perssond/coreograph/unet_coreograph.html">perssond/coreograph/unet_coreograph</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.html">
					End-to-End Tissue Microarray Image Analysis with Galaxy-ME
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/perssond/basic_illumination/basic_illumination.html">perssond/basic_illumination/basic_illumination</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.html">
					End-to-End Tissue Microarray Image Analysis with Galaxy-ME
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/perssond/ashlar/ashlar.html">perssond/ashlar/ashlar</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.html">
					End-to-End Tissue Microarray Image Analysis with Galaxy-ME
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/imgteam/bfconvert/ip_convertimage.html">imgteam/bfconvert/ip_convertimage</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.html">
					End-to-End Tissue Microarray Image Analysis with Galaxy-ME
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/goeckslab/vitessce_spatial/vitessce_spatial.html">goeckslab/vitessce_spatial/vitessce_spatial</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.html">
					End-to-End Tissue Microarray Image Analysis with Galaxy-ME
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/goeckslab/scimap_phenotyping/scimap_phenotyping.html">goeckslab/scimap_phenotyping/scimap_phenotyping</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.html">
					End-to-End Tissue Microarray Image Analysis with Galaxy-ME
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/goeckslab/scimap_mcmicro_to_anndata/scimap_mcmicro_to_anndata.html">goeckslab/scimap_mcmicro_to_anndata/scimap_mcmicro_to_anndata</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.html">
					End-to-End Tissue Microarray Image Analysis with Galaxy-ME
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/goeckslab/rename_tiff_channels/rename_tiff_channels.html">goeckslab/rename_tiff_channels/rename_tiff_channels</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.html">
					End-to-End Tissue Microarray Image Analysis with Galaxy-ME
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/goeckslab/mesmer/mesmer.html">goeckslab/mesmer/mesmer</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.html">
					End-to-End Tissue Microarray Image Analysis with Galaxy-ME
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/imgteam/spot_detection_2d/ip_spot_detection_2d.html">imgteam/spot_detection_2d/ip_spot_detection_2d</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/detection-of-mitoflashes/tutorial.html">
					Tracking of mitochondria and capturing mitoflashes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/imgteam/points_association_nn/ip_points_association_nn.html">imgteam/points_association_nn/ip_points_association_nn</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/detection-of-mitoflashes/tutorial.html">
					Tracking of mitochondria and capturing mitoflashes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/imgteam/curve_fitting/ip_curve_fitting.html">imgteam/curve_fitting/ip_curve_fitting</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/detection-of-mitoflashes/tutorial.html">
					Tracking of mitochondria and capturing mitoflashes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_track_objects/cp_track_objects.html">bgruening/cp_track_objects/cp_track_objects</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html">
					Object tracking using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_tile/cp_tile.html">bgruening/cp_tile/cp_tile</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html">
					Object tracking using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_overlay_outlines/cp_overlay_outlines.html">bgruening/cp_overlay_outlines/cp_overlay_outlines</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html">
					Object tracking using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_color_to_gray/cp_color_to_gray.html">bgruening/cp_color_to_gray/cp_color_to_gray</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html">
					Object tracking using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/idr_download_by_ids/idr_download_by_ids.html">iuc/idr_download_by_ids/idr_download_by_ids</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_relate_objects/cp_relate_objects.html">bgruening/cp_relate_objects/cp_relate_objects</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_measure_texture/cp_measure_texture.html">bgruening/cp_measure_texture/cp_measure_texture</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_measure_image_quality/cp_measure_image_quality.html">bgruening/cp_measure_image_quality/cp_measure_image_quality</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_measure_image_intensity/cp_measure_image_intensity.html">bgruening/cp_measure_image_intensity/cp_measure_image_intensity</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_measure_image_area_occupied/cp_measure_image_area_occupied.html">bgruening/cp_measure_image_area_occupied/cp_measure_image_area_occupied</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_measure_granularity/cp_measure_granularity.html">bgruening/cp_measure_granularity/cp_measure_granularity</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_mask_image/cp_mask_image.html">bgruening/cp_mask_image/cp_mask_image</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_image_math/cp_image_math.html">bgruening/cp_image_math/cp_image_math</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_gray_to_color/cp_gray_to_color.html">bgruening/cp_gray_to_color/cp_gray_to_color</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_enhance_or_suppress_features/cp_enhance_or_suppress_features.html">bgruening/cp_enhance_or_suppress_features/cp_enhance_or_suppress_features</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_display_data_on_image/cp_display_data_on_image.html">bgruening/cp_display_data_on_image/cp_display_data_on_image</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cp_convert_objects_to_image/cp_convert_objects_to_image.html">bgruening/cp_convert_objects_to_image/cp_convert_objects_to_image</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">
					Nucleoli segmentation and feature extraction using CellProfiler
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ufz/omero_roi_import/omero_roi_import.html">ufz/omero_roi_import/omero_roi_import</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/omero-suite/tutorial.html">
					Overview of the Galaxy OMERO-suite - Upload images and metadata in OMERO using Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ufz/omero_metadata_import/omero_metadata_import.html">ufz/omero_metadata_import/omero_metadata_import</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/omero-suite/tutorial.html">
					Overview of the Galaxy OMERO-suite - Upload images and metadata in OMERO using Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ufz/omero_import/omero_import.html">ufz/omero_import/omero_import</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/omero-suite/tutorial.html">
					Overview of the Galaxy OMERO-suite - Upload images and metadata in OMERO using Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ufz/omero_get_value/omero_get_value.html">ufz/omero_get_value/omero_get_value</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/omero-suite/tutorial.html">
					Overview of the Galaxy OMERO-suite - Upload images and metadata in OMERO using Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ufz/omero_get_id/omero_get_id.html">ufz/omero_get_id/omero_get_id</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/omero-suite/tutorial.html">
					Overview of the Galaxy OMERO-suite - Upload images and metadata in OMERO using Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ufz/omero_filter/omero_filter.html">ufz/omero_filter/omero_filter</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/omero-suite/tutorial.html">
					Overview of the Galaxy OMERO-suite - Upload images and metadata in OMERO using Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/imgteam/overlay_images/ip_overlay_images.html">imgteam/overlay_images/ip_overlay_images</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html">
					Introduction to Image Analysis using Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/imgteam/image_info/ip_imageinfo.html">imgteam/image_info/ip_imageinfo</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html">
					Introduction to Image Analysis using Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/imgteam/count_objects/ip_count_objects.html">imgteam/count_objects/ip_count_objects</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html">
					Introduction to Image Analysis using Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/imgteam/2d_histogram_equalization/ip_histogram_equalization.html">imgteam/2d_histogram_equalization/ip_histogram_equalization</a></td>
		<td>
			<ul>
				
				<li>
					Imaging /
					<a href="/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html">
					Introduction to Image Analysis using Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/tduigou/rrparser/rrparser.html">tduigou/rrparser/rrparser</a></td>
		<td>
			<ul>
				
				<li>
					Synthetic Biology /
					<a href="/training-material/topics/synthetic-biology/tutorials/retrosynthesis_analysis/tutorial.html">
					Generating theoretical possible pathways for the production of Lycopene in E.Coli using Retrosynthesis tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/tduigou/rpextractsink/rpextractsink.html">tduigou/rpextractsink/rpextractsink</a></td>
		<td>
			<ul>
				
				<li>
					Synthetic Biology /
					<a href="/training-material/topics/synthetic-biology/tutorials/retrosynthesis_analysis/tutorial.html">
					Generating theoretical possible pathways for the production of Lycopene in E.Coli using Retrosynthesis tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/tduigou/rpcompletion/rpcompletion.html">tduigou/rpcompletion/rpcompletion</a></td>
		<td>
			<ul>
				
				<li>
					Synthetic Biology /
					<a href="/training-material/topics/synthetic-biology/tutorials/retrosynthesis_analysis/tutorial.html">
					Generating theoretical possible pathways for the production of Lycopene in E.Coli using Retrosynthesis tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/tduigou/rp2paths/rp2paths.html">tduigou/rp2paths/rp2paths</a></td>
		<td>
			<ul>
				
				<li>
					Synthetic Biology /
					<a href="/training-material/topics/synthetic-biology/tutorials/retrosynthesis_analysis/tutorial.html">
					Generating theoretical possible pathways for the production of Lycopene in E.Coli using Retrosynthesis tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/tduigou/retropath2/retropath2.html">tduigou/retropath2/retropath2</a></td>
		<td>
			<ul>
				
				<li>
					Synthetic Biology /
					<a href="/training-material/topics/synthetic-biology/tutorials/retrosynthesis_analysis/tutorial.html">
					Generating theoretical possible pathways for the production of Lycopene in E.Coli using Retrosynthesis tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/tduigou/get_sbml_model/get_sbml_model.html">tduigou/get_sbml_model/get_sbml_model</a></td>
		<td>
			<ul>
				
				<li>
					Synthetic Biology /
					<a href="/training-material/topics/synthetic-biology/tutorials/retrosynthesis_analysis/tutorial.html">
					Generating theoretical possible pathways for the production of Lycopene in E.Coli using Retrosynthesis tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/tduigou/rpthermo/rpthermo.html">tduigou/rpthermo/rpthermo</a></td>
		<td>
			<ul>
				
				<li>
					Synthetic Biology /
					<a href="/training-material/topics/synthetic-biology/tutorials/pathway_analysis/tutorial.html">
					Evaluating and ranking a set of pathways based on multiple metrics
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/tduigou/rpscore/rpscore.html">tduigou/rpscore/rpscore</a></td>
		<td>
			<ul>
				
				<li>
					Synthetic Biology /
					<a href="/training-material/topics/synthetic-biology/tutorials/pathway_analysis/tutorial.html">
					Evaluating and ranking a set of pathways based on multiple metrics
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/tduigou/rpranker/rpranker.html">tduigou/rpranker/rpranker</a></td>
		<td>
			<ul>
				
				<li>
					Synthetic Biology /
					<a href="/training-material/topics/synthetic-biology/tutorials/pathway_analysis/tutorial.html">
					Evaluating and ranking a set of pathways based on multiple metrics
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/rpfba/rpfba.html">iuc/rpfba/rpfba</a></td>
		<td>
			<ul>
				
				<li>
					Synthetic Biology /
					<a href="/training-material/topics/synthetic-biology/tutorials/pathway_analysis/tutorial.html">
					Evaluating and ranking a set of pathways based on multiple metrics
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/tduigou/selenzy/selenzy-wrapper.html">tduigou/selenzy/selenzy-wrapper</a></td>
		<td>
			<ul>
				
				<li>
					Synthetic Biology /
					<a href="/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.html">
					Designing plasmids encoding predicted pathways by using the BASIC assembly method
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/tduigou/rpbasicdesign/rpbasicdesign.html">tduigou/rpbasicdesign/rpbasicdesign</a></td>
		<td>
			<ul>
				
				<li>
					Synthetic Biology /
					<a href="/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.html">
					Designing plasmids encoding predicted pathways by using the BASIC assembly method
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/tduigou/dnabot/dnabot.html">tduigou/dnabot/dnabot</a></td>
		<td>
			<ul>
				
				<li>
					Synthetic Biology /
					<a href="/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.html">
					Designing plasmids encoding predicted pathways by using the BASIC assembly method
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/krakentools_beta_diversity/krakentools_beta_diversity.html">iuc/krakentools_beta_diversity/krakentools_beta_diversity</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/diversity/tutorial.html">
					Calculating α and β diversity from microbiome taxonomic data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/krakentools_alpha_diversity/krakentools_alpha_diversity.html">iuc/krakentools_alpha_diversity/krakentools_alpha_diversity</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/diversity/tutorial.html">
					Calculating α and β diversity from microbiome taxonomic data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/phyloseq_from_dada2/phyloseq_from_dada2.html">iuc/phyloseq_from_dada2/phyloseq_from_dada2</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/dada2_seqcounts/dada2_seqCounts.html">iuc/dada2_seqcounts/dada2_seqCounts</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/dada2_removebimeradenovo/dada2_removeBimeraDenovo.html">iuc/dada2_removebimeradenovo/dada2_removeBimeraDenovo</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/dada2_plotqualityprofile/dada2_plotQualityProfile.html">iuc/dada2_plotqualityprofile/dada2_plotQualityProfile</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/dada2_mergepairs/dada2_mergePairs.html">iuc/dada2_mergepairs/dada2_mergePairs</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/dada2_makesequencetable/dada2_makeSequenceTable.html">iuc/dada2_makesequencetable/dada2_makeSequenceTable</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/dada2_learnerrors/dada2_learnErrors.html">iuc/dada2_learnerrors/dada2_learnErrors</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/dada2_filterandtrim/dada2_filterAndTrim.html">iuc/dada2_filterandtrim/dada2_filterAndTrim</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/dada2_dada/dada2_dada.html">iuc/dada2_dada/dada2_dada</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/dada2_assigntaxonomyaddspecies/dada2_assignTaxonomyAddspecies.html">iuc/dada2_assigntaxonomyaddspecies/dada2_assignTaxonomyAddspecies</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_phyloseq.html">interactive_tool_phyloseq</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html">
					Building an amplicon sequence variant (ASV) table from 16S data using DADA2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/kraken_biom/kraken_biom.html">iuc/kraken_biom/kraken_biom</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/taxonomic-profiling/tutorial.html">
					Taxonomic Profiling and Visualization of Metagenomic Data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_phinch.html">interactive_tool_phinch</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/taxonomic-profiling/tutorial.html">
					Taxonomic Profiling and Visualization of Metagenomic Data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_pavian.html">interactive_tool_pavian</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/taxonomic-profiling/tutorial.html">
					Taxonomic Profiling and Visualization of Metagenomic Data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/ggplot2_histogram/ggplot2_histogram.html">iuc/ggplot2_histogram/ggplot2_histogram</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/estrain/add_column_headers/add_column_headers.html">estrain/add_column_headers/add_column_headers</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/count_gff_features/count_gff_features.html">devteam/count_gff_features/count_gff_features</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/CONVERTER_fasta_to_tabular.html">CONVERTER_fasta_to_tabular</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/metaplasmidome_query/tutorial.html">
					Query an annotated mobile genetic element database to identify and annotate genetic elements (e.g. plasmids) in metagenomics data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/datamash_reverse/datamash_reverse.html">iuc/datamash_reverse/datamash_reverse</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/nanopore-16S-metagenomics/tutorial.html">
					16S Microbial analysis with Nanopore data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/samtools_depth/samtools_depth.html">iuc/samtools_depth/samtools_depth</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/samtools_coverage/samtools_coverage.html">iuc/samtools_coverage/samtools_coverage</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mlst/mlst.html">iuc/mlst/mlst</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/krakentools_extract_kraken_reads/krakentools_extract_kraken_reads.html">iuc/krakentools_extract_kraken_reads/krakentools_extract_kraken_reads</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/ggplot2_heatmap/ggplot2_heatmap.html">iuc/ggplot2_heatmap/ggplot2_heatmap</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/clair3/clair3.html">iuc/clair3/clair3</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bedtools/bedtools_getfastabed.html">iuc/bedtools/bedtools_getfastabed</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bamtools_split_mapped/bamtools_split_mapped.html">iuc/bamtools_split_mapped/bamtools_split_mapped</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/abricate/abricate.html">iuc/abricate/abricate</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
					Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/earlhaminst/lotus2/lotus2.html">earlhaminst/lotus2/lotus2</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/lotus2-identifying-fungi/tutorial.html">
					Identifying Mycorrhizal Fungi from ITS2 sequencing using LotuS2
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_taxonomy_to_krona/mothur_taxonomy_to_krona.html">iuc/mothur_taxonomy_to_krona/mothur_taxonomy_to_krona</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_seq_error/mothur_seq_error.html">iuc/mothur_seq_error/mothur_seq_error</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_make_contigs/mothur_make_contigs.html">iuc/mothur_make_contigs/mothur_make_contigs</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_get_groups/mothur_get_groups.html">iuc/mothur_get_groups/mothur_get_groups</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html">
					16S Microbial Analysis with mothur (extended)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/plasflow/PlasFlow.html">iuc/plasflow/PlasFlow</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/plasmid-metagenomics-nanopore/tutorial.html">
					Antibiotic resistance detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/miniasm/miniasm.html">iuc/miniasm/miniasm</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/plasmid-metagenomics-nanopore/tutorial.html">
					Antibiotic resistance detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/circos/circos_wiggle_to_scatter.html">iuc/circos/circos_wiggle_to_scatter</a></td>
		<td>
			<ul>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/circos-microbial/tutorial.html">
					Ploting a Microbial Genome with Circos
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/circos/circos_interval_to_tile.html">iuc/circos/circos_interval_to_tile</a></td>
		<td>
			<ul>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/circos-microbial/tutorial.html">
					Ploting a Microbial Genome with Circos
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/circos/circos_interval_to_text.html">iuc/circos/circos_interval_to_text</a></td>
		<td>
			<ul>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/circos-microbial/tutorial.html">
					Ploting a Microbial Genome with Circos
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/circos/circos_gc_skew.html">iuc/circos/circos_gc_skew</a></td>
		<td>
			<ul>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/circos-microbial/tutorial.html">
					Ploting a Microbial Genome with Circos
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/gff2bed1.html">gff2bed1</a></td>
		<td>
			<ul>
				
				<li>
					Visualisation /
					<a href="/training-material/topics/visualisation/tutorials/circos-microbial/tutorial.html">
					Ploting a Microbial Genome with Circos
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/trinotate/trinotate.html">iuc/trinotate/trinotate</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/trinity_samples_qccheck/trinity_samples_qccheck.html">iuc/trinity_samples_qccheck/trinity_samples_qccheck</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/trinity_run_de_analysis/trinity_run_de_analysis.html">iuc/trinity_run_de_analysis/trinity_run_de_analysis</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/trinity_filter_low_expr_transcripts/trinity_filter_low_expr_transcripts.html">iuc/trinity_filter_low_expr_transcripts/trinity_filter_low_expr_transcripts</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/trinity_define_clusters_by_cutting_tree/trinity_define_clusters_by_cutting_tree.html">iuc/trinity_define_clusters_by_cutting_tree/trinity_define_clusters_by_cutting_tree</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/trinity_contig_exn50_statistic/trinity_contig_exn50_statistic.html">iuc/trinity_contig_exn50_statistic/trinity_contig_exn50_statistic</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/trinity_analyze_diff_expr/trinity_analyze_diff_expr.html">iuc/trinity_analyze_diff_expr/trinity_analyze_diff_expr</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/trinity_align_and_estimate_abundance/trinity_align_and_estimate_abundance.html">iuc/trinity_align_and_estimate_abundance/trinity_align_and_estimate_abundance</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/trinity_abundance_estimates_to_matrix/trinity_abundance_estimates_to_matrix.html">iuc/trinity_abundance_estimates_to_matrix/trinity_abundance_estimates_to_matrix</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/trinity/trinity.html">iuc/trinity/trinity</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/transdecoder/transdecoder.html">iuc/transdecoder/transdecoder</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/describe_samples/describe_samples.html">iuc/describe_samples/describe_samples</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
					De novo transcriptome assembly, annotation, and differential expression analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bedtools/bedtools_bamtofastq.html">iuc/bedtools/bedtools_bamtofastq</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/srna/tutorial.html">
					Differential abundance testing of small RNAs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/samtool_filter2/samtool_filter2.html">devteam/samtool_filter2/samtool_filter2</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/srna/tutorial.html">
					Differential abundance testing of small RNAs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/fastq_manipulation/fastq_manipulation.html">devteam/fastq_manipulation/fastq_manipulation</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/srna/tutorial.html">
					Differential abundance testing of small RNAs
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/chira_quantify/chira_quantify.html">iuc/chira_quantify/chira_quantify</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-interactome/tutorial.html">
					RNA-RNA interactome data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/chira_merge/chira_merge.html">iuc/chira_merge/chira_merge</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-interactome/tutorial.html">
					RNA-RNA interactome data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/chira_map/chira_map.html">iuc/chira_map/chira_map</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-interactome/tutorial.html">
					RNA-RNA interactome data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/chira_extract/chira_extract.html">iuc/chira_extract/chira_extract</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-interactome/tutorial.html">
					RNA-RNA interactome data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/chira_collapse/chira_collapse.html">iuc/chira_collapse/chira_collapse</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-interactome/tutorial.html">
					RNA-RNA interactome data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/rnateam/targetfinder/targetfinder.html">rnateam/targetfinder/targetfinder</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/rnateam/mirdeep2_quantifier/rbc_mirdeep2_quantifier.html">rnateam/mirdeep2_quantifier/rbc_mirdeep2_quantifier</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/rnateam/mirdeep2_mapper/rbc_mirdeep2_mapper.html">rnateam/mirdeep2_mapper/rbc_mirdeep2_mapper</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/peterjc/sample_seqs/sample_seqs.html">peterjc/sample_seqs/sample_seqs</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
					Whole transcriptome analysis of Arabidopsis thaliana
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/heinz/heinz_visualization.html">iuc/heinz/heinz_visualization</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html">
					Network analysis with Heinz
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/heinz/heinz_scoring.html">iuc/heinz/heinz_scoring</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html">
					Network analysis with Heinz
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/heinz/heinz_bum.html">iuc/heinz/heinz_bum</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html">
					Network analysis with Heinz
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/heinz/heinz.html">iuc/heinz/heinz</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html">
					Network analysis with Heinz
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/rnateam/blockclust/blockclust.html">rnateam/blockclust/blockclust</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.html">
					Small Non-coding RNA Clustering using BlockClust
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/rnateam/blockbuster/blockbuster.html">rnateam/blockbuster/blockbuster</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.html">
					Small Non-coding RNA Clustering using BlockClust
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/samtools_sort/samtools_sort.html">devteam/samtools_sort/samtools_sort</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.html">
					Small Non-coding RNA Clustering using BlockClust
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/goslimmer/goslimmer.html">iuc/goslimmer/goslimmer</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/goenrichment/tutorial.html">
					GO Enrichment Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nilesh/rseqc/rseqc_junction_saturation.html">nilesh/rseqc/rseqc_junction_saturation</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nilesh/rseqc/rseqc_junction_annotation.html">nilesh/rseqc/rseqc_junction_annotation</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nilesh/rseqc/rseqc_inner_distance.html">nilesh/rseqc/rseqc_inner_distance</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/rnaquast/rna_quast.html">iuc/rnaquast/rna_quast</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/isoformswitchanalyzer/isoformswitchanalyzer.html">iuc/isoformswitchanalyzer/isoformswitchanalyzer</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/text_processing/tp_uniq_tool.html">bgruening/text_processing/tp_uniq_tool</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/pfamscan/pfamscan.html">bgruening/pfamscan/pfamscan</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/cpat/cpat.html">bgruening/cpat/cpat</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html">
					Genome-wide alternative splicing analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/fgsea/fgsea.html">iuc/fgsea/fgsea</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html">
					3: RNA-seq genes to pathways
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/egsea/egsea.html">iuc/egsea/egsea</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html">
					3: RNA-seq genes to pathways
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/rnateam/rcas/rcas.html">rnateam/rcas/rcas</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/rnateam/peakachu/peakachu.html">rnateam/peakachu/peakachu</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/umi_tools_dedup/umi_tools_dedup.html">iuc/umi_tools_dedup/umi_tools_dedup</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bedtools/bedtools_slopbed.html">iuc/bedtools/bedtools_slopbed</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bedtools/bedtools_genomecoveragebed_bedgraph.html">iuc/bedtools/bedtools_genomecoveragebed_bedgraph</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bctools_extract_alignment_ends/bctools_extract_alignment_ends.html">iuc/bctools_extract_alignment_ends/bctools_extract_alignment_ends</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html">
					CLIP-Seq data analysis from pre-processing to motif detection
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/iedb_api/iedb_api.html">iuc/iedb_api/iedb_api</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-7-hla-binding-novel-peptides/tutorial.html">
					Neoantigen 7: IEDB binding PepQuery Validated Neopeptides
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/proteore/proteore_reactome/reactome_analysis.html">proteore/proteore_reactome/reactome_analysis</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteome_annotation/tutorial.html">
					Annotating a protein list identified by LC-MS/MS experiments
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/proteore/proteore_clusterprofiler/cluter_profiler.html">proteore/proteore_clusterprofiler/cluter_profiler</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteome_annotation/tutorial.html">
					Annotating a protein list identified by LC-MS/MS experiments
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/flashlfq/flashlfq.html">galaxyp/flashlfq/flashlfq</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html">
					metaQuantome 1: Data creation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/validate_fasta_database/validate_fasta_database.html">galaxyp/validate_fasta_database/validate_fasta_database</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-3-fragpipe-discovery/tutorial.html">
					Neoantigen 3: Database merge and FragPipe discovery
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/fragpipe/fragpipe.html">galaxyp/fragpipe/fragpipe</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-3-fragpipe-discovery/tutorial.html">
					Neoantigen 3: Database merge and FragPipe discovery
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/PeakPickerHiRes.html">PeakPickerHiRes</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html">
					Peptide and Protein ID using SearchGUI and PeptideShaker
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/FileConverter.html">FileConverter</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html">
					Peptide and Protein ID using SearchGUI and PeptideShaker
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_peakpickerhires/PeakPickerHiRes.html">galaxyp/openms_peakpickerhires/PeakPickerHiRes</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_msgfplusadapter/MSGFPlusAdapter.html">galaxyp/openms_msgfplusadapter/MSGFPlusAdapter</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_idmerger/IDMerger.html">galaxyp/openms_idmerger/IDMerger</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_idfilter/IDFilter.html">galaxyp/openms_idfilter/IDFilter</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_consensusid/ConsensusID.html">galaxyp/openms_consensusid/ConsensusID</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">
					Peptide and Protein ID using OpenMS tools
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_decoydatabase/DecoyDatabase.html">galaxyp/openms_decoydatabase/DecoyDatabase</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/database-handling/tutorial.html">
					Protein FASTA Database Handling
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/pyprophet_score/pyprophet_score.html">galaxyp/pyprophet_score/pyprophet_score</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html">
					DIA Analysis using OpenSwathWorkflow
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/pyprophet_protein/pyprophet_protein.html">galaxyp/pyprophet_protein/pyprophet_protein</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html">
					DIA Analysis using OpenSwathWorkflow
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/pyprophet_peptide/pyprophet_peptide.html">galaxyp/pyprophet_peptide/pyprophet_peptide</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html">
					DIA Analysis using OpenSwathWorkflow
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/pyprophet_merge/pyprophet_merge.html">galaxyp/pyprophet_merge/pyprophet_merge</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html">
					DIA Analysis using OpenSwathWorkflow
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/pyprophet_export/pyprophet_export.html">galaxyp/pyprophet_export/pyprophet_export</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html">
					DIA Analysis using OpenSwathWorkflow
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_openswathworkflow/OpenSwathWorkflow.html">galaxyp/openms_openswathworkflow/OpenSwathWorkflow</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html">
					DIA Analysis using OpenSwathWorkflow
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/peptide_genomic_coordinate/peptide_genomic_coordinate.html">galaxyp/peptide_genomic_coordinate/peptide_genomic_coordinate</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.html">
					Proteogenomics 3: Novel peptide analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_targetedfileconverter/TargetedFileConverter.html">galaxyp/openms_targetedfileconverter/TargetedFileConverter</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html">
					Library Generation for DIA Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_openswathdecoygenerator/OpenSwathDecoyGenerator.html">galaxyp/openms_openswathdecoygenerator/OpenSwathDecoyGenerator</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html">
					Library Generation for DIA Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_openswathassaygenerator/OpenSwathAssayGenerator.html">galaxyp/openms_openswathassaygenerator/OpenSwathAssayGenerator</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html">
					Library Generation for DIA Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/diapysef/diapysef.html">galaxyp/diapysef/diapysef</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html">
					Library Generation for DIA Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/mz_to_sqlite/mz_to_sqlite.html">galaxyp/mz_to_sqlite/mz_to_sqlite</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html">
					Proteogenomics 2: Database Search
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/recetox/bioconductor_scp/bioconductor_scp.html">recetox/bioconductor_scp/bioconductor_scp</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/bioconductor-scp/tutorial.html">
					Single Cell Proteomics data analysis with bioconductor-scp
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/arriba_get_filters/arriba_get_filters.html">iuc/arriba_get_filters/arriba_get_filters</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-1-fusion-database-generation/tutorial.html">
					Neoantigen 1: Fusion-Database-Generation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/arriba/arriba.html">iuc/arriba/arriba</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-1-fusion-database-generation/tutorial.html">
					Neoantigen 1: Fusion-Database-Generation
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/PeptideIndexer.html">PeptideIndexer</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ntails/tutorial.html">
					Detection and quantitation of N-termini (degradomics) via N-TAILS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/MzTabExporter.html">MzTabExporter</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ntails/tutorial.html">
					Detection and quantitation of N-termini (degradomics) via N-TAILS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/MSGFPlusAdapter.html">MSGFPlusAdapter</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ntails/tutorial.html">
					Detection and quantitation of N-termini (degradomics) via N-TAILS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/IDMerger.html">IDMerger</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ntails/tutorial.html">
					Detection and quantitation of N-termini (degradomics) via N-TAILS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/IDMapper.html">IDMapper</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ntails/tutorial.html">
					Detection and quantitation of N-termini (degradomics) via N-TAILS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/IDFilter.html">IDFilter</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ntails/tutorial.html">
					Detection and quantitation of N-termini (degradomics) via N-TAILS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/IDConflictResolver.html">IDConflictResolver</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ntails/tutorial.html">
					Detection and quantitation of N-termini (degradomics) via N-TAILS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/HighResPrecursorMassCorrector.html">HighResPrecursorMassCorrector</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ntails/tutorial.html">
					Detection and quantitation of N-termini (degradomics) via N-TAILS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/FileMerger.html">FileMerger</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ntails/tutorial.html">
					Detection and quantitation of N-termini (degradomics) via N-TAILS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/FileFilter.html">FileFilter</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ntails/tutorial.html">
					Detection and quantitation of N-termini (degradomics) via N-TAILS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/FeatureFinderMultiplex.html">FeatureFinderMultiplex</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ntails/tutorial.html">
					Detection and quantitation of N-termini (degradomics) via N-TAILS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/FalseDiscoveryRate.html">FalseDiscoveryRate</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ntails/tutorial.html">
					Detection and quantitation of N-termini (degradomics) via N-TAILS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ConsensusID.html">ConsensusID</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ntails/tutorial.html">
					Detection and quantitation of N-termini (degradomics) via N-TAILS
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/jay/pdaug_ml_models/pdaug_ml_models.html">jay/pdaug_ml_models/pdaug_ml_models</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/ml-modeling-of-anti-cancer-peptides/tutorial.html">
					Machine Learning Modeling of Anticancer Peptides
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/encyclopedia_searchtolib/encyclopedia_searchtolib.html">galaxyp/encyclopedia_searchtolib/encyclopedia_searchtolib</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/encyclopedia/tutorial.html">
					EncyclopeDIA
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/encyclopedia_quantify/encyclopedia_quantify.html">galaxyp/encyclopedia_quantify/encyclopedia_quantify</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/encyclopedia/tutorial.html">
					EncyclopeDIA
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/jay/pdaug_tsvtofasta/pdaug_tsvtofasta.html">jay/pdaug_tsvtofasta/pdaug_tsvtofasta</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/peptide-library-data-analysis/tutorial.html">
					Peptide Library Data Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/jay/pdaug_peptide_sequence_analysis/pdaug_peptide_sequence_analysis.html">jay/pdaug_peptide_sequence_analysis/pdaug_peptide_sequence_analysis</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/peptide-library-data-analysis/tutorial.html">
					Peptide Library Data Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/jay/pdaug_peptide_data_access/pdaug_peptide_data_access.html">jay/pdaug_peptide_data_access/pdaug_peptide_data_access</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/peptide-library-data-analysis/tutorial.html">
					Peptide Library Data Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/jay/pdaug_fishers_plot/pdaug_fishers_plot.html">jay/pdaug_fishers_plot/pdaug_fishers_plot</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/peptide-library-data-analysis/tutorial.html">
					Peptide Library Data Analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/proteore/proteore_prot_features/prot_features.html">proteore/proteore_prot_features/prot_features</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html">
					Biomarker candidate identification
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/proteore/proteore_ms_observation_pepatlas/retr_pepatlas1.html">proteore/proteore_ms_observation_pepatlas/retr_pepatlas1</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html">
					Biomarker candidate identification
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/peterjc/tmhmm_and_signalp/wolf_psort.html">peterjc/tmhmm_and_signalp/wolf_psort</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html">
					Secretome Prediction
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/term_id_vs_term_name.html">term_id_vs_term_name</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html">
					Secretome Prediction
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/get_subontology_from.html">get_subontology_from</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html">
					Secretome Prediction
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_proteinquantifier/ProteinQuantifier.html">galaxyp/openms_proteinquantifier/ProteinQuantifier</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_multiplexresolver/MultiplexResolver.html">galaxyp/openms_multiplexresolver/MultiplexResolver</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_idmapper/IDMapper.html">galaxyp/openms_idmapper/IDMapper</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_idconflictresolver/IDConflictResolver.html">galaxyp/openms_idconflictresolver/IDConflictResolver</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxyp/openms_featurefindermultiplex/FeatureFinderMultiplex.html">galaxyp/openms_featurefindermultiplex/FeatureFinderMultiplex</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
					Peptide and Protein Quantification via Stable Isotope Labelling (SIL)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/seq2hla/seq2hla.html">iuc/seq2hla/seq2hla</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-6-predicting-hla-binding/tutorial.html">
					Neoantigen 6: Predicting HLA Binding
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/optitype/optitype.html">iuc/optitype/optitype</a></td>
		<td>
			<ul>
				
				<li>
					Proteomics /
					<a href="/training-material/topics/proteomics/tutorials/neoantigen-6-predicting-hla-binding/tutorial.html">
					Neoantigen 6: Predicting HLA Binding
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/galaxy-australia/smudgeplot/smudgeplot.html">galaxy-australia/smudgeplot/smudgeplot</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/blobtoolkit/blobtoolkit.html">bgruening/blobtoolkit/blobtoolkit</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/interactive_tool_blobtoolkit.html">interactive_tool_blobtoolkit</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html">
					ERGA post-assembly QC
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/shovill/shovill.html">iuc/shovill/shovill</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html">
					Genome Assembly of a bacterial genome (MRSA) sequenced using Illumina MiSeq Data
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/yahs/yahs.html">iuc/yahs/yahs</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/salsa/salsa.html">iuc/salsa/salsa</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/purge_dups/purge_dups.html">iuc/purge_dups/purge_dups</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/hifiasm/hifiasm.html">bgruening/hifiasm/hifiasm</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/bionano_scaffold/bionano_scaffold.html">bgruening/bionano_scaffold/bionano_scaffold</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
					Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/polypolish/polypolish.html">iuc/polypolish/polypolish</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html">
					Genome Assembly of MRSA from Oxford Nanopore MinION data (and optionally Illumina data)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/filtlong/filtlong.html">iuc/filtlong/filtlong</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html">
					Genome Assembly of MRSA from Oxford Nanopore MinION data (and optionally Illumina data)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/CONVERTER_bz2_to_uncompressed.html">CONVERTER_bz2_to_uncompressed</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html">
					Genome Assembly of MRSA from Oxford Nanopore MinION data (and optionally Illumina data)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/compress_file/compress_file.html">iuc/compress_file/compress_file</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mitochondrion-assembly/tutorial.html">
					Assembly of the mitochondrial genome from PacBio HiFi reads
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/mitohifi/mitohifi.html">bgruening/mitohifi/mitohifi</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/mitochondrion-assembly/tutorial.html">
					Assembly of the mitochondrial genome from PacBio HiFi reads
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/seqtk/seqtk_sample.html">iuc/seqtk/seqtk_sample</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
					Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/parse_mito_blast/parse_mito_blast.html">iuc/parse_mito_blast/parse_mito_blast</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-decontamination/tutorial.html">
					Decontamination of a genome assembly
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/ncbi_blast_plus/ncbi_dustmasker_wrapper.html">devteam/ncbi_blast_plus/ncbi_dustmasker_wrapper</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/assembly-decontamination/tutorial.html">
					Decontamination of a genome assembly
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/simon-gladman/velvetoptimiser/velvetoptimiser.html">simon-gladman/velvetoptimiser/velvetoptimiser</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/debruijn-graph-assembly/tutorial.html">
					De Bruijn Graph Assembly
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nml/spades/spades.html">nml/spades/spades</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/debruijn-graph-assembly/tutorial.html">
					De Bruijn Graph Assembly
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/pilon/pilon.html">iuc/pilon/pilon</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/tutorial.html">
					Chloroplast genome assembly
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/circos/circos_interval_to_tiles.html">iuc/circos/circos_interval_to_tiles</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/circos/circos_aln_to_links.html">iuc/circos/circos_aln_to_links</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bedtools/bedtools_complementbed.html">iuc/bedtools/bedtools_complementbed</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/lastz/lastz_wrapper_2.html">devteam/lastz/lastz_wrapper_2</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html">
					Making sense of a newly assembled genome
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/idr/idr.html">iuc/idr/idr</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html">
					CUT&RUN data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/kpbioteam/ewastools/minfi_analysis.html">kpbioteam/ewastools/minfi_analysis</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html">
					Infinium Human Methylation BeadChip
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/kpbioteam/clusterprofiler_go/clusterprofiler_go.html">kpbioteam/clusterprofiler_go/clusterprofiler_go</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html">
					Infinium Human Methylation BeadChip
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/kpbioteam/clusterprofiler_bitr/clusterprofiler_bitr.html">kpbioteam/clusterprofiler_bitr/clusterprofiler_bitr</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html">
					Infinium Human Methylation BeadChip
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/kpbioteam/chipeakanno_annopeaks/chipeakanno_annopeaks.html">kpbioteam/chipeakanno_annopeaks/chipeakanno_annopeaks</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html">
					Infinium Human Methylation BeadChip
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/crs4/bwa_mem/bwa_mem.html">crs4/bwa_mem/bwa_mem</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html">
					Hi-C analysis of Drosophila melanogaster cells using HiCExplorer
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/hicexplorer_hicplottads/hicexplorer_hicplottads.html">bgruening/hicexplorer_hicplottads/hicexplorer_hicplottads</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html">
					Hi-C analysis of Drosophila melanogaster cells using HiCExplorer
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/hicexplorer_hicplotmatrix/hicexplorer_hicplotmatrix.html">bgruening/hicexplorer_hicplotmatrix/hicexplorer_hicplotmatrix</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html">
					Hi-C analysis of Drosophila melanogaster cells using HiCExplorer
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/hicexplorer_hicpca/hicexplorer_hicpca.html">bgruening/hicexplorer_hicpca/hicexplorer_hicpca</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html">
					Hi-C analysis of Drosophila melanogaster cells using HiCExplorer
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/hicexplorer_hicmergematrixbins/hicexplorer_hicmergematrixbins.html">bgruening/hicexplorer_hicmergematrixbins/hicexplorer_hicmergematrixbins</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html">
					Hi-C analysis of Drosophila melanogaster cells using HiCExplorer
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/hicexplorer_hicfindtads/hicexplorer_hicfindtads.html">bgruening/hicexplorer_hicfindtads/hicexplorer_hicfindtads</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html">
					Hi-C analysis of Drosophila melanogaster cells using HiCExplorer
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/hicexplorer_hiccorrectmatrix/hicexplorer_hiccorrectmatrix.html">bgruening/hicexplorer_hiccorrectmatrix/hicexplorer_hiccorrectmatrix</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html">
					Hi-C analysis of Drosophila melanogaster cells using HiCExplorer
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/hicexplorer_hicbuildmatrix/hicexplorer_hicbuildmatrix.html">bgruening/hicexplorer_hicbuildmatrix/hicexplorer_hicbuildmatrix</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html">
					Hi-C analysis of Drosophila melanogaster cells using HiCExplorer
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/rnateam/metilene/metilene.html">rnateam/metilene/metilene</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/methylation-seq/tutorial.html">
					DNA Methylation data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bwameth/bwameth.html">iuc/bwameth/bwameth</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/methylation-seq/tutorial.html">
					DNA Methylation data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/pileometh/pileometh.html">bgruening/pileometh/pileometh</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/methylation-seq/tutorial.html">
					DNA Methylation data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/deeptools_plot_profile/deeptools_plot_profile.html">bgruening/deeptools_plot_profile/deeptools_plot_profile</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/methylation-seq/tutorial.html">
					DNA Methylation data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/deeptools_correct_gc_bias/deeptools_correct_gc_bias.html">bgruening/deeptools_correct_gc_bias/deeptools_correct_gc_bias</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html">
					Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bedtools/bedtools_mergebed.html">iuc/bedtools/bedtools_mergebed</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/bbgbigwig/bbgtobigwig.html">iuc/bbgbigwig/bbgtobigwig</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/fubar/jbrowse2/jbrowse2.html">fubar/jbrowse2/jbrowse2</a></td>
		<td>
			<ul>
				
				<li>
					Epigenetics /
					<a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
					Formation of the Super-Structures on the Inactive X
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/transit_gumbel/transit_gumbel.html">iuc/transit_gumbel/transit_gumbel</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html">
					Essential genes detection with Transposon insertion sequencing
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/gff_to_prot/gff_to_prot.html">iuc/gff_to_prot/gff_to_prot</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html">
					Essential genes detection with Transposon insertion sequencing
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/bowtie_wrappers/bowtie_wrapper.html">devteam/bowtie_wrappers/bowtie_wrapper</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html">
					Essential genes detection with Transposon insertion sequencing
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/find_subsequences/bg_find_subsequences.html">bgruening/find_subsequences/bg_find_subsequences</a></td>
		<td>
			<ul>
				
				<li>
					Genome Annotation /
					<a href="/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html">
					Essential genes detection with Transposon insertion sequencing
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nick/sequence_content_trimmer/sequence_content_trimmer.html">nick/sequence_content_trimmer/sequence_content_trimmer</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dunovo/tutorial.html">
					Calling very rare variants
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nick/dunovo/make_families.html">nick/dunovo/make_families</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dunovo/tutorial.html">
					Calling very rare variants
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nick/dunovo/dunovo.html">nick/dunovo/dunovo</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dunovo/tutorial.html">
					Calling very rare variants
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nick/dunovo/correct_barcodes.html">nick/dunovo/correct_barcodes</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dunovo/tutorial.html">
					Calling very rare variants
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nick/dunovo/align_families.html">nick/dunovo/align_families</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dunovo/tutorial.html">
					Calling very rare variants
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/nick/allele_counts/allele_counts_1.html">nick/allele_counts/allele_counts_1</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dunovo/tutorial.html">
					Calling very rare variants
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/blankenberg/naive_variant_caller/naive_variant_caller.html">blankenberg/naive_variant_caller/naive_variant_caller</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/dunovo/tutorial.html">
					Calling very rare variants
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/pathview/pathview.html">iuc/pathview/pathview</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/length_and_gc_content/length_and_gc_content.html">iuc/length_and_gc_content/length_and_gc_content</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/dexseq/dexseq_count.html">iuc/dexseq/dexseq_count</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/dexseq/dexseq.html">iuc/dexseq/dexseq</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/deg_annotate/deg_annotate.html">iuc/deg_annotate/deg_annotate</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/__TAG_FROM_FILE__.html">__TAG_FROM_FILE__</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/ChangeCase.html">ChangeCase</a></td>
		<td>
			<ul>
				
				<li>
					Transcriptomics /
					<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">
					Reference-based RNA-Seq data analysis
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/bgruening/sklearn_numeric_clustering/sklearn_numeric_clustering.html">bgruening/sklearn_numeric_clustering/sklearn_numeric_clustering</a></td>
		<td>
			<ul>
				
				<li>
					Statistics and machine learning /
					<a href="/training-material/topics/statistics/tutorials/clustering_machinelearning/tutorial.html">
					Clustering in Machine Learning
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/velvet/velveth.html">devteam/velvet/velveth</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/general-introduction/tutorial.html">
					An Introduction to Genome Assembly
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/velvet/velvetg.html">devteam/velvet/velvetg</a></td>
		<td>
			<ul>
				
				<li>
					Assembly /
					<a href="/training-material/topics/assembly/tutorials/general-introduction/tutorial.html">
					An Introduction to Genome Assembly
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/saskia-hiltemann/krona_text/krona-text.html">saskia-hiltemann/krona_text/krona-text</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/mothur_make_group/mothur_make_group.html">iuc/mothur_make_group/mothur_make_group</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/metaphlan2krona/metaphlan2krona.html">iuc/metaphlan2krona/metaphlan2krona</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/metaphlan2/metaphlan2.html">iuc/metaphlan2/metaphlan2</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/humann2_renorm_table/humann2_renorm_table.html">iuc/humann2_renorm_table/humann2_renorm_table</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/humann2_regroup_table/humann2_regroup_table.html">iuc/humann2_regroup_table/humann2_regroup_table</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/iuc/humann2/humann2.html">iuc/humann2/humann2</a></td>
		<td>
			<ul>
				
				<li>
					Microbiome /
					<a href="/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html">
					Analyses of metagenomics data - The global picture
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/fastq_quality_filter/cshl_fastq_quality_filter.html">devteam/fastq_quality_filter/cshl_fastq_quality_filter</a></td>
		<td>
			<ul>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-short/tutorial.html">
					A short introduction to Galaxy
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/get_flanks/get_flanks1.html">devteam/get_flanks/get_flanks1</a></td>
		<td>
			<ul>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html">
					From peaks to genes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/CONVERTER_interval_to_bed_0.html">CONVERTER_interval_to_bed_0</a></td>
		<td>
			<ul>
				
				<li>
					Introduction to Galaxy Analyses /
					<a href="/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html">
					From peaks to genes
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/vcffilter/vcffilter2.html">devteam/vcffilter/vcffilter2</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html">
					Calling variants in non-diploid systems
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	
	
	<tr>
		<td><a href="/training-material/by-tool/devteam/vcf2tsv/vcf2tsv.html">devteam/vcf2tsv/vcf2tsv</a></td>
		<td>
			<ul>
				
				<li>
					Variant Analysis /
					<a href="/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html">
					Calling variants in non-diploid systems
					</a>
				</li>
				
			</ul>
		</td>
	</tr>
	
	
	</tbody>

</table>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/stats/top-tools.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/stats/top-tools.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/ae9d750ff4d4c69ea19d387b01201377aa82bb0b">ae9d750</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>
